// @appdirect/sfb-theme-components/src/atoms/tooltip/stories/Tooltip.stories.js
import React from 'react';
import Tooltip from '../Tooltip';
import IconComponent from '../../icon/Icon';

const placeholder = (
    <ul style={{ padding: 0, margin: 0 }}>
        <li>item 1</li>
        <li>item 2</li>
    </ul>
);

export default {
    title: 'Atoms/Tooltip',
    component: Tooltip
};

export const Default = {
    render: () => (
        <Tooltip
            id="default"
            placeholder="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eget iaculis nibh."
        >
            Tooltip
        </Tooltip>
    ),

    name: 'Default'
};

export const Icon = {
    render: () => (
        <Tooltip id="icon" placeholder="Add to favorites">
            <IconComponent type="star" width="20px" />
        </Tooltip>
    ),

    name: 'Icon'
};

export const MultipleTooltipsInParagraph = {
    render: () => (
        <p>
            {'Lorem '}
            <Tooltip id="placeholder-1" placeholder="ipsum">
                ipsum
            </Tooltip>
            {' dolor sit amet, consectetur elit. Donec, rutrum vel vitae, blandit vel est.'}
        </p>
    ),

    name: 'Multiple tooltips in paragraph'
};

export const HtmlTooltip = {
    render: () => (
        <Tooltip id="html" placeholder={placeholder} position="right">
            Tooltip
        </Tooltip>
    ),
    name: 'HTML tooltip'
};
