// @appdirect/sfb-theme-components/src/components/header/atoms/header-navigation-bar/stories/HeaderNavigationBar.stories.js
import React from 'react';
import HeaderNavigationBar from '../HeaderNavigationBar';
import { headerNavigationBarDataMock,
    headerNavigationBarDataMockWithSearchBar } from '../../../../../../mocks/components/header';
import { headerNavigationBarSchemaForm } from '../../../../../constants/components/schemas/header';
import { getArgTypes, getArgs } from '../../../../../tools/controlsHelper';

export default {
    title: 'Global Components/Header/Atoms/Header Navigation Bar',
    component: HeaderNavigationBar,
    argTypes: getArgTypes(headerNavigationBarSchemaForm),
    parameters: {
        layout: 'fullscreen'
    }
};

export const Default = {
    render: args => (
        <HeaderNavigationBar data={headerNavigationBarDataMock} settings={args} />
    ),
    args: getArgs(headerNavigationBarSchemaForm)
};

export const WithCategoryMenu = {
    render: args => (
        <HeaderNavigationBar data={headerNavigationBarDataMock} settings={args} />
    ),
    args: getArgs(headerNavigationBarSchemaForm)
};

export const WithSearchBar = {
    render: args => (
        <HeaderNavigationBar
            data={headerNavigationBarDataMockWithSearchBar}
            settings={args}
        />
    ),
    args: getArgs(headerNavigationBarSchemaForm)
};
