// @appdirect/sfb-theme-components/src/atoms/sticky-banner-container/stories/StickyBannerContainer.stories.js
import React from 'react';
import StickyBannerContainer from '../StickyBannerContainer';
import SelectionPreview from '../../selection-preview/SelectionPreview';
import { ProductEditionListComponent as ProductEditionList } from '../../product-edition/list/ProductEditionList';
import { editionsMockData as mockData } from '../../../../mocks/product/productEditions';
import { selectedAddonsMock } from '../../../../mocks/product/configuratorSelectedAddons';
import { productCardMock } from '../../../../mocks/product/productCard';

const editionsData = {
    editions: mockData.items,
    i18n: mockData.i18n,
    callToActions: {
        items: [
            {
                label: 'Buy Now'
            }
        ]
    }
};

export default {
    title: 'Atoms (Preview)/Sticky Banner Container',
    component: StickyBannerContainer
};

export const Default = {
    render: () => (
        <StickyBannerContainer
            stickyBannerContent={<div>This is the sticky banner content.</div>}
            content={<div>This is the content of the rest of the layout.</div>}
        />
    ),

    name: 'Default'
};

export const Mobile = {
    render: () => (
        <div
            style={{
                width: '320px'
            }}
        >
            <StickyBannerContainer
                stickyBannerContent={<div>This is the sticky banner content.</div>}
                content={<div>This is the content of the rest of the layout.</div>}
            />
        </div>
    ),

    name: 'Mobile'
};

export const Configurator = {
    render: () => (
        <div
            style={{
                height: '500px'
            }}
        >
            <StickyBannerContainer
                stickyBannerContent={(
                    <SelectionPreview
                        data={{
                            configurator: selectedAddonsMock.configurator,
                            selectedEdition: selectedAddonsMock.configurator.selectedEdition,
                            callToActions: selectedAddonsMock.callToActions,
                            productImgSrc: productCardMock.logoSrc
                        }}
                    />
                )}
                content={<ProductEditionList data={editionsData} configuratorMode />}
            />
        </div>
    ),

    name: 'Configurator'
};
