// @appdirect/sfb-theme-components/mocks/components/featureDetails.js
import { featuresMock } from '../product/features';
import { gSuiteMock, tSheetsMock } from '../product/features/items';

export const featureDetailsMocks = {
    features: featuresMock
};

export const gSuiteFeatureDetailsMock = {
    features: {
        ...featuresMock,
        items: gSuiteMock
    }
};

export const tSheetsFeatureDetailsMock = {
    features: {
        ...featuresMock,
        items: tSheetsMock
    }
};

export const featureDetailsWithHTMLMock = {
    features: {
        ...featuresMock,
        items: [
            ...tSheetsMock,
            {
                ...tSheetsMock[0],
                description: '<b>All you need to do your best work</b>, together in one package that works seamlessly from your computer, phone or tablet',
                belowCharacteristics: [
                    {
                        id: null,
                        title: 'Create schedules from any device',
                        position: 0,
                        abovePicture: false,
                        highlights: [
                            {
                                highlight: 'Easy <b>scheduling</b> allows admins to create and update employee schedules from desktops, laptops, and mobile devices. No more dry-erase boards and printed Excel spreadsheets.',
                                displayCheck: false
                            }
                        ]
                    }
                ]
            }
        ]
    }
};
