// @appdirect/sfb-theme-components/mocks/components/hero-banner.js
export const heroBannerSettings = {
    style: 'hero',
    height: 'large',
    isTaglineVisible: true,
    isTitleVisible: true,
    isTextVisible: true,
    isButtonVisible: true,
    isSearchBarVisible: false,
    tagline: 'Tagline goes here',
    title: 'Longer headline to turn your visitors into customers',
    text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    imageSize: 'scaleToFitPage',
    taglineColor: '#FFFFFF',
    titleColor: '#FFFFFF',
    descriptionColor: '#FFFFFF',
    backgroundColor: '#5F5F5F',
    shadowColor: '#C3C3C3',
    textAlignment: 'left',
    textPosition: 'topLeft',
    textSize: 'medium',
    buttonSize: 'default',
    buttonType: 'primary',
    buttonLabel: 'Learn more',
    searchBarWidth: 521,
    searchCornerRadius: 3,
    searchPlaceholder: 'Search the marketplace',
    searchPlaceholderTextColor: '#C3C3C3',
    searchIcon: null,
    searchBorderColor: null,
    searchBackgroundColor: '#FFFFFF',
    searchButtonColor: '#00769D',
    searchIconColor: '#FFFFFF'
};

const { style, ...restOfSettings } = heroBannerSettings;
export const heroStyleSettings = restOfSettings;

export const backgroundImage = {
    url: 'https://images.unsplash.com/photo-1542903660-eedba2cda473?q=80&w=2070&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
    alt: 'tech image'
};

export const foregroundImage = {
    url: 'https://images.unsplash.com/photo-1542831371-29b0f74f9713?q=80&w=2070&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
    alt: 'another tech image'
};

const { textPosition,
    searchBarWidth,
    ...restOfHeroBannerSettings } = heroBannerSettings;

export const imageStyleSettings = {
    ...restOfHeroBannerSettings,
    foregroundImage,
    foregroundImagePosition: 'left',
    foregroundImageSize: 'scaleToFill',
    textPositionImageStyle: 'top',
    searchBarWidthImageStyle: 412
};
