// @appdirect/sfb-theme-components/mocks/components/productImageViewer.js
import { screenshotsMock } from '../product/screenshots';
import { productOverviewMock } from '../product/overview';
import { youtubeMock, vimeoMock } from '../product/overview/callToActions';

export const defaultMock = {
    screenshots: screenshotsMock,
    overview: productOverviewMock
};

export const zoomDemoMock = {
    screenshots: {
        items: [
            {
                name: 'Horizontal image',
                imageUrl: 'https://picsum.photos/id/1020/800/600',
                imageUrlSrcset: {
                    default: 'https://picsum.photos/id/1020/800/600',
                    '1x': 'https://picsum.photos/id/1020/800/600',
                    '2x': 'https://picsum.photos/id/1020/1500/1200'
                }
            },
            {
                name: 'Vertical image',
                imageUrl: 'https://picsum.photos/id/1011/600/800',
                imageUrlSrcset: {
                    default: 'https://picsum.photos/id/1011/600/800',
                    '1x': 'https://picsum.photos/id/1011/600/800',
                    '2x': 'https://picsum.photos/id/1011/1200/1500'
                }
            },
            {
                name: 'Square image',
                imageUrl: 'https://picsum.photos/id/1029/600/600',
                imageUrlSrcset: {
                    default: 'https://picsum.photos/id/1029/600/600',
                    '1x': 'https://picsum.photos/id/1029/600/600',
                    '2x': 'https://picsum.photos/id/1029/1500/1500'
                }
            },
            {
                name: '1100 image',
                imageUrl: 'https://picsum.photos/id/1/1100/660',
                imageUrlSrcset: {
                    default: 'https://picsum.photos/id/1/1100/660',
                    '1x': 'https://picsum.photos/id/1/1100/660',
                    '2x': 'https://picsum.photos/id/1/1100/660'
                }
            }
        ],
        i18n: {
            title: 'Screenshots',
            sidebarTitle: 'View Screenshots'
        }
    },
    overview: productOverviewMock
};

export const youtubeDemoMock = {
    ...defaultMock,
    overview: {
        ...defaultMock.overview,
        callToActions: {
            ...defaultMock.overview.callToActions,
            demo: youtubeMock
        }
    }
};

export const vimeoDemoMock = {
    ...defaultMock,
    overview: {
        ...defaultMock.overview,
        callToActions: {
            ...defaultMock.overview.callToActions,
            demo: vimeoMock
        }
    }
};

export const youtubeDemoWithoutScreenshotsMock = {
    ...defaultMock,
    screenshots: {},
    overview: {
        ...defaultMock.overview,
        callToActions: {
            ...defaultMock.overview.callToActions,
            demo: youtubeMock
        }
    }
};

export const tallImagesMock = {
    screenshots: {
        items: [
            {
                name: '500x600',
                imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/screenshot/img7295648017623061531.png',
                imageUrlSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/screenshot/img7295648017623061531.png',
                    '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/screenshot/img2441025428261698960.png',
                    '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/screenshot/img7295648017623061531.png'
                }
            },
            {
                name: '500x600',
                imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/screenshot/img4342220252299883492.png',
                imageUrlSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/screenshot/img4342220252299883492.png',
                    '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/screenshot/img2699469929175347469.png',
                    '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/screenshot/img4342220252299883492.png'
                }
            }
        ],
        i18n: {
            title: 'Screenshots',
            sidebarTitle: 'View Screenshots'
        }
    },
    overview: {
        title: 'Congratulations!',
        description: 'You now have access to AT&T Productivity Suite where you can finish the sign-up process for your Webex Meetings with AT&T trial.',
        callToActions: {
            demo: {
                label: 'Watch Demo',
                href: 'https://vimeo.com/157214359',
                iFrameFriendlyUrl: '//player.vimeo.com/video/157214359'
            }
        },
        image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/overview/img4619154643990594698.png',
        imageSrcSet: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/overview/img4619154643990594698.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/overview/img4619154643990594698.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/overview/img4619154643990594698-2x.png'
        },
        showThirdPartyLegalDisclosure: false,
        documentationUrl: 'https://help.appwise.com/',
        downloadDocumentationUrl: null,
        pricingUrl: '/en-US/apps/330408/webex-meetings-with-att---trial/editions',
        customers: [],
        benefits: [],
        i18n: {
            customers: 'Featured Customers',
            featuresAndBenefits: 'Features and Benefits',
            demo: 'Demo',
            overviewImage: 'Overview Image',
            documentation: 'Read Documentation',
            startingFrom: 'Starting From',
            seePricingPlans: 'See Pricing Plans',
            availableAddons: 'Available Add-ons',
            availableBundles: 'Bundle Package Includes'
        }
    }
};
