// @appdirect/sfb-theme-components/src/constants/atoms/schema/flex-container.js
import { DROPDOWN, SLIDER } from '../../schemaComponentTypes';
import { LEFT, RIGHT, CENTER, SPACE_AROUND, SPACE_BETWEEN, SPACE_EVENLY } from '../../alignment';
import FlexContainer from '../../../atoms/flex-container/FlexContainer';

const ALIGN_OPTIONS = [LEFT, CENTER, RIGHT, SPACE_AROUND, SPACE_BETWEEN, SPACE_EVENLY];

const flexContainerForm = {
    rowGap: {
        title: 'uieditor.sfbComponent.flexContainer.rowGap.title',
        type: SLIDER,
        defaultValue: 0,
        options: {
            range: true,
            min: 0,
            max: 35,
            step: 1
        }
    },
    columnGap: {
        title: 'uieditor.sfbComponent.flexContainer.columnGap.title',
        type: SLIDER,
        defaultValue: 0,
        options: {
            range: true,
            min: 0,
            max: 35,
            step: 1
        }
    },
    contentAlignment: {
        title: 'uieditor.sfbComponent.flexContainer.contentAlignment.title',
        type: DROPDOWN,
        defaultValue: ALIGN_OPTIONS[0].value,
        options: ALIGN_OPTIONS
    }
};

FlexContainer.schema = () => ({
    name: 'FlexContainer',
    title: 'uieditor.sfbComponent.flexContainer.title',
    form: flexContainerForm
});

export default FlexContainer;
