// @appdirect/sfb-theme-components/src/atoms/product-addon/list/stories/ProductAddonList.stories.js
import React from 'react';
import { addonsMock, addonsWithDiscountsMock } from '../../../../../mocks/product/description/addons';
import { ProductAddonListComponent as ProductAddonList } from '../ProductAddonList';

const addonsData = {
    addonProducts: {
        items: addonsMock,
        i18n: {
            addons: 'Add-ons available',
            selectAddonsTitle: 'Add-ons available for this plan',
            inclusions: 'Included'
        }
    }
};

export default {
    title: 'Atoms/Product Addon List',
    component: ProductAddonList
};

export const Default = {
    render: () => <ProductAddonList data={addonsData} />,
    name: 'Default'
};

export const WithDiscount = {
    render: () => (
        <ProductAddonList
            data={{
                ...addonsData,

                addonProducts: {
                    ...addonsData.addonProducts,
                    items: addonsWithDiscountsMock
                }
            }}
        />
    ),

    name: 'With discount'
};

export const Carousel = {
    render: () => (
        <div
            style={{
                maxWidth: 600,
                margin: '0 auto'
            }}
        >
            <ProductAddonList data={addonsData} carouselEnabled />
        </div>
    ),

    name: 'Carousel'
};

export const Selectable = {
    render: () => (
        <div
            style={{
                maxWidth: 600,
                margin: '0 auto'
            }}
        >
            <ProductAddonList
                data={{
                    ...addonsData,

                    addonProducts: {
                        ...addonsData.addonProducts,
                        items: addonsData.addonProducts.items.slice(0, 3)
                    }
                }}
                selectedAddons={[addonsData.addonProducts.items[0]]}
                selectable
            />
        </div>
    ),

    name: 'Selectable'
};
