// @appdirect/sfb-theme-components/src/constants/accessibility.js
export const KEYBOARD_KEYS = {
    esc: 'Escape',
    tab: 'Tab',
    rightKey: 'ArrowRight',
    leftKey: 'ArrowLeft',
    upKey: 'ArrowUp',
    downKey: 'ArrowDown'
};

export const KEYBOARD_KEY_CODES = {
    tab: 9,
    return: 13,
    esc: 27,
    space: 32,
    pageUp: 33,
    pageDown: 34,
    end: 35,
    home: 36,
    left: 37,
    up: 38,
    right: 39,
    down: 40
};

export const FOCUSABLE_SELECTORS = [
    'a[href]:not([tabindex^="-"])',
    'area[href]:not([tabindex^="-"])',
    'input:not([type="hidden"]):not([type="radio"]):not([disabled]):not([tabindex^="-"])',
    'input[type="radio"]:not([disabled]):not([tabindex^="-"]):checked',
    'select:not([disabled]):not([tabindex^="-"])',
    'textarea:not([disabled]):not([tabindex^="-"])',
    'button:not([disabled]):not([tabindex^="-"])',
    'iframe:not([tabindex^="-"])',
    'audio[controls]:not([tabindex^="-"])',
    'video[controls]:not([tabindex^="-"])',
    '[contenteditable]:not([tabindex^="-"])',
    '[tabindex]:not([tabindex^="-"])'
];
