// @appdirect/sfb-theme-components/src/constants/actions.js
export const CTA_ADD_TO_CART = 'ADD_TO_CART';
export const CTA_BUY = 'BUY';
export const CTA_BUY_MORE = 'BUY_MORE';
export const CTA_CANCEL = 'CANCEL';
export const CTA_DOMAIN_ADD_TO_CART = 'DOMAIN_ADD_TO_CART';
export const CTA_FREE = 'FREE';
export const CTA_FREE_TRIAL = 'FREE_TRIAL';
export const CTA_LEAD_GEN = 'LEAD_GEN';
export const CTA_MANAGE_APP = 'MANAGE_APP';
export const CTA_NO_ACTION = 'NO_ACTION';
export const CTA_REFERABLE = 'REFERABLE';
export const CTA_UPDATE_CART_ITEM = 'UPDATE_CART_ITEM';
export const CTA_UPGRADE = 'CAN_MOSI_UPGRADE';
export const CTA_CONFIGURE = 'CONFIGURE';
export const CTA_UNDER_MAINTENANCE = 'UNDER_MAINTENANCE';
export const CTA_FREE_TRIAL_CONFIGURE = 'FREE_TRIAL_CONFIGURE';
export const CTA_BUY_CROSS_SELL = 'BUY_CROSS_SELL';
export const CTA_ADD_TO_CART_CROSS_SELL = 'ADD_TO_CART_CROSS_SELL';
export const CTA_UPDATE_CART_ITEM_CROSS_SELL = 'UPDATE_CART_ITEM_CROSS_SELL';

export const ALLOWED_CTA_TYPES = [
    CTA_ADD_TO_CART,
    CTA_BUY,
    CTA_BUY_MORE,
    CTA_CANCEL,
    CTA_CONFIGURE,
    CTA_DOMAIN_ADD_TO_CART,
    CTA_FREE,
    CTA_FREE_TRIAL,
    CTA_LEAD_GEN,
    CTA_MANAGE_APP,
    CTA_NO_ACTION,
    CTA_REFERABLE,
    CTA_UPDATE_CART_ITEM,
    CTA_UPGRADE,
    CTA_FREE_TRIAL_CONFIGURE,
    CTA_ADD_TO_CART_CROSS_SELL,
    CTA_UPDATE_CART_ITEM_CROSS_SELL,
    CTA_UNDER_MAINTENANCE
];

export const CART_URL = '/checkout/cart';

export const DOMAINS_CART_STORAGE_KEY = 'standaloneDomainsCart:';
