// @appdirect/sfb-theme-components/src/constants/alignment.js
export const LEFT = {
    value: 'left',
    label: 'uieditor.sfbComponent.dropdown.option.left'
};

export const RIGHT = {
    value: 'right',
    label: 'uieditor.sfbComponent.dropdown.option.right'
};

export const CENTER = {
    value: 'center',
    label: 'uieditor.sfbComponent.dropdown.option.center'
};

export const ALTERNATING = {
    value: 'alternating',
    label: 'uieditor.sfbComponent.dropdown.option.alternating'
};

export const TOP = {
    value: 'top',
    label: 'uieditor.sfbComponent.dropdown.option.top'
};

export const MIDDLE = {
    value: 'middle',
    label: 'uieditor.sfbComponent.dropdown.option.middle'
};

export const BOTTOM = {
    value: 'bottom',
    label: 'uieditor.sfbComponent.dropdown.option.bottom'
};

export const NONE = {
    value: 'none',
    label: 'uieditor.sfbComponent.dropdown.option.none'
};

export const TOP_LEFT = {
    value: 'topLeft',
    label: 'uieditor.sfbComponent.dropdown.option.topLeft'
};

export const TOP_RIGHT = {
    value: 'topRight',
    label: 'uieditor.sfbComponent.dropdown.option.topRight'
};

export const BOTTOM_LEFT = {
    value: 'bottomLeft',
    label: 'uieditor.sfbComponent.dropdown.option.bottomLeft'
};

export const BOTTOM_RIGHT = {
    value: 'bottomRight',
    label: 'uieditor.sfbComponent.dropdown.option.bottomRight'
};

export const MIDDLE_LEFT = {
    value: 'middleLeft',
    label: 'uieditor.sfbComponent.dropdown.option.middleLeft'
};

export const MIDDLE_RIGHT = {
    value: 'middleRight',
    label: 'uieditor.sfbComponent.dropdown.option.middleRight'
};

export const TOP_CENTER = {
    value: 'topCenter',
    label: 'uieditor.sfbComponent.dropdown.option.topCenter'
};

export const MIDDLE_CENTER = {
    value: 'middleCenter',
    label: 'uieditor.sfbComponent.dropdown.option.middleCenter'
};

export const BOTTOM_CENTER = {
    value: 'bottomCenter',
    label: 'uieditor.sfbComponent.dropdown.option.bottomCenter'
};

export const SPACE_AROUND = {
    value: 'space-around',
    label: 'uieditor.sfbComponent.dropdown.option.spaceAround'
};

export const SPACE_BETWEEN = {
    value: 'space-between',
    label: 'uieditor.sfbComponent.dropdown.option.spaceBetween'
};

export const SPACE_EVENLY = {
    value: 'space-evenly',
    label: 'uieditor.sfbComponent.dropdown.option.spaceEvenly'
};
