// @appdirect/sfb-theme-components/src/constants/profileTabs.js
export const PROFILE_TABS = {
    overview: {
        label: '',
        defaultLabel: 'uieditor.tabs.overview',
        href: '/en-US/apps/163915/talech/overview',
        value: 'overview',
        selected: true,
        isEditMode: false,
        isVisible: true
    },
    features: {
        label: '',
        defaultLabel: 'uieditor.tabs.features',
        href: '/en-US/apps/163915/talech/features',
        value: 'features',
        selected: false,
        isEditMode: false,
        isVisible: true
    },
    reviews: {
        label: '',
        defaultLabel: 'uieditor.tabs.review',
        href: '/en-US/apps/163915/talech/reviews',
        value: 'reviews',
        selected: false,
        isEditMode: false,
        isVisible: true
    },
    questions: {
        label: '',
        defaultLabel: 'uieditor.tabs.questions',
        href: '/en-US/apps/163915/talech/questions',
        value: 'questions',
        selected: false,
        isEditMode: false,
        isVisible: true
    },
    policies: {
        label: '',
        defaultLabel: 'uieditor.tabs.support',
        href: '/en-US/apps/163915/talech/support',
        value: 'support',
        selected: false,
        isEditMode: false,
        isVisible: true
    },
    resources: {
        label: '',
        defaultLabel: 'uieditor.tabs.resources',
        href: '/en-US/apps/163915/talech/resources',
        value: 'resources',
        selected: false,
        isEditMode: false,
        isVisible: true
    },
    editions: {
        label: '',
        defaultLabel: 'uieditor.tabs.editions',
        href: '/en-US/apps/163915/talech/editions',
        value: 'editions',
        selected: false,
        isEditMode: false,
        isVisible: true
    }
};
