// @appdirect/sfb-theme-components/src/constants/sizes.js
export const XSMALL = {
    value: 'xsmall',
    label: 'uieditor.sfbComponent.dropdown.option.xsmall'
};

export const SMALL = {
    value: 'small',
    label: 'uieditor.sfbComponent.dropdown.option.small'
};

export const MEDIUM = {
    value: 'medium',
    label: 'uieditor.sfbComponent.dropdown.option.medium'
};

export const XLARGE = {
    value: 'xlarge',
    label: 'uieditor.sfbComponent.dropdown.option.xlarge'
};

export const LARGE = {
    value: 'large',
    label: 'uieditor.sfbComponent.dropdown.option.large'
};

export const XSMALL_255 = {
    value: 'xsmall',
    // FIXME: SFB-4203 - allow passing param to translation
    label: 'Extra small 255px'
};

export const SMALL_280 = {
    value: 'small',
    // FIXME: SFB-4203 - allow passing param to translation
    label: 'Small 280px'
};

export const MEDIUM_350 = {
    value: 'medium',
    // FIXME: SFB-4203 - allow passing param to translation
    label: 'Medium 350px'
};

export const LARGE_466 = {
    value: 'large',
    // FIXME: SFB-4203 - allow passing param to translation
    label: 'Large 466px'
};

export const XLARGE_600 = {
    value: 'xlarge',
    // FIXME: SFB-4203 - allow passing param to translation
    label: 'Extra large 600px'
};
