// @appdirect/sfb-theme-components/.storybook/main.js
const fs = require('fs');
// This package will be deprecated in the future: https://www.npmjs.com/package/@storybook/addon-mdx-gfm
// Storybook recommends using remark-gfm instead: https://storybook.js.org/docs/writing-docs/mdx#markdown-tables-arent-rendering-correctly
const remarkGfm = require('remark-gfm');

const isToolkit = __dirname.includes('node_modules');

const customComponentsPath = () => {
	if (isToolkit && fs.existsSync(`${__dirname}/../../../../customComponents`)) {
		// TODO: FIX THIS
		return ['../../../../customComponents/**/*.story.@(jsx|mdx)']
	};
	return [];
};

module.exports = {
    stories: [
		'../src/**/*.stories.(js|jsx)',
		'../src/**/*.mdx',
		'../docs/**/*.mdx',
		...customComponentsPath()
	],

    addons: [
				'@storybook/addon-controls',
				'storybook-dark-mode',
        '@storybook/addon-a11y',
        '@storybook/addon-viewport',
        {
            name: './grid-addon/register.js',
            title: 'Grid',
        },
				{
					name: '@storybook/addon-docs',
					options: {
						mdxPluginOptions: {
							mdxCompileOptions: {
								remarkPlugins: [remarkGfm],
							},
						},
					},
				},
    ],

    // treat .mjs in node_modules/ as CJS/ESM instead of forcing one type
    webpackFinal: async (config) => {
		config.module.rules.push({
			test: /\.mjs$/,
			include: /node_modules/,
			type: "javascript/auto",
			use: {
				loader: "babel-loader",
				options: {
					// transpile
					presets: ["@babel/preset-env"],
					sourceType: "unambiguous",
				},
			},
		});
		return config;
	},

    framework: {
        name: '@storybook/react-webpack5',
        options: {}
    },

		staticDirs: [
			'../docs/images'
		],

    docs: {
        autodocs: true
    }
};
