// @appdirect/sfb-theme-components/mocks/nav/primary-nav/categories.js
export const categoriesMock = [
    {
        id: 67285,
        name: 'Cool Apps',
        subCategories: [
            {
                id: 45281,
                name: 'Chilly Apps',
                numTaggedProducts: 2,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/chilly-apps/45281',
                hrefListing: '/en-US/listing?scat=45281'
            },
            {
                id: 45285,
                name: 'Freezing Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            }
        ],
        numTaggedProducts: 5,
        selected: false,
        href: '/en-US/browse/category/cool-apps/67285',
        hrefListing: '/en-US/listing?cat=67285',
        childSelected: false
    },
    {
        id: 67289,
        name: 'Wild Apps',
        subCategories: [],
        numTaggedProducts: 4,
        selected: false,
        href: '/en-US/browse/category/wild-apps/67289',
        hrefListing: '/en-US/listing?cat=67289',
        childSelected: false
    }
];

export const categoriesMockWithMoreParentItems = [
    {
        id: 67285,
        name: 'Cool Apps',
        subCategories: [
            {
                id: 45281,
                name: 'Chilly Apps',
                numTaggedProducts: 2,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/chilly-apps/45281',
                hrefListing: '/en-US/listing?scat=45281'
            }
        ],
        numTaggedProducts: 5,
        selected: false,
        href: '/en-US/browse/category/cool-apps/67285',
        hrefListing: '/en-US/listing?cat=67285',
        childSelected: false
    },
    {
        id: 67289,
        name: 'Wild Apps',
        subCategories: [],
        numTaggedProducts: 4,
        selected: false,
        href: '/en-US/browse/category/wild-apps/67289',
        hrefListing: '/en-US/listing?cat=67289',
        childSelected: false
    },
    {
        id: 67290,
        name: 'Extra Apps',
        subCategories: [],
        numTaggedProducts: 4,
        selected: false,
        href: '/en-US/browse/category/extra-apps/67290',
        hrefListing: '/en-US/listing?cat=67290',
        childSelected: false
    }
];

// More complex mock for testing multi-level navigation in Storybook
export const categoriesMultiLevelMock = [
    {
        ...categoriesMock[0],
        subCategories: [
            {
                id: 45281,
                name: 'Chilly Apps',
                numTaggedProducts: 2,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/chilly-apps/45281',
                hrefListing: '/en-US/listing?scat=45281'
            },
            {
                id: 45285,
                name: 'Freezing Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            }
        ]
    },
    {
        ...categoriesMock[1],
        subCategories: [
            {
                id: 11171,
                name: 'Another level of apps',
                numTaggedProducts: 4,
                selected: false,
                href: '/en-US/browse/category/wild-apps/67289',
                hrefListing: '/en-US/listing?cat=67289',
                childSelected: false
            },
            {
                id: 74358,
                name: 'Another other level',
                subCategories: [],
                numTaggedProducts: 4,
                selected: false,
                href: '/en-US/browse/category/wild-apps/67289',
                hrefListing: '/en-US/listing?cat=67289',
                childSelected: false
            },
            {
                id: 96582,
                name: 'Fun Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            },
            {
                id: 99650,
                name: 'Funky Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            },
            {
                id: 88528,
                name: 'Exciting Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            },
            {
                id: 66369,
                name: 'Cooking Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            },
            {
                id: 14417,
                name: 'Sports Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            },
            {
                id: 55528,
                name: 'Entertaining Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            },
            {
                id: 11520,
                name: 'Hilarious Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            },
            {
                id: 74582,
                name: 'Mysterious Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            },
            {
                id: 78989,
                name: 'Lol Apps',
                numTaggedProducts: 1,
                selected: false,
                href: '/en-US/browse/category/cool-apps/67285/freezing-apps/45285',
                hrefListing: '/en-US/listing?scat=45285'
            }
        ]
    }
];
