// @appdirect/sfb-theme-components/src/atoms/buttons-list/stories/ButtonsList.stories.js
import React from 'react';
import ButtonsList from '../ButtonsList';
import { callToActionsMock,
    callToActionsCaptionMock,
    productBannerSkipConfiguratorCallToActionsMock } from '../../../../mocks/product/callToActions';

export default {
    title: 'Atoms/ButtonsList',
    component: ButtonsList
};

export const Default = {
    render: () => <ButtonsList data={callToActionsMock} />,
    name: 'Default'
};

export const WithCaption = {
    render: () => <ButtonsList data={callToActionsCaptionMock} />,
    name: 'With Caption'
};

export const BuyAndAddToCart = {
    render: () => (
        <ButtonsList
            data={{
                callToActions: productBannerSkipConfiguratorCallToActionsMock
            }}
        />
    ),

    name: 'Buy and Add to Cart'
};

export const BuyAndAddedToCart = {
    render: () => (
        <ButtonsList
            data={{
                callToActions: productBannerSkipConfiguratorCallToActionsMock,
                isAddedToCart: true,

                i18n: {
                    cart: {
                        addedToCart: 'Added to Cart'
                    }
                }
            }}
        />
    ),

    name: 'Buy and Added to Cart'
};
