// @appdirect/sfb-theme-components/src/namespaces/global/index.js
import '../../tools/derivedColorsObserver';

// import global stylesheet for the theme's website
import '../../styles/global.scss';

export { default as React } from 'react';
export { default as ReactDOM } from 'react-dom';

export { default as AtomsShowcase } from '../../components/atoms-showcase/AtomsShowcase';
export { default as Footer } from '../../components/footer/Footer';
export { default as Header } from '../../components/header/Header';
export { default as HeaderAnnouncementBar } from '../../components/header/atoms/header-announcement-bar/HeaderAnnouncementBar';
export { default as HeaderAuxiliaryBar } from '../../components/header/atoms/header-auxiliary-bar/HeaderAuxiliaryBar';
export { default as HeaderHomeBar } from '../../components/header/atoms/header-home-bar/HeaderHomeBar';
export { default as HeaderNavigationBar } from '../../components/header/atoms/header-navigation-bar/HeaderNavigationBar';
export { default as HeaderSearchBar } from '../../components/header/atoms/common/header-search-bar/HeaderSearchBar';
export { default as HeaderCategoriesMenu } from '../../components/header/atoms/common/header-categories-menu/HeaderCategoriesMenu';
export { default as GlobalAlerts } from '../../components/global-alerts/GlobalAlerts';
export { default as SecondaryNav } from '../../components/secondary-nav/SecondaryNav';
export { default as StickyButtonPageScroll } from '../../components/sticky-button-page-scroll/StickyButtonPageScroll';
