// @appdirect/sfb-theme-components/src/atoms/link-menu-item/stories/LinkMenuItem.stories.js
import React from 'react';
import LinkMenuItem from '../LinkMenuItem';
import { footerMenusMock } from '../../../../mocks/components/footer';
import { createNamespace } from '../../../tools/namingTools';

const namespaceString = 'SomeMenu';
const n = createNamespace(namespaceString);
const { menu } = footerMenusMock.BOTTOM_BAR_MENU;

export default {
    title: 'Atoms/LinkMenuItem',
    component: LinkMenuItem
};

export const Default = {
    render: () => (
        <LinkMenuItem
            menuItem={menu[0]}
            key={menu[0].uuid}
            namespaceString={namespaceString}
            mainItemClassNames={n('col').withTestId().props}
            mainItemSubListClassNames={n('list').withTestId().props}
            childItemClassNames={n('item').withTestId().props}
        />
    ),

    name: 'Default'
};

export const FullMenu = {
    render: () => menu.map(item => (
        <LinkMenuItem
            menuItem={item}
            key={item.uuid}
            namespaceString={namespaceString}
            mainItemClassNames={n('col').withTestId().props}
            mainItemSubListClassNames={n('list').withTestId().props}
            childItemClassNames={n('item').withTestId().props}
        />
    )),

    name: 'Full menu'
};

export const FullMenuHorizontal = {
    render: () => (
        <div
            style={{
                display: 'flex',
                flexDirection: 'row',
                justifyContent: 'space-between'
            }}
        >
            {menu.map(item => (
                <LinkMenuItem
                    menuItem={item}
                    key={item.uuid}
                    namespaceString={namespaceString}
                    mainItemClassNames={n('col').withTestId().props}
                    mainItemSubListClassNames={n('list').withTestId().props}
                    childItemClassNames={n('item').withTestId().props}
                />
            ))}
        </div>
    ),

    name: 'Full menu - horizontal'
};
