// @appdirect/sfb-theme-components/mocks/config/index.js
import { userLoggedOutMock, userLoggedInMock } from './user';
import { logoMock } from './logo';
import { localeMock } from './locale';

export const configMock = {
    user: userLoggedOutMock,
    marketplaceName: 'sfbmarketplace',
    logo: logoMock,
    channelSettings: {
        partner: 'SFBMARKETPLACE',
        capabilitie: [
            'CUSTOMER_APPLICATION_MANAGEMENT',
            'DEVELOPER',
            'CHECKOUT',
            'PRODUCT_PURCHASE'
        ],
        showProfiles: true
    },
    channelCanonicalBaseUrl: 'https://sfbmarketplace.byappdirect.com',
    customLogo: 'https://d3bql97l1ytoxn.cloudfront.net/marketplace_logo/defaultlogo.png?988e84efc1a891b025e6d5c7eff56d27',
    customFavicon: null,
    privacyPolicyUrl: null,
    support: {
        contactUrl: null,
        email: null,
        phone: null,
        url: null
    },
    reviewRequirePurchase: false,
    termsUrl: null,
    ownerBaseUrlMap: null,
    importAppsEnabled: false,
    isCheckoutV2Enabled: true,
    isAddToCartCTAEnabled: false,
    listingEndpoint: '/en-US/listing',
    locale: localeMock
};

export const configUserLoggedInMock = {
    ...configMock,
    user: userLoggedInMock
};
