// @appdirect/sfb-theme-components/mocks/productLists/popularApps.js
export const popularAppsMock = {
    name: 'Popular Apps',
    url: '/en-US/listing?order=POPULARITY',
    items: [
        {
            id: 42937,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/thumbs_64/img63732497944352803.png?0b64020ae947c80340ac4f9716c1d01b',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/thumbs_64/img63732497944352803.png?0b64020ae947c80340ac4f9716c1d01b',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/thumbs_64/img63732497944352803-2x.png?197ad9cbc5bc0b718c0df6c3c3300bfa',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/thumbs_64/img63732497944352803.png?0b64020ae947c80340ac4f9716c1d01b'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/thumbs_112/img1099429097555068136.png?0b64020ae947c80340ac4f9716c1d01b',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/overview/img1301783796595200436.png?18e59c26ffbf1b84953937d376a29fdc',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '622a27d2-09ac-46d1-87e2-3ffa764e8bce',
            description: 'AppWise is a business productivity tool that provides Unified Search across 40+ leading applications like Salesforce, Gmail, Office 365, Box and Asana.   AppWise has enterprise-grade security built-in from the ground up, and it\'s intuitive enough be setup by end-users in seconds.',
            overview: 'With Unified Search across over 40+ leading cloud applications, easily find all your emails, documents and tasks in a couple of clicks.',
            vendorName: 'AppWise, Inc.',
            showRating: true,
            hidePricings: false,
            free: true,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'AppWise',
            blurb: 'Search everything, in one shot',
            url: '/en-US/apps/42937/appwise',
            startingPrice: {
                formattedFullPrice: 'Free',
                formattedPrice: 'Free',
                price: 'Free',
                priceCaption: '',
                frequency: {
                    label: '',
                    shortLabel: '',
                    id: ''
                }
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 67289,
                    name: 'Wild Apps',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [],
                    description: null
                }
            ],
            rating: 5,
            numRatings: 4,
            badges: []
        },
        {
            id: 10030,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_64/img8466608210627164533.png?e0df123fedcc5e96c83a248797c76c6a',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_64/img8466608210627164533.png?e0df123fedcc5e96c83a248797c76c6a',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_64/img8466608210627164533-2x.png?7708f1f649c53812026dcea2d7c1bbf2',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_64/img8466608210627164533.png?e0df123fedcc5e96c83a248797c76c6a'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_112/img3745051225877065498.png?e0df123fedcc5e96c83a248797c76c6a',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/overview/img5121553238902495902.png?f752784a1cf710d1d845ab7937ba0962',
            productRibbon: {
                text: 'Core App',
                textColor: '#FFFFFF',
                ribbonColor: '#D42020'
            },
            merchandisingBadges: [
                {
                    uuid: '57af2145-5543-4fc1-b947-15e08d4575ed',
                    name: 'Single Sign-On',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/57af2145-5543-4fc1-b947-15e08d4575ed/img8223107847585143017.png?289aa01be11853c6ef0f8735471e2ba2'
                },
                {
                    uuid: '08d06719-1027-4952-914c-fe0d3de89d06',
                    name: 'Unified Invoice',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/08d06719-1027-4952-914c-fe0d3de89d06/img5117470399285071356.png?8cf2543bdae1af7a6f287688d07c1d45'
                },
                {
                    uuid: 'f57bd864-c1e3-4eff-af8d-592ee0a5165c',
                    name: 'Access Management',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/f57bd864-c1e3-4eff-af8d-592ee0a5165c/img765251955674935544.png?ab226874a0f6e57f753b26e96591efe3'
                }
            ],
            uuid: 'a14a89d1-e62e-4138-b579-2c2a6f79fce7',
            description: 'AppInsights dashboards automatically collect the latest data about your company into one place.',
            overview: 'AppInsights help you make faster and more informed\nbusiness decisions by letting you view your KPIs on a real-time dashboard, all in\none place.',
            vendorName: 'AppInsights',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'AppInsights',
            blurb: 'Real-time business dashboard solution',
            url: '/en-US/apps/10030/appinsights',
            startingPrice: {
                formattedFullPrice: 'Free',
                formattedPrice: 'Free',
                price: 'Free',
                priceCaption: '',
                frequency: {
                    label: '',
                    shortLabel: '',
                    id: ''
                }
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 67285,
                    name: 'Cool Apps',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 45285,
                            name: 'Freezing Apps',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 5,
            numRatings: 1,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 142378,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/142378/thumbs_64/img61982019133247767.png?da2eec94dd3a6a553bae47de42161366',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/142378/thumbs_64/img61982019133247767.png?da2eec94dd3a6a553bae47de42161366',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/142378/thumbs_64/img61982019133247767-2x.png?d4c81ddd7000c0c047cf231336a8369f',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/142378/thumbs_64/img61982019133247767.png?da2eec94dd3a6a553bae47de42161366'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/142378/thumbs_112/img4806238366532364489.png?da2eec94dd3a6a553bae47de42161366',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/142378/overview/img955268652153096299.png?56d002f35b195df857554d1d0e9c1887',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '779adc9e-2ff1-4193-b56e-eb044f260e11',
            description: 'Launch a website, blog or online store with BaseKit. Start with a beautiful template and make changes to text, colours, images, fonts and more, whenever it suits you. The BaseKit sitebuilder is completely mobile, meaning you can update your site from your phone, tablet or computer.',
            overview: 'BaseKit is a website building tool aimed at small businesses. All BaseKit websites, blogs and online stores can be created and updated from a mobile phone, tablet or desktop computer.',
            vendorName: 'BaseKit',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'BaseKit sitebuilder',
            blurb: 'The mobile sitebuilder for small businesses',
            url: '/en-US/apps/142378/basekit-sitebuilder',
            startingPrice: {
                formattedFullPrice: 'Free',
                formattedPrice: 'Free',
                price: 'Free',
                priceCaption: '',
                frequency: {
                    label: '',
                    shortLabel: '',
                    id: ''
                }
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 67289,
                    name: 'Wild Apps',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [],
                    description: null
                }
            ],
            rating: 5,
            numRatings: 1,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 132264,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/thumbs_64/img1819752633064435461.png?571b7d3cc01faad2f256d7a949720d77',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/thumbs_64/img1819752633064435461.png?571b7d3cc01faad2f256d7a949720d77',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/thumbs_64/img1819752633064435461-2x.png?d9710879e13184fda08d80b67210174c',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/thumbs_64/img1819752633064435461.png?571b7d3cc01faad2f256d7a949720d77'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/thumbs_112/img4134068526255962732.png?571b7d3cc01faad2f256d7a949720d77',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/overview/img1164992687341006935.png?5b3a4de2d0819cb9f8a0e64124a6c7cb',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: 'a374a9b2-d9df-438c-98f6-3de024262b8e',
            description: 'ServiceM8 is a cloud-based solution that helps field service organizations manage their business. ServiceM8 enables SMB plumbers, electricians and similar businesses to provide great customer experiences, increase productivity, improve cash flow and win more business.\n\nServiceM8 is an end to end solution that helps from the client’s first call, to quoting, invoicing and taking payment. Providing real-time information on job progress, where and when employees are at client sites and direct integration with major accounting packages.',
            overview: 'ServiceM8 is a cloud\u00adbased solution that helps field service organizations manage their business. ServiceM8 guides users from creating job quotes to invoicing and payment collection, and provides real\u00adtime information on what jobs are active, what quotes have been issued to clients, and where and when employees are at client sites.',
            vendorName: 'ServiceM8',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'ServiceM8',
            blurb: 'Field Service for Small Business',
            url: '/en-US/apps/132264/servicem8',
            startingPrice: {
                formattedFullPrice: '$25.00/Mo',
                formattedPrice: '$25.00',
                price: '25.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 67289,
                    name: 'Wild Apps',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [],
                    description: null
                }
            ],
            rating: 4,
            numRatings: 2,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 97079,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/thumbs_64/img7650639399717409102.png?4bcfffdceb0e142540092fdc054e02e0',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/thumbs_64/img7650639399717409102.png?4bcfffdceb0e142540092fdc054e02e0',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/thumbs_64/img7650639399717409102-2x.png?4bcfffdceb0e142540092fdc054e02e0',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/thumbs_64/img7650639399717409102.png?4bcfffdceb0e142540092fdc054e02e0'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/thumbs_112/img7753574590922565364.png?380ac3995aecd95dc7e1334d55aa153d',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/overview/img8875042054204981544.png?354410d3f134c5e06c3b7e9bc77900c8',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: 'ed7435bc-3586-4198-ae11-ee04706274f6',
            description: 'GoDaddy domain description',
            overview: 'Every great website is built on a great domain. Having your own domain shows the world that you take your business seriously, makes it easy for customers to find you and builds your brand online. Get your domain today with GoDaddy.',
            vendorName: 'GoDaddy',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: false,
            discounted: false,
            pending: false,
            name: 'GoDaddy Domain Registration',
            blurb: 'GoDaddy Domain Registration',
            url: '/en-US/apps/97079/godaddy-domain-registration',
            startingPrice: {
                formattedFullPrice: '$1.98/yr',
                formattedPrice: '$1.98',
                price: '1.98',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Year',
                    shortLabel: 'yr',
                    id: 'YEARLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [],
            rating: null,
            numRatings: 0,
            badges: []
        },
        {
            id: 138165,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/138165/thumbs_64/img2369481905243208254.png?96272fa021c408d65600ae59e5ec3c6b',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/138165/thumbs_64/img2369481905243208254.png?96272fa021c408d65600ae59e5ec3c6b',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/138165/thumbs_64/img2369481905243208254-2x.png?4b2aa93303af6e4235838d516d398872',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/138165/thumbs_64/img2369481905243208254.png?96272fa021c408d65600ae59e5ec3c6b'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/138165/thumbs_112/img273212679483395195.png?96272fa021c408d65600ae59e5ec3c6b',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/138165/overview/img3304889738969130919.png?426e104b2475c7337c5cbedd3cdfcaa0',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '9b799d4e-dcf4-41da-b14d-5fcd27d0743e',
            description: 'Simplify your training workflow. You can create, repurpose, schedule, deliver and report on training assets from Velpic\'s innovative eLearning platform, anytime anywhere. Increase learning engagement and retention while raising productivity, substantially reducing costs and shrinking your ecological footprint.',
            overview: 'Simplify your training workflow. You can create, repurpose, schedule, deliver and report on video training assets from our innovative eLearning platform, anytime anywhere. Increase learning engagement and retention while raising productivity, substantially reducing costs and shrinking your ecological footprint.',
            vendorName: 'Velpic Limited',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'Velpic LMS | Video Authoring | Streaming Portal',
            blurb: 'One online training platform to do it all',
            url: '/en-US/apps/138165/velpic-lms-video-authoring-streaming-portal',
            startingPrice: {
                formattedFullPrice: '$230.00/Mo',
                formattedPrice: '$230.00',
                price: '230.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 67285,
                    name: 'Cool Apps',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 45281,
                            name: 'Chilly Apps',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: null,
            numRatings: null,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 190503,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/190503/thumbs_64/img5952879707386504480.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/190503/thumbs_64/img5952879707386504480.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/190503/thumbs_64/img5952879707386504480-2x.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/190503/thumbs_64/img5952879707386504480.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/190503/thumbs_112/img1841316472454197765.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/190503/overview/img7789327393588503465.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '5a590b65-0790-43f6-81cc-bce13562d03c',
            description: 'All AppDirect platform documentation',
            overview: 'All AppDirect platform documentation for all AppDirect products, including AppMarket, AppReseller, AppDevices, AppBilling, AppDistribution, AppIdentity, AppInsights, and AppWise.',
            vendorName: 'AppDirect',
            showRating: true,
            hidePricings: false,
            free: true,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'Documentation Center',
            blurb: 'AppDirect platform documentation',
            url: '/en-US/apps/190503/documentation-center',
            startingPrice: {
                formattedFullPrice: '',
                formattedPrice: '',
                price: '',
                priceCaption: '',
                frequency: {
                    label: '',
                    shortLabel: '',
                    id: ''
                }
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 67285,
                    name: 'Cool Apps',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 45281,
                            name: 'Chilly Apps',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: null,
            numRatings: null,
            badges: []
        },
        {
            id: 293048,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293048/thumbs_64/image5315487358905934939.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293048/thumbs_64/image5315487358905934939.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293048/thumbs_64/image5315487358905934939-2x.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293048/thumbs_64/image5315487358905934939.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293048/thumbs_112/image668294021629676051.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293048/overview/image3372691994973109686.png',
            productRibbon: {
                text: 'Core App',
                textColor: '#FFFFFF',
                ribbonColor: '#D42020'
            },
            merchandisingBadges: [
                {
                    uuid: 'f57bd864-c1e3-4eff-af8d-592ee0a5165c',
                    name: 'Access Management',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/f57bd864-c1e3-4eff-af8d-592ee0a5165c/img765251955674935544.png?ab226874a0f6e57f753b26e96591efe3'
                },
                {
                    uuid: '57af2145-5543-4fc1-b947-15e08d4575ed',
                    name: 'Single Sign-On',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/57af2145-5543-4fc1-b947-15e08d4575ed/img8223107847585143017.png?289aa01be11853c6ef0f8735471e2ba2'
                },
                {
                    uuid: '08d06719-1027-4952-914c-fe0d3de89d06',
                    name: 'Unified Invoice',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/08d06719-1027-4952-914c-fe0d3de89d06/img5117470399285071356.png?8cf2543bdae1af7a6f287688d07c1d45'
                }
            ],
            uuid: '99ea13a6-679d-419f-9d87-d6c25b9026de',
            description: 'ServiceM8 is a cloud-based solution that helps field service organizations manage their business. ServiceM8 enables SMB plumbers, electricians and similar businesses to provide great customer experiences, increase productivity, improve cash flow and win more business.\n\nServiceM8 is an end to end solution that helps from the client’s first call, to quoting, invoicing and taking payment. Providing real-time information on job progress, where and when employees are at client sites and direct integration with Quickbooks Online.',
            overview: 'ServiceM8 is a cloud\u00adbased solution that helps field service organizations manage their business. ServiceM8 guides users from creating job quotes to invoicing and payment collection, and provides real\u00adtime information on what jobs are active, what quotes have been issued to clients, and where and when employees are at client sites.',
            vendorName: 'ServiceM8',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: false,
            discounted: false,
            pending: false,
            name: 'Buy Now - Service M8 Clone',
            blurb: 'Field Service for Small Business',
            url: '/en-US/apps/293048/buy-now---service-m8-clone',
            startingPrice: {
                formattedFullPrice: '$79.00/Mo',
                formattedPrice: '$79.00',
                price: '79.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 67285,
                    name: 'Cool Apps',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [],
                    description: null
                }
            ],
            rating: null,
            numRatings: null,
            badges: []
        },
        {
            id: 293052,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293052/thumbs_64/image7032528054555174273.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293052/thumbs_64/image7032528054555174273.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293052/thumbs_64/image7032528054555174273-2x.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293052/thumbs_64/image7032528054555174273.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293052/thumbs_112/image8500941481135593911.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293052/overview/image6695220627385330091.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: 'efbcd6b3-e527-44cf-b478-3ae66d069030',
            description: 'ServiceM8 is a cloud-based solution that helps field service organizations manage their business. ServiceM8 enables SMB plumbers, electricians and similar businesses to provide great customer experiences, increase productivity, improve cash flow and win more business.\n\nServiceM8 is an end to end solution that helps from the client’s first call, to quoting, invoicing and taking payment. Providing real-time information on job progress, where and when employees are at client sites and direct integration with major accounting packages.',
            overview: 'ServiceM8 is a cloud\u00adbased solution that helps field service organizations manage their business. ServiceM8 guides users from creating job quotes to invoicing and payment collection, and provides real\u00adtime information on what jobs are active, what quotes have been issued to clients, and where and when employees are at client sites.',
            vendorName: 'ServiceM8',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: false,
            discounted: false,
            pending: false,
            name: 'Buy Edition  - Service M8 Clone',
            blurb: 'Field Service for Small Business',
            url: '/en-US/apps/293052/buy-edition---service-m8-clone',
            startingPrice: {
                formattedFullPrice: '$25.00/Mo',
                formattedPrice: '$25.00',
                price: '25.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 67289,
                    name: 'Wild Apps',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [],
                    description: null
                }
            ],
            rating: null,
            numRatings: null,
            badges: []
        },
        {
            id: 323839,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/323839/thumbs_64/img1091250565832542134.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/323839/thumbs_64/img1091250565832542134.png'
            },
            profileLogoUrl: null,
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/323839/overview/img6369519769140680804.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '90d6eb6e-e3e1-44f4-b193-6fe80346895b',
            description: null,
            overview: 'This is a bundle description',
            vendorName: 'sfbmarketplace',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: false,
            discounted: false,
            pending: false,
            name: 'New Bundle',
            blurb: 'welcome to the jungle',
            url: '/en-US/bundles/323839/new-bundle',
            startingPrice: {
                formattedFullPrice: '',
                formattedPrice: '',
                price: '',
                priceCaption: '',
                frequency: {
                    label: '',
                    shortLabel: '',
                    id: ''
                }
            },
            tags: [],
            rating: null,
            numRatings: null,
            badges: []
        },
        {
            id: 329614,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329614/thumbs_64/image185549883960186896.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329614/thumbs_64/image185549883960186896.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329614/thumbs_64/image185549883960186896-2x.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329614/thumbs_64/image185549883960186896.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329614/thumbs_112/image3302534366432164044.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329614/overview/image597961320931219815.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '5a114f83-502c-4443-9e43-afd060ce3977',
            description: 'Homebase makes managing hourly work easier for local businesses. With employee time tracking, scheduling, team communication, and hiring, managers and employees can spend less time on paperwork and more time on growing their business.',
            overview: 'Use Homebase for free employee scheduling, free timesheets, free time clock, and free labor cost reporting. Perfect for businesses of all sizes.',
            vendorName: 'Homebase',
            showRating: true,
            hidePricings: false,
            free: true,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'Homebase',
            blurb: 'Free Time Clock & Scheduling',
            url: '/en-US/apps/329614/homebase',
            startingPrice: {
                formattedFullPrice: 'Free',
                formattedPrice: 'Free',
                price: 'Free',
                priceCaption: '',
                frequency: {
                    label: '',
                    shortLabel: '',
                    id: ''
                }
            },
            tags: [],
            rating: null,
            numRatings: null,
            badges: []
        },
        {
            id: 330408,
            iconUrl: null,
            iconSrcset: {},
            profileLogoUrl: null,
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/330408/overview/img4619154643990594698.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '98fd9d5d-7e6d-4d5d-ae98-5e8298c131d6',
            description: 'You now have access to AT&T Productivity Suite where you can finish the sign-up process for your Webex Meetings with AT&T trial.',
            overview: null,
            vendorName: 'sfbmarketplace',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: false,
            discounted: false,
            pending: false,
            name: 'Webex Meetings with AT&T - Trial',
            blurb: '',
            url: '/en-US/apps/330408/webex-meetings-with-att---trial',
            startingPrice: {
                formattedFullPrice: '$10.00/Mo',
                formattedPrice: '$10.00',
                price: '10.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [],
            rating: null,
            numRatings: null,
            badges: []
        }
    ]
};
