// @appdirect/sfb-theme-components/src/atoms/menu-dropdown/stories/menuDropdown.stories.js
import React from 'react';
import MenuDropdown from '../menuDropdown';

const items = [
    {
        isDisabled: false,
        isVisible: true,
        link: {
            value: 'http://www.google.ca/',
            isLocalized: true,
            type: 'INPUT'
        },
        openInNewTab: false,
        title: {
            value: 'Untitled 1',
            isLocalized: true,
            type: 'INPUT'
        },
        type: 'CUSTOM_LINK',
        uuid: '9be402d7-9aca-44e1-a5eb-b1c15327f64d'
    },
    {
        isDisabled: false,
        isVisible: true,
        link: {
            value: 'http://www.google.ca/',
            isLocalized: true,
            type: 'INPUT'
        },
        openInNewTab: false,
        title: {
            value: 'Untitled 2',
            isLocalized: true,
            type: 'INPUT'
        },
        type: 'CUSTOM_LINK',
        uuid: '9be402d7-9aca-44e1-a5eb-b1c15327f68d'
    }
];

export default {
    title: 'Atoms/Menu Dropdown',
    component: MenuDropdown
};

export const Default = {
    render: () => <MenuDropdown items={items} />,
    name: 'Default'
};
