// @appdirect/sfb-theme-components/src/atoms/status-message/stories/StatusMessage.stories.js
import React from 'react';
import StatusMessage from '../StatusMessage';
import { allowedTypes } from '../../icon/Icon';

export default {
    title: 'Atoms/Status Message',
    component: StatusMessage,
    render: args => <StatusMessage {...args} />,
    argTypes: {
        title: {
            control: 'text',
            description: 'The title of the status message'
        },
        details: {
            control: 'text',
            description: 'The details of the status message'
        },
        icon: {
            control: 'select',
            options: allowedTypes,
            description: 'The icon of the status message'
        },
        scrollIntoView: {
            control: 'boolean',
            description: 'Whether to scroll into view'
        }
    },
    parameters: {
        controls: {
            include: ['title', 'details', 'icon', 'scrollIntoView']
        }
    }
};

export const Default = {
    args: {
        title: 'Your search did not return results',
        details: 'Please try to search for other keywords',
        icon: 'search',
        scrollIntoView: true
    }
};
