// @appdirect/sfb-theme-components/mocks/listingProducts/pagination.js
const i18nMock = {
    previous: 'Previous Page',
    next: 'Next Page',
    title: 'Pagination',
    page: 'Page',
    goToPage: 'Go to page',
    currentPage: 'Current page'
};

export const paginationMock = {
    nbPages: 79,
    currentPage: 1,
    simplePagesList: '1 of 79',
    pagesList: [
        {
            type: 'previous',
            status: 'disabled'
        },
        {
            type: 'page',
            nb: 1,
            status: 'selected',
            url: '/en-US/listing?page=1'
        },
        {
            type: 'page',
            nb: 2,
            status: 'enabled',
            url: '/en-US/listing?page=2'
        },
        {
            type: 'page',
            nb: 3,
            status: 'enabled',
            url: '/en-US/listing?page=3'
        },
        {
            type: 'page',
            nb: 4,
            status: 'enabled',
            url: '/en-US/listing?page=4'
        },
        {
            type: 'page',
            nb: 5,
            status: 'enabled',
            url: '/en-US/listing?page=5'
        },
        {
            type: 'ellipsis'
        },
        {
            type: 'page',
            nb: 79,
            url: '/en-US/listing?page=79'
        },
        {
            type: 'next',
            status: 'enabled',
            url: '/en-US/listing?page=2'
        }
    ],
    itemsPerPage: 12,
    totalCount: 943,
    start: 0,
    i18n: i18nMock
};

export const paginationShortMock = {
    nbPages: 3,
    currentPage: 1,
    simplePagesList: '1 of 3',
    pagesList: [
        {
            type: 'previous',
            status: 'disabled'
        },
        {
            type: 'page',
            nb: 1,
            status: 'selected',
            url: '/en-US/listing?page=1'
        },
        {
            type: 'page',
            nb: 2,
            status: 'enabled',
            url: '/en-US/listing?page=2'
        },
        {
            type: 'page',
            nb: 3,
            status: 'enabled',
            url: '/en-US/listing?page=3'
        },
        {
            type: 'next',
            status: 'enabled',
            url: '/en-US/listing?page=2'
        }
    ],
    itemsPerPage: 12,
    totalCount: 25,
    start: 0,
    i18n: i18nMock
};
