// @appdirect/sfb-theme-components/src/components/product-associated-domain/stories/ProductAssociatedDomain.stories.js
import React from 'react';
import ProductAssociatedDomain, { ProductAssociatedDomainComponent } from '../ProductAssociatedDomain';
import { productDomainAppMock } from '../../../../mocks/product/productDomainApp';
import { fetchSuggestionsMock,
    fetchEmptySuggestionsMock,
    failedFetchSuggestionsMock } from '../../../../mocks/product/domainSuggestionFetch';

export default {
    title: 'Components/Product Associated Domain',
    component: ProductAssociatedDomain,

    parameters: {
        layout: 'fullscreen'
    }
};

export const Default = {
    render: () => (
        <div
            style={{
                width: '100%'
            }}
        >
            <ProductAssociatedDomainComponent data={productDomainAppMock} fetchSuggestions={fetchSuggestionsMock} />
        </div>
    ),

    name: 'Default'
};

export const EmptySuggestion = {
    render: () => (
        <div
            style={{
                width: '100%'
            }}
        >
            <ProductAssociatedDomainComponent
                data={productDomainAppMock}
                fetchSuggestions={fetchEmptySuggestionsMock}
            />
        </div>
    ),

    name: 'Empty Suggestion'
};

export const FailingFetch = {
    render: () => (
        <div
            style={{
                width: '100%'
            }}
        >
            <ProductAssociatedDomainComponent
                data={productDomainAppMock}
                fetchSuggestions={failedFetchSuggestionsMock}
            />
        </div>
    ),

    name: 'Failing fetch'
};
