// @appdirect/sfb-theme-components/src/atoms/product-card/stories/ProductCard.stories.js
import React from 'react';
import ProductCard from '../ProductCard';
import { relatedContentMock } from '../../../../mocks/product/related-content';
import { XSMALL, LARGE } from '../../../constants/sizes';
import { productCardMock, productCardMockWithoutLogo } from '../../../../mocks/product/productCard';

export default {
    title: 'Atoms/Product Card',
    component: ProductCard
};

export const Default = {
    render: () => <ProductCard cardData={productCardMock} reviewData={relatedContentMock} />,
    name: 'Default'
};

export const ImageSizes = {
    render: () => (
        <div
            style={{
                margin: '50px 0'
            }}
        >
            <ProductCard
                cardData={{
                    ...productCardMock,
                    size: LARGE.value
                }}
                settings={{
                    productNameSize: XSMALL.value,
                    descTextSize: XSMALL.value
                }}
            />
        </div>
    ),

    name: 'Image sizes'
};

export const WithoutVendor = {
    render: () => (
        <ProductCard
            cardData={productCardMock}
            reviewData={{
                details: {
                    ...relatedContentMock.details,
                    vendor: {}
                }
            }}
        />
    ),

    name: 'Without vendor'
};

export const WithRatings = {
    render: () => (
        <ProductCard
            cardData={productCardMock}
            reviewData={relatedContentMock}
            settings={{
                showRatings: true
            }}
        />
    ),

    name: 'With ratings'
};

export const WithPlaceholderImage = {
    render: () => <ProductCard cardData={productCardMockWithoutLogo} reviewData={relatedContentMock} />,
    name: 'With placeholder image'
};
