// @appdirect/sfb-theme-components/src/components/product-configurator/stories/ProductConfigurator.stories.js
import React from 'react';
import ProductConfigurator from '../ProductConfigurator';
import { productConfiguratorMock,
    productConfiguratorNoAddToCartCartMock,
    productConfiguratorBackFromCartMock,
    productConfiguratorNoSelectedEditionsMock,
    productConfiguratorWithoutAddons,
    productConfiguratorUpgradeMock } from '../../../../mocks/product/productConfigurator';

export default {
    title: 'Components/Product Configurator',
    component: ProductConfigurator
};

export const Default = {
    render: () => <ProductConfigurator data={productConfiguratorMock} />,
    name: 'Default'
};

export const WithoutAddToCart = {
    render: () => <ProductConfigurator data={productConfiguratorNoAddToCartCartMock} />,
    name: 'Without add to cart'
};

export const BackFromCart = {
    render: () => <ProductConfigurator data={productConfiguratorBackFromCartMock} />,
    name: 'Back from cart'
};

export const WithoutSelectedEdition = {
    render: () => <ProductConfigurator data={productConfiguratorNoSelectedEditionsMock} />,
    name: 'Without selected edition'
};

export const WithoutAddons = {
    render: () => <ProductConfigurator data={productConfiguratorWithoutAddons} />,
    name: 'Without addons'
};

export const UpgradeButton = {
    render: () => <ProductConfigurator data={productConfiguratorUpgradeMock} />,
    name: 'Upgrade Button'
};
