// @appdirect/sfb-theme-components/mocks/product/callToActions.js
export const callToActionsMock = {
    callToActions: {
        items: [
            {
                label: 'First Button',
                tooltip: 'Tooltip',
                isPrimary: true,
                enabled: true,
                url: 'google.ca',
                actionType: 'BUY',
                a11yLabel: 'Click here to go a the first page'
            },
            {
                label: 'Second Button',
                tooltip: '',
                isPrimary: false,
                enabled: false,
                url: '',
                actionType: '',
                a11yLabel: 'Click here to go a the second page'
            },
            {
                label: 'Upgrade',
                tooltip: '',
                isPrimary: false,
                enabled: false,
                url: '',
                actionType: 'CAN_MOSI_UPGRADE',
                a11yLabel: 'Upgrade the AppInsights product'
            }
        ]
    }
};

const updateCart = {
    actionType: 'UPDATE_CART_ITEM',
    enabled: true,
    popup: {},
    isPrimary: false,
    label: 'Save Changes',
    tooltip: '',
    url: 'undefined/api/internal/storefront/v1/cta?productId=241515&type=UPDATE_CART_ITEM&editionId=452529&cartItemId=905697de-44e1-43bd-a2ae-befcad6de7eb',
    cartItemId: '905697de-44e1-43bd-a2ae-befcad6de7eb',
    urlCreator: {
        base: 'undefined/api/internal/storefront/v1/cta?productId=241515&type=UPDATE_CART_ITEM&cartItemId=905697de-44e1-43bd-a2ae-befcad6de7eb',
        actionType: 'UPDATE_CART_ITEM',
        editionParam: 'eppid',
        addonParam: 'appid'
    }
};

const buyNow = {
    actionType: 'BUY_NOW',
    enabled: true,
    popup: {},
    isPrimary: true,
    label: 'Buy Now',
    tooltip: '',
    url: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=1&type=BUY_NOW',
    urlCreator: {
        base: 'http: //marketplace.com/api/internal/storefront/v1/cta',
        actionType: 'BUY_NOW',
        editionParam: 'editionId',
        addonParam: 'addonId'
    }
};

export const addToCart = {
    label: 'Add to Cart',
    tooltip: '',
    isPrimary: false,
    enabled: true,
    url: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=273407&type=UPDATE_CART_ITEM&editionId=493724&cartItemId=4cf04569-3cc7-44ee-b823-b34f4a840b6c',
    actionType: 'ADD_TO_CART',
    popup: {},
    cartItemId: '4cf04569-3cc7-44ee-b823-b34f4a840b6c',
    urlCreator: {
        base: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=273407&type=UPDATE_CART_ITEM&cartItemId=4cf04569-3cc7-44ee-b823-b34f4a840b6c',
        actionType: 'ADD_TO_CART',
        editionParam: 'eppid',
        addonParam: 'appid'
    }
};

const upgrade = {
    label: 'Upgrade',
    tooltip: '',
    isPrimary: false,
    enabled: true,
    url: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=273407&type=UPDATE_CART_ITEM&editionId=493724&cartItemId=4cf04569-3cc7-44ee-b823-b34f4a840b6c',
    actionType: 'CAN_MOSI_UPGRADE',
    popup: {
        title: 'Upgrading to Office 365 Business',
        text: 'By upgrading from this product:',
        items: [
            {
                id: '-1',
                fromApplicationName: 'Office 365 Small Business',
                fromEditionName: 'First Edition',
                url: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=273407&type=UPDATE_CART_ITEM&cartItemId=4cf04569-3cc7-44ee-b823-b34f4a840b6c'
            }
        ],
        i18n: {
            application: 'Application',
            edition: 'Edition',
            actions: 'Action',
            select: 'Select'
        }
    },
    cartItemId: '4cf04569-3cc7-44ee-b823-b34f4a840b6c',
    urlCreator: {
        base: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=273407&type=UPDATE_CART_ITEM&cartItemId=4cf04569-3cc7-44ee-b823-b34f4a840b6c',
        actionType: 'CAN_MOSI_UPGRADE',
        editionParam: 'eppid',
        addonParam: 'appid'
    }
};

const lead = {
    label: 'Contact Us',
    tooltip: '',
    isPrimary: false,
    enabled: true,
    leadPageUrl: '/en_us/add-lead',
    url: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=273407&type=UPDATE_CART_ITEM&editionId=493724&cartItemId=4cf04569-3cc7-44ee-b823-b34f4a840b6c',
    actionType: 'LEAD_GEN',
    popup: {},
    cartItemId: '4cf04569-3cc7-44ee-b823-b34f4a840b6c',
    urlCreator: {
        base: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=273407&type=UPDATE_CART_ITEM&cartItemId=4cf04569-3cc7-44ee-b823-b34f4a840b6c',
        actionType: 'ADD_TO_CART',
        editionParam: 'eppid',
        addonParam: 'appid'
    }
};

const buySkipConfigurator = {
    label: 'Buy Now',
    tooltip: '',
    isPrimary: true,
    enabled: true,
    url: 'https://sfbmarketplace.byappdirect.com/api/internal/storefront/v1/cta?productId=357225&type=BUY',
    actionType: 'BUY',
    popup: {}
};

export const addToCartSkipConfigurator = {
    label: 'Add to Cart',
    tooltip: '',
    isPrimary: false,
    enabled: true,
    url: 'https://sfbmarketplace.byappdirect.com/api/internal/storefront/v1/cta?productId=357225&type=ADD_TO_CART',
    actionType: 'ADD_TO_CART',
    popup: {}
};

export const editionAddToCartSkipConfigurator1 = {
    label: 'Add to Cart',
    tooltip: '',
    isPrimary: true,
    eppid: 'ad44b0f4-a661-4b54-b755-e267bdaee544',
    enabled: true,
    url: 'https://sfbmarketplace.byappdirect.com/api/internal/storefront/v1/cta?productId=357231&type=ADD_TO_CART&editionId=709503&eppid=ad44b0f4-a661-4b54-b755-e267bdaee544',
    actionType: 'ADD_TO_CART',
    popup: {},
    isSecondary: false
};

export const editionAddToCartSkipConfigurator2 = {
    label: 'Add to Cart',
    tooltip: '',
    isPrimary: false,
    eppid: '7eeadbb1-7672-4a99-8d3e-293efdacc912',
    enabled: true,
    url: 'https://sfbmarketplace.byappdirect.com/api/internal/storefront/v1/cta?productId=357231&type=ADD_TO_CART&editionId=709509&eppid=7eeadbb1-7672-4a99-8d3e-293efdacc912',
    actionType: 'ADD_TO_CART',
    popup: {},
    isSecondary: true
};

export const editionSelect = {
    label: 'Select',
    tooltip: '',
    isPrimary: true,
    eppid: '76068349-1098-4904-a91e-a038f8b3a9a0',
    enabled: true,
    url: 'https://engage19billing.test.devappdirect.me/api/internal/storefront/v1/cta?productId=300570&type=CONFIGURE&editionId=555483&eppid=76068349-1098-4904-a91e-a038f8b3a9a0',
    actionType: 'CONFIGURE',
    popup: {},
    isSecondary: false
};

export const productBannerSkipConfiguratorCallToActionsMock = {
    items: [
        buySkipConfigurator,
        addToCartSkipConfigurator
    ]
};

export const configuratorUpdateCartCallToActionsMock = {
    items: [
        updateCart
    ]

};

export const configuratorDefaultCallToActionsMock = {
    items: [
        addToCart,
        buyNow
    ]
};

export const configuratorNoAddToCartCallToActionsMock = {
    items: [
        buyNow
    ]
};

export const configuratorUpgradeCallToActionsMock = {
    items: [
        upgrade
    ]
};

export const callToActionsMultiButtonMock = {
    callToActions: {
        items: [
            {
                label: 'First Button',
                tooltip: 'Tooltip',
                isPrimary: true,
                enabled: true,
                url: 'google.ca',
                actionType: ''
            },
            {
                label: 'Second Button',
                tooltip: '',
                isPrimary: false,
                enabled: false,
                url: '',
                actionType: ''
            },
            lead,
            {
                actionType: 'NO_ACTION'
            }
        ]
    }
};

export const callToActionsCaptionMock = {
    callToActions: {
        items: [
            {
                label: 'First Button',
                tooltip: 'Tooltip',
                isPrimary: true,
                enabled: true,
                url: 'google.ca',
                actionType: ''
            },
            {
                label: 'Second Button',
                tooltip: '',
                isPrimary: false,
                enabled: false,
                url: '',
                actionType: ''
            }
        ],
        caption: {
            text: 'or, <a href="google.com">Choose an edition to purchase</a>'
        }
    }
};

export const callToActionsNoActionMock = {
    callToActions: {
        items: [
            {
                actionType: 'NO_ACTION'
            }
        ]
    }
};

export const callToActionsReferable = {
    callToActions: {
        items: [
            {
                actionType: 'REFERABLE',
                label: 'Learn More'
            }
        ]
    }
};

export const callToActionsBuy = {
    callToActions: {
        items: [
            {
                actionType: 'BUY',
                enabled: true,
                popup: {},
                isPrimary: true,
                label: 'Buy Now',
                tooltip: '',
                url: 'http://marketplace.com/api/internal/storefront/v1/cta?productId=1&type=BUY',
                urlCreator: {
                    base: 'http://marketplace.com/api/internal/storefront/v1/cta',
                    actionType: 'BUY',
                    editionParam: 'editionId',
                    addonParam: 'addonId'
                }
            }
        ]
    }
};
