// @appdirect/sfb-theme-components/mocks/product/configuratorSelectedEdition.js
export const selectedEditionsMock = {
    configurator: {
        productId: 0,
        selectedEdition: {
            name: 'G Suite Business - No Contract',
            description: 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nulla ex consectetur dignissimos.',
            id: 404624,
            primary: false,
            plans: [{
                costs: {
                    unitPrice: [{
                        id: 'USER',
                        usageRange: [{
                            formattedPrice: '$7.80',
                            formattedFullPrice: '$7.80 / User / Month',
                            formattedFullPriceElements: [
                                '$7.80',
                                '/ User',
                                '/ Month'
                            ],
                            unit: {
                                min: 1,
                                max: null,
                                label: 'User',
                                dependency: '',
                                id: 'USER',
                                limitLabel: 'For 1 User And Above'
                            },
                            frequency: {
                                shortLabel: 'Mo',
                                label: 'Month',
                                id: 'MONTHLY'
                            },
                            price: {
                                symbol: '$',
                                currency: 'USD',
                                amount: 12,
                                value: 12,
                                originalValue: 12,
                                discountValue: 0,
                                editionStartingPrice: 12,
                                formatted: {
                                    originalPrice: '$7.80',
                                    originalFullPrice: '$7.80 / User / Month',
                                    originalFullPriceElements: [
                                        '$7.80',
                                        '/ User',
                                        '/ Month'
                                    ],
                                    meteredUsage: 'For 1 User And Above'
                                }
                            }
                        }]
                    }]
                },
                setupFees: [],
                contractFees: [],
                contractLength: null,
                formattedContract: false,
                discount: false,
                uuid: '4e6237b7-7a74-4ee5-b050-aaafed2e44e9'
            }],
            bullets: [{
                content: 'Unlimited cloud storage (or 1TB per user if fewer than 5 users)',
                tooltip: null
            }, {
                content: 'Smart search across G Suite with Cloud Search',
                tooltip: null
            }, {
                content: 'Archive and set retention policies for emails and chats',
                tooltip: null
            }, {
                content: 'eDiscovery for emails, chats, and files',
                tooltip: null
            }, {
                content: 'Audit reports to track user activity',
                tooltip: null
            }, {
                content: '*This is a monthly-billed plan, with no contract.',
                tooltip: null
            }],
            callToAction: {
                isPrimary: false,
                isSecondary: true
            },
            audience: [{
                content: 'Enhanced office suite with unlimited storage and archiving'
            }],
            items: [],
            invisible: false,
            leadGen: false
        },
        backAction: {
            label: 'Back to Office 365',
            url: '/en-US/apps/577/office/editions'
        },
        i18n: {
            selectedEdition: 'Selected Edition',
            selectedAddons: 'Selected Add-ons',
            selectedProduct: 'Selected products',
            startingAt: 'Starting at'
        }
    }
};
