// @appdirect/sfb-theme-components/mocks/product/domainApp.js
import { domainsMock } from '../application/domains';

const domainAppMock = {
    application: {
        domains: domainsMock,
        config: {
            id: 97079,
            uuid: 'ed7435bc-3586-4198-ae11-ee04706274f6',
            type: 'DOMAIN_RESELLER',
            isAssignedToUser: false,
            hidePricings: false,
            linkedImportableApplicationUuid: null,
            assignLeadsToDevelopers: false
        },
        summary: {
            title: 'GoDaddy Domain Registration',
            details: 'GoDaddy Domain Registration',
            overview: 'Every great website is built on a great domain. Having your own domain shows the world that you take your business seriously, makes it easy for customers to find you and builds your brand online. Get your domain today with GoDaddy.',
            logoSrc: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/thumbs_112/img7753574590922565364.png?380ac3995aecd95dc7e1334d55aa153d',
            logoSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/thumbs_112/img7753574590922565364.png?380ac3995aecd95dc7e1334d55aa153d',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/thumbs_112/img7753574590922565364-2x.png?4c8a7d350eaa178672e6d3613dfeae7a',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/97079/thumbs_112/img7753574590922565364.png?380ac3995aecd95dc7e1334d55aa153d'
            },
            productRibbon: null,
            merchandisingBadges: []
        },
        callToActions: {
            items: [
                {
                    label: 'Buy',
                    tooltip: '',
                    isPrimary: false,
                    enabled: true,
                    url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=97079&type=BUY&editionId=404624',
                    actionType: 'BUY',
                    popup: {}
                }
            ],
            caption: {
                text: '',
                tooltip: ''
            }
        },
        categories: {
            list: []
        },
        tags: [],
        startingPrice: {
            formattedPrice: '$1.98',
            price: '1.98',
            priceCaption: 'Starting At',
            frequency: {
                label: 'Year',
                shortLabel: 'yr',
                id: 'YEARLY'
            },
            symbol: '$',
            currency: 'USD'
        },
        editions: {
            bullets: [],
            items: [
                {
                    name: 'Recurring Edition',
                    id: 220419,
                    primary: true,
                    plans: [
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$16.58',
                                    formattedFullPrice: '$16.58 / Year',
                                    formattedFullPriceElements: [
                                        '$16.58',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 16.58,
                                        value: 16.58,
                                        originalValue: 16.58,
                                        discountValue: 0,
                                        editionStartingPrice: 16.58,
                                        formatted: {
                                            originalPrice: '$16.58',
                                            originalFullPrice: '$16.58 / Year',
                                            originalFullPriceElements: [
                                                '$16.58',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: 'ee12da47-4be3-40a2-97f2-2739c0d90791'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$17.98',
                                    formattedFullPrice: '$17.98 / Year',
                                    formattedFullPriceElements: [
                                        '$17.98',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 17.98,
                                        value: 17.98,
                                        originalValue: 17.98,
                                        discountValue: 0,
                                        editionStartingPrice: 17.98,
                                        formatted: {
                                            originalPrice: '$17.98',
                                            originalFullPrice: '$17.98 / Year',
                                            originalFullPriceElements: [
                                                '$17.98',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '3f07c3c4-afc1-4f2d-9eba-3416cea034a7'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$49.99',
                                    formattedFullPrice: '$49.99 / Year',
                                    formattedFullPriceElements: [
                                        '$49.99',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 49.99,
                                        value: 49.99,
                                        originalValue: 49.99,
                                        discountValue: 0,
                                        editionStartingPrice: 49.99,
                                        formatted: {
                                            originalPrice: '$49.99',
                                            originalFullPrice: '$49.99 / Year',
                                            originalFullPriceElements: [
                                                '$49.99',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '387dcda3-43c2-44dd-9c21-19ad26d0e2f4'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$1.98',
                                    formattedFullPrice: '$1.98 / Year',
                                    formattedFullPriceElements: [
                                        '$1.98',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 1.98,
                                        value: 1.98,
                                        originalValue: 1.98,
                                        discountValue: 0,
                                        editionStartingPrice: 1.98,
                                        formatted: {
                                            originalPrice: '$1.98',
                                            originalFullPrice: '$1.98 / Year',
                                            originalFullPriceElements: [
                                                '$1.98',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: 'ec6aee07-88a4-43a5-8a02-e63530bf2f93'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$34.99',
                                    formattedFullPrice: '$34.99 / Year',
                                    formattedFullPriceElements: [
                                        '$34.99',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 34.99,
                                        value: 34.99,
                                        originalValue: 34.99,
                                        discountValue: 0,
                                        editionStartingPrice: 34.99,
                                        formatted: {
                                            originalPrice: '$34.99',
                                            originalFullPrice: '$34.99 / Year',
                                            originalFullPriceElements: [
                                                '$34.99',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '6a64b617-a506-4f47-a38b-fd4c78e8fd07'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$20.94',
                                    formattedFullPrice: '$20.94 / Year',
                                    formattedFullPriceElements: [
                                        '$20.94',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 20.94,
                                        value: 20.94,
                                        originalValue: 20.94,
                                        discountValue: 0,
                                        editionStartingPrice: 20.94,
                                        formatted: {
                                            originalPrice: '$20.94',
                                            originalFullPrice: '$20.94 / Year',
                                            originalFullPriceElements: [
                                                '$20.94',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: 'ac94db8e-1b2e-439d-bece-80f636ae7abd'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$79.99',
                                    formattedFullPrice: '$79.99 / Year',
                                    formattedFullPriceElements: [
                                        '$79.99',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 79.99,
                                        value: 79.99,
                                        originalValue: 79.99,
                                        discountValue: 0,
                                        editionStartingPrice: 79.99,
                                        formatted: {
                                            originalPrice: '$79.99',
                                            originalFullPrice: '$79.99 / Year',
                                            originalFullPriceElements: [
                                                '$79.99',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: 'b67ce995-cff9-4309-bf7b-a97ce8a1ef18'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$39.98',
                                    formattedFullPrice: '$39.98 / Year',
                                    formattedFullPriceElements: [
                                        '$39.98',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 39.98,
                                        value: 39.98,
                                        originalValue: 39.98,
                                        discountValue: 0,
                                        editionStartingPrice: 39.98,
                                        formatted: {
                                            originalPrice: '$39.98',
                                            originalFullPrice: '$39.98 / Year',
                                            originalFullPriceElements: [
                                                '$39.98',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '4cf06c14-36c5-475e-9522-42504a8e89b2'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$16.99',
                                    formattedFullPrice: '$16.99 / Year',
                                    formattedFullPriceElements: [
                                        '$16.99',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 16.99,
                                        value: 16.99,
                                        originalValue: 16.99,
                                        discountValue: 0,
                                        editionStartingPrice: 16.99,
                                        formatted: {
                                            originalPrice: '$16.99',
                                            originalFullPrice: '$16.99 / Year',
                                            originalFullPriceElements: [
                                                '$16.99',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: 'b50dff13-fcf7-4f90-8d5c-de78e6070ce1'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$16.99',
                                    formattedFullPrice: '$16.99 / Year',
                                    formattedFullPriceElements: [
                                        '$16.99',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 16.99,
                                        value: 16.99,
                                        originalValue: 16.99,
                                        discountValue: 0,
                                        editionStartingPrice: 16.99,
                                        formatted: {
                                            originalPrice: '$16.99',
                                            originalFullPrice: '$16.99 / Year',
                                            originalFullPriceElements: [
                                                '$16.99',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '3e4212c3-95d8-4c92-b8ea-c366071001b2'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$168.98',
                                    formattedFullPrice: '$168.98 / Year',
                                    formattedFullPriceElements: [
                                        '$168.98',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 168.98,
                                        value: 168.98,
                                        originalValue: 168.98,
                                        discountValue: 0,
                                        editionStartingPrice: 168.98,
                                        formatted: {
                                            originalPrice: '$168.98',
                                            originalFullPrice: '$168.98 / Year',
                                            originalFullPriceElements: [
                                                '$168.98',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '0eade469-2121-4cb6-a6e5-fb9cad9ed3e1'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$41.56',
                                    formattedFullPrice: '$41.56 / Two Years',
                                    formattedFullPriceElements: [
                                        '$41.56',
                                        null,
                                        '/ Two Years'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'two yr',
                                        label: 'Two Years',
                                        id: 'TWO_YEARS'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 41.56,
                                        value: 41.56,
                                        originalValue: 41.56,
                                        discountValue: 0,
                                        editionStartingPrice: 41.56,
                                        formatted: {
                                            originalPrice: '$41.56',
                                            originalFullPrice: '$41.56 / Two Years',
                                            originalFullPriceElements: [
                                                '$41.56',
                                                null,
                                                '/ Two Years'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '2 year contract',
                            discount: false,
                            uuid: '597437c0-9ae5-4dfa-938e-b16beced9068'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$33.98',
                                    formattedFullPrice: '$33.98 / Year',
                                    formattedFullPriceElements: [
                                        '$33.98',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 33.98,
                                        value: 33.98,
                                        originalValue: 33.98,
                                        discountValue: 0,
                                        editionStartingPrice: 33.98,
                                        formatted: {
                                            originalPrice: '$33.98',
                                            originalFullPrice: '$33.98 / Year',
                                            originalFullPriceElements: [
                                                '$33.98',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '0fce3d60-2a5f-44da-80d6-fb6468a810fe'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$79.98',
                                    formattedFullPrice: '$79.98 / Year',
                                    formattedFullPriceElements: [
                                        '$79.98',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 79.98,
                                        value: 79.98,
                                        originalValue: 79.98,
                                        discountValue: 0,
                                        editionStartingPrice: 79.98,
                                        formatted: {
                                            originalPrice: '$79.98',
                                            originalFullPrice: '$79.98 / Year',
                                            originalFullPriceElements: [
                                                '$79.98',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '5f5dad37-c45a-471a-83b7-6503d59632da'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$1,898.00',
                                    formattedFullPrice: '$1,898.00 / Year',
                                    formattedFullPriceElements: [
                                        '$1,898.00',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 1898,
                                        value: 1898,
                                        originalValue: 1898,
                                        discountValue: 0,
                                        editionStartingPrice: 1898,
                                        formatted: {
                                            originalPrice: '$1,898.00',
                                            originalFullPrice: '$1,898.00 / Year',
                                            originalFullPriceElements: [
                                                '$1,898.00',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '9f718e0c-171c-4060-ab46-05a7d8f988ff'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$12.99',
                                    formattedFullPrice: '$12.99 / Year',
                                    formattedFullPriceElements: [
                                        '$12.99',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 12.99,
                                        value: 12.99,
                                        originalValue: 12.99,
                                        discountValue: 0,
                                        editionStartingPrice: 12.99,
                                        formatted: {
                                            originalPrice: '$12.99',
                                            originalFullPrice: '$12.99 / Year',
                                            originalFullPriceElements: [
                                                '$12.99',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: 'c367b965-ab0d-4053-b813-0a56f3f848e8'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$12.99',
                                    formattedFullPrice: '$12.99 / Year',
                                    formattedFullPriceElements: [
                                        '$12.99',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 12.99,
                                        value: 12.99,
                                        originalValue: 12.99,
                                        discountValue: 0,
                                        editionStartingPrice: 12.99,
                                        formatted: {
                                            originalPrice: '$12.99',
                                            originalFullPrice: '$12.99 / Year',
                                            originalFullPriceElements: [
                                                '$12.99',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '4e76636d-13fe-4f4a-9a64-82319f648663'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$39.98',
                                    formattedFullPrice: '$39.98 / Year',
                                    formattedFullPriceElements: [
                                        '$39.98',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 39.98,
                                        value: 39.98,
                                        originalValue: 39.98,
                                        discountValue: 0,
                                        editionStartingPrice: 39.98,
                                        formatted: {
                                            originalPrice: '$39.98',
                                            originalFullPrice: '$39.98 / Year',
                                            originalFullPriceElements: [
                                                '$39.98',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '2bc5da1d-8346-4ce9-9a75-258a53460ed6'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$38.99',
                                    formattedFullPrice: '$38.99 / Year',
                                    formattedFullPriceElements: [
                                        '$38.99',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 38.99,
                                        value: 38.99,
                                        originalValue: 38.99,
                                        discountValue: 0,
                                        editionStartingPrice: 38.99,
                                        formatted: {
                                            originalPrice: '$38.99',
                                            originalFullPrice: '$38.99 / Year',
                                            originalFullPriceElements: [
                                                '$38.99',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '490d83e7-f630-483f-887b-1b7188b8ce95'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$20.76',
                                    formattedFullPrice: '$20.76 / Two Years',
                                    formattedFullPriceElements: [
                                        '$20.76',
                                        null,
                                        '/ Two Years'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'two yr',
                                        label: 'Two Years',
                                        id: 'TWO_YEARS'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 20.76,
                                        value: 20.76,
                                        originalValue: 20.76,
                                        discountValue: 0,
                                        editionStartingPrice: 20.76,
                                        formatted: {
                                            originalPrice: '$20.76',
                                            originalFullPrice: '$20.76 / Two Years',
                                            originalFullPriceElements: [
                                                '$20.76',
                                                null,
                                                '/ Two Years'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '2 year contract',
                            discount: false,
                            uuid: 'a9c2b14b-5ccb-411e-b244-2ac4e6a35a23'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$14.00',
                                    formattedFullPrice: '$14.00 / Year',
                                    formattedFullPriceElements: [
                                        '$14.00',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 14,
                                        value: 14,
                                        originalValue: 14,
                                        discountValue: 0,
                                        editionStartingPrice: 14,
                                        formatted: {
                                            originalPrice: '$14.00',
                                            originalFullPrice: '$14.00 / Year',
                                            originalFullPriceElements: [
                                                '$14.00',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: 'a998a48e-83d1-45de-9a4f-d2de65ea1aaa'
                        },
                        {
                            costs: {
                                unitPrice: [],
                                flatPrice: {
                                    formattedPrice: '$10.38',
                                    formattedFullPrice: '$10.38 / Year',
                                    formattedFullPriceElements: [
                                        '$10.38',
                                        null,
                                        '/ Year'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        dependency: '',
                                        id: 'NOT_APPLICABLE'
                                    },
                                    frequency: {
                                        shortLabel: 'yr',
                                        label: 'Year',
                                        id: 'YEARLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 10.38,
                                        value: 10.38,
                                        originalValue: 10.38,
                                        discountValue: 0,
                                        editionStartingPrice: 10.38,
                                        formatted: {
                                            originalPrice: '$10.38',
                                            originalFullPrice: '$10.38 / Year',
                                            originalFullPriceElements: [
                                                '$10.38',
                                                null,
                                                '/ Year'
                                            ]
                                        }
                                    }
                                }
                            },
                            setupFees: [],
                            contractFees: [],
                            contractLength: 1,
                            formattedContract: '1 year contract',
                            discount: false,
                            uuid: '2c48408c-8b95-40d0-b02b-6a214c8f6672'
                        }
                    ],
                    bullets: [],
                    callToAction: {
                        isPrimary: true,
                        isSecondary: false
                    },
                    audience: [],
                    items: [
                        {
                            content: '1 User'
                        }
                    ],
                    invisible: false,
                    leadGen: false,
                    revenueType: 'RECURRING'
                }
            ],
            footnotes: [],
            i18n: {
                bulletsHeader: 'All Plans Include',
                inclusions: 'Included',
                audience: 'Ideal For',
                footnotes: 'Notes',
                invisible: 'Hidden from store customers',
                tierSeparator: 'or',
                setupFee: 'Setup fee',
                showMoreDetails: 'Show More Info',
                hideMoreDetails: 'Hide',
                mostPopular: 'Most Popular',
                availablePricing: 'Available Pricing:',
                select: 'Select',
                selected: 'Selected',
                hideDetails: 'Hide Details',
                viewDetails: 'View Details'
            }
        },
        actionMessages: {
            requestPurchaseSuccess: 'Your request to purchase GoDaddy Domain Registration was sent.',
            requestPurchaseFail: 'Your request to purchase GoDaddy Domain Registration was not sent, please try again.',
            requestAccessSuccess: 'You have successfully requested access to GoDaddy Domain Registration.',
            requestAccessFail: 'Your request to access GoDaddy Domain Registration was not sent, please try again.'
        },
        overview: {
            title: null,
            description: '',
            callToActions: {},
            image: null,
            imageSrcSet: {},
            showThirdPartyLegalDisclosure: false,
            documentationUrl: null,
            downloadDocumentationUrl: null,
            pricingUrl: '/en-US/apps/97079/godaddy-domain-registration/editions',
            customers: [],
            benefits: [],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation'
            }
        }
    }
};

export const domainAppMockCheckoutV2 = {
    ...domainAppMock,
    config: {
        canRouteToV2: true,
        channelSettings: {
            isAddToCartCTAEnabled: true
        }
    }
};

export const domainAppMockCheckoutV2DisabledCta = {
    application: {
        ...domainAppMock.application,
        callToActions: {
            items: [
                {
                    actionType: 'NO_ACTION'
                }
            ]
        }
    },
    config: {
        canRouteToV2: true
    }
};

export const domainAppMockCheckoutV2MultipleCtas = {
    application: {
        ...domainAppMock.application,
        callToActions: {
            items: [
                {
                    label: 'Buy',
                    tooltip: '',
                    isPrimary: false,
                    enabled: true,
                    url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=97079&type=BUY&editionId=404624',
                    actionType: 'BUY',
                    popup: {}
                },
                {
                    label: 'Manage',
                    tooltip: '',
                    isPrimary: false,
                    enabled: true,
                    url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=97079&type=MANAGE&editionId=404624',
                    actionType: 'MANAGE',
                    popup: {}
                }
            ]
        }
    },
    config: {
        canRouteToV2: true,
        channelSettings: {
            isAddToCartCTAEnabled: true
        }
    }
};

export const domainAppMockCheckoutV1 = {
    ...domainAppMock,
    config: {
        canRouteToV2: false,
        user: {
            isLoggedIn: true
        }
    }
};
