// @appdirect/sfb-theme-components/mocks/product/productFeatures.js
export const gSuiteMock = {
    items: [
        {
            id: 3373512,
            label: 'Overview',
            description: 'All you need to do your best work, together in one package that works seamlessly from your computer, phone or tablet.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img4481318113966540770.png',
            aboveCharacteristics: [

            ],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Make decisions faster, face to face.',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Use shared calendars to see when others are available and schedule meetings with automatic email invites.',
                            displayCheck: true
                        },
                        {
                            highlight: 'With one click, turn your meeting into a video conference from any camera-enabled computer, phone, or tablet.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Share your screen to review your work as a team, and make decisions on the spot.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Collaborate in real-time.',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Easily work on documents, spreadsheets, and slides across your devices, with or without internet.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Work in a single document with teammates or people outside your company. See edits as others type, communicate through built-in chat, and ask questions in comments.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Multiple people can work at the same time, and every change is saved automatically.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Store and share files in the cloud.',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Keep all your work in one place with secure access from your computer, phone, or tablet.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Quickly invite others to view, download, and collaborate on any file – no email attachment needed.',
                            displayCheck: true
                        },
                        {
                            highlight: 'File updates are automatically saved and stored in Drive, so everyone can always have access to the latest version.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Secure your data and devices.',
                    position: 3,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Protect your company’s data with security options like 2-step verification and single-sign-on, and use mobile management to keep your data safe in the case of a lost device or employee turnover.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Archive email messages and on-the-record chats, and control how long they are retained.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Easily configure security settings from a centralized administration console, and call or email Google support for help 24/7.',
                            displayCheck: true
                        }
                    ]
                }
            ]
        },
        {
            id: 3373488,
            label: 'Gmail',
            description: 'G Suite helps build customer trust with professional email addresses for your domain. Create as many group mailing lists as your company needs, like sales@yourcompany.com.\n\nGmail provides unlimited of storage, advanced spam filtering, lightning-fast message search, integrated text, voice, and video chat, and other productivity innovations such as Smary Reply. Gmail is accessible securely from almost any mobile device with a data connection, including Android, Apple iPhone®, BlackBerry®, and Windows Phone®. Users familiar with Microsoft Outlook® can continue using it via Gmail.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img266670161965067340.png',
            aboveCharacteristics: [

            ],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Email wherever you work',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Email from any computer or mobile device with a data connection',
                            displayCheck: true
                        },
                        {
                            highlight: 'Keep working even when you’re disconnected with offline support',
                            displayCheck: true
                        },
                        {
                            highlight: "View your email whether you're at your desk, in a meeting, or on a plane; it’s always there",
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Work fast, save time',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Gmail is designed to make you more productive',
                            displayCheck: true
                        },
                        {
                            highlight: '25GB of storage means you never have to delete anything',
                            displayCheck: true
                        },
                        {
                            highlight: 'Powerful search lets you find everything, and labels and filters help you stay organized',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Connect with people',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: "Your inbox isn't just about messages, it's about people too",
                            displayCheck: true
                        },
                        {
                            highlight: 'Text, voice, and video chat lets you see who’s online and connect instantly',
                            displayCheck: true
                        },
                        {
                            highlight: 'See your contacts’ profile photos, recent updates and shared docs next to each email',
                            displayCheck: true
                        }
                    ]
                }
            ]
        },
        {
            id: 3373516,
            label: 'Hangouts',
            description: 'Connect with your team from anywhere. With easy-to-join video calls, you can meet face to face without the added cost of travel.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img2461819673124624682.png',
            aboveCharacteristics: [

            ],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Enjoy frictionless meetings.',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Meet takes the headaches out of joining a video call at work. Just set up a meeting and share a link. No worrying about whether teammates, clients, or customers have the right accounts or plug-ins. With a fast, lightweight interface and smart participant management, multi-person video calls are a breeze.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Designed for every kind of business.',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Meet is fully integrated with G Suite, so you can join meetings directly from a Calendar event or email invite. All of the important event details are right there when you need them, whether you’re joining from a computer, phone, or conference room.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Join meetings on the go.',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'With Meet’s specially designed iOS and Android apps, you can see your meetings for the day with all the important information from Calendar, then join with a tap. G Suite’s Enterprise edition also creates a dial-in phone number for each meeting, so every guest has a great experience – even on the road without wifi or data.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Dial-in phone numbers for every meeting.',
                    position: 3,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'With G Suite’s Enterprise edition, every meeting automatically includes its own dial-in details. Guests can dial in quickly with just two taps on their smartphone.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Integrated with G Suite.',
                    position: 4,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Google Calendar information is available directly within the meeting so you can quickly catch up on the agenda and meeting details.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Connected to meeting rooms.',
                    position: 5,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Chromebox for Meetings connects any meeting room or huddle space to a video meeting with a single click. Simple to use and designed for deploying at scale, all units can be set up in minutes and managed online.',
                            displayCheck: true
                        }
                    ]
                }
            ]
        },
        {
            id: 3373492,
            label: 'Calendar',
            description: 'Google Calendar helps employees organize their schedules and create and manage shared calendars with colleagues. Multiple calendars can be overlayed to view free and busy times at a glance, and the “Quick Add” feature understands natural language so you can easily create appointments by typing event information into a simple box. Google Calendar is available from any web browser and all major mobile platforms, and can be accessed from Microsoft Outlook.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img729941106562180417.png',
            aboveCharacteristics: [

            ],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Stay organized and on schedule',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Organize your day with Calendar and get event reminders on your phone or in your inbox',
                            displayCheck: true
                        },
                        {
                            highlight: 'Attach files or docs to your event so you have the right materials when your meeting starts',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Find time with your team',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Calendar sharing makes it easy to find time with the people you work with and the smart scheduling feature suggests meeting times that work for everyone.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Publish calendars to the web',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Create an event calendar and embed it on your website or set up appointment slots so customers can choose the best time for them.',
                            displayCheck: true
                        }
                    ]
                }
            ]
        },
        {
            id: 3373496,
            label: 'Drive',
            description: 'In Google Drive, you can create new documents, spreadsheets and presentations instantly. Work together at the same time, on the same doc, and see changes as they appear.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img5240507078084403910.png',
            aboveCharacteristics: [

            ],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Access your files anywhere',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Google Drive on your Mac, PC or mobile device gives you a single place for up-to-date versions of your files from anywhere',
                            displayCheck: true
                        },
                        {
                            highlight: 'You can also open 40+ different file types with Drive, including PDFs and MPEG4s, and work on Microsoft Word files straight from Drive.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Bring your files to life',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Avoid multiple file versions and share individual files or whole folders with individual people, your entire team or even customers, vendors and partners',
                            displayCheck: true
                        },
                        {
                            highlight: 'Create and reply to comments on files to get feedback or add ideas.',
                            displayCheck: true
                        },
                        {
                            highlight: 'With Google-powered machine learning, Drive makes your company’s data more useful by predicting what you’ll need before you need it. Beyond easily finding your content with features like natural language search and optical character recognition (OCR), Quick Access in Drive uses signals like your regular work patterns and scheduled meetings to serve up what’s important – so you don’t have to search.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Buy what you need & grow flexibly',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Start with 5 GB of included storage for each of your users. Need more? For as little as $4/month for 20 GB, administrators can centrally purchase and manage up to 16TB (Yes, that’s 16,000 GB!) of additional storage for each user. This feature is coming soon to Staples App Center.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Unlimited',
                    position: 3,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Unlimited cloud storage, so you’ll always have enough space for all your files.',
                            displayCheck: true
                        }
                    ]
                }
            ]
        },
        {
            id: 3373500,
            label: 'Docs, Sheets, Slides',
            description: 'G Suite enables multi-person, real-time collaboration for documents, spreadsheets, presentations and drawings, so groups can work more efficiently and avoid the hassles of attachments and version-control issues.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img2592743314694850303.png',
            aboveCharacteristics: [

            ],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Commenting, chat and real-time editing',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Work in a single document with teammates or people outside your company. See edits as others type, communicate through built-in chat and ask questions through including comments.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Import your documents to make them instantly editable, including Microsoft® Word and PDF files. Export your work in .docx, .pdf, .odt, .rtf, .txt or .html format.',
                            displayCheck: true
                        },
                        {
                            highlight: "Track changes made to your documents and undo anything you choose. Previous versions are kept indefinitely and they don't count toward your storage.",
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Advanced, fast, online spreadsheets',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Perform calculations with powerful functions and formulas in spreadsheets. You can also add sophisticated charts, generate pivot table reports, add filters and more.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Import and convert existing spreadsheets to make them instantly editable, including Microsoft Excel® files. Export Sheets in .xlsx, .csv, .html, .ods, .pdf or .txt form.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Create, edit and share spreadsheets from your phone or tablet, no matter where you happen to be. Work in Sheets even when you’re offline.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Beautiful presentations created together',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Create and edit polished presentations in your browser—no software required. Multiple people can work at the same time so everyone always has the latest version.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Work on a single presentation online with your business partner, your whole team or external contacts. You control who gets permission to edit, view or just add comments.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Start from scratch or expedite the process by choosing a template. You can enhance your presentations with videos, images, drawings and smooth transitions.',
                            displayCheck: true
                        }
                    ]
                }
            ]
        },
        {
            id: 3373520,
            label: 'Forms',
            description: 'Create custom forms for surveys and questionnaires at no extra cost. Gather everything in a spreadsheet and analyze data right in Google Sheets.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img6300464783401748289.png',
            aboveCharacteristics: [

            ],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Create a form as easily as creating a document',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Select from multiple question types, drag-and-drop to reorder questions and customize values as easily as pasting a list.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Send professional looking surveys to customers',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Great looking surveys help you connect with customers to gain valuable insights. Add images, videos, and custom logic to give people who respond a great survey experience.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Analyze responses with automatic summaries',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Watch responses appear in real time. You can also access the raw data and analyze it with Google Sheets or other software.',
                            displayCheck: true
                        }
                    ]
                }
            ]
        },
        {
            id: 3373524,
            label: 'Admin',
            description: "Easily add users, manage devices and configure security and settings so your data stays safe. Administration shouldn't need a manual.",
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img6087781783495484922.png',
            aboveCharacteristics: [

            ],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Security and control',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Centralized administration makes setup and management fast and easy. Add and remove users, set up groups, and add security options like 2-step verification and SSO.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Mobile device management',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Distribute apps to employees and keep data secure on employee’s iOS and Android devices. Use mobile management to distribute apps and get employees up and running quickly, check usage, manage security settings, and lock or wipe devices remotely.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: '24/7 support and reliability',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: "Call or email Google support 24/7. Combine our support team, with our world-class data center networks, and we've definitely got you covered.",
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'An archive for emails and chats',
                    position: 3,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Control how long your email messages and on-the-record chats are retained. Specify policies for your entire domain or based on organizational units, date ranges and specific terms.',
                            displayCheck: true
                        }
                    ]
                }
            ]
        },
        {
            id: 3373528,
            label: 'Search, Sites, Keep',
            description: 'Additional tools to help you work more productively.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img3049401810947748530.png',
            aboveCharacteristics: [

            ],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Google Search -- built for G Suite',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Use the power of Google to search across your company’s content in G Suite. From Gmail and Drive to Docs, Sheets, Slides, Calendar, and more, Google Cloud Search answers your questions and delivers relevant suggestions to help you throughout the day.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Cloud Search is just like Google Search - built for G Suite. Search emails, docs, events, the corporate directory, and more. Find whatever you’re looking for, wherever it’s located.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Open the Cloud Search app to get timely and relevant assist cards that help keep you organized throughout the day. Google intelligence helps you stay on top of meetings, docs, emails, and more.',
                            displayCheck: true
                        },
                        {
                            highlight: 'Search your company directory. See colleagues’ contact details, plus events and files you have in common. Send an email, make a call, or start a Hangout with just a click or a tap.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Sites - effortlessly create impactful team sites',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Collaboratively create engaging, high-quality sites for your team, project or event. The sites look great on every screen, from desktop to smartphone. All without learning design or programming.',
                            displayCheck: true
                        },
                        {
                            highlight: "Create a site, no programming or design skills needed. Sites makes it a simple to exhibit your team's work, with easy access to all your content from G Suite - whether it's a Drive folder, Doc, or even shared Calendar.",
                            displayCheck: true
                        },
                        {
                            highlight: 'Carefully created themes help make your content stand out. Sites intelligently optimizes your work so it looks great on desktop, tablet, and mobile.',
                            displayCheck: true
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Keep - capture what’s important and get more done.',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Keep organized. Capture inspiration and to-dos effortlessly. Collaborate on notes with teammates and set reminders to stay on track. Everything syncs across your devices, so what’s important is always in reach.',
                            displayCheck: true
                        },
                        {
                            highlight: 'With Keep, it’s easy to collaborate with your colleagues on notes, lists, photos, audio, and drawings. Capture brainstorm ideas quickly, keep them on hand while you work, and watch to-dos get checked off in real time.',
                            displayCheck: true
                        },
                        {
                            highlight: "Access, create, and edit notes wherever you go — from your computer, phone, or tablet — even when there's no connection. Every edit you make is automatically saved and updated across all devices.",
                            displayCheck: true
                        }
                    ]
                }
            ]
        }
    ],
    i18n: {
        next: 'Next',
        prev: 'Previous'
    }
};

export const tSheetsMock = {
    items: [
        {
            id: 3153277,
            label: 'Integration Details',
            description: 'TSheets Time Tracking integrates with RUN Powered by ADP to provide easy tracking of employee hours. Once your employees track time, then you choose when to sync data to ADP Powered by RUN. When you sync, TSheets will send employee hours and paid time off data to ADP Powered by RUN, and employee names and company contact information will be updated in TSheets.',
            aboveCharacteristics: [],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'What ADP Applications will be integrated?',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'TSheets Time Tracking will integrate with RUN Powered by ADP.',
                            displayCheck: false
                        }
                    ]
                },
                {
                    id: null,
                    title: 'What type of integration is it?',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'TSheets utilizes single sign-on for easy user access between applications. TSheets has implemented the Data Connector and the sync is performed by the Payroll or HR Admin.',
                            displayCheck: false
                        }
                    ]
                },
                {
                    id: null,
                    title: 'What data will be sent from TSheets to ADP?',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'TSheets will send the following data for each employee that has recorded time worked: <br />\n1. Time (hours)<br />\n2. Overtime hours<br />\n3. Paid time off hours',
                            displayCheck: false
                        }
                    ]
                },
                {
                    id: null,
                    title: 'What ADP data will be read by TSheets?',
                    position: 3,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: '1. Company name<br />\n2. Company website<br />\n3. Employee first name<br />\n4. Employee last name<br />\n5. Employee phone number<br />\n6. Employee email address<br />\n7. Employee ID<br />\n8. Employee pay frequency<br />\n9. Employee pay type<br />\n10. Earnings codes',
                            displayCheck: false
                        }
                    ]
                }
            ]
        },
        {
            id: 3153253,
            label: 'Mobile App',
            description: 'Track, submit, and approve accurate employee hours directly from the TSheets mobile app.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/181904/feature/img2268731364879272893.png',
            aboveCharacteristics: [],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Eliminate manual time entry',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Don’t put up with inaccurate and messy paper timesheets. Complete payroll in a fraction of the time and be confident that it’s error-free. There should be no guesswork when it comes to payroll.',
                            displayCheck: false
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Track time on any device',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: "Whether you've got employees in multiple locations, at the office, or out in the field, the TSheets mobile app has everything your employees need to track time accurately from anywhere — including on mobile devices they use daily.",
                            displayCheck: false
                        }
                    ]
                }
            ]
        },
        {
            id: 3153273,
            label: 'GPS Tracking',
            description: 'Get real-time insight into your mobile workforce.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/181904/feature/img6036482751320671963.png',
            aboveCharacteristics: [],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'GPS tracking helps you track and schedule employees on the move',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'If you have a mobile workforce, this feature was made for you. Get real-time insight into who’s on the clock, where they’re working, and what they’re working on with the TSheets GPS time tracker, available on the mobile app. Once employees clock in, the mobile app logs their location and updates it throughout the workday. It’s transparency and accuracy you can depend on actively and retroactively, without draining cell phone batteries or data.',
                            displayCheck: false
                        }
                    ]
                }
            ]
        },
        {
            id: 3153261,
            label: 'Job Costing',
            description: 'Gain accurate insights to estimate project costs.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/181904/feature/img5458115302449367123.png',
            aboveCharacteristics: [],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Gain accurate insights to estimate project costs',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'With TSheets, you can to acquire data for job costing to gain accurate insights to estimate the cost of a project. Track time against any job to determine the resources to invest, the best-suited employees, and a fair price for you and your customers.',
                            displayCheck: false
                        }
                    ]
                }
            ]
        },
        {
            id: 3153265,
            label: 'Time Clock Kiosk',
            description: 'A fast and easy way for employees to clock in, on site.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/181904/feature/img7477760359665209785.png',
            aboveCharacteristics: [],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Ideal for employees at the same location',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'TSheets Time Clock Kiosk is the ideal time-capture solution for employees who clock in at the same location or on a single device. Enable a time clock on any desktop, laptop, tablet, or cellphone with an internet connection. Employees clock in and out and take breaks with a unique four-digit PIN. They can even track time against different projects or clients, add notes, and submit time for payroll from the same time clock.',
                            displayCheck: false
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Facial recognition ensures the right person is clocking in',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Enable photo capture and facial recognition on your time clock kiosk and automatically capture employee photos when they clock in and out. TSheets software flags timesheet photos for review when faces don’t match — curbing buddy punching and time theft. Best of all, this biometric solution needs no additional hardware.',
                            displayCheck: false
                        }
                    ]
                }
            ]
        },
        {
            id: 3153257,
            label: "Who's Working",
            description: "Know exactly who's working and where at any point in the day.",
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/181904/feature/img8160420411590118765.png',
            aboveCharacteristics: [],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Ideal for a workforce that is always on the move',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: "If your workforce is always on the move or based at several locations, just open the “Who's Working” window on the TSheets web dashboard or mobile app, and you'll immediately see who's on the clock, where they are, and how long they've been at work. You can also sort the list by hours worked, clock-in time, employee name, or location for an in-depth, real-time snapshot of your business.",
                            displayCheck: false
                        }
                    ]
                }
            ]
        },
        {
            id: 3153269,
            label: 'Scheduling',
            description: 'Create and share schedules from any device.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/181904/feature/img4657100236174244076.png',
            aboveCharacteristics: [],
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Create schedules from any device',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Easy scheduling allows admins to create and update employee schedules from desktops, laptops, and mobile devices. No more dry-erase boards and printed Excel spreadsheets.',
                            displayCheck: false
                        }
                    ]
                },
                {
                    id: null,
                    title: 'Assign and share schedules with your employees',
                    position: 1,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'TSheets scheduling software makes it faster and easier than ever to assign and share job and shift schedules with employees and keep your workforce in-the-know.',
                            displayCheck: false
                        }
                    ]
                },
                {
                    id: null,
                    title: 'All the functionality you desire',
                    position: 2,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'TSheets Scheduling syncs with major calendar programs, and admins can set up notifications to alert employees when shifts are changed or canceled.',
                            displayCheck: false
                        }
                    ]
                }
            ]
        }
    ],
    i18n: {
        next: 'Next',
        prev: 'Previous'
    }
};

export const productMockWithHTML = {
    ...tSheetsMock,
    items: [
        {
            ...tSheetsMock.items[0],
            description: '<b>All you need to do your best work</b>, together in one package that works seamlessly from your computer, phone or tablet',
            belowCharacteristics: [
                {
                    id: null,
                    title: 'Create schedules from any device',
                    position: 0,
                    abovePicture: false,
                    highlights: [
                        {
                            highlight: 'Easy <b>scheduling</b> allows admins to create and update employee schedules from desktops, laptops, and mobile devices. No more dry-erase boards and printed Excel spreadsheets.',
                            displayCheck: false
                        }
                    ]
                }
            ]
        }
    ]
};
