// @appdirect/sfb-theme-components/mocks/product/productScreenshots.js
export const screenshotsMockData = {
    screenshots: [{
        name: 'Business email through Gmail',
        imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img6432077439382128397.png',
        imageUrlSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img6432077439382128397.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img8286148827278342495.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img6432077439382128397.png'
        }
    }, {
        name: 'Video and voice conferencing',
        imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img6606457208412499340.png',
        imageUrlSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img6606457208412499340.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img902230549307187830.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img6606457208412499340.png'
        }
    }, {
        name: 'Security and administration controls',
        imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img1755005389433420487.png',
        imageUrlSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img1755005389433420487.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img4061457645332640452.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img1755005389433420487.png'
        }
    }, {
        name: 'Windows Server Client Access License (CAL)*',
        imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img8613964566043315063.png',
        imageUrlSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img8613964566043315063.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img3978961145785895803.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img8613964566043315063.png'
        }
    }, {
        name: 'Encryption for all files and storage locations, cloud-based file tracking',
        imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img934590527213828283.png',
        imageUrlSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img934590527213828283.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img4249104149208966111.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/screenshot/img934590527213828283.png'
        }
    }]
};

export const irregularImagesMockData = [
    {
        name: 'Get real-time insight into your mobile workforce.',
        imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/feature/img6300464783401748289.png'
    },
    {
        name: 'GPS tracking helps you track and schedule employees on the move',
        imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/181904/feature/img6036482751320671963.png'
    },
    {
        name: 'Know exactly who\'s working and where at any point in the day.',
        imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/53185/overview/img112226600830447841.png'
    }
];
