// @appdirect/sfb-theme-components/mocks/product/productBundles.js
export const productBundleMock = {
    bundle: {
        summary: {
            title: 'Super Bundle',
            description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
            images: [
                'https://via.placeholder.com/600'
            ]
        },
        callToActions: {
            items: [
                {
                    label: 'Buy Now',
                    tooltip: '',
                    isPrimary: true,
                    enabled: true,
                    url: 'undefined/api/internal/storefront/v1/cta?productId=120477&type=BUY',
                    actionType: 'BUY',
                    popup: {}
                },
                {
                    label: 'Contact Us',
                    tooltip: '',
                    isPrimary: false,
                    enabled: true,
                    url: '',
                    actionType: 'LEAD_GEN'
                },
                {
                    label: 'Upgrade',
                    tooltip: '',
                    isPrimary: false,
                    enabled: true,
                    url: '',
                    actionType: 'CAN_MOSI_UPGRADE'
                }
            ],
            caption: {
                text: '',
                tooltip: ''
            }
        },
        overview: {
            title: 'Bundle Blurb',
            description: 'The bundle description appears on the bundle profile and listing in your marketplace. We recommend keeping it around 150 words.',
            image: 'https://placehold.co/600'
        },
        pricing: {
            editions: [
                {
                    productName: 'PX-EVA Base Platform',
                    id: 652311,
                    fullName: 'PX-EVA Base Platform - Recurring Edition',
                    name: 'Recurring Edition',
                    value: '$70.00',
                    frequency: 'per Year'
                },
                {
                    productName: 'PX-EVA Budgeting Module',
                    id: 652315,
                    fullName: 'PX-EVA Budgeting Module - Recurring Edition',
                    name: 'Recurring Edition',
                    value: '$24.00',
                    frequency: 'per Year'
                },
                {
                    productName: 'PX-EVA Sustainability Module',
                    id: 652320,
                    fullName: 'PX-EVA Sustainability Module - Recurring Edition',
                    name: 'Recurring Edition',
                    value: '$24.00',
                    frequency: 'per Year'
                },
                {
                    productName: 'PX-EVA Real-time Monitoring Module',
                    id: 652324,
                    fullName: 'PX-EVA Real-time Monitoring Module - Recurring Edition',
                    name: 'Recurring Edition',
                    value: '$180.00',
                    frequency: 'per Year'
                },
                {
                    productName: 'PX-EVA Measurement & Verification Module',
                    id: 652329,
                    fullName: 'PX-EVA Measurement & Verification Module - Recurring Edition',
                    name: 'Recurring Edition',
                    value: '$24.00',
                    frequency: 'per Year'
                },
                {
                    productName: 'PX-EVA Peak Load Management Module',
                    id: 652333,
                    fullName: 'PX-EVA Peak Load Management Module - Recurring Edition',
                    name: 'Recurring Edition',
                    value: '$24.00',
                    frequency: 'per Year'
                },
                {
                    productName: 'PX-EVA Automated Utility Data Capture',
                    id: 652338,
                    fullName: 'PX-EVA Automated Utility Data Capture - Recurring Edition',
                    name: 'Recurring Edition',
                    value: '$36.00',
                    frequency: 'per Year'
                }
            ]
        },
        ribbon: null,
        tags: [],
        i18n: {
            bundleIncludes: 'Bundle Package Includes',
            bundles: 'Bundles',
            viewProduct: 'View Product',
            pricing: 'Starting Price',
            demo: 'Demo',
            clickBundleSummary: 'Click for this bundle\'s summary',
            previous: 'Go to previous bundle',
            next: 'Go to next bundle'
        },
        items: [
            {
                id: 329889,
                focused: true,
                url: '/en-US/apps/329889/px-eva-base-platform',
                name: 'PX-EVA Base Platform',
                image: null,
                card: {
                    title: 'PX-EVA Base Platform',
                    details: 'PX EVA',
                    overview: 'PX EVA',
                    logoSrc: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png',
                    logoSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263-2x.png',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png'
                    },
                    logoAlt: 'Logo of PX-EVA Base Platform',
                    productRibbon: null,
                    merchandisingBadges: []
                },
                customAttributes: [
                    {
                        type: 'TEXT',
                        attributeKey: 'ERP-Id',
                        label: 'ERP Id',
                        value: 'VISTA01'
                    }
                ],
                edition: {
                    name: 'Recurring Edition'
                },
                overview: {
                    title: 'Splash Title',
                    description: 'Splash Description',
                    callToActions: {},
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329889/overview/img680213046346428498.png',
                    imageSrcSet: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329889/overview/img680213046346428498.png',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329889/overview/img680213046346428498.png',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329889/overview/img680213046346428498-2x.png'
                    },
                    imageAlt: 'Eaton sample budget preview',
                    showThirdPartyLegalDisclosure: false,
                    documentationUrl: null,
                    downloadDocumentationUrl: null,
                    pricingUrl: 'undefined/editions',
                    customers: [],
                    benefits: [],
                    i18n: {
                        customers: 'Featured Customers',
                        featuresAndBenefits: 'Features and Benefits',
                        demo: 'Demo',
                        overviewImage: 'Overview Image',
                        documentation: 'Read Documentation'
                    }
                },
                resources: {
                    reseller: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    public: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    i18n: {
                        Resources: 'Resources',
                        Videos: 'Videos',
                        Documents: 'Documents',
                        ResellerResources: 'Reseller Resources',
                        Download: 'Download'
                    }
                },
                screenshots: {
                    items: [],
                    i18n: {
                        title: 'Screenshots',
                        sidebarTitle: 'View Screenshots'
                    }
                },
                support: {
                    termsUrl: null,
                    privacyUrl: null,
                    email: null,
                    phone: null,
                    knowledgebaseUrl: null,
                    description: '',
                    i18n: {
                        termsAndConditions: 'Terms & Conditions',
                        title: 'Additional Information',
                        termsUrl: 'Terms of Service',
                        privacyUrl: 'Privacy Policy',
                        support: 'Support',
                        email: 'Email',
                        description: 'Description',
                        phone: 'Phone',
                        knowledgebaseUrl: 'Knowledge Base'
                    }
                },
                tags: [],
                bundle: {
                    id: 329919,
                    url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329889'
                }
            },
            {
                id: 329893,
                focused: false,
                url: '/en-US/apps/329893/px-eva-budgeting-module',
                name: 'PX-EVA Budgeting Module',
                image: null,
                card: {
                    title: 'PX-EVA Budgeting Module',
                    details: '',
                    overview: '',
                    logoSrc: null,
                    logoSrcset: {},
                    productRibbon: null,
                    merchandisingBadges: []
                },
                customAttributes: [
                    {
                        type: 'TEXT',
                        attributeKey: 'ERP-Id',
                        label: 'ERP Id',
                        value: 'VISTA01'
                    }
                ],
                edition: {
                    name: 'Recurring Edition'
                },
                overview: {
                    title: 'Splash Title',
                    description: 'Splash Desc',
                    callToActions: {},
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329893/overview/img4781544673053016721.png',
                    imageSrcSet: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329893/overview/img4781544673053016721.png',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329893/overview/img4781544673053016721.png',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329893/overview/img4781544673053016721-2x.png'
                    },
                    imageAlt: 'Eaton sample budget preview',
                    showThirdPartyLegalDisclosure: false,
                    documentationUrl: null,
                    downloadDocumentationUrl: null,
                    pricingUrl: 'undefined/editions',
                    customers: [],
                    benefits: [],
                    i18n: {
                        customers: 'Featured Customers',
                        featuresAndBenefits: 'Features and Benefits',
                        demo: 'Demo',
                        overviewImage: 'Overview Image',
                        documentation: 'Read Documentation'
                    }
                },
                resources: {
                    reseller: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    public: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    i18n: {
                        Resources: 'Resources',
                        Videos: 'Videos',
                        Documents: 'Documents',
                        ResellerResources: 'Reseller Resources',
                        Download: 'Download'
                    }
                },
                screenshots: {
                    items: [],
                    i18n: {
                        title: 'Screenshots',
                        sidebarTitle: 'View Screenshots'
                    }
                },
                support: {
                    termsUrl: null,
                    privacyUrl: null,
                    email: null,
                    phone: null,
                    knowledgebaseUrl: null,
                    description: '',
                    i18n: {
                        termsAndConditions: 'Terms & Conditions',
                        title: 'Additional Information',
                        termsUrl: 'Terms of Service',
                        privacyUrl: 'Privacy Policy',
                        support: 'Support',
                        email: 'Email',
                        description: 'Description',
                        phone: 'Phone',
                        knowledgebaseUrl: 'Knowledge Base'
                    }
                },
                tags: [],
                bundle: {
                    id: 329919,
                    url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329893'
                }
            },
            {
                id: 329898,
                focused: false,
                url: '/en-US/apps/329898/px-eva-sustainability-module',
                name: 'PX-EVA Sustainability Module',
                image: null,
                card: {
                    title: 'PX-EVA Sustainability Module',
                    details: '',
                    overview: '',
                    logoSrc: null,
                    logoSrcset: {},
                    productRibbon: null,
                    merchandisingBadges: []
                },
                customAttributes: [
                    {
                        type: 'TEXT',
                        attributeKey: 'ERP-Id',
                        label: 'ERP Id',
                        value: 'VISTA01'
                    }
                ],
                edition: {
                    name: 'Recurring Edition'
                },
                overview: {
                    title: 'Splash Title',
                    description: 'splash',
                    callToActions: {},
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329898/overview/img6582364165551198837.png',
                    imageSrcSet: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329898/overview/img6582364165551198837.png',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329898/overview/img6582364165551198837.png',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329898/overview/img6582364165551198837-2x.png'
                    },
                    imageAlt: 'Eaton sample budget',
                    showThirdPartyLegalDisclosure: false,
                    documentationUrl: null,
                    downloadDocumentationUrl: null,
                    pricingUrl: 'undefined/editions',
                    customers: [],
                    benefits: [],
                    i18n: {
                        customers: 'Featured Customers',
                        featuresAndBenefits: 'Features and Benefits',
                        demo: 'Demo',
                        overviewImage: 'Overview Image',
                        documentation: 'Read Documentation'
                    }
                },
                resources: {
                    reseller: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    public: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    i18n: {
                        Resources: 'Resources',
                        Videos: 'Videos',
                        Documents: 'Documents',
                        ResellerResources: 'Reseller Resources',
                        Download: 'Download'
                    }
                },
                screenshots: {
                    items: [],
                    i18n: {
                        title: 'Screenshots',
                        sidebarTitle: 'View Screenshots'
                    }
                },
                support: {
                    termsUrl: null,
                    privacyUrl: null,
                    email: null,
                    phone: null,
                    knowledgebaseUrl: null,
                    description: '',
                    i18n: {
                        termsAndConditions: 'Terms & Conditions',
                        title: 'Additional Information',
                        termsUrl: 'Terms of Service',
                        privacyUrl: 'Privacy Policy',
                        support: 'Support',
                        email: 'Email',
                        description: 'Description',
                        phone: 'Phone',
                        knowledgebaseUrl: 'Knowledge Base'
                    }
                },
                tags: [],
                bundle: {
                    id: 329919,
                    url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329898'
                }
            },
            {
                id: 329902,
                focused: false,
                url: '/en-US/apps/329902/px-eva-real-time-monitoring-module',
                name: 'PX-EVA Real-time Monitoring Module',
                image: null,
                card: {
                    title: 'PX-EVA Real-time Monitoring Module',
                    details: '',
                    overview: '',
                    logoSrc: null,
                    logoSrcset: {},
                    productRibbon: null,
                    merchandisingBadges: []
                },
                customAttributes: [],
                edition: {
                    name: 'Recurring Edition'
                },
                overview: {
                    title: null,
                    description: '',
                    callToActions: {},
                    image: null,
                    imageSrcSet: {},
                    showThirdPartyLegalDisclosure: false,
                    documentationUrl: null,
                    downloadDocumentationUrl: null,
                    pricingUrl: 'undefined/editions',
                    customers: [],
                    benefits: [],
                    i18n: {
                        customers: 'Featured Customers',
                        featuresAndBenefits: 'Features and Benefits',
                        demo: 'Demo',
                        overviewImage: 'Overview Image',
                        documentation: 'Read Documentation'
                    }
                },
                resources: {
                    reseller: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    public: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    i18n: {
                        Resources: 'Resources',
                        Videos: 'Videos',
                        Documents: 'Documents',
                        ResellerResources: 'Reseller Resources',
                        Download: 'Download'
                    }
                },
                screenshots: {
                    items: [],
                    i18n: {
                        title: 'Screenshots',
                        sidebarTitle: 'View Screenshots'
                    }
                },
                support: {
                    termsUrl: null,
                    privacyUrl: null,
                    email: null,
                    phone: null,
                    knowledgebaseUrl: null,
                    description: '',
                    i18n: {
                        termsAndConditions: 'Terms & Conditions',
                        title: 'Additional Information',
                        termsUrl: 'Terms of Service',
                        privacyUrl: 'Privacy Policy',
                        support: 'Support',
                        email: 'Email',
                        description: 'Description',
                        phone: 'Phone',
                        knowledgebaseUrl: 'Knowledge Base'
                    }
                },
                tags: [],
                bundle: {
                    id: 329919,
                    url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329902'
                }
            },
            {
                id: 329907,
                focused: false,
                url: '/en-US/apps/329907/px-eva-measurement-verification-module',
                name: 'PX-EVA Measurement & Verification Module',
                image: null,
                card: {
                    title: 'PX-EVA Measurement & Verification Module',
                    details: '',
                    overview: '',
                    logoSrc: null,
                    logoSrcset: {},
                    productRibbon: null,
                    merchandisingBadges: []
                },
                customAttributes: [],
                edition: {
                    name: 'Recurring Edition'
                },
                overview: {
                    title: null,
                    description: '',
                    callToActions: {},
                    image: null,
                    imageSrcSet: {},
                    showThirdPartyLegalDisclosure: false,
                    documentationUrl: null,
                    downloadDocumentationUrl: null,
                    pricingUrl: 'undefined/editions',
                    customers: [],
                    benefits: [],
                    i18n: {
                        customers: 'Featured Customers',
                        featuresAndBenefits: 'Features and Benefits',
                        demo: 'Demo',
                        overviewImage: 'Overview Image',
                        documentation: 'Read Documentation'
                    }
                },
                resources: {
                    reseller: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    public: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    i18n: {
                        Resources: 'Resources',
                        Videos: 'Videos',
                        Documents: 'Documents',
                        ResellerResources: 'Reseller Resources',
                        Download: 'Download'
                    }
                },
                screenshots: {
                    items: [],
                    i18n: {
                        title: 'Screenshots',
                        sidebarTitle: 'View Screenshots'
                    }
                },
                support: {
                    termsUrl: null,
                    privacyUrl: null,
                    email: null,
                    phone: null,
                    knowledgebaseUrl: null,
                    description: '',
                    i18n: {
                        termsAndConditions: 'Terms & Conditions',
                        title: 'Additional Information',
                        termsUrl: 'Terms of Service',
                        privacyUrl: 'Privacy Policy',
                        support: 'Support',
                        email: 'Email',
                        description: 'Description',
                        phone: 'Phone',
                        knowledgebaseUrl: 'Knowledge Base'
                    }
                },
                tags: [],
                bundle: {
                    id: 329919,
                    url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329907'
                }
            },
            {
                id: 329911,
                focused: false,
                url: '/en-US/apps/329911/px-eva-peak-load-management-module',
                name: 'PX-EVA Peak Load Management Module',
                image: null,
                card: {
                    title: 'PX-EVA Peak Load Management Module',
                    details: '',
                    overview: '',
                    logoSrc: null,
                    logoSrcset: {},
                    productRibbon: null,
                    merchandisingBadges: []
                },
                customAttributes: [],
                edition: {
                    name: 'Recurring Edition'
                },
                overview: {
                    title: null,
                    description: '',
                    callToActions: {},
                    image: null,
                    imageSrcSet: {},
                    showThirdPartyLegalDisclosure: false,
                    documentationUrl: null,
                    downloadDocumentationUrl: null,
                    pricingUrl: 'undefined/editions',
                    customers: [],
                    benefits: [],
                    i18n: {
                        customers: 'Featured Customers',
                        featuresAndBenefits: 'Features and Benefits',
                        demo: 'Demo',
                        overviewImage: 'Overview Image',
                        documentation: 'Read Documentation'
                    }
                },
                resources: {
                    reseller: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    public: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    i18n: {
                        Resources: 'Resources',
                        Videos: 'Videos',
                        Documents: 'Documents',
                        ResellerResources: 'Reseller Resources',
                        Download: 'Download'
                    }
                },
                screenshots: {
                    items: [],
                    i18n: {
                        title: 'Screenshots',
                        sidebarTitle: 'View Screenshots'
                    }
                },
                support: {
                    termsUrl: null,
                    privacyUrl: null,
                    email: null,
                    phone: null,
                    knowledgebaseUrl: null,
                    description: '',
                    i18n: {
                        termsAndConditions: 'Terms & Conditions',
                        title: 'Additional Information',
                        termsUrl: 'Terms of Service',
                        privacyUrl: 'Privacy Policy',
                        support: 'Support',
                        email: 'Email',
                        description: 'Description',
                        phone: 'Phone',
                        knowledgebaseUrl: 'Knowledge Base'
                    }
                },
                tags: [],
                bundle: {
                    id: 329919,
                    url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329911'
                }
            },
            {
                id: 329916,
                focused: false,
                url: '/en-US/apps/329916/px-eva-automated-utility-data-capture',
                name: 'PX-EVA Automated Utility Data Capture',
                image: null,
                card: {
                    title: 'PX-EVA Automated Utility Data Capture',
                    details: '',
                    overview: '',
                    logoSrc: null,
                    logoSrcset: {},
                    productRibbon: null,
                    merchandisingBadges: []
                },
                customAttributes: [],
                edition: {
                    name: 'Recurring Edition'
                },
                overview: {
                    title: null,
                    description: '',
                    callToActions: {},
                    image: null,
                    imageSrcSet: {},
                    showThirdPartyLegalDisclosure: false,
                    documentationUrl: null,
                    downloadDocumentationUrl: null,
                    pricingUrl: 'undefined/editions',
                    customers: [],
                    benefits: [],
                    i18n: {
                        customers: 'Featured Customers',
                        featuresAndBenefits: 'Features and Benefits',
                        demo: 'Demo',
                        overviewImage: 'Overview Image',
                        documentation: 'Read Documentation'
                    }
                },
                resources: {
                    reseller: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    public: {
                        videos: {
                            items: []
                        },
                        documents: {
                            items: []
                        }
                    },
                    i18n: {
                        Resources: 'Resources',
                        Videos: 'Videos',
                        Documents: 'Documents',
                        ResellerResources: 'Reseller Resources',
                        Download: 'Download'
                    }
                },
                screenshots: {
                    items: [],
                    i18n: {
                        title: 'Screenshots',
                        sidebarTitle: 'View Screenshots'
                    }
                },
                support: {
                    termsUrl: null,
                    privacyUrl: null,
                    email: null,
                    phone: null,
                    knowledgebaseUrl: null,
                    description: '',
                    i18n: {
                        termsAndConditions: 'Terms & Conditions',
                        title: 'Additional Information',
                        termsUrl: 'Terms of Service',
                        privacyUrl: 'Privacy Policy',
                        support: 'Support',
                        email: 'Email',
                        description: 'Description',
                        phone: 'Phone',
                        knowledgebaseUrl: 'Knowledge Base'
                    }
                },
                tags: [],
                bundle: {
                    id: 329919,
                    url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329916'
                }
            }
        ]
    }
};
