// @appdirect/sfb-theme-components/mocks/product/productListing.js
export const paginationMock = {
    nbPages: 79,
    currentPage: 1,
    simplePagesList: '1 of 79',
    pagesList: [
        {
            type: 'previous',
            status: 'disabled'
        },
        {
            type: 'page',
            nb: 1,
            status: 'selected',
            url: '/en-US/listing?page=1'
        },
        {
            type: 'page',
            nb: 2,
            status: 'enabled',
            url: '/en-US/listing?page=2'
        },
        {
            type: 'page',
            nb: 3,
            status: 'enabled',
            url: '/en-US/listing?page=3'
        },
        {
            type: 'page',
            nb: 4,
            status: 'enabled',
            url: '/en-US/listing?page=4'
        },
        {
            type: 'page',
            nb: 5,
            status: 'enabled',
            url: '/en-US/listing?page=5'
        },
        {
            type: 'ellipsis'
        },
        {
            type: 'page',
            nb: 79,
            url: '/en-US/listing?page=79'
        },
        {
            type: 'next',
            status: 'enabled',
            url: '/en-US/listing?page=2'
        }
    ],
    itemsPerPage: 12,
    totalCount: 943,
    start: 0
};

export const sortingOptionsMock = {
    options: [
        {
            value: '',
            label: 'Relevance',
            selected: false,
            url: '/en-US/listing?order='
        },
        {
            value: 'POPULARITY',
            label: 'Popularity',
            selected: true,
            url: '/en-US/listing?order=POPULARITY'
        },
        {
            value: 'NEWEST_FIRST',
            label: 'Newest',
            selected: false,
            url: '/en-US/listing?order=NEWEST_FIRST'
        },
        {
            value: 'ALPHABETICAL',
            label: 'Alphabetical',
            selected: false,
            url: '/en-US/listing?order=ALPHABETICAL'
        },
        {
            value: 'RATING',
            label: 'Rating',
            selected: false,
            url: '/en-US/listing?order=RATING'
        }
    ],
    selected: {
        value: '',
        label: 'Relevance',
        selected: true,
        url: '/en-US/listing?order='
    }
};

export const productListingMock = {
    items: [
        {
            id: 42937,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_64/img63732497944352803.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_64/img63732497944352803.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_64/img63732497944352803-2x.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_64/img63732497944352803.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_112/img1099429097555068136.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/overview/img1301783796595200436.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '622a27d2-09ac-46d1-87e2-3ffa764e8bce',
            description: 'AppWise is a business productivity tool that provides Unified Search across 40+ leading applications like Salesforce, Gmail, Office 365, Box and Asana. AppWise has enterprise-grade security built-in from the ground up, and it\'s intuitive enough be setup by end-users in seconds.',
            overview: 'With Unified Search across over 40+ leading cloud applications, easily find all your emails, documents and tasks in a couple of clicks.',
            vendorName: 'Xendo, Inc.',
            showRating: true,
            hidePricings: false,
            free: true,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'AppWise',
            version: '1.16.32',
            blurb: 'Search everything, in one shot',
            url: '/en-US/apps/42937/appwise',
            startingPrice: {
                formattedFullPrice: '$19.00/Mo',
                formattedPrice: '$19.00',
                price: '19.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 429,
                    name: 'Office Productivity',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 232,
                    name: 'Private Cloud',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 517,
                            name: 'Private Cloud',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 520,
                            name: 'Public Cloud',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 422,
                    name: 'Sales',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2223,
                            name: 'Customer Relationship Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 3203,
                    name: 'Customer Service',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 3591,
                            name: 'Help Desk & Ticketing',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 1350,
                    name: 'Language availability',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 4606,
                            name: 'English (US)',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: 'Currently available or Core Application Catalog only.'
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 327,
                    name: 'Supported Devices',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 811,
                            name: 'Android',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 815,
                            name: 'iPhone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 819,
                            name: 'iPad',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 823,
                            name: 'Linux',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 827,
                            name: 'Mac',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 831,
                            name: 'Mobile Web App',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 835,
                            name: 'Web-based',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 839,
                            name: 'Windows',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 843,
                            name: 'Windows Phone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 3207,
                    name: 'Marketing',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 374,
                    name: 'Integration',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 920,
                            name: 'Single Sign-On',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 941,
                            name: 'User Provisioning',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 5,
            ratingsSummary: {
                title: 'Rating',
                caption: '5/5 Based on 139 reviews'
            },
            numRatings: 139,
            badges: [
                {
                    id: '1',
                    label: 'Top seller',
                    type: 'success'
                },
                {
                    id: '2',
                    label: 'Best of 2021',
                    type: 'emphasis'
                },
                {
                    id: '3',
                    label: 'Buyers\' favorite',
                    type: 'pending'
                }
            ]
        },
        {
            id: 13960,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/13960/thumbs_64/img2319685171585638435.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/13960/thumbs_64/img2319685171585638435.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/13960/thumbs_112/img5599356460088711979.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/13960/overview/img5871220726025334391.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '63a587b6-62d2-4dbd-86b6-8d31104651d8',
            description: 'Install on any server, in the cloud, or on premises. Connect to any SQL or NoSQL database, file storage system, or external service. Develop modern applications for the phone, tablet, or desktop without writing server software. Mobilize your enterprise data in minutes!',
            overview: 'The DreamFactory Services Platform (DSP) is an open source software package that provides a REST API for mobile enterprise application development. DreamFactory enables the enterprise developer to quickly hook up a wide variety of backend systems and expose these assets with a comprehensive palette of RESTful services.',
            vendorName: 'Dreamfactory Software, Inc.',
            showRating: true,
            hidePricings: false,
            free: true,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'DreamFactory',
            blurb: 'Open Source REST API Platform',
            url: '/en-US/apps/13960/dreamfactory',
            startingPrice: {
                formattedFullPrice: 'Free',
                formattedPrice: 'Free',
                price: 'Free',
                priceCaption: '',
                frequency: {
                    label: '',
                    shortLabel: '',
                    id: ''
                }
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 428,
                    name: 'Developer Tools',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2250,
                            name: 'Mobile Development',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 2917,
                            name: 'Open Source',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 5,
            ratingsSummary: {
                title: 'Rating',
                caption: '5/5 Based on 9 reviews'
            },
            numRatings: 9,
            badges: [
                {
                    id: 'ON_SALE',
                    label: 'On Sale',
                    type: 'emphasis'
                }
            ]
        },
        {
            id: 416,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/416/thumbs_64/img2718378498022851128.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/416/thumbs_64/img2718378498022851128.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/416/thumbs_112/img7278442252620726419.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/416/overview/img9138010324290200954.png',
            productRibbon: {
                text: 'CORE APP',
                textColor: '#FFFFFF',
                ribbonColor: '#00769D'
            },
            merchandisingBadges: [
                {
                    uuid: '76eb89e4-def9-4322-9a10-5a8d9f78df89',
                    name: 'iOS Compatible',
                    image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/76eb89e4-def9-4322-9a10-5a8d9f78df89/img4805831920228314845.png?12294cbcc5425724b161137570276e41'
                },
                {
                    uuid: '4e5a0070-28de-11e8-bd43-0aea64b20f8f',
                    name: 'Monkey',
                    image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/4e5a0070-28de-11e8-bd43-0aea64b20f8f/img5143663096997530867.png?23ad4994e6947eed95fc58810c363fb1'
                },
                {
                    uuid: 'dsf87g87y-64e854t65-65d4rg-6s84rdg68',
                    name: 'Single Sign-On',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/ba3dc7bc-b3ca-4d45-90dd-6a0a083ab872/img1508692810840088346.png?a98c1e4ae512ec64db825b0d31435b08'
                },
                {
                    uuid: 'er98gt89-654dfb684354gf-65sd4f-6sd85f4',
                    name: 'Access Management',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/0af0e01c-5c06-4bbb-9da1-214428362bdf/img7972380461864714752.png?0a566604eb8930fa19c2275dddec67fd'
                },
                {
                    uuid: 'rd654rd3g4-68d4rg64-gd68r4g684dr-drg684684',
                    name: 'Unified Invoice',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/9cfc700d-71a8-4207-8c5d-7d46f98c6dc2/img7504959842234640276.png?95543cdf906d7fcb30020958f73c09fe'
                },
                {
                    uuid: '76eb89e4-def9-4322-9a10-5a8d9f78df85',
                    name: 'iOS Compatible v2',
                    image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/76eb89e4-def9-4322-9a10-5a8d9f78df89/img4805831920228314845.png?12294cbcc5425724b161137570276e41'
                },
                {
                    uuid: 'rd654rd3g4-68d4rg64-gd68r4g684dr-drg684685',
                    name: 'Unified Invoice 2',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/9cfc700d-71a8-4207-8c5d-7d46f98c6dc2/img7504959842234640276.png?95543cdf906d7fcb30020958f73c09fe'
                }
            ],
            uuid: '61f2f1a6-ae52-11e3-ac06-22000a1dd446',
            description: 'Financial accounting software for small business. It has everything you\nneed to manage, invoicing, purchases, expenses and payments. Sales\npipeline and customer management. Includes social media and accounting integration.',
            overview: 'CRM, Projects, Service Desk and Financial software for small/medium sized businesses.\nSmartphone compatible, works on iPhone, Android etc.  Branded Sales\nInvoices, Purchase Orders, Custom Invoice Design Templates, Send\nInvoices &amp; Reminders by Email, Debtor/Creditor Management,\nCustomer/Supplier Management, Cashflow Projections, Create Detailed\nBudgets + Compare to Actual, Export Custom Management Reports to PDF,\nProfit &amp; Loss Report, Balance sheet, Trial Balance, Nominal Ledger',
            vendorName: 'Yendo',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'Yendo',
            blurb: 'Small Business ERP',
            url: '/en-US/apps/416/yendo',
            startingPrice: {
                formattedFullPrice: '$19.00/Mo',
                formattedPrice: '$19.00',
                price: '19.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 327,
                    name: 'Supported Devices',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 835,
                            name: 'Web-based',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 374,
                    name: 'Integration',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 920,
                            name: 'Single Sign-On',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 1350,
                    name: 'Language availability',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 4606,
                            name: 'English (US)',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: 'Currently available or Core Application Catalog only.'
                }
            ],
            rating: 3,
            ratingsSummary: {
                title: 'Rating',
                caption: '3/5 Based on 5 reviews'
            },
            numRatings: 5,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                },
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 53185,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/53185/thumbs_64/img1596724994380261585.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/53185/thumbs_64/img1596724994380261585.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/53185/thumbs_112/img5514891912140543937.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/53185/overview/img112226600830447841.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: 'ddce318a-5e78-4c50-9efd-e9ef922e7d39',
            description: 'NextMinute is the app that your business can’t do without. It will save you time and money by helping to better manage your jobs and organise your business, your teams and your operations. \n\nYou’ll spend less time on paperwork and more time on the job. Designed with trade and field services in mind, NextMinute is an easy-to-use tool that allows you to plan, schedule, communicate, track time, quote, order, monitor, and invoice on the go, anytime, through your mobile, tablet or laptop.',
            overview: 'NextMinute is essential online job management software that allows you to run your mobile work-force more efficiently.',
            vendorName: 'NextMinute',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'NextMinute',
            blurb: 'Workforce Job Management',
            url: '/en-US/apps/53185/nextminute',
            startingPrice: {
                formattedFullPrice: '$28.00/Mo',
                formattedPrice: '$28.00',
                price: '28.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 430,
                    name: 'Communication',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 3651,
                            name: 'Mobile',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 374,
                    name: 'Integration',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 920,
                            name: 'Single Sign-On',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 926,
                            name: 'Data Visualization',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 932,
                            name: 'Buyable',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 941,
                            name: 'User Provisioning',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 422,
                    name: 'Sales',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2222,
                            name: 'Contact Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 2226,
                            name: 'Quoting',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 425,
                    name: 'Customer Management',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2234,
                            name: 'Customer Relationship Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 2237,
                            name: 'Scheduling & Appointments',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 424,
                    name: 'Online Backup & Storage',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 3211,
                    name: 'Operations',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 3699,
                            name: 'Business Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 3703,
                            name: 'Business Workflow',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 3707,
                            name: 'Field Activity & Service Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 3719,
                            name: 'Scheduling',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 3723,
                            name: 'Service Dispatch',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 327,
                    name: 'Supported Devices',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 811,
                            name: 'Android',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 815,
                            name: 'iPhone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 819,
                            name: 'iPad',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 823,
                            name: 'Linux',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 827,
                            name: 'Mac',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 831,
                            name: 'Mobile Web App',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 835,
                            name: 'Web-based',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 839,
                            name: 'Windows',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 843,
                            name: 'Windows Phone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 429,
                    name: 'Office Productivity',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2256,
                            name: 'Project Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 2259,
                            name: 'Meeting Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 997,
                    name: 'BI & Analytics',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 3515,
                            name: 'Business Intelligence',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 3523,
                            name: 'Reporting & Dashboard',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 5,
            ratingsSummary: {
                title: 'Rating',
                caption: '5/5 Based on 4 reviews'
            },
            numRatings: 4,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 577,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/thumbs_64/img8352473687030874440.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/thumbs_64/img8352473687030874440.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/thumbs_64/img8352473687030874440-2x.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/thumbs_64/img8352473687030874440.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/thumbs_112/img679544425794048368.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/overview/img3187797120366660068.png',
            productRibbon: {
                text: 'CORE APP',
                textColor: '#FFFFFF',
                ribbonColor: '#00769D'
            },
            merchandisingBadges: [],
            uuid: '61f3e71e-ae52-11e3-ac06-22000a1dd446',
            description: 'G Suite offers simple, powerful communication and collaboration tools for your small or medium business – all hosted by Google to streamline setup, minimize maintenance, and reduce IT costs.',
            overview: 'G Suite from Google offers simple, powerful communication and collaboration tools for you small or medium business – all hosted by Google to streamline setup, minimize maintenance, and reduce IT costs.',
            vendorName: 'google',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: false,
            discounted: false,
            pending: false,
            name: 'G Suite',
            blurb: 'Web-Based Email, Calendar, and Documents for Teams',
            url: '/en-US/apps/577/g-suite',
            startingPrice: {
                formattedFullPrice: '$6.00/Mo',
                formattedPrice: '$6.00',
                price: '6.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 1350,
                    name: 'Language availability',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 4606,
                            name: 'English (US)',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4610,
                            name: 'Chinese',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4618,
                            name: 'Danish',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4626,
                            name: 'French',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4638,
                            name: 'German',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4642,
                            name: 'Italian',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4646,
                            name: 'Polish',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4650,
                            name: 'Portuguese',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4658,
                            name: 'Spanish (European)',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4662,
                            name: 'Swedish',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4674,
                            name: 'English (AU)',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4678,
                            name: 'English (UK)',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4682,
                            name: 'Spanish (Mexican)',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4686,
                            name: 'Dutch',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 4690,
                            name: 'Mandarin',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: 'Currently available or Core Application Catalog only.'
                },
                {
                    type: 'CATEGORY',
                    id: 429,
                    name: 'Office Productivity',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2258,
                            name: 'Office Tools',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 327,
                    name: 'Supported Devices',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 811,
                            name: 'Android',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 815,
                            name: 'iPhone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 835,
                            name: 'Web-based',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 843,
                            name: 'Windows Phone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 1276,
                    name: 'Catalog',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 4320,
                            name: 'Core',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 2,
            ratingsSummary: {
                title: 'Rating',
                caption: '2/5 Based on 3 reviews'
            },
            numRatings: 3,
            badges: [
                {
                    id: 'COMING_SOON',
                    label: 'Coming Soon',
                    type: 'pending'
                }
            ]
        },
        {
            id: 420,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/420/thumbs_64/img1031575413466193557.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/420/thumbs_64/img1031575413466193557.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/420/thumbs_112/img238523466168921695.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/420/overview/img3962260781981072079.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '61f2fb74-ae52-11e3-ac06-22000a1dd446',
            description: 'Be truly social and give SnapEngage a try today!',
            overview: 'Turn your visitors into customers and your customers into friends.\n\nSnapEngage lets you chat with your website visitors in real-time using a single line of JavaScript. You can use your preferred instant messaging client (connected to Google Talk or Skype) to instantly talk to your clients, guide them to certain pages and improve their overall experience exponentially. \n\nBe truly social and give SnapEngage a try today!',
            vendorName: 'SnapEngage',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'SnapEngage',
            blurb: 'Live Chat for your website',
            url: '/en-US/apps/420/snapengage',
            startingPrice: {
                formattedFullPrice: '$49.00/Mo',
                formattedPrice: '$49.00',
                price: '49.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 3203,
                    name: 'Customer Service',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 3599,
                            name: 'Live Chat',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 374,
                    name: 'Integration',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 920,
                            name: 'Single Sign-On',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 932,
                            name: 'Buyable',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 938,
                            name: 'Importable',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 941,
                            name: 'User Provisioning',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 425,
                    name: 'Customer Management',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2236,
                            name: 'Live Chat',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 327,
                    name: 'Supported Devices',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 827,
                            name: 'Mac',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 831,
                            name: 'Mobile Web App',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 835,
                            name: 'Web-based',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 4,
            ratingsSummary: {
                title: 'Rating',
                caption: '4/5 Based on 33 reviews'
            },
            numRatings: 33,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 473,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/473/thumbs_64/img993721259211857590.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/473/thumbs_64/img993721259211857590.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/473/thumbs_112/img4507210906661651287.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/473/overview/UGwrIe5UKx7fbv98.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '61f3504c-ae52-11e3-ac06-22000a1dd446',
            description: 'With WORKetc, manage every aspect of your business from organizing contacts and gathering leads to collaborating on projects and sending out the final bill.  Gmail &amp; Outlook integration.  Native Android, iPhone and mobile web apps.',
            overview: 'WORKetc combines CRM, project management, billing, sales, help desk and\nmore with a powerful automation engine to drive your small business to\nthe next level.',
            vendorName: 'WORKetc CRM',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'WORKetc CRM + Projects + Billing',
            blurb: 'Complete small business management platform.',
            url: '/en-US/apps/473/worketc-crm-projects-billing',
            startingPrice: {
                formattedFullPrice: '$195.00/Mo',
                formattedPrice: '$195.00',
                price: '195.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 429,
                    name: 'Office Productivity',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2256,
                            name: 'Project Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 5,
            ratingsSummary: {
                title: 'Rating',
                caption: '5/5 Based on 3 reviews'
            },
            numRatings: 3,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 575,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/575/thumbs_64/img3861498039405875069.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/575/thumbs_64/img3861498039405875069.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/575/thumbs_112/img748634416651988262.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/575/overview/img5546371407940253707.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '61f3e3e0-ae52-11e3-ac06-22000a1dd446',
            description: 'Field activity management tool to Track Field Reps and Manage Field Activities',
            overview: 'Use our intuitive forms, photos and geo-tagging functionality to improve\nyour field activity execution.',
            vendorName: 'Repsly, Inc.',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'Repsly - Track Field Reps and Manage Field Activities',
            blurb: 'Field Activity Management Tool',
            url: '/en-US/apps/575/repsly---track-field-reps-and-manage-field-activities',
            startingPrice: {
                formattedFullPrice: '$5.00/Mo',
                formattedPrice: '$5.00',
                price: '5.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 422,
                    name: 'Sales',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2221,
                            name: 'Configure Price Quote',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 374,
                    name: 'Integration',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 920,
                            name: 'Single Sign-On',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 932,
                            name: 'Buyable',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 941,
                            name: 'User Provisioning',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 426,
                    name: 'Human Resources',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 3643,
                            name: 'Workforce Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 327,
                    name: 'Supported Devices',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 811,
                            name: 'Android',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 815,
                            name: 'iPhone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 819,
                            name: 'iPad',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 823,
                            name: 'Linux',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 827,
                            name: 'Mac',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 831,
                            name: 'Mobile Web App',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 835,
                            name: 'Web-based',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 839,
                            name: 'Windows',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 843,
                            name: 'Windows Phone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 3211,
                    name: 'Operations',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 3707,
                            name: 'Field Activity & Service Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 3715,
                            name: 'Order Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 5,
            ratingsSummary: {
                title: 'Rating',
                caption: '5/5 Based on 13 reviews'
            },
            numRatings: 13,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 544,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/544/thumbs_64/img7826440015608931882.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/544/thumbs_64/img7826440015608931882.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/544/thumbs_112/img5194893520778687861.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/544/overview/fH2VJAnSYFWkEpfY.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '61f3c1da-ae52-11e3-ac06-22000a1dd446',
            description: 'The Galaxie Broadband Player is a simple and cost-effective music solution for businesses. A no-hassle, professional music service for commercial use.',
            overview: 'A turn-key, music service for businesses of all kinds and sizes. 45 channels of commercial free music, 24/7.  Galaxie offers a wide variety of music styles and genres for any mood or any situation.  Use Galaxie in your office, your store, your hotel, your restaurant, your place of business.',
            vendorName: 'Stingray Digital',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: false,
            discounted: false,
            pending: false,
            name: 'Galaxie',
            blurb: 'Online, continuous music service',
            url: '/en-US/apps/544/galaxie',
            startingPrice: {
                formattedFullPrice: '$4.99/Mo',
                formattedPrice: '$4.99',
                price: '4.99',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 374,
                    name: 'Integration',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 920,
                            name: 'Single Sign-On',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 932,
                            name: 'Buyable',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 938,
                            name: 'Importable',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 941,
                            name: 'User Provisioning',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 327,
                    name: 'Supported Devices',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 811,
                            name: 'Android',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 827,
                            name: 'Mac',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 835,
                            name: 'Web-based',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 839,
                            name: 'Windows',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 429,
                    name: 'Office Productivity',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2267,
                            name: 'Multimedia',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 5,
            ratingsSummary: {
                title: 'Rating',
                caption: '5/5 Based on 2 reviews'
            },
            numRatings: 2,
            badges: []
        },
        {
            id: 1967,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/1967/thumbs_64/img7999893482096127889.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/1967/thumbs_64/img7999893482096127889.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/1967/thumbs_112/img7694988283024196889.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/1967/overview/img5384984704344421474.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '61fc47ba-ae52-11e3-ac06-22000a1dd446',
            description: 'SendGrid\'s cloud-based email infrastructure relieves businesses of the\ncost and complexity of maintaining custom email systems. SendGrid\nprovides reliable delivery, scalability and real-time analytics along\nwith flexible API\'s that make custom integration a breeze. SendGrid\nsolves the technical challenges, eliminating your email headaches, so\nthat you can focus on your core product and meet the email demands of\nyour business.',
            overview: 'Sendgrid\'s cloud-based email infrastructure relieves businesses of the\ncost and complexity of maintaining email systems.',
            vendorName: 'Sendgrid',
            showRating: true,
            hidePricings: false,
            free: true,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'SendGrid',
            blurb: 'Email Delivery. Simplified.',
            url: '/en-US/apps/1967/sendgrid',
            startingPrice: {
                formattedFullPrice: 'Free',
                formattedPrice: 'Free',
                price: 'Free',
                priceCaption: '',
                frequency: {
                    label: '',
                    shortLabel: '',
                    id: ''
                }
            },
            tags: [
                {
                    type: 'CATEGORY',
                    id: 428,
                    name: 'Developer Tools',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2252,
                            name: 'Email Delivery',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 33,
                    name: 'App Collections',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 73,
                            name: 'Bundles',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 74,
                            name: 'Deals',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 75,
                            name: 'Staff Picks',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 5,
            ratingsSummary: {
                title: 'Rating',
                caption: '5/5 Based on 2 reviews'
            },
            numRatings: 2,
            badges: [
                {
                    id: 'COMING_SOON',
                    label: 'Coming Soon',
                    type: 'pending'
                }
            ]
        },
        {
            id: 4646,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/4646/thumbs_64/img5961978868787183796.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/4646/thumbs_64/img5961978868787183796.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/4646/thumbs_112/img1203436419493684257.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/4646/overview/img7378320452838770255.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '620e059a-ae52-11e3-ac06-22000a1dd446',
            description: 'Convert website traffic into business with Agendize suite of online\ninteraction tools. Online Scheduling, Live Chat, Click to Call and more!',
            overview: 'An all-in-one customer service platform that makes it easy for customers\nto call, chat and book appointments from your website.',
            vendorName: 'Agendize',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'Agendize',
            blurb: 'Agendize brings in more calls, chats, and appointments from your online presence',
            url: '/en-US/apps/4646/agendize',
            startingPrice: {
                formattedFullPrice: '$20.00/Mo',
                formattedPrice: '$20.00',
                price: '20.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 374,
                    name: 'Integration',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 920,
                            name: 'Single Sign-On',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 932,
                            name: 'Buyable',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 938,
                            name: 'Importable',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 941,
                            name: 'User Provisioning',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 327,
                    name: 'Supported Devices',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 811,
                            name: 'Android',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 835,
                            name: 'Web-based',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 4,
            ratingsSummary: {
                title: 'Rating',
                caption: '4/5 Based on 24 reviews'
            },
            numRatings: 24,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        },
        {
            id: 6813,
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/6813/thumbs_64/img4176382378711157175.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/6813/thumbs_64/img4176382378711157175.png'
            },
            profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/6813/thumbs_112/img2759293882919465131.png',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/6813/overview/img1656219502364525042.png',
            productRibbon: null,
            merchandisingBadges: [],
            uuid: '62196890-ae52-11e3-ac06-22000a1dd446',
            description: 'Simple operating, effective use: TimeTac is the user-friendly software for web-based time tracking for the self-employed and businesses with potential.',
            overview: 'Simple operating, effective use: TimeTac is the user-friendly software for web-based time tracking for the self-employed and businesses with potential.',
            vendorName: 'TimeTac GmbH',
            showRating: true,
            hidePricings: false,
            free: false,
            referable: false,
            buyable: true,
            freeTrialOrEditionPresent: true,
            discounted: false,
            pending: false,
            name: 'TimeTac TimeTracking',
            blurb: 'Web-based time tracking for PC/Mac, smartphone and terminal',
            url: '/en-US/apps/6813/timetac-timetracking',
            startingPrice: {
                formattedFullPrice: '$19.00/Mo',
                formattedPrice: '$19.00',
                price: '19.00',
                priceCaption: 'Starting At',
                frequency: {
                    label: 'Month',
                    shortLabel: 'Mo',
                    id: 'MONTHLY'
                },
                symbol: '$',
                currency: 'USD'
            },
            tags: [
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 374,
                    name: 'Integration',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 920,
                            name: 'Single Sign-On',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 232,
                    name: 'Private Cloud',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 517,
                            name: 'Private Cloud',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 520,
                            name: 'Public Cloud',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'ATTRIBUTE_GROUP',
                    id: 327,
                    name: 'Supported Devices',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'ATTRIBUTE',
                            id: 811,
                            name: 'Android',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 815,
                            name: 'iPhone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 819,
                            name: 'iPad',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 823,
                            name: 'Linux',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 827,
                            name: 'Mac',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 831,
                            name: 'Mobile Web App',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 835,
                            name: 'Web-based',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 839,
                            name: 'Windows',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'ATTRIBUTE',
                            id: 843,
                            name: 'Windows Phone',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                },
                {
                    type: 'CATEGORY',
                    id: 426,
                    name: 'Human Resources',
                    badge: null,
                    showBadge: false,
                    showOnNavigation: true,
                    children: [
                        {
                            type: 'SUB_CATEGORY',
                            id: 2238,
                            name: 'Time Tracking',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        },
                        {
                            type: 'SUB_CATEGORY',
                            id: 2239,
                            name: 'Timesheets & Vacation Management',
                            badge: null,
                            showBadge: false,
                            showOnNavigation: true,
                            children: null,
                            description: null
                        }
                    ],
                    description: null
                }
            ],
            rating: 4,
            ratingsSummary: {
                title: 'Rating',
                caption: '4/5 Based on 78 reviews'
            },
            numRatings: 78,
            badges: [
                {
                    id: 'FREE_TRIAL',
                    label: 'Free Trial',
                    type: 'success'
                }
            ]
        }
    ],
    query: {},
    homeUrl: '/',
    pagination: paginationMock,
    compare: {
        baseUrl: '/en-US/compare'
    },
    sorting: sortingOptionsMock,
    i18n: {
        title: 'Find Applications',
        viewProfile: 'View Profile',
        results: 'Results',
        startingAt: 'Starting At',
        searchLabel: 'Showing 943 Applications',
        compare: 'Compare',
        selectToCompare: 'Select to compare',
        selectAppsCompare: 'Select 2-4 products to compare',
        learnMore: 'Learn More',
        filters: 'Filters',
        sortBy: 'Sorted By',
        viewMode: 'View Mode',
        grid: 'Grid',
        list: 'List',
        hideFilters: 'Hide Filters',
        showFilters: 'Show Filters',
        version: 'Version',
        soldBy: 'Sold by',
        backToTop: 'Back to Top',
        previous: 'Previous',
        next: 'Next'
    }
};

export const productMockWithLongDescription = {
    ...productListingMock.items[0],
    overview: 'Vivamus ultrices, felis tincidunt vestibulum volutpat, elit leo scelerisque dolor, et aliquet arcu ligula quis diam. Duis nisl dui, vestibulum quis hendrerit sit amet, pharetra a magna. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque ante felis, vulputate ut mi ac, fringilla luctus tortor. Aliquam tempor convallis nisi sit amet luctus. Fusce eu felis ut ipsum aliquet maximus. Morbi maximus ut est id ultricies. Nulla vel nibh mauris. Aenean sed ipsum lobortis, dapibus justo vel, fringilla urna. Vivamus fermentum sem eget ultrices ultrices. Pellentesque tempor massa nec felis faucibus, vel pellentesque sapien faucibus. Duis luctus eros lorem, ac varius turpis consequat quis. Pellentesque vehicula id nisl vel ornare. Pellentesque suscipit, turpis non egestas imperdiet, orci magna scelerisque mauris, sit amet pellentesque dui lorem at libero. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae.'
};

export const productMockWithHTML = {
    ...productListingMock.items[0],
    overview: '<p>Vivamus ultrices, <b>felis tincidunt vestibulum</b> volutpat, elit <span style="color:red;">leo scelerisque dolor</span>, et aliquet arcu ligula quis diam.</p><p>Duis nisl dui, vestibulum quis hendrerit sit amet, pharetra a magna. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque ante felis, vulputate ut mi ac, fringilla luctus tortor. Aliquam tempor convallis nisi sit amet luctus. Fusce eu felis ut ipsum aliquet maximus. Morbi maximus ut est id ultricies. Nulla vel nibh mauris. Aenean sed ipsum lobortis, dapibus justo vel, fringilla urna. Vivamus fermentum sem eget ultrices ultrices. Pellentesque tempor massa nec felis faucibus, vel pellentesque sapien faucibus. Duis luctus eros lorem, ac varius turpis consequat quis. Pellentesque vehicula id nisl vel ornare. Pellentesque suscipit, turpis non egestas imperdiet, orci magna scelerisque mauris, sit amet pellentesque dui lorem at libero. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; </p>'
};
