// @appdirect/sfb-theme-components/mocks/product/productBanner.js
import { summaryMock } from './summary';
import { callToActionsCaptionMock,
    callToActionsMock,
    callToActionsMultiButtonMock,
    callToActionsNoActionMock,
    callToActionsReferable,
    productBannerSkipConfiguratorCallToActionsMock } from './callToActions';
import { productEditionsNonStackableNoAddons } from './productEditions';
import { relatedContentMock } from './related-content';
import { productBundleMock } from './productBundles';
import { DOMAIN_APP } from '../../src/constants/application';

const i18nMock = {
    backToResults: 'Back to Results',
    allProducts: 'All Products',
    cart: {
        addedToCart: 'Added to Cart'
    }
};

export const productBannerMock = {
    summary: summaryMock,
    callToActions: callToActionsMock.callToActions,
    configurator: {
        url: 'http://url'
    },
    categories: {
        label: 'Categories',
        items: [
            {
                label: 'Analytics',
                href: '/en-US/listing?cat=74889'
            },
            {
                label: 'Mobility IoT',
                outline: true,
                href: '/en-US/listing?scat=51954'
            }
        ]
    },
    listingPageUrl: '/en-US/listing',
    relatedContent: relatedContentMock,
    i18n: i18nMock
};

export const productBannerMultiButtonMock = {
    summary: summaryMock,
    callToActions: callToActionsMultiButtonMock.callToActions,
    configurator: {
        url: 'http://url'
    },
    relatedContent: relatedContentMock
};

export const productBannerCaptionMock = {
    summary: summaryMock,
    callToActions: callToActionsCaptionMock.callToActions,
    configurator: {
        url: 'http://url'
    },
    relatedContent: relatedContentMock
};

export const productBannerNoImageMock = {
    ...productBannerMock,
    summary: {
        ...productBannerMock.summary,
        logoSrc: null,
        logoSrcset: {}
    }
};

export const productBannerNoActionMock = {
    ...productBannerMock,
    callToActions: callToActionsNoActionMock.callToActions
};

export const productBannerDomainAppMock = {
    ...productBannerMock,
    applicationType: DOMAIN_APP
};

export const productBannerReferableCta = {
    ...productBannerMock,
    config: {
        referUrl: 'https://appdirect.com'
    },
    callToActions: callToActionsReferable.callToActions
};

export const bundleBannerMock = {
    summary: productBundleMock.bundle.summary,
    callToActions: productBundleMock.bundle.callToActions,
    listingPageUrl: '/en-US/listing',
    i18n: i18nMock
};

export const productBannerSkipConfiguratorMock = {
    config: {
        uuid: 'e98b3cf4-2d97-4e12-91d4-3845ebfb8283'
    },
    summary: summaryMock,
    editions: productEditionsNonStackableNoAddons,
    callToActions: productBannerSkipConfiguratorCallToActionsMock,
    i18n: i18nMock
};
