// @appdirect/sfb-theme-components/mocks/product/productEditions.js
import { editionAddToCartSkipConfigurator1,
    editionAddToCartSkipConfigurator2,
    editionSelect } from './callToActions';

import { monthlyPlanUsd } from '../application/editions/plans';

export const editionWithDiscountedUnit = {
    name: 'Monthly + per user + added fees',
    id: 668924,
    primary: true,
    plans: [
        {
            costs: {
                unitPrice: [
                    {
                        id: 'USER',
                        usageRange: [
                            {
                                formattedPrice: '$30.00',
                                formattedFullPrice: '$30.00 / User / Month',
                                formattedFullPriceElements: [
                                    '$30.00',
                                    '/ User',
                                    '/ Month'
                                ],
                                unit: {
                                    min: 1,
                                    max: 100,
                                    label: 'User',
                                    dependency: '',
                                    id: 'USER',
                                    limitLabel: 'For 1-100 Users'
                                },
                                frequency: {
                                    shortLabel: 'Mo',
                                    label: 'Month',
                                    id: 'MONTHLY'
                                },
                                price: {
                                    symbol: '$',
                                    currency: 'USD',
                                    amount: 50,
                                    value: 30,
                                    originalValue: 50,
                                    discountValue: 20,
                                    editionStartingPrice: 130,
                                    formatted: {
                                        originalPrice: '$50.00',
                                        originalFullPrice: '$50.00 / User / Month',
                                        originalFullPriceElements: [
                                            '$50.00',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        discount: '-$20.00',
                                        discountOff: '$20.00 off per User',
                                        meteredUsage: 'For 1-100 Users'
                                    }
                                }
                            }
                        ]
                    }
                ],
                flatPrice: {
                    formattedPrice: '$100.00',
                    formattedFullPrice: '$100.00 / Month',
                    formattedFullPriceElements: [
                        '$100.00',
                        null,
                        '/ Month'
                    ],
                    unit: {
                        min: 0,
                        max: null,
                        dependency: '',
                        id: 'NOT_APPLICABLE'
                    },
                    frequency: {
                        shortLabel: 'Mo',
                        label: 'Month',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: 'USD',
                        amount: 100,
                        value: 100,
                        originalValue: 100,
                        discountValue: 0,
                        editionStartingPrice: 150,
                        formatted: {
                            originalPrice: '$100.00',
                            originalFullPrice: '$100.00 / Month',
                            originalFullPriceElements: [
                                '$100.00',
                                null,
                                '/ Month'
                            ]
                        }
                    }
                }
            },
            setupFees: [
                {
                    formattedPrice: '$30.00',
                    formattedFullPrice: '$30.00 Setup fee',
                    formattedFullPriceElements: [
                        '$30.00',
                        'Setup fee',
                        ''
                    ],
                    unit: {
                        min: 0,
                        max: null,
                        label: 'Setup fee',
                        dependency: '',
                        id: 'ONE_TIME_SETUP'
                    },
                    frequency: {
                        shortLabel: '',
                        label: '',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: 'USD',
                        amount: 30,
                        value: 30,
                        originalValue: 30,
                        discountValue: 0,
                        editionStartingPrice: 150,
                        formatted: {
                            originalPrice: '$30.00',
                            originalFullPrice: '$30.00 Setup fee',
                            originalFullPriceElements: [
                                '$30.00',
                                'Setup fee',
                                ''
                            ]
                        }
                    }
                }
            ],
            contractFees: [
                {
                    formattedPrice: '$40.00',
                    formattedFullPrice: '$40.00 Contract Fee',
                    formattedFullPriceElements: [
                        '$40.00',
                        'Contract Fee',
                        ''
                    ],
                    unit: {
                        min: 0,
                        max: null,
                        label: 'Contract Fee',
                        dependency: '',
                        id: 'CONTRACT_FEE'
                    },
                    frequency: {
                        shortLabel: '',
                        label: '',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: 'USD',
                        amount: 40,
                        value: 40,
                        originalValue: 40,
                        discountValue: 0,
                        editionStartingPrice: 150,
                        formatted: {
                            originalPrice: '$40.00',
                            originalFullPrice: '$40.00 Contract Fee',
                            originalFullPriceElements: [
                                '$40.00',
                                'Contract Fee',
                                ''
                            ]
                        }
                    }
                }
            ],
            contractLength: 12,
            formattedContract: '12 month contract',
            discount: {
                description: 'discount!',
                percentage: null,
                startDate: null,
                expirationDate: null,
                numOfBillingCycles: null,
                availableRedemptions: null,
                amount: {
                    USD: 20
                },
                pricingUnit: 'USER'
            },
            uuid: '48a37f7c-4030-4d40-8c87-5262966ffd94'
        }
    ],
    bullets: [],
    callToAction: {
        label: 'Buy Now',
        tooltip: '',
        isPrimary: true,
        eppid: '48a37f7c-4030-4d40-8c87-5262966ffd94',
        enabled: true,
        url: 'https://sfbmarketplace.byappdirect.com/api/internal/storefront/v1/cta?productId=338726&type=BUY&editionId=668924&eppid=48a37f7c-4030-4d40-8c87-5262966ffd94',
        actionType: 'BUY',
        popup: {},
        isSecondary: false
    },
    audience: [],
    items: [
        {
            content: '1 User'
        }
    ],
    invisible: false,
    leadGen: false,
    revenueType: 'RECURRING'
};

export const editionsMockData = {
    productEditionRefId: 'edition-1234',
    bullets: [{
        content: 'Business email through Gmail',
        tooltip: null
    }, {
        content: 'Video and voice conferencing',
        tooltip: null
    }, {
        content: 'Shared calendars',
        tooltip: null
    }, {
        content: 'Documents, spreadsheets, and presentations',
        tooltip: null
    }, {
        content: '24/7 support by phone, email, and online',
        tooltip: null
    }, {
        content: 'Security and administration controls',
        tooltip: null
    }, {
        content: 'Secure single sign-on to cloud and on-premises apps, multi-factor authentication, conditional access, advanced security reporting',
        tooltip: null
    }, {
        content: 'Encryption for all files and storage locations, cloud-based file tracking',
        tooltip: null
    }, {
        content: 'Protection from advanced targeted attacks by applying user and entity behaviour analytics',
        tooltip: null
    }, {
        content: 'Windows Server Client Access License (CAL)*',
        tooltip: null
    }],
    items: [{
        name: 'G Suite Basic - No Contract',
        id: 404624,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$7.80',
                        formattedFullPrice: '$7.80 / User / Month',
                        formattedFullPriceElements: ['$7.80', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: '$',
                            amount: 7.8,
                            value: 7.8,
                            originalValue: 7.8,
                            discountValue: 0,
                            editionStartingPrice: 7.8,
                            formatted: {
                                originalPrice: '$7.80',
                                originalFullPrice: '$7.80 / User / Month',
                                originalFullPriceElements: ['$7.80', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [{
                formattedPrice: 'Free',
                formattedFullPrice: 'Free Setup fee',
                formattedFullPriceElements: ['Free', 'Setup fee', ''],
                unit: { min: 0, max: null, label: 'Setup fee', dependency: '', id: 'ONE_TIME_SETUP' },
                frequency: { shortLabel: '', label: '', id: 'YEARLY' },
                price: {
                    symbol: '$',
                    currency: 'USD',
                    value: null,
                    originalValue: null,
                    discountValue: 0,
                    editionStartingPrice: null,
                    formatted: {
                        originalPrice: '$0.00',
                        originalFullPrice: '$0.00 Setup fee',
                        originalFullPriceElements: ['$0.00', 'Setup fee', '']
                    }
                }
            }],
            contractFees: [{
                formattedPrice: '$1.00',
                formattedFullPrice: '$1.00 Contract Fee',
                formattedFullPriceElements: ['$1.00', 'Contract Fee', ''],
                unit: { min: 0, max: null, label: 'Contract Fee', dependency: '', id: 'CONTRACT_FEE' },
                frequency: { shortLabel: '', label: '', id: 'MONTHLY' },
                price: {
                    symbol: '$',
                    currency: 'USD',
                    amount: 1,
                    value: 1,
                    originalValue: 1,
                    discountValue: 0,
                    editionStartingPrice: 10,
                    formatted: {
                        originalPrice: '$1.00',
                        originalFullPrice: '$1.00 Contract Fee',
                        originalFullPriceElements: ['$1.00', 'Contract Fee', '']
                    }
                }
            }],
            contractLength: 12312,
            formattedContract: '12312 month contract',
            discount: false,
            uuid: 'bbf4439f-834d-4fdb-abd1-d15ba9f01fad'
        }],
        bullets: [{
            content: '30GB cloud storage',
            tooltip: null
        }, {
            content: '*This is a monthly-billed plan, with no contract.',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=577&type=BUY&editionId=404624',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [{
            content: 'Professional office suite with 30GB storage'
        }, {
            content: 'Most small and medium businesses'
        }],
        items: [],
        invisible: false,
        leadGen: false
    }, {
        name: 'Starter',
        id: 28155,
        primary: true,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'SMS_TEXT_MESSAGE',
                    usageRange: [{
                        formattedPrice: '$0.01',
                        formattedFullPrice: '$0.01 / SMS Text Message / Month',
                        formattedFullPriceElements: ['$0.01', '/ SMS Text Message', '/ Month'],
                        unit: {
                            min: 0,
                            max: null,
                            label: 'SMS Text Message',
                            dependency: '',
                            id: 'SMS_TEXT_MESSAGE'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 0.02,
                            value: 0.01,
                            originalValue: 0.02,
                            discountValue: 0.01,
                            editionStartingPrice: 2,
                            formatted: {
                                originalPrice: '$0.02',
                                originalFullPrice: '$0.02 / SMS Text Message / Month',
                                originalFullPriceElements: ['$0.02', '/ SMS Text Message', '/ Month'],
                                discount: '-$0.01',
                                discountOff: '50% off'
                            }
                        }
                    }]
                }],
                flatPrice: {
                    formattedPrice: '$2.00',
                    formattedFullPrice: '$2.00 / Month',
                    formattedFullPriceElements: ['$2.00', null, '/ Month'],
                    unit: {
                        min: 0,
                        max: null,
                        dependency: '',
                        id: 'NOT_APPLICABLE'
                    },
                    frequency: {
                        shortLabel: 'Mo',
                        label: 'Month',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: 'USD',
                        amount: 2,
                        value: 2,
                        originalValue: 2,
                        discountValue: 0,
                        editionStartingPrice: 2,
                        formatted: {
                            originalPrice: '$2.00',
                            originalFullPrice: '$2.00 / Month',
                            originalFullPriceElements: ['$2.00', null, '/ Month']
                        }
                    }
                }
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: 'ce14c14c-5fc5-4bf9-b3ed-1bfc5353341e'
        }],
        bullets: [{
            content: 'The very best of Deputy\'s scheduling features',
            tooltip: 'Offers optimised scheduling, Communication and Tasking'
        }],
        callToAction: {
            label: 'Contact Us',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            leadPageUrl: '/ad-lead',
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=18480&type=BUY&editionId=28155',
            actionType: 'LEAD_GEN',
            popup: {},
            isSecondary: true
        },
        audience: [{
            content: 'Small businesses'
        }],
        items: [{
            content: 'Unlimited Users'
        }],
        invisible: false,
        leadGen: false
    }, {
        name: 'Premium',
        id: 28158,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'EMPLOYEE',
                    usageRange: [{
                        formattedPrice: '$4.00',
                        formattedFullPrice: '$4.00 / Employee / Month',
                        formattedFullPriceElements: ['$4.00', '/ Employee', '/ Month'],
                        unit: {
                            min: 2,
                            max: null,
                            label: 'Employee',
                            dependency: '',
                            id: 'EMPLOYEE',
                            limitLabel: 'For 1 Employee And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 4,
                            value: 4,
                            originalValue: 4,
                            discountValue: 0,
                            editionStartingPrice: 4,
                            formatted: {
                                originalPrice: '$4.00',
                                originalFullPrice: '$4.00 / Employee / Month',
                                originalFullPriceElements: ['$4.00', '/ Employee', '/ Month'],
                                meteredUsage: 'For 1 Employee And Above'
                            }
                        }
                    }]
                }, {
                    id: 'SMS_TEXT_MESSAGE',
                    usageRange: [{
                        formattedPrice: '$0.01',
                        formattedFullPrice: '$0.01 / SMS Text Message / Month',
                        formattedFullPriceElements: ['$0.01', '/ SMS Text Message', '/ Month'],
                        unit: {
                            min: 0,
                            max: null,
                            label: 'SMS Text Message',
                            dependency: '',
                            id: 'SMS_TEXT_MESSAGE'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 0.01,
                            value: 0.01,
                            originalValue: 0.01,
                            discountValue: 0,
                            editionStartingPrice: 4,
                            formatted: {
                                originalPrice: '$0.01',
                                originalFullPrice: '$0.01 / SMS Text Message / Month',
                                originalFullPriceElements: ['$0.01', '/ SMS Text Message', '/ Month']
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: 'f109bab8-812b-4e14-ac70-cc07d1dbaa06'
        }, {
            costs: {
                unitPrice: [{
                    id: 'SERVER',
                    usageRange: [{
                        formattedPrice: '$134.10',
                        formattedFullPrice: '$134.10 / Server / Month',
                        formattedFullPriceElements: ['$134.10', '/ Server', '/ Month'],
                        unit: {
                            min: 2,
                            max: null,
                            label: 'Server',
                            dependency: '',
                            id: 'SERVER',
                            limitLabel: 'For 1 Server And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 149,
                            value: 134.1,
                            originalValue: 149,
                            discountValue: 14.9,
                            editionStartingPrice: 134.1,
                            formatted: {
                                originalPrice: '$149.00',
                                originalFullPrice: '$149.00 / Server / Month',
                                originalFullPriceElements: ['$149.00', '/ Server', '/ Month'],
                                discount: '-$14.90',
                                discountOff: '10% off',
                                meteredUsage: 'For 1 Server And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: 12,
            formattedContract: '12 month contract',
            discount: {},
            uuid: '7472c606-cccd-11e2-bff0-12313d1adc2b'
        }],
        bullets: [{
            content: 'The ultimate in staff management, with optimised scheduling, accurate time & attendance, effective team communication and tasking',
            tooltip: 'Give your team the very best tools with Deputy'
        }],
        callToAction: {
            label: 'Try For 30 Days',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=18480&type=FREE_TRIAL&editionId=28158',
            actionType: 'FREE_TRIAL',
            popup: {},
            isSecondary: true
        },
        audience: [{
            content: 'Most small and medium businesses'
        }],
        items: [{
            content: 'Unlimited Users'
        }],
        invisible: false,
        leadGen: false,
        freeTrialDuration: 3,
        freeTrialType: 'Day'
    }, {
        name: 'G Suite Business',
        id: 404636,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$15.60',
                        formattedFullPrice: '$15.60 / User / Month',
                        formattedFullPriceElements: ['$15.60', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: '$',
                            amount: 15.6,
                            value: 15.6,
                            originalValue: 15.6,
                            discountValue: 0,
                            editionStartingPrice: 15.6,
                            formatted: {
                                originalPrice: '$15.60',
                                originalFullPrice: '$15.60 / User / Month',
                                originalFullPriceElements: ['$15.60', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: 12,
            formattedContract: '12 month contract',
            discount: false,
            uuid: 'b0959bf4-b60e-4448-9fe1-9f98383f7f68'
        }],
        bullets: [{
            content: 'Unlimited cloud storage (or 1TB per user if fewer than 5 users)',
            tooltip: null
        }, {
            content: 'Smart search across G Suite with Cloud Search',
            tooltip: null
        }, {
            content: 'Archive and set retention policies for emails and chats',
            tooltip: null
        }, {
            content: 'eDiscovery for emails, chats, and files',
            tooltip: null
        }, {
            content: 'Audit reports to track user activity',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=577&type=BUY&editionId=404636',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [{
            content: 'Enhanced office suite with unlimited storage and archiving'
        }],
        items: [],
        invisible: false,
        leadGen: false
    }, {
        name: 'G Suite Enterprise - No Contract',
        id: 404640,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$34.00',
                        formattedFullPrice: '$34.00 / User / Month',
                        formattedFullPriceElements: ['$34.00', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: '$',
                            amount: 34,
                            value: 34,
                            originalValue: 34,
                            discountValue: 0,
                            editionStartingPrice: 34,
                            formatted: {
                                originalPrice: '$34.00',
                                originalFullPrice: '$34.00 / User / Month',
                                originalFullPriceElements: ['$34.00', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: '4dadecca-252d-4c2d-816f-3ba4ce1b378d'
        }],
        bullets: [{
            content: 'Unlimited cloud storage (or 1TB per user if fewer than 5 users)',
            tooltip: null
        }, {
            content: 'Smart search across G Suite with Cloud Search',
            tooltip: null
        }, {
            content: 'Archive and set retention policies for emails and chats',
            tooltip: null
        }, {
            content: 'eDiscovery for emails, chats, and files',
            tooltip: null
        }, {
            content: 'Audit reports to track user activity',
            tooltip: null
        }, {
            content: 'Data loss prevention for Gmail',
            tooltip: null
        }, {
            content: 'Data loss prevention for Drive',
            tooltip: null
        }, {
            content: 'Hosted S/MIME for Gmail',
            tooltip: null
        }, {
            content: 'Integrate Gmail with compliant third-party archiving tools',
            tooltip: null
        }, {
            content: 'Enterprise-grade access control with security key enforcement',
            tooltip: null
        }, {
            content: 'Gmail log analysis in BigQuery',
            tooltip: null
        }, {
            content: '*This is a monthly-billed plan, with no contract.',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=577&type=BUY&editionId=404640',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [{
            content: 'Premium office suite with advanced controls and capabilities'
        }],
        items: [],
        invisible: false,
        leadGen: false
    }, {
        name: 'Free',
        id: 83472,
        primary: false,
        plans: [
            {
                costs: {},
                setupFees: [],
                contractFees: [],
                contractLength: null,
                formattedContract: false,
                discount: false,
                uuid: '21534fba-fb72-4aee-9c9f-6cbca7d7c36d'
            }
        ],
        bullets: [
            {
                content: 'Full-text search',
                tooltip: 'Full-text search with OCR for scanned documents.'
            },
            {
                content: 'Unlimited accounts',
                tooltip: 'Connect any number of apps, including different accounts associated with same service'
            },
            {
                content: 'Unlimited content (last 30 days)',
                tooltip: null
            }
        ],
        callToAction: {
            label: 'Try the Free Edition',
            tooltip: '',
            isPrimary: false,
            eppid: '21534fba-fb72-4aee-9c9f-6cbca7d7c36d',
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=42937&type=FREE&editionId=83472&eppid=21534fba-fb72-4aee-9c9f-6cbca7d7c36d',
            actionType: 'FREE',
            popup: {},
            isSecondary: false
        },
        audience: [],
        items: [
            {
                content: 'Unlimited Users'
            }
        ],
        invisible: false,
        leadGen: false,
        revenueType: 'FREE'
    }, {
        name: 'G Suite Enterprise',
        id: 404644,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$34.00',
                        formattedFullPrice: '$34.00 / User / Month',
                        formattedFullPriceElements: ['$34.00', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: '$',
                            amount: 34,
                            value: 34,
                            originalValue: 34,
                            discountValue: 0,
                            editionStartingPrice: 34,
                            formatted: {
                                originalPrice: '$34.00',
                                originalFullPrice: '$34.00 / User / Month',
                                originalFullPriceElements: ['$34.00', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: 12,
            formattedContract: '12 month contract',
            discount: false,
            uuid: '8770026d-3388-4afd-90ab-04612cd6d536'
        }],
        bullets: [{
            content: 'Unlimited cloud storage (or 1TB per user if fewer than 5 users)',
            tooltip: null
        }, {
            content: 'Smart search across G Suite with Cloud Search',
            tooltip: null
        }, {
            content: 'Archive and set retention policies for emails and chats',
            tooltip: null
        }, {
            content: 'eDiscovery for emails, chats, and files',
            tooltip: null
        }, {
            content: 'Audit reports to track user activity',
            tooltip: null
        }, {
            content: 'Data loss prevention for Gmail',
            tooltip: null
        }, {
            content: 'Data loss prevention for Drive',
            tooltip: null
        }, {
            content: 'Hosted S/MIME for Gmail',
            tooltip: null
        }, {
            content: 'Integrate Gmail with compliant third-party archiving tools',
            tooltip: null
        }, {
            content: 'Enterprise-grade access control with security key enforcement',
            tooltip: null
        }, {
            content: 'Gmail log analysis in BigQuery',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=577&type=BUY&editionId=404644',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [{
            content: 'Premium office suite with advanced controls and capabilities'
        }],
        items: [],
        invisible: false,
        leadGen: false
    }, {
        name: 'Office 365 Enterprise E1',
        id: 1055,
        primary: false,
        plans: [{
            costs: {
                flatPrice: {
                    formattedPrice: 'Free',
                    formattedFullPrice: 'Free / User / Month',
                    formattedFullPriceElements: ['Free', '/ User', '/ Month'],
                    unit: {
                        min: 1,
                        max: null,
                        label: 'User',
                        dependency: '',
                        id: 'USER',
                        limitLabel: 'For 1 User And Above'
                    },
                    frequency: {
                        shortLabel: 'Mo',
                        label: 'Month',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: '$',
                        amount: 0,
                        value: 0,
                        originalValue: 0,
                        discountValue: 0,
                        editionStartingPrice: 0,
                        formatted: {
                            originalPrice: '$0.00',
                            originalFullPrice: '$0.00 / User / Month',
                            originalFullPriceElements: ['$0.00', '/ User', '/ Month'],
                            meteredUsage: 'For 1 User And Above'
                        }
                    }
                }
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: 'ccabaa5e-cd04-4037-8119-72441c8f4028'
        }],
        bullets: [{
            content: 'The Office suite for PC and Mac with apps for tablets and phones, plus email, instant messaging, HD video conferencing, 1 TB personal file storage and sharing, and available add-ons like PSTN calling.'
        }],
        callToAction: {
            isPrimary: false,
            isSecondary: true
        },
        audience: [],
        items: [],
        invisible: false,
        leadGen: false
    }, {
        name: 'Enterprise',
        id: 28164,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'EMPLOYEE',
                    usageRange: [{
                        formattedPrice: '$8.00',
                        formattedFullPrice: '$8.00 / Employee / Month',
                        formattedFullPriceElements: ['$8.00', '/ Employee', '/ Month'],
                        unit: {
                            min: 0,
                            max: null,
                            label: 'Employee',
                            dependency: '',
                            id: 'EMPLOYEE'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 8,
                            value: 8,
                            originalValue: 8,
                            discountValue: 0,
                            editionStartingPrice: 0,
                            formatted: {
                                originalPrice: '$8.00',
                                originalFullPrice: '$8.00 / Employee / Month',
                                originalFullPriceElements: ['$8.00', '/ Employee', '/ Month']
                            }
                        }
                    }]
                }, {
                    id: 'SMS_TEXT_MESSAGE',
                    usageRange: [{
                        formattedPrice: '$0.01',
                        formattedFullPrice: '$0.01 / SMS Text Message / Month',
                        formattedFullPriceElements: ['$0.01', '/ SMS Text Message', '/ Month'],
                        unit: {
                            min: 0,
                            max: 10,
                            label: 'Gigabyte',
                            dependency: '',
                            id: 'GIGABYTE',
                            limitLabel: 'For The First 10 Gigabytes'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 0.01,
                            value: 0.01,
                            originalValue: 0.01,
                            discountValue: 0,
                            editionStartingPrice: 0,
                            formatted: {
                                originalPrice: '$0.01',
                                originalFullPrice: '$0.01 / SMS Text Message / Month',
                                originalFullPriceElements: ['$0.01', '/ SMS Text Message', '/ Month']
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: 'adb05ae5-d975-4955-81b1-bc625e4d1246'
        }],
        bullets: [{
            content: 'All the best of Deputy Premium with extended reporting and customisation ability',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=18480&type=BUY&editionId=28164',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [{
            content: 'Suitable for large businesses and franchises'
        }],
        items: [{
            content: 'Unlimited Users'
        }],
        invisible: false,
        leadGen: false
    }, {
        name: 'Flexible Weekly',
        id: 28161,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'EMPLOYEE_PER_WEEK',
                    usageRange: [{
                        formattedPrice: '$1.50',
                        formattedFullPrice: '$1.50 / Employee per week / Month',
                        formattedFullPriceElements: ['$1.50', '/ Employee per week', '/ Month'
                        ],
                        unit: {
                            min: 0,
                            max: 2,
                            label: 'Employee per week',
                            dependency: '',
                            id: 'EMPLOYEE_PER_WEEK',
                            limitLabel: 'For 2 Employees per week And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 1.5,
                            value: 1.5,
                            originalValue: 1.5,
                            discountValue: 0,
                            editionStartingPrice: 1.5,
                            formatted: {
                                originalPrice: '$1.50',
                                originalFullPrice: '$1.50 / Employee per week / Month',
                                originalFullPriceElements: ['$1.50', '/ Employee per week', '/ Month'],
                                meteredUsage: 'For 1 Employee per week And Above'
                            }
                        }
                    }]
                }, {
                    id: 'SMS_TEXT_MESSAGE',
                    usageRange: [{
                        formattedPrice: '$0.01',
                        formattedFullPrice: '$0.01 / SMS Text Message / Month',
                        formattedFullPriceElements: ['$0.01', '/ SMS Text Message', '/ Month'],
                        unit: {
                            min: 0,
                            max: null,
                            label: 'SMS Text Message',
                            dependency: '',
                            id: 'SMS_TEXT_MESSAGE'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 0.01,
                            value: 0.01,
                            originalValue: 0.01,
                            discountValue: 0,
                            editionStartingPrice: 1.5,
                            formatted: {
                                originalPrice: '$0.01',
                                originalFullPrice: '$0.01 / SMS Text Message / Month',
                                originalFullPriceElements: ['$0.01', '/ SMS Text Message', '/ Month']
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: 'cb2f76a0-dae7-483a-be3f-8bbf37d7914a'
        }],
        bullets: [{
            content: 'Pay for active staff only',
            tooltip: 'Ideal for seasonal, catering and event companies'
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=18480&type=BUY&editionId=28161',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [{
            content: 'Seasonal, catering and event companies'
        }],
        items: [{
            content: 'Unlimited Users'
        }],
        invisible: false,
        leadGen: false
    }, {
        name: 'Office 365 Enterprise E4',
        id: 2108,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$10.00',
                        formattedFullPrice: '$10.00 / User / Month',
                        formattedFullPriceElements: ['$10.00', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: '$',
                            amount: 10,
                            value: 10,
                            originalValue: 10,
                            discountValue: 0,
                            editionStartingPrice: 10,
                            formatted: {
                                originalPrice: '$10.00',
                                originalFullPrice: '$10.00 / User / Month',
                                originalFullPriceElements: ['$10.00', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: '72f07fed-6888-4aac-82ee-2bf6c4fa47ee'
        }],
        bullets: [{
            content: 'The Office suite for PC and Mac with apps for tablets and phones, plus email, instant messaging, HD video conferencing, 1 TB personal file storage and sharing, and available add-ons like PSTN calling.'
        }],
        callToAction: {
            isPrimary: false,
            isSecondary: true
        },
        audience: [],
        items: [{
            content: '0 Users'
        }],
        invisible: false,
        leadGen: false
    }, {
        name: 'Office 365 Enterprise E3',
        id: 1056,
        primary: false,
        plans: [{
            costs: {
                flatPrice: {
                    formattedPrice: 'Free',
                    formattedFullPrice: 'Free / User / Month',
                    formattedFullPriceElements: ['Free', '/ User', '/ Month'],
                    unit: {
                        min: 1,
                        max: null,
                        label: 'User',
                        dependency: '',
                        id: 'USER',
                        limitLabel: 'For 1 User And Above'
                    },
                    frequency: {
                        shortLabel: 'Mo',
                        label: 'Month',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: '$',
                        amount: 0,
                        value: 0,
                        originalValue: 0,
                        discountValue: 0,
                        editionStartingPrice: 0,
                        formatted: {
                            originalPrice: '$0.00',
                            originalFullPrice: '$0.00 / User / Month',
                            originalFullPriceElements: ['$0.00', '/ User', '/ Month'],
                            meteredUsage: 'For 1 User And Above'
                        }
                    }
                }
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: '2c96a9cb-8608-4e05-bc11-0338eb5dbfb6'
        }],
        bullets: [{
            content: 'The Office suite for PC and Mac with apps for tablets and phones, plus email, instant messaging, HD video conferencing, 1 TB personal file storage and sharing, and available add-ons like PSTN calling.'
        }],
        callToAction: {
            isPrimary: false,
            isSecondary: true
        },
        audience: [],
        items: [],
        invisible: false,
        leadGen: false
    }, {
        name: 'Office 365 ProPlus',
        id: 1057,
        primary: false,
        plans: [{
            costs: {
                flatPrice: {
                    formattedPrice: 'Free',
                    formattedFullPrice: 'Free / User / Month',
                    formattedFullPriceElements: ['Free', '/ User', '/ Month'],
                    unit: {
                        min: 1,
                        max: null,
                        label: 'User',
                        dependency: '',
                        id: 'USER',
                        limitLabel: 'For 1 User And Above'
                    },
                    frequency: {
                        shortLabel: 'Mo',
                        label: 'Month',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: '$',
                        amount: 0,
                        value: 0,
                        originalValue: 0,
                        discountValue: 0,
                        editionStartingPrice: 0,
                        formatted: {
                            originalPrice: '$0.00',
                            originalFullPrice: '$0.00 / User / Month',
                            originalFullPriceElements: ['$0.00', '/ User', '/ Month'],
                            meteredUsage: 'For 1 User And Above'
                        }
                    }
                }
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: '189a3e6d-d436-47ea-b0e7-0d44402d5c53'
        }],
        bullets: [{
            content: 'The Office suite for PC and Mac with apps for tablets and phones, plus email, instant messaging, HD video conferencing, 1 TB personal file storage and sharing, and available add-ons like PSTN calling.'
        }],
        callToAction: {
            isPrimary: false,
            isSecondary: true
        },
        audience: [],
        items: [],
        invisible: false,
        leadGen: false
    }, {
        name: 'Pro (Annual)',
        id: 1699,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'SERVER',
                    usageRange: [{
                        formattedPrice: '$134.10',
                        formattedFullPrice: '$134.10 / Server / Month',
                        formattedFullPriceElements: ['$134.10', '/ Server', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'Server',
                            dependency: '',
                            id: 'SERVER',
                            limitLabel: 'For 1 Server And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 149,
                            value: 134.1,
                            originalValue: 149,
                            discountValue: 14.9,
                            editionStartingPrice: 134.1,
                            formatted: {
                                originalPrice: '$149.00',
                                originalFullPrice: '$149.00 / Server / Month',
                                originalFullPriceElements: ['$149.00', '/ Server', '/ Month'],
                                discount: '-$14.90',
                                discountOff: '10% off',
                                meteredUsage: 'For 1 Server And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: 12,
            formattedContract: '12 month contract',
            discount: {},
            uuid: '7472c606-cccd-11e2-bff0-12313d1adc2b'
        }],
        bullets: [{
            content: '$149/month/server (with annual commitment)',
            tooltip: null
        }, {
            content: '24x7 Transaction Analysis and Breakdown',
            tooltip: null
        }, {
            content: 'Real User Analysis and Breakdown',
            tooltip: null
        }, {
            content: '24x7 Server Monitoring',
            tooltip: null
        }, {
            content: 'Database Analysis and Breakdown',
            tooltip: null
        }, {
            content: 'Error Detection and Analysis',
            tooltip: null
        }, {
            content: 'Reporting and Optimization',
            tooltip: null
        }, {
            content: 'Proactive Alerting',
            tooltip: null
        }, {
            content: 'Data Retention: Unlimited',
            tooltip: null
        }, {
            content: 'Great volume discounts available with 5 or more servers. Call for custom quote 888-643-8776.',
            tooltip: null
        }],
        callToAction: {
            label: 'Download Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=1140&type=DOWNLOAD&editionId=1699',
            actionType: 'DOWNLOAD',
            popup: {},
            isSecondary: true
        },
        audience: [],
        items: [{
            content: 'Unlimited Users'
        }],
        invisible: false,
        leadGen: false
    }, {
        name: 'Pro (Monthly)',
        id: 1700,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'SERVER',
                    usageRange: [{
                        formattedPrice: '$179.10',
                        formattedFullPrice: '$179.10 / Server / Month',
                        formattedFullPriceElements: ['$179.10', '/ Server', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'Server',
                            dependency: '',
                            id: 'SERVER',
                            limitLabel: 'For 1 Server And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 199,
                            value: 179.1,
                            originalValue: 199,
                            discountValue: 19.900000000000002,
                            editionStartingPrice: 179.1,
                            formatted: {
                                originalPrice: '$199.00',
                                originalFullPrice: '$199.00 / Server / Month',
                                originalFullPriceElements: ['$199.00', '/ Server', '/ Month'],
                                discount: '-$19.90',
                                discountOff: '10% off',
                                meteredUsage: 'For 1 Server And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: {},
            uuid: '7472c8ae-cccd-11e2-bff0-12313d1adc2b'
        }],
        bullets: [{
            content: 'Month-to-Month: $199/server/month (no annual commitment)',
            tooltip: null
        }, {
            content: '24x7 Transaction Analysis and Breakdown',
            tooltip: null
        }, {
            content: 'Real User Analysis and Breakdown',
            tooltip: null
        }, {
            content: '24x7 Server Monitoring',
            tooltip: null
        }, {
            content: 'Database Analysis and Breakdown',
            tooltip: null
        }, {
            content: 'Error Detection and Analysis',
            tooltip: null
        }, {
            content: 'Reporting and Optimization',
            tooltip: null
        }, {
            content: 'Proactive Alerting',
            tooltip: null
        }, {
            content: 'Data Retention: Unlimited',
            tooltip: null
        }, {
            content: 'Great volume discounts available with 5 or more servers. Call for custom quote 888-643-8776.',
            tooltip: null
        }],
        callToAction: {
            label: 'Download Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=1140&type=DOWNLOAD&editionId=1700',
            actionType: 'DOWNLOAD',
            popup: {},
            isSecondary: true
        },
        audience: [],
        items: [{
            content: 'Unlimited Users'
        }],
        invisible: false,
        leadGen: false
    }, {
        name: 'Bronze',
        id: 2605,
        primary: false,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'EMAIL',
                    usageRange: [{
                        formattedPrice: '$0.00',
                        formattedFullPrice: '$0.00 / Email / Month',
                        formattedFullPriceElements: ['$0.00', '/ Email', '/ Month'],
                        unit: {
                            min: 0,
                            max: null,
                            label: 'Email',
                            dependency: '',
                            id: 'EMAIL'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 0.00025,
                            value: 0.00025,
                            originalValue: 0.00025,
                            discountValue: 0,
                            editionStartingPrice: 9.95,
                            formatted: {
                                originalPrice: '$0.00',
                                originalFullPrice: '$0.00 / Email / Month',
                                originalFullPriceElements: ['$0.00', '/ Email', '/ Month']
                            }
                        }
                    }]
                }, {
                    id: 'CONTACTS_1000',
                    usageRange: [{
                        formattedPrice: 'Free',
                        formattedFullPrice: 'Free / 1000 Contacts / Month',
                        formattedFullPriceElements: ['Free', '/ 1000 Contacts', '/ Month'],
                        unit: {
                            min: 0,
                            max: null,
                            label: '1000 Contacts',
                            dependency: '',
                            id: 'CONTACTS_1000'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 0,
                            value: 0,
                            originalValue: 0,
                            discountValue: 0,
                            editionStartingPrice: 9.95,
                            formatted: {
                                originalPrice: '$0.00',
                                originalFullPrice: '$0.00 / 1000 Contacts / Month',
                                originalFullPriceElements: ['$0.00', '/ 1000 Contacts', '/ Month']
                            }
                        }
                    }]
                }],
                flatPrice: {
                    formattedPrice: '$9.95',
                    formattedFullPrice: '$9.95 / Month',
                    formattedFullPriceElements: ['$9.95', null, '/ Month'],
                    unit: {
                        min: 0,
                        max: null,
                        dependency: '',
                        id: 'NOT_APPLICABLE'
                    },
                    frequency: {
                        shortLabel: 'Mo',
                        label: 'Month',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: 'USD',
                        amount: 9.95,
                        value: 9.95,
                        originalValue: 9.95,
                        discountValue: 0,
                        editionStartingPrice: 9.95,
                        formatted: {
                            originalPrice: '$9.95',
                            originalFullPrice: '$9.95 / Month',
                            originalFullPriceElements: ['$9.95', null, '/ Month']
                        }
                    }
                }
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: '74975318-cccd-11e2-bff0-12313d1adc2b'
        }],
        bullets: [{
            content: '40,000 Emails Credits/Month',
            tooltip: 'A credit is used each time we receive a "request" from your application or server to send an email. For example, if 1000 people register new accounts on your website, and you send a confirmation email to each new user, you\'ll expend 100 credits. Or, if you send a newsletter to a list of 5,000 customers you\'ll use 5,000 credits.'
        }, {
            content: '$ 0.00100 Per Email Thereafter',
            tooltip: 'The prices of Bronze, Silver, Gold and Platinum plans allow for a specific number of emil credits per month - indicated in the second row of this table. If you exceed the alloted number of credits on your plan, you will be charged on a per email basis at the rate indicated to the right. Notice that our most expensive per email rate is $1 per thousand emails - not bad, eh?!'
        }, {
            content: 'DKIM',
            tooltip: null
        }, {
            content: 'SPF',
            tooltip: null
        }, {
            content: 'Content Inspection',
            tooltip: null
        }, {
            content: 'Reputation Monitoring',
            tooltip: null
        }, {
            content: 'Spam Filter Testing',
            tooltip: null
        }, {
            content: 'ISP Monitoring',
            tooltip: null
        }, {
            content: 'Link Customization',
            tooltip: null
        }, {
            content: 'SMTP Relay',
            tooltip: null
        }, {
            content: 'SMTP API',
            tooltip: null
        }, {
            content: 'Web API',
            tooltip: null
        }, {
            content: 'Premium Phone Support',
            tooltip: null
        }, {
            content: 'Live Chat Support',
            tooltip: null
        }, {
            content: 'Email Support',
            tooltip: null
        }, {
            content: 'Online Support Portal',
            tooltip: null
        }, {
            content: 'Forums',
            tooltip: null
        }, {
            content: 'Documentation',
            tooltip: null
        }, {
            content: 'Knowledge Base',
            tooltip: null
        }, {
            content: 'Twitter',
            tooltip: null
        }, {
            content: 'Reporting/ Analytics',
            tooltip: null
        }, {
            content: 'Real Time Analytics Reporting',
            tooltip: null
        }, {
            content: 'Automated Email Reporting',
            tooltip: null
        }, {
            content: 'Requests',
            tooltip: null
        }, {
            content: 'Delivered',
            tooltip: null
        }, {
            content: 'Bounces',
            tooltip: null
        }, {
            content: 'Repeat Bounces',
            tooltip: null
        }, {
            content: 'Spam Reports',
            tooltip: null
        }, {
            content: 'Repeat Spam Reports',
            tooltip: null
        }, {
            content: 'Invalid Email Requests',
            tooltip: null
        }, {
            content: 'Blocks',
            tooltip: null
        }, {
            content: 'Email Activity',
            tooltip: null
        }, {
            content: 'Open Tracking',
            tooltip: null
        }, {
            content: 'Unique Open Tracking',
            tooltip: null
        }, {
            content: 'Click Tracking',
            tooltip: null
        }, {
            content: 'Unique Click Tracking',
            tooltip: null
        }, {
            content: 'Unsubscribe Tracking',
            tooltip: null
        }, {
            content: 'Sub-user Panel',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: true,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=1967&type=BUY&editionId=2605',
            actionType: 'BUY',
            popup: {},
            isSecondary: false
        },
        audience: [],
        items: [{
            content: '1 User'
        }],
        invisible: false,
        leadGen: false
    }, {
        name: 'Recurring Plan',
        id: 20294,
        primary: false,
        plans: [
            {
                costs: {
                    unitPrice: [
                        {
                            id: 'USER',
                            usageRange: [
                                {
                                    formattedPrice: '$1.00',
                                    formattedFullPrice: '$1.00 / User / Day',
                                    formattedFullPriceElements: [
                                        '$1.00',
                                        '/ User',
                                        '/ Day'
                                    ],
                                    unit: {
                                        min: 1,
                                        max: null,
                                        label: 'User',
                                        dependency: '',
                                        id: 'USER',
                                        limitLabel: 'For 1 User And Above'
                                    },
                                    frequency: {
                                        shortLabel: 'Day',
                                        label: 'Day',
                                        id: 'DAILY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 1,
                                        value: 1,
                                        originalValue: 1,
                                        discountValue: 0,
                                        editionStartingPrice: 2,
                                        formatted: {
                                            originalPrice: '$1.00',
                                            originalFullPrice: '$1.00 / User / Day',
                                            originalFullPriceElements: [
                                                '$1.00',
                                                '/ User',
                                                '/ Day'
                                            ],
                                            meteredUsage: 'For 1 User And Above'
                                        }
                                    }
                                }
                            ]
                        },
                        {
                            id: 'GIGABYTE',
                            usageRange: [
                                {
                                    formattedPrice: '$1.00',
                                    formattedFullPrice: '$1.00 / Gigabyte / Day',
                                    formattedFullPriceElements: [
                                        '$1.00',
                                        '/ Gigabyte',
                                        '/ Day'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        label: 'Gigabyte',
                                        dependency: '',
                                        id: 'GIGABYTE'
                                    },
                                    frequency: {
                                        shortLabel: 'Day',
                                        label: 'Day',
                                        id: 'DAILY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 1,
                                        value: 1,
                                        originalValue: 1,
                                        discountValue: 0,
                                        editionStartingPrice: 2,
                                        formatted: {
                                            originalPrice: '$1.00',
                                            originalFullPrice: '$1.00 / Gigabyte / Day',
                                            originalFullPriceElements: [
                                                '$1.00',
                                                '/ Gigabyte',
                                                '/ Day'
                                            ]
                                        }
                                    }
                                }
                            ]
                        }
                    ],
                    flatPrice: {
                        formattedPrice: '-$1.00',
                        formattedFullPrice: '-$1.00 / Day',
                        formattedFullPriceElements: [
                            '-$1.00',
                            null,
                            '/ Day'
                        ],
                        unit: {
                            min: 0,
                            max: null,
                            dependency: '',
                            id: 'NOT_APPLICABLE'
                        },
                        frequency: {
                            shortLabel: 'Day',
                            label: 'Day',
                            id: 'DAILY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 1,
                            value: -1,
                            originalValue: 1,
                            discountValue: 2,
                            editionStartingPrice: 0,
                            formatted: {
                                originalPrice: '$1.00',
                                originalFullPrice: '$1.00 / Day',
                                originalFullPriceElements: [
                                    '$1.00',
                                    null,
                                    '/ Day'
                                ],
                                discount: '-$2.00',
                                discountOff: '$2.00 off'
                            }
                        }
                    }
                },
                setupFees: [
                    {
                        formattedPrice: '$8.00',
                        formattedFullPrice: '$8.00 Setup fee',
                        formattedFullPriceElements: [
                            '$8.00',
                            'Setup fee',
                            ''
                        ],
                        unit: {
                            min: 0,
                            max: null,
                            label: 'Setup fee',
                            dependency: '',
                            id: 'ONE_TIME_SETUP'
                        },
                        frequency: {
                            shortLabel: '',
                            label: '',
                            id: 'DAILY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 10,
                            value: 8,
                            originalValue: 10,
                            discountValue: 2,
                            editionStartingPrice: 0,
                            formatted: {
                                originalPrice: '$10.00',
                                originalFullPrice: '$10.00 Setup fee',
                                originalFullPriceElements: [
                                    '$10.00',
                                    'Setup fee',
                                    ''
                                ],
                                discount: '-$2.00',
                                discountOff: '$2.00 off'
                            }
                        }
                    }
                ],
                contractFees: [],
                contractLength: 2,
                formattedContract: '2 day contract',
                discount: {
                    description: 'test',
                    percentage: null,
                    startDate: null,
                    expirationDate: null,
                    numOfBillingCycles: null,
                    availableRedemptions: null,
                    amount: {
                        USD: 2
                    },
                    pricingUnit: null
                },
                uuid: '60ff7d58-c8b8-11e2-8fe5-22000a1d96ab'
            }
        ],
        bullets: [
            {
                content: 'Bullet point 1',
                tooltip: 'Bullet point 1'
            },
            {
                content: 'Bullet point 2',
                tooltip: 'Bullet point 2'
            }
        ],
        callToAction: {
            label: 'Select',
            tooltip: '',
            isPrimary: false,
            eppid: '60ff7d58-c8b8-11e2-8fe5-22000a1d96ab',
            enabled: true,
            url: 'https://testmarketplace.appdirect.com/api/internal/storefront/v1/cta?productId=15861&type=CONFIGURE&editionId=20294&eppid=60ff7d58-c8b8-11e2-8fe5-22000a1d96ab',
            actionType: 'CONFIGURE',
            popup: {},
            isSecondary: true
        },
        audience: [],
        items: [
            {
                content: '0 Users'
            }
        ],
        invisible: false,
        leadGen: false,
        revenueType: 'RECURRING'
    },
    {
        name: '1 month minimum contract',
        id: 44587,
        primary: false,
        plans: [
            {
                costs: {
                    unitPrice: [],
                    flatPrice: {
                        formattedPrice: '$8.00',
                        formattedFullPrice: '$8.00 / Month',
                        formattedFullPriceElements: [
                            '$8.00',
                            null,
                            '/ Month'
                        ],
                        unit: {
                            min: 0,
                            max: null,
                            dependency: '',
                            id: 'NOT_APPLICABLE'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 10,
                            value: 8,
                            originalValue: 10,
                            discountValue: 2,
                            editionStartingPrice: 8,
                            formatted: {
                                originalPrice: '$10.00',
                                originalFullPrice: '$10.00 / Month',
                                originalFullPriceElements: [
                                    '$10.00',
                                    null,
                                    '/ Month'
                                ],
                                discount: '-$2.00',
                                discountOff: '$2.00 off'
                            }
                        }
                    }
                },
                setupFees: [
                    {
                        formattedPrice: '$3.00',
                        formattedFullPrice: '$3.00 Setup fee',
                        formattedFullPriceElements: [
                            '$3.00',
                            'Setup fee',
                            ''
                        ],
                        unit: {
                            min: 0,
                            max: null,
                            label: 'Setup fee',
                            dependency: '',
                            id: 'ONE_TIME_SETUP'
                        },
                        frequency: {
                            shortLabel: '',
                            label: '',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 5,
                            value: 3,
                            originalValue: 5,
                            discountValue: 2,
                            editionStartingPrice: 8,
                            formatted: {
                                originalPrice: '$5.00',
                                originalFullPrice: '$5.00 Setup fee',
                                originalFullPriceElements: [
                                    '$5.00',
                                    'Setup fee',
                                    ''
                                ],
                                discount: '-$2.00',
                                discountOff: '$2.00 off'
                            }
                        }
                    }
                ],
                contractFees: [
                    {
                        formattedPrice: '$3.00',
                        formattedFullPrice: '$3.00 Contract Fee',
                        formattedFullPriceElements: [
                            '$3.00',
                            'Contract Fee',
                            ''
                        ],
                        unit: {
                            min: 0,
                            max: null,
                            label: 'Contract Fee',
                            dependency: '',
                            id: 'CONTRACT_FEE'
                        },
                        frequency: {
                            shortLabel: '',
                            label: '',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 5,
                            value: 3,
                            originalValue: 5,
                            discountValue: 2,
                            editionStartingPrice: 8,
                            formatted: {
                                originalPrice: '$5.00',
                                originalFullPrice: '$5.00 Contract Fee',
                                originalFullPriceElements: [
                                    '$5.00',
                                    'Contract Fee',
                                    ''
                                ],
                                discount: '-$2.00',
                                discountOff: '$2.00 off'
                            }
                        }
                    }
                ],
                contractLength: 1,
                formattedContract: '1 month contract',
                discount: {
                    description: 'test',
                    percentage: null,
                    startDate: null,
                    expirationDate: null,
                    numOfBillingCycles: null,
                    availableRedemptions: null,
                    amount: {
                        USD: 2
                    },
                    pricingUnit: null
                },
                uuid: 'e9cae023-11b5-45d6-b860-9253257727e7'
            }
        ],
        bullets: [],
        callToAction: {
            label: 'Select',
            tooltip: '',
            isPrimary: false,
            eppid: 'e9cae023-11b5-45d6-b860-9253257727e7',
            enabled: true,
            url: 'https://testmarketplace.appdirect.com/api/internal/storefront/v1/cta?productId=15861&type=CONFIGURE&editionId=44587&eppid=e9cae023-11b5-45d6-b860-9253257727e7',
            actionType: 'CONFIGURE',
            popup: {},
            isSecondary: true
        },
        audience: [],
        items: [
            {
                content: '1 User'
            }
        ],
        invisible: false,
        leadGen: false,
        revenueType: 'RECURRING'
    }],
    footnotes: [
        'Active employees are those who are working for you. In our Flexi plan, you do not pay for for inactive employees. We deem an employee active on the following occurrences taking place: A shift has been published for the employee. A timesheet has been submitted for the employee. The employee has performed management functions such as Rostering, Managing Timesheets, Exporting Payroll information, Managing Tasks, Creating Journal Entries or Reviewing Reports.',
        'Terminated/Discarded employees are not charged in any of the plans. Discarded staff cannot be rostered or have any interaction in Deputy. However, their history is retained and you can re-instate them at any time.'
    ],
    i18n: {
        bulletsHeader: 'All Plans Include',
        bulletsLabel: 'Plan\'s Features',
        inclusions: 'Included',
        audience: 'Ideal For',
        footnotes: 'Notes',
        invisible: 'Hidden from store customers',
        tierSeparator: 'or',
        setupFee: 'Setup Fee',
        mostPopular: 'Most Popular',
        availablePricing: 'Available Pricing:',
        select: 'Select',
        selected: 'Selected',
        viewDetails: 'View Details',
        hideDetails: 'Hide Details',
        selectEditionTitle: 'Select an edition for this product',
        showMoreDetails: 'Show More Info',
        hideMoreDetails: 'Hide',
        freeTrial: 'free trial',
        viewAddons: 'View Add-ons',
        additionalProductsRequired: 'Additional products required:',
        additionalProductRequired: 'Additional product required:'
    }
};

export const editionsSkipConfiguratorMockData = [
    {
        name: 'Edition 1',
        id: 12345,
        primary: true,
        callToAction: editionAddToCartSkipConfigurator1,
        plans: monthlyPlanUsd
    },
    {
        name: 'Edition 2',
        id: 12346,
        primary: false,
        callToAction: editionAddToCartSkipConfigurator2,
        plans: monthlyPlanUsd
    }
];

export const editionSelectMockData = [
    {
        name: 'Edition 1',
        id: 12345,
        primary: true,
        callToAction: editionSelect,
        plans: monthlyPlanUsd
    }
];

export const productEditionListMockData = {
    editions: editionsMockData.items,
    i18n: editionsMockData.i18n,
    callToActions: {
        items: [{
            label: 'Buy Now'
        }]
    }
};

export const productEditionListSkipConfiguratorMockData = {
    editions: editionsSkipConfiguratorMockData,
    i18n: editionsMockData.i18n,
    callToActions: {
        items: [{
            label: 'Buy Now'
        }]
    }
};

export const productEditionsNonStackableNoAddons = {
    bullets: [],
    items: [
        {
            name: 'Recurring Edition',
            id: 593051,
            productEditionRefId: '5ef8a923-2588-49d6-9994-1440a64fc4f4',
            primary: true,
            plans: [
                {
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: '23945eae-40fe-4140-ac71-3f8fd575b10c'
                }
            ],
            bullets: [],
            callToAction: {
                label: 'Add to Cart',
                tooltip: '',
                isPrimary: true,
                eppid: '23945eae-40fe-4140-ac71-3f8fd575b10c',
                enabled: true,
                url: 'https://engage19billing.test.devappdirect.me/api/internal/storefront/v1/cta?productId=322076&type=ADD_TO_CART&editionId=593051&eppid=23945eae-40fe-4140-ac71-3f8fd575b10c',
                actionType: 'ADD_TO_CART',
                popup: {},
                isSecondary: false
            },
            audience: [],
            items: [
                {
                    content: '0 Users'
                }
            ],
            invisible: false,
            leadGen: false,
            revenueType: 'RECURRING'
        },
        {
            name: 'Edition 3',
            id: 593057,
            productEditionRefId: '1e2c4810-9a0a-4387-9bab-4cc4dc0598e6',
            primary: false,
            plans: [
                {
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: '7502c355-00d2-4f3c-83bb-aab503edf08b'
                }
            ],
            bullets: [],
            callToAction: {
                label: 'Add to Cart',
                tooltip: '',
                isPrimary: false,
                eppid: '7502c355-00d2-4f3c-83bb-aab503edf08b',
                enabled: true,
                url: 'https://engage19billing.test.devappdirect.me/api/internal/storefront/v1/cta?productId=322076&type=ADD_TO_CART&editionId=593057&eppid=7502c355-00d2-4f3c-83bb-aab503edf08b',
                actionType: 'ADD_TO_CART',
                popup: {},
                isSecondary: true
            },
            audience: [],
            items: [
                {
                    content: '1 User'
                }
            ],
            invisible: false,
            leadGen: false,
            revenueType: 'RECURRING'
        },
        {
            name: 'Edition 2',
            id: 593054,
            productEditionRefId: '2594a703-427c-46f0-9ed8-23aa88712de5',
            primary: false,
            plans: [
                {
                    costs: {
                        unitPrice: [],
                        flatPrice: {
                            formattedPrice: '$20.00',
                            formattedFullPrice: '$20.00 / Month',
                            formattedFullPriceElements: [
                                '$20.00',
                                null,
                                '/ Month'
                            ],
                            unit: {
                                min: 0,
                                max: null,
                                dependency: '',
                                id: 'NOT_APPLICABLE'
                            },
                            frequency: {
                                shortLabel: 'Mo',
                                label: 'Month',
                                id: 'MONTHLY'
                            },
                            price: {
                                symbol: '$',
                                currency: 'USD',
                                amount: 20,
                                value: 20,
                                originalValue: 20,
                                discountValue: 0,
                                editionStartingPrice: 20,
                                formatted: {
                                    originalPrice: '$20.00',
                                    originalFullPrice: '$20.00 / Month',
                                    originalFullPriceElements: [
                                        '$20.00',
                                        null,
                                        '/ Month'
                                    ]
                                }
                            }
                        }
                    },
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: 'f2dcc6a4-56e9-45da-bc29-da8189ed040a'
                }
            ],
            bullets: [],
            callToAction: {
                label: 'Add to Cart',
                tooltip: '',
                isPrimary: false,
                eppid: 'f2dcc6a4-56e9-45da-bc29-da8189ed040a',
                enabled: true,
                url: 'https://engage19billing.test.devappdirect.me/api/internal/storefront/v1/cta?productId=322076&type=ADD_TO_CART&editionId=593054&eppid=f2dcc6a4-56e9-45da-bc29-da8189ed040a',
                actionType: 'ADD_TO_CART',
                popup: {},
                isSecondary: true
            },
            audience: [],
            items: [
                {
                    content: '1 User'
                }
            ],
            invisible: false,
            leadGen: false,
            revenueType: 'RECURRING'
        }
    ],
    footnotes: []
};

export const productEditionsNonStackableWithAddons = {
    items: [
        {
            name: 'Edition 2',
            id: 555486,
            productEditionRefId: '1c1eb1c1-cfc3-4e84-b3cf-0c5b2682c18a',
            primary: false,
            plans: [
                {
                    ...monthlyPlanUsd,
                    uuid: '23f104a9-12a6-4656-a35c-cfdf44555883'
                }
            ],
            bullets: [],
            callToAction: {
                ...editionSelect,
                isPrimary: false,
                isSecondary: true,
                eppid: '23f104a9-12a6-4656-a35c-cfdf44555883',
                url: 'https://engage19billing.test.devappdirect.me/api/internal/storefront/v1/cta?productId=300570&type=CONFIGURE&editionId=555486&eppid=23f104a9-12a6-4656-a35c-cfdf44555883'
            },
            audience: [],
            items: [
                {
                    content: '1 User'
                }
            ],
            invisible: false,
            leadGen: false,
            revenueType: 'RECURRING'
        },
        {
            name: 'Edition 1',
            id: 555483,
            productEditionRefId: '2946b0a7-7284-4bf2-8d35-8e425248562b',
            primary: true,
            plans: [
                {
                    ...monthlyPlanUsd,
                    uuid: '76068349-1098-4904-a91e-a038f8b3a9a0'
                }
            ],
            bullets: [],
            callToAction: {
                ...editionSelect,
                eppid: '76068349-1098-4904-a91e-a038f8b3a9a0',
                url: 'https://engage19billing.test.devappdirect.me/api/internal/storefront/v1/cta?productId=300570&type=CONFIGURE&editionId=555483&eppid=76068349-1098-4904-a91e-a038f8b3a9a0'
            },
            audience: [],
            items: [
                {
                    content: '0 Users'
                }
            ],
            invisible: false,
            leadGen: false,
            revenueType: 'RECURRING'
        }
    ]
};
