// @appdirect/sfb-theme-components/mocks/product/questions.js
import { productCardMock } from './productCard';

export const questionsMock = {
    summary: productCardMock,
    isLoggedIn: true,
    questions: {
        actionEndpoint: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions',
        questionFormReturnURL: '',
        addQuestionURL: '/en-US/apps/358695/gtreasury/add-question',
        items: [
            {
                id: 68,
                title: 'What is there in this system as far as inventory control?',
                comment: 'What is there in this system as far as inventory control. It says you can do PO\'s, we are using QB and i would love to go cloud based but I really need all the features of QB. Would really like to know what the capabilities of inventory and price lists this system has.',
                author: {
                    uuid: '5b94f0ee-a637-4ef7-9289-8defc3113bb9',
                    name: 'Bryon Holmes',
                    profilePicUrl: null
                },
                lastModified: 1315855986000,
                answerCount: 0,
                byDate: 'by Bryon Holmes on September 12th 2011',
                isCurrentUserQuestion: false,
                answers: [],
                actionEndpoints: {
                    entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/71',
                    comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/71/answers'
                },
                i18n: {
                    showAnswer: 'Show 1 Answer',
                    hideAnswer: 'Hide 1 Answer'
                }
            },
            {
                id: 69,
                title: 'What is there in this system as far as inventory control?',
                comment: 'What is there in this system as far as inventory control. It says you can do PO\'s, we are using QB and i would love to go cloud based but I really need all the features of QB. Would really like to know what the capabilities of inventory and price lists this system has.',
                author: {
                    uuid: '5b94f0ee-a637-4ef7-9289-8defc3113bb9',
                    name: 'Bryon Holmes',
                    profilePicUrl: null
                },
                lastModified: 1315855986000,
                answerCount: 0,
                byDate: 'by Bryon Holmes on September 12th 2011',
                isCurrentUserQuestion: false,
                answers: [],
                actionEndpoints: {
                    entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/71',
                    comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/71/answers'
                },
                i18n: {
                    showAnswer: 'Show 1 Answer',
                    hideAnswer: 'Hide 1 Answer'
                }
            },
            {
                id: 71,
                title: 'Inventory',
                comment: 'What is there in this system as far as inventory control. It says you can do PO\'s, we are using QB and i would love to go cloud based but I really need all the features of QB. Would really like to know what the capabilities of inventory and price lists this system has.',
                author: {
                    uuid: '5b94f0ee-a637-4ef7-9289-8defc3113bb9',
                    name: 'Bryon Holmes',
                    profilePicUrl: null
                },
                lastModified: 1315855986000,
                answerCount: 0,
                byDate: 'by Bryon Holmes on September 12th 2011',
                isCurrentUserQuestion: false,
                answers: [],
                actionEndpoints: {
                    entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/71',
                    comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/71/answers'
                },
                i18n: {
                    showAnswer: 'Show 1 Answer',
                    hideAnswer: 'Hide 1 Answer'
                }
            },
            {
                id: 66,
                title: 'Yendo link to my bank account?',
                comment: 'Does Yendo link to my bank account to automatically populate payments recieved into Yendo or do I have to upload a file from my bank into Yendo to get a view of my financials?',
                author: {
                    uuid: '4cd702ba-8836-4e68-970a-97796e856554',
                    name: 'Jason Ortepi',
                    profilePicUrl: 'https://d3bql97l1ytoxn.cloudfront.net/profilePics/img3155664473746228434.png?21d8e9b7af8252e5baebd7387c767433'
                },
                lastModified: 1305726632000,
                answerCount: 0,
                byDate: 'by Jason Ortepi on May 18th 2011',
                isCurrentUserQuestion: false,
                answers: [],
                actionEndpoints: {
                    entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/66',
                    comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/66/answers'
                },
                i18n: {
                    showAnswer: 'Show 1 Answer',
                    hideAnswer: 'Hide 1 Answer'
                }
            },
            {
                id: 65,
                title: 'Muti-company and Multi-Currency',
                comment: 'We\'re a small startup with offices in the US and Argentina. Can Yendo manage different companies, FS integration and different currencies?',
                author: {
                    uuid: 'a45289f2-a7f9-4678-8638-b416bc18ddfa',
                    name: 'Hernan Haro',
                    profilePicUrl: null
                },
                lastModified: 1303301435000,
                answerCount: 3,
                byDate: 'by Hernan Haro on April 20th 2011',
                isCurrentUserQuestion: false,
                answers: [
                    {
                        id: 59,
                        comment: 'At present we don not support multi-company or multi-currency logins. This will be changing soon as we are planning to add it in the next few months.',
                        author: {
                            uuid: 'aaca95b5-e7b5-4bdc-b611-191914bb1aa6',
                            name: 'Yendo Admin',
                            profilePicUrl: null
                        },
                        lastModified: 1305659263000,
                        official: false,
                        isCurrentUserAnswer: false,
                        actionEndpoint: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/65/answers/59',
                        i18n: {
                            byDate: 'by Yendo Admin on May 17th 2011'
                        }
                    },
                    {
                        id: 60,
                        comment: 'At present we don not support multi-company or multi-currency logins. This will be changing soon as we are planning to add it in the next few months.',
                        author: {
                            uuid: 'aaca95b5-e7b5-4bdc-b611-191914bb1aa6',
                            name: 'Yendo Admin',
                            profilePicUrl: null
                        },
                        lastModified: 1305659263000,
                        official: false,
                        isCurrentUserAnswer: false,
                        actionEndpoint: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/65/answers/59',
                        i18n: {
                            byDate: 'by Yendo Admin on May 17th 2011'
                        }
                    },
                    {
                        id: 61,
                        comment: 'At present we don not support multi-company or multi-currency logins. This will be changing soon as we are planning to add it in the next few months. At present we don not support multi-company or multi-currency logins. This will be changing soon as we are planning to add it in the next few months. At present we don not support multi-company or multi-currency logins. This will be changing soon as we are planning to add it in the next few months. At present we don not support multi-company or multi-currency logins. This will be changing soon as we are planning to add it in the next few months.',
                        author: {
                            uuid: 'aaca95b5-e7b5-4bdc-b611-191914bb1aa6',
                            name: 'Yendo Admin',
                            profilePicUrl: null
                        },
                        lastModified: 1305659263000,
                        official: false,
                        isCurrentUserAnswer: false,
                        actionEndpoint: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/65/answers/59',
                        i18n: {
                            byDate: 'by Yendo Admin on May 17th 2011'
                        }
                    }
                ],
                actionEndpoints: {
                    entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/65',
                    comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/65/answers'
                },
                i18n: {
                    showAnswer: 'Show 3 Answers',
                    hideAnswer: 'Hide 3 Answers'
                }
            },
            {
                id: 52,
                title: 'GST/HST Reporting in Yendo',
                comment: 'Is Yendo capable of providing GST/HST reports for CRA? Ideally this information would be similar to that shown on a GST/HST remittance form submitted to Canada Revenue Agency.',
                author: {
                    uuid: '92329407-f07d-4d57-9944-26667ad990ce',
                    name: 'Earl Rudolfo',
                    profilePicUrl: null
                },
                lastModified: 1297874252000,
                answerCount: 1,
                byDate: 'by Earl Rudolfo on February 16th 2011',
                isCurrentUserQuestion: false,
                answers: [
                    {
                        id: 60,
                        comment: 'Yendo provides all the information you need to complete your GST/HST return (GST34). There is a general tax report that will provide the amounts for sales/purchases and the amounts allocated to each tax band.\nThis should be everything you need to complete the tax return.',
                        author: {
                            uuid: 'aaca95b5-e7b5-4bdc-b611-191914bb1aa6',
                            name: 'Yendo Admin',
                            profilePicUrl: null
                        },
                        lastModified: 1305659683000,
                        official: false,
                        isCurrentUserAnswer: false,
                        actionEndpoint: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/52/answers/60',
                        i18n: {
                            byDate: 'by Yendo Admin on May 17th 2011'
                        }
                    }
                ],
                actionEndpoints: {
                    entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/52',
                    comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/questions/52/answers'
                },
                i18n: {
                    showAnswer: 'Show 1 Answer',
                    hideAnswer: 'Hide 1 Answer'
                }
            }
        ],
        i18n: {
            headers: {
                noQuestion: 'No questions have been asked.',
                whatWouldYou: 'What would you like to know about this product?',
                nbQuestions: '4 Questions',
                questions: 'Questions'
            },
            labels: {
                search: 'Search',
                askQuestion: 'Ask a question',
                developerAnswer: 'Developer Answer',
                answer: 'Answer',
                close: 'Close the ask a question form',
                edit: 'Edit',
                delete: 'Delete',
                cancel: 'Cancel',
                confirm: 'Confirm',
                resultsStr: '4 Results',
                question: 'Question',
                questions: 'Questions',
                description: 'Description',
                submit: 'Submit',
                signUp: 'Sign Up to Ask a Question',
                submitComment: 'Submit Answer',
                writeYourAnswerHere: 'Write your answer here',
                answerQuestion: 'Answer Question',
                displayQuestions: 'Display all questions',
                hideQuestions: 'Hide questions',
                goToAskAQuestion: 'Go to "ask a question" form',
                for: 'For',
                submitQuestion: 'Submit Question'
            },
            messages: {
                confirm: {
                    deleteEntry: 'Please confirm deleting this question.',
                    deleteComment: 'Please confirm deleting this answer.'
                },
                success: {
                    deleteEntry: 'Question successfully deleted',
                    deleteComment: 'Answer successfully deleted',
                    postEntry: 'Thank you for asking a question!  It will be posted shortly.',
                    postComment: 'Thank you for your answer!  It will be posted shortly.'
                },
                error: {
                    deleteEntry: 'Unable to delete question.',
                    deleteComment: 'Unable to delete answer.',
                    postEntry: 'You must log in to ask a question.',
                    postComment: 'You must log in to ask a question.',
                    textareaMaxLength: 'Must be at most 2048 characters',
                    fieldRequired: 'Required'
                }
            }
        }
    }
};
