// @appdirect/sfb-theme-components/mocks/product/reviews.js
export const structuredReviewMock = {
    id: '1',
    rating: 4,
    title: 'Microsoft 365 is a great tool for any office environment',
    sections: [
        {
            title: 'What do you like best?',
            content: 'I love how easily accessible Microsoft 365 is. It can be accessed anywhere on the go, and on any platform. It is extremely easy to share documents among coworkers, where we are all able to edit it in real time.'
        },
        {
            title: 'What do you dislike?',
            content: 'I prefer the layout of the original Microsoft Office. Microsoft 365 doesn\'t include many of the features and tools included in the original Microsoft Office. It also can be quite a bit harder to create and edit documents. I used this for a REST interface on my SQL DB, it creates all the services for you, so I was using stored procedures. You can do user management and role based access, there is a way to do record level access, looking into that. No problems so far its great. Microsoft 365 has helped my department stay better connected with one another, and given us the ability to easily share documents and reports amongst one another. Microsoft 365 helps to keep all of our Microsoft applications accessible in one place, which in turn helps save time and energy. It does a great job of keeping records of who last edited a document including time stamps. I just recently learned about this platform, but it wasn\'t until I downloaded the free trial and looked into it that I started to understand everything this platform does.'
        },
        {
            title: 'What problems are you solving with the product? What benefits have you realized?',
            content: 'Microsoft 365 has helped my department stay better connected with one another, and given us the ability to easily share documents and reports amongst one another. Microsoft 365 helps to keep all of our Microsoft applications accessible in one place, which in turn helps save time and energy. It does a great job of keeping records of who last edited a document including time stamps.'
        }
    ],
    reviewAuthor: {
        name: 'Amanda Smith',
        title: 'Financial Services',
        company: {
            name: 'ACME Inc.',
            type: 'Education'
        },
        avatarUrl: 'https://images.g2crowd.com/uploads/avatar/image/211817/thumb_square_efaf01e9f1fa302cf1e5e4bcfb776cb0.jpeg',
        validatedReviewer: true
    },
    reviewTags: [
        {
            type: 'VERIFIED_CURRENT_USER',
            label: 'Verified Current User',
            description: 'Verified Current User'
        },
        {
            type: 'SYNDICATOR_REVIEW_SOURCE',
            label: 'Validated User',
            description: 'Validated User'
        },
        {
            type: 'FREE_FORM',
            label: 'Free Form Text',
            description: 'Free Form Description'
        }
    ],
    reviewUrl: 'hhttps://www.g2.com/products/office-365/reviews',
    syndication: {
        externalProvider: 'G2'
    },
    lastModified: 1433347864000,
    isCurrentUserReview: false,
    actionEndpoints: {
        entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2992',
        comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2992/comments'
    },
    i18n: {
        byDate: 'by Richter33552 Richter33552 on June 3rd 2015',
        date: 'June 3, 2015',
        showCommentCount: 'Show 0 Comments',
        hideCommentCount: 'Hide 0 Comments',
        validatedUserTag: 'Validated Reviewer'
    },
    reviewLastModified: '2020-10-27T01:57:08.986Z'
};

export const genericReviewMock = {
    id: '2992',
    rating: 5,
    title: 'Hook up your SQL DB no problems',
    sections: [
        {
            title: 'What do you like best?',
            content: 'Hook up your SQL DB no problems.'
        }
    ],
    content: 'I used this for a REST interface on my SQL DB, it creates all the services for you, so I was using stored procedures. You can do user management and role based access, there is a way to do record level access, looking into that. No problems so far its great. Microsoft 365 has helped my department stay better connected with one another, and given us the ability to easily share documents and reports amongst one another. Microsoft 365 helps to keep all of our Microsoft applications accessible in one place, which in turn helps save time and energy. It does a great job of keeping records of who last edited a document including time stamps. I just recently learned about this platform, but it wasn\'t until I downloaded the free trial and looked into it that I started to understand everything this platform does. The bottom line is it\'s saving me a ton of time, and I can connect to a db with a couple of clicks. It\'s one of those rare times where it really isn\'t all marketing hype, the darned thing really is that good. I\'d honestly give their support group more than five stars if that were possible. We were one of the early adopters of the DreamFactory Services Platform after a bunch of discussions with our consultants. After considering a bunch of other options, we finally agreed on DSP primarily because of security. We do not need to trust an additional cloud provider - I was able to install on Amazon and connect my Azure account easily. We have not been disappointed. They have listened to our feedback and given us most of what we have asked for. Support is top notch and speedy.\n\nThe roadmap looks promising. My team cannot wait for the next version to come out to integrate and release our app update. \n\nA big thank you to the DreamFactory team and good luck on the next version.',
    reviewAuthor: {
        name: 'Richard Tremblay',
        avatarUrl: 'https://images.g2crowd.com/uploads/avatar/image/229471/thumb_square_bf15358a6c620c2f00fdeb88e3021959.jpeg'
    },
    lastModified: 1433347864000,
    comments: [],
    actionEndpoints: {
        entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2992',
        comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2992/comments'
    },
    i18n: {
        byDate: 'by Richard Tremblay on June 3rd 2015',
        date: 'June 3, 2015',
        showCommentCount: 'Show 0 Comments',
        hideCommentCount: 'Hide 0 Comments'
    },
    reviewLastModified: '2020-10-27T01:57:08.986Z'
};

export const reviewsMock = {
    // this is a mock invented because there was no available data in toolkit
    ratings: [
        {
            rating: 5,
            text: '5 star',
            count: 221
        },
        {
            rating: 4,
            text: '4 star',
            count: 143
        },
        {
            rating: 3,
            text: '3 star',
            count: 387
        },
        {
            rating: 2,
            text: '2 star',
            count: 76
        },
        {
            rating: 1,
            text: '1 star',
            count: 0
        }
    ],
    addReviewUrl: '/add-review',
    adTenant: 'APPDIRECT',
    baseUrl: 'http://localhost:3000',
    productId: 'cf6ab187-628b-4571-8089-d97319d68801',
    items: [
        structuredReviewMock,
        genericReviewMock,
        {
            id: '2428',
            rating: 5,
            title: 'Well done open source project',
            sections: [
                {
                    title: 'What do you like best?',
                    content: 'The DreamFactory project has been pleasant to work with. I first installed it on my Mac locally to try out connecting to a SQL Server database and that was easy (just followed one of their blog posts). It automatically generates a REST API for SQL that is nicely documented in the browser. Now our team will use the API to build a few mobile apps that use the data in SQL Server. Nice that we can deploy DreamFactory on OpenShift when we\'re ready to go live with our first few of apps. '
                }
            ],
            reviewAuthor: {
                name: 'Elistrong356 Elistrong356',
                avatarUrl: null
            },
            lastModified: 1414684586000,
            comments: [],
            actionEndpoints: {
                entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2428',
                comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2428/comments'
            },
            i18n: {
                byDate: 'by Elistrong356 Elistrong356 on October 30th 2014',
                date: 'October 30, 2014',
                showCommentCount: 'Show 0 Comments',
                hideCommentCount: 'Hide 0 Comments'
            },
            reviewLastModified: '2020-10-27T01:57:08.986Z',
            reviewTags: []
        },
        {
            id: '2425',
            rating: 5,
            title: 'Brilliant!',
            sections: [
                {
                    title: 'What do you like best?',
                    content: 'This is the \'killer\' app for the mobile age!'
                }
            ],
            reviewAuthor: {
                name: 'Xbadcock Xbadcock',
                avatarUrl: null
            },
            lastModified: 1414677318000,
            comments: [],
            actionEndpoints: {
                entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2425',
                comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2425/comments'
            },
            i18n: {
                byDate: 'by Xbadcock Xbadcock on October 30th 2014',
                date: 'October 30, 2014',
                showCommentCount: 'Show 0 Comments',
                hideCommentCount: 'Hide 0 Comments'
            },
            reviewLastModified: '2020-10-27T01:57:08.986Z'
        },
        {
            id: '2422',
            rating: 5,
            title: 'Overall a Solid platform',
            sections: [
                {
                    title: 'What do you like best?',
                    content: 'So far Dreamfactory has been a good backend for our iOS/Android app.  No major complaints really. The API and SDK\'s are pretty easy to figure out once you understand how they\'re laid out. The few times I got stuck their support was able to get me going again. Open source is a bonus too.'
                }
            ],
            reviewAuthor: {
                name: 'Lgdonoldson Lgdonoldson',
                avatarUrl: null
            },
            lastModified: 1414672737000,
            comments: [],
            actionEndpoints: {
                entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2422',
                comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2422/comments'
            },
            i18n: {
                byDate: 'by Lgdonoldson Lgdonoldson on October 30th 2014',
                date: 'October 30, 2014',
                showCommentCount: 'Show 0 Comments',
                hideCommentCount: 'Hide 0 Comments'
            },
            reviewLastModified: '2020-10-27T01:57:08.986Z',
            reviewTags: []
        },
        {
            id: '2419',
            rating: 5,
            title: 'Awesome interface',
            sections: [
                {
                    title: 'What do you like best?',
                    content: 'I just recently learned about this platform, but it wasn\'t until I downloaded the free trial and looked into it that I started to understand everything this platform does. The bottom line is it\'s saving me a ton of time, and I can connect to a db with a couple of clicks. It\'s one of those rare times where it really isn\'t all marketing hype, the darned thing really is that good. I\'d honestly give their support group more than five stars if that were possible. '
                }
            ],
            reviewAuthor: {
                name: 'Helipilot50 Helipilot50',
                avatarUrl: null
            },
            lastModified: 1414641270000,
            comments: [],
            actionEndpoints: {
                entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2419',
                comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2419/comments'
            },
            i18n: {
                byDate: 'by Helipilot50 Helipilot50 on October 30th 2014',
                date: 'October 30, 2014',
                showCommentCount: 'Show 0 Comments',
                hideCommentCount: 'Hide 0 Comments'
            },
            reviewLastModified: '2020-10-27T01:57:08.986Z'
        },
        {
            id: '2416',
            rating: 5,
            title: 'This is just what we needed',
            sections: [
                {
                    title: 'What do you like best?',
                    content: 'I cannot begin to tell you how long we have searched for something like this. You guys have literally saved our lives. Thank you so much for this wonderful platform!'
                }
            ],
            reviewAuthor: {
                name: 'Seagram777 Seagram777',
                avatarUrl: null
            },
            lastModified: 1414634646000,
            comments: [],
            actionEndpoints: {
                entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2416',
                comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2416/comments'
            },
            i18n: {
                byDate: 'by Seagram777 Seagram777 on October 30th 2014',
                date: 'October 30, 2014',
                showCommentCount: 'Show 0 Comments',
                hideCommentCount: 'Hide 0 Comments'
            },
            reviewLastModified: '2020-10-27T01:57:08.986Z'
        },
        {
            id: '2413',
            rating: 5,
            title: 'Easy to install and use on my platform of my choice',
            sections: [
                {
                    title: 'What do you like best?',
                    content: 'We were one of the early adopters of the DreamFactory Services Platform after a bunch of discussions with our consultants. After considering a bunch of other options, we finally agreed on DSP primarily because of security. We do not need to trust an additional cloud provider - I was able to install on Amazon and connect my Azure account easily. We have not been disappointed. They have listened to our feedback and given us most of what we have asked for. Support is top notch and speedy.\n\nThe roadmap looks promising. My team cannot wait for the next version to come out to integrate and release our app update. \n\nA big thank you to the DreamFactory team and good luck on the next version.\n\nKen'
                }
            ],
            reviewAuthor: {
                name: 'Kenkissinger Kenkissinger',
                avatarUrl: null
            },
            lastModified: 1414515456000,
            comments: [],
            actionEndpoints: {
                entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2413',
                comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2413/comments'
            },
            i18n: {
                byDate: 'by Kenkissinger Kenkissinger on October 28th 2014',
                date: 'October 28, 2014',
                showCommentCount: 'Show 0 Comments',
                hideCommentCount: 'Hide 0 Comments'
            },
            reviewLastModified: '2020-10-27T01:57:08.986Z'
        },
        {
            id: '65',
            rating: 5,
            title: 'Easy to Use',
            sections: [
                {
                    title: 'What do you like best?',
                    content: 'I\'ve used this for a little over a year and in general just find it easy and straightforward.'
                }
            ],
            reviewAuthor: {
                name: 'Michael Venn',
                avatarUrl: null
            },
            lastModified: 1310026578000,
            comments: [
                {
                    id: '956',
                    content: 'who is GetApp?',
                    reviewAuthor: {
                        name: 'joe nomo',
                        avatarUrl: null
                    },
                    lastModified: 1367437176000,
                    official: false,
                    isCurrentUserComment: false,
                    actionEndpoint: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/reviews/65/comments/956',
                    i18n: {
                        byDate: 'by joe nomo on May 1st 2013'
                    }
                }
            ],
            actionEndpoints: {
                entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/reviews/65',
                comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/416/reviews/65/comments'
            },
            i18n: {
                byDate: 'by Michael Venn on July 7th 2011',
                date: 'July 7, 2011',
                showCommentCount: 'Show 1 Comment',
                hideCommentCount: 'Hide 1 Comment'
            },
            reviewLastModified: '2020-10-27T01:57:08.986Z'
        },
        {
            id: '2406',
            rating: 5,
            title: 'Great Mobile Platform',
            sections: [
                {
                    title: 'What do you like best?',
                    content: 'we have been using DreamFactory for a few weeks, this is a great open source platform for mobile app dev. You can hook up any databases or other services, they create all the REST APIs for you. We are using the Angular SDK, all the other client platforms are also there but we haven\'t tried those yet.'
                }
            ],
            reviewAuthor: {
                name: 'Kentmcgreggor Kentmcgreggor',
                avatarUrl: null
            },
            lastModified: 1414421740000,
            comments: [],
            actionEndpoints: {
                entry: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2406',
                comments: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews/2406/comments'
            },
            i18n: {
                byDate: 'by Kentmcgreggor Kentmcgreggor on October 27th 2014',
                date: 'October 27, 2014',
                showCommentCount: 'Show 0 Comments',
                hideCommentCount: 'Hide 0 Comments'
            },
            reviewLastModified: '2020-10-27T01:57:08.986Z'
        }
    ],
    featured: [
        {
            link: 'https://techcrunch.com/2016/10/19/why-oh-why-are-they-calling-it-g-suite/',
            linkDescription: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc augue nisi, gravida pretium sollicitudin vitae, vehicula ut nulla. In est sem, volutpat sit amet nunc sit amet, hendrerit scelerisque lacus. Proin tristique laoreet tellus, in ultrices enim tristique nec.',
            source: {
                id: 3,
                name: 'TechCrunch',
                logoUrl: 'https://placehold.co/200'
            }
        },
        {
            link: 'https://www.google.ca/',
            linkDescription: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc augue nisi, gravida pretium sollicitudin vitae, vehicula ut nulla. In est sem, volutpat sit amet nunc sit amet, hendrerit scelerisque lacus. Proin tristique laoreet tellus, in ultrices enim tristique nec.',
            source: {
                id: 4,
                name: 'Lorem Ipsum',
                logoUrl: 'https://placehold.co/200'
            }
        }
    ],
    rating: 5,
    totalCount: 10,
    totalCustomerReviews: 38,
    averageRating: 3.8,
    currentUserReview: {},
    actionEndpoint: 'https://marketplace.appdirect.com/api/marketplace/v1/products/13960/reviews',
    i18n: {
        labels: {
            content: 'Comment',
            edit: 'Edit',
            delete: 'Delete',
            cancel: 'Cancel',
            confirm: 'Confirm',
            submit: 'Submit',
            featuredReviews: 'Featured Reviews',
            noReviews: 'No customer reviews have been written.',
            myReview: 'My Review',
            custReviewsResults: '8 Results',
            averageReview: 'Average rating',
            myRating: 'My Rating',
            signup: 'Sign Up to Write a Review',
            readMore: 'Read More',
            writeReview: 'Write a Review',
            reviews: 'Reviews',
            reviewDetails: 'Review details',
            description: 'Description',
            title: 'Title',
            customerReviews: 'Customer Reviews',
            singleStar: ' star',
            multiStars: ' stars',
            ratioOutOfNumReviewTotal: '5/5 based on 8 reviews',
            commentOnReview: 'Comment on Review',
            submitComment: 'Submit Comment',
            writeYourCommentsHere: 'Write your comments here.',

            currentPage: '1-{end} of {total} reviews',
            basedOnReviews: 'Based on {n} reviews',
            sortBy: 'Sort by',
            latest: 'Most recent',
            oldest: 'Least recent',
            highestRated: 'Highest rated',
            lowestRated: 'Lowest rated',
            showFullReview: 'Read full review',
            hideFullReview: 'Show less',
            validatedUserTag: 'Validated Reviewer',
            clearFilter: 'clear filter'
        },
        messages: {
            confirm: {
                deleteEntry: 'Please confirm deleting this review.',
                deleteComment: 'Please confirm deleting this comment.'
            },
            success: {
                deleteEntry: 'Review successfully deleted',
                deleteComment: 'Comment successfully deleted',
                postEntry: 'Thank you for the review, it will be posted shortly',
                postComment: 'Thank you for your comment!  It will be posted shortly.'
            },
            error: {
                deleteEntry: 'Unable to delete review.',
                deleteComment: 'Unable to delete comment.',
                postEntry: 'You must log in to write a review.',
                textareaMaxLength: 'Must be at most 2048 characters',
                fieldRequired: 'Required',
                mustBeBetween: 'Must be between 1 and 5',
                ratingRequired: 'Error: Please select a rating option for your review'
            }
        }
    }
};
