// @appdirect/sfb-theme-components/mocks/product/testimonials.js
export const testimonialMock = {
    results: [
        {
            id: 1,
            author: 'Amy E. Hood',
            job: 'Microsoft CFO',
            testimonial: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur nec sodales leo. Nulla commodo lacus fringilla arcu tempor convallis. Nunc semper mi ut tincidunt finibus. Nunc sodales, massa et rhoncus pretium, tortor justo tincidunt est, posuere scelerisque augue turpis non lorem. Nam id convallis neque. Sed a urna in dolor mattis dignissim vitae et ante. Sed molestie quis tortor eget tincidunt. Morbi enim justo, interdum at est at, auctor fringilla nibh.',
            iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/thumbs_64/img8352473687030874440.png',
            iconSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/thumbs_64/img8352473687030874440.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/thumbs_64/img8352473687030874440-2x.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/577/thumbs_64/img8352473687030874440.png'
            },
            iconAlt: 'Microsoft'
        },
        {
            id: 2,
            author: 'Amy E. Hood',
            job: 'Microsoft CFO',
            testimonial: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur nec sodales leo. Nulla commodo lacus fringilla arcu tempor convallis. Nunc semper mi ut tincidunt finibus. Nunc sodales, massa et rhoncus pretium, tortor justo tincidunt est, posuere scelerisque augue turpis non lorem. Nam id convallis neque. Sed a urna in dolor mattis dignissim vitae et ante. Sed molestie quis tortor eget tincidunt. Morbi enim justo, interdum at est at, auctor fringilla nibh.'
        }
    ],
    i18n: {
        button: 'Read the article'
    }
};
