// @appdirect/sfb-theme-components/mocks/product/top-features.js
export const featuresMock = {
    overview: {
        benefits: [
            {
                title: 'Test Product',
                description: 'Lorem Ipsum is simply dummy text ...',
                descriptionLong: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.'
            },
            {
                title: 'Test Product',
                description: 'Lorem Ipsum is simply dummy text ...',
                descriptionLong: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.'
            },
            {
                title: 'Test Product',
                description: 'Lorem Ipsum is simply dummy text ...',
                descriptionLong: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.'
            },
            {
                title: 'Test Product',
                description: 'Lorem Ipsum is simply dummy text ...',
                descriptionLong: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.'
            },
            {
                title: 'Test Product',
                description: 'Lorem Ipsum is simply dummy text ...',
                descriptionLong: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.'
            },
            {
                title: 'Test Product',
                description: 'Lorem Ipsum is simply dummy text ...',
                descriptionLong: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.'
            }
        ],
        callToActions: {
            tour: {
                label: 'Take the tour',
                alternateLabel: 'More Features',
                a11yLabel: 'More features about this product',
                color: 'primary',
                href: '/en-US/apps/544/galaxie/features'
            }
        },
        i18n: {
            featuresAndBenefits: 'Top Features'
        }
    }
};

export const featureCardMock = {
    title: 'Test Product',
    description: 'Lorem Ipsum is simply dummy text ...'
};

export const featureCardWithImageMock = {
    title: 'Test Product',
    description: 'Lorem Ipsum is simply dummy text ...',
    image: {
        alt: 'Lorem Ipsum',
        url: 'https://placehold.co/100',
        urlSrcset: {
            default: 'https://placehold.co/100',
            '2x': 'https://placehold.co/200'
        }
    }
};

export const featureCardWithHTMLinDescription = {
    ...featureCardWithImageMock,
    description: 'Palo santo <span style="color:red;">tbh synth</span> brooklyn ugh normcore <b>3 wolf moon</b> VHS sriracha coloring book etsy. +1 kogi taiyaki shoreditch, unicorn pork belly man bun activated charcoal chartreuse put a bird on it asymmetrical semiotics.'
};
