// @appdirect/sfb-theme-components/src/components/product-featured-quotes/stories/ProductFeaturedQuotes.stories.js
import React from 'react';
import ProductFeaturedQuotes from '../ProductFeaturedQuotes';
import { reviewsMock } from '../../../../mocks/product/reviews';
import { LAYOUT_HORIZONTAL } from '../../../constants/layouts';

export default {
    title: 'Components/Product Featured Quotes',
    component: ProductFeaturedQuotes
};

export const Default = {
    render: () => (
        <div
            style={{
                width: 300
            }}
        >
            <ProductFeaturedQuotes data={reviewsMock} />
        </div>
    )
};

export const Horizontal = {
    render: () => (
        <ProductFeaturedQuotes
            data={{
                ...reviewsMock,
                layout: LAYOUT_HORIZONTAL
            }}
        />
    )
};

export const CustomLogoSize = {
    render: () => (
        <ProductFeaturedQuotes
            data={{
                ...reviewsMock,
                logoSize: 120
            }}
        />
    )
};

export const CustomLogoSpacing = {
    render: () => (
        <ProductFeaturedQuotes
            data={{
                ...reviewsMock,
                logoSpacing: 50
            }}
        />
    )
};
