// @appdirect/sfb-theme-components/src/components/product-review-detail/stories/ProductReviewDetail.stories.js
import React from 'react';
import ProductReviewDetail from '../ProductReviewDetail';
import { genericReviewMock, structuredReviewMock } from '../../../../mocks/product/reviews';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

export default {
    title: 'Components/Product Review Detail',
    component: ProductReviewDetail,
    argTypes: getArgTypes(ProductReviewDetail.schema().form),
    parameters: {
        controls: {
            include: Object.keys(ProductReviewDetail.schema().form)
        }
    }
};

export const Default = {
    render: args => (
        <ProductReviewDetail
            data={{
                ...structuredReviewMock
            }}
            settings={args}
        />
    ),
    args: getArgs(ProductReviewDetail.schema().form)
};

export const WithSingleText = {
    render: args => (
        <ProductReviewDetail
            data={{
                ...genericReviewMock
            }}
            settings={args}
        />
    ),
    args: getArgs(ProductReviewDetail.schema().form)
};
