// @appdirect/sfb-theme-components/src/constants/components/prop-types/footer.js
import PropTypes from 'prop-types';
import { localeShape, marketplaceNamePropTypes } from '../../prop-types/config';
import { FOOTER_LAYOUT_OPTIONS, FOOTER_SIZES_OPTIONS, FOOTER_LOGO_POSITION_OPTIONS, FOOTER_SOCIAL_ICON_POSITION_OPTIONS, FOOTER_NEWSLETTER_BUTTON_STYLE_OPTIONS, FOOTER_NEWSLETTER_POSITION_OPTIONS, FOOTER_NEWSLETTER_LOGO_POSITION_OPTIONS } from '../footer';
import { LOGO_OPTIONS } from '../../logo';

// Globals Props
export const footerSocialPropTypes = {
    items: PropTypes.arrayOf(PropTypes.shape({
        isVisible: PropTypes.bool,
        label: PropTypes.string,
        values: PropTypes.object
    }))
};

export const footerCopyrightPropTypes = {
    text: PropTypes.string
};

export const footerFeaturesPropTypes = {
    items: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.string,
        text: PropTypes.string,
        icon: PropTypes.string,
        link: PropTypes.string
    }))
};

export const i18nFooterSocialPropTypes = {
    followUs: PropTypes.string
};

export const i18nFooterNewsletterPropTypes = {
    newsletterTitle: PropTypes.string,
    newsletterDescription: PropTypes.string,
    newsletterButtonLabel: PropTypes.string
};

export const i18nBottomBarPropTypes = {
    navigationTitle: PropTypes.string
};

export const i18nFooterMenuPropTypes = {
    menuTitle: PropTypes.string
};

export const footerNewsletterPropTypes = {
    buttonTheme: PropTypes.oneOf(FOOTER_NEWSLETTER_BUTTON_STYLE_OPTIONS.map(({ value }) => value)),
    url: PropTypes.string,
    i18n: PropTypes.shape(i18nFooterNewsletterPropTypes)
};

export const footerDataPropTypes = {
    locale: PropTypes.shape(localeShape),
    marketplaceName: marketplaceNamePropTypes,
    i18n: PropTypes.shape({
        ...i18nFooterSocialPropTypes,
        ...i18nBottomBarPropTypes,
        ...i18nFooterMenuPropTypes
    }),
    pageId: PropTypes.string
};

export const footerSettingsPropTypes = {
    layout: PropTypes.oneOf(FOOTER_LAYOUT_OPTIONS.map(({ value }) => value)),
    size: PropTypes.oneOf(FOOTER_SIZES_OPTIONS.map(({ value }) => value)),
    copyright: PropTypes.string,
    selectedLogo: PropTypes.oneOf(LOGO_OPTIONS),
    logoSize: PropTypes.number,
    logoPosition: PropTypes.oneOf(
        FOOTER_LOGO_POSITION_OPTIONS.map(({ value }) => value)
    ),
    newsletterLogoPosition: PropTypes.oneOf(
        FOOTER_NEWSLETTER_LOGO_POSITION_OPTIONS.map(({ value }) => value)
    ),
    footerBackground: PropTypes.string,
    footerText: PropTypes.string,
    bottomBarBackground: PropTypes.string,
    bottomBarText: PropTypes.string,
    socials: PropTypes.shape(footerSocialPropTypes),
    displaySocialMedia: PropTypes.bool,
    displayDefaultSocialIcons: PropTypes.bool,
    socialIconColor: PropTypes.string,
    socialIconPosition: PropTypes.oneOf(
        FOOTER_SOCIAL_ICON_POSITION_OPTIONS.map(({ value }) => value)
    ),
    newsletterTitle: PropTypes.string,
    newsletterText: PropTypes.string,
    newsletterButtonLabel: PropTypes.string,
    newsletterButtonLink: PropTypes.string,
    newsletterButtonStyle: PropTypes.oneOf(
        FOOTER_NEWSLETTER_BUTTON_STYLE_OPTIONS.map(({ value }) => value)
    ),
    newsletterSocialPosition: PropTypes.oneOf(
        FOOTER_NEWSLETTER_POSITION_OPTIONS.map(({ value }) => value)
    ),
    features: PropTypes.shape(footerFeaturesPropTypes)
};

// Footer Layouts Props
export const footerNewsletterSocialPropTypes = {};
