// @appdirect/sfb-theme-components/src/constants/components/prop-types/product-slider.js
import PropTypes from 'prop-types';
import { i18nShape } from '../../prop-types/productLists/i18n';
import { DYNAMICALLY_CURATED_CONTENT_TYPE, MANUALLY_CURATED_CONTENT_TYPE } from '../slider';
import { productSliderSettingsShape as sliderSettingShape } from './slider';

export const productSliderDataShape = {
    i18n: PropTypes.shape(i18nShape)
};

export const productSliderSettingsShape = {
    ...sliderSettingShape,
    productsList: PropTypes.shape({
        name: PropTypes.string,
        url: PropTypes.string,
        products: PropTypes.arrayOf(
            PropTypes.shape({
                title: PropTypes.string
            })
        ),
        items: PropTypes.arrayOf(
            PropTypes.shape({
                title: PropTypes.string
            })
        ),
        contentType: PropTypes.oneOf([MANUALLY_CURATED_CONTENT_TYPE, DYNAMICALLY_CURATED_CONTENT_TYPE])
    }),
    maxRows: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    buttonLabel: PropTypes.oneOfType([PropTypes.string, PropTypes.shape({
        value: PropTypes.string
    })]),
    sliderOptions: PropTypes.bool
};
