// @appdirect/sfb-theme-components/src/constants/components/prop-types/product-image-viewer.js
import PropTypes from 'prop-types';

export const sliderArrowShape = {
    i18nWithDefaults: PropTypes.shape({
        previous: PropTypes.string,
        next: PropTypes.string
    }),
    caretOnly: PropTypes.string,
    currentSlide: PropTypes.number,
    slideCount: PropTypes.number
};

export const screenshotShape = {
    name: PropTypes.string,
    imageUrl: PropTypes.string,
    imageUrlSrcset: PropTypes.shape({
        default: PropTypes.string,
        '1x': PropTypes.string,
        '2x': PropTypes.string
    }),
    imageAlt: PropTypes.string
};

export const overviewShape = {
    callToActions: PropTypes.shape({
        demo: PropTypes.shape({
            href: PropTypes.string,
            iFrameFriendlyUrl: PropTypes.string,
            label: PropTypes.string
        })
    }),
    title: PropTypes.string,
    description: PropTypes.string,
    image: PropTypes.string,
    imageSrcSet: PropTypes.shape({
        default: PropTypes.string,
        '1x': PropTypes.string,
        '2x': PropTypes.string
    }),
    pricingUrl: PropTypes.string,
    i18n: PropTypes.shape({
        overview: PropTypes.string
    })
};
