// @appdirect/sfb-theme-components/src/constants/components/prop-types/product-additional-info.js
import PropTypes from 'prop-types';
import { overviewPropTypes } from '../../prop-types/application/overview';
import { supportPropTypes } from '../../prop-types/application/support';
import { customAttributesPropTypes } from '../../prop-types/application/custom-attributes';
import { relatedContentPropTypes } from '../../prop-types/application/related-content';
import { COLUMN_COUNT_OPTIONS, INFO_POSITION_OPTIONS } from '../product-additional-info';

export const productAdditionalInfoDataPropTypes = {
    support: PropTypes.shape(supportPropTypes),
    relatedContent: PropTypes.shape(relatedContentPropTypes),
    customAttributes: PropTypes.arrayOf(PropTypes.shape(customAttributesPropTypes)),
    overview: PropTypes.shape(overviewPropTypes)
};

export const productAdditionalInfoSettingsPropTypes = {
    columnCount: PropTypes.oneOf(COLUMN_COUNT_OPTIONS.map(({ value }) => value)),
    infoPosition: PropTypes.oneOf(INFO_POSITION_OPTIONS.map(({ value }) => value)),
    headerBackground: PropTypes.string,
    sideBarBackground: PropTypes.string,
    sideBarBorder: PropTypes.string
};
