// @appdirect/sfb-theme-components/src/constants/components/prop-types/tile.js
import PropTypes from 'prop-types';

import { ratingsSummaryShape } from '../../prop-types/common';

export const tileShape = PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    vendorName: PropTypes.string,
    blurb: PropTypes.string,
    description: PropTypes.string,
    hidePricings: PropTypes.bool,
    startingPrice: PropTypes.shape({
        priceCaption: PropTypes.string,
        formattedFullPrice: PropTypes.string
    }),
    rating: PropTypes.number,
    ratingCaption: PropTypes.string,
    numRatings: PropTypes.number,
    url: PropTypes.string,
    iconUrl: PropTypes.string,
    referUrl: PropTypes.string,
    iconSrcset: PropTypes.shape({
        default: PropTypes.string,
        '1x': PropTypes.string,
        '2x': PropTypes.string
    }),
    tags: PropTypes.arrayOf(
        PropTypes.shape({
            type: PropTypes.string,
            name: PropTypes.string
        })
    ),
    productRibbon: PropTypes.shape({
        text: PropTypes.string,
        textColor: PropTypes.string,
        ribbonColor: PropTypes.string
    }),
    badges: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string,
            label: PropTypes.string,
            type: PropTypes.string
        })
    ),
    merchandisingBadges: PropTypes.arrayOf(
        PropTypes.shape({
            uuid: PropTypes.string,
            name: PropTypes.string,
            image: PropTypes.string
        })
    ),
    ratingsSummary: PropTypes.shape(ratingsSummaryShape)
});
