// @appdirect/sfb-theme-components/src/atoms/feature-card/stories/FeatureCard.stories.js
import React from 'react';
import { featureCardMock,
    featureCardWithImageMock,
    featureCardWithHTMLinDescription } from '../../../../mocks/product/top-features';
import FeatureCard from '../FeatureCard';

export default {
    title: 'Atoms/Feature Card',
    component: FeatureCard
};

export const Default = {
    render: () => (
        <div
            style={{
                width: 300
            }}
        >
            <FeatureCard {...featureCardWithImageMock} />
        </div>
    ),

    name: 'Default'
};

export const WithSmallImage = {
    render: () => (
        <div
            style={{
                width: 300
            }}
        >
            <FeatureCard {...featureCardWithImageMock} imageSize="small" />
        </div>
    ),

    name: 'With small image'
};

export const WithoutImage = {
    render: () => (
        <div
            style={{
                width: 300
            }}
        >
            <FeatureCard {...featureCardMock} />
        </div>
    ),

    name: 'Without image'
};

export const WithHeading = {
    render: () => (
        <div
            style={{
                width: 300,
                '--titleText': 'red'
            }}
        >
            <FeatureCard {...featureCardMock} withHeading />
        </div>
    ),

    name: 'With heading'
};

export const WithHtmlInTheDescription = {
    render: () => (
        <div
            style={{
                width: 300
            }}
        >
            <FeatureCard {...featureCardWithHTMLinDescription} />
        </div>
    ),

    name: 'With HTML in the description'
};
