// @appdirect/sfb-theme-components/src/atoms/bullet/stories/Bullet.stories.js
import React from 'react';
import Bullet from '../Bullet';
import { allowedTypes } from '../../icon/Icon';

export default {
    title: 'Atoms/Bullet',
    component: Bullet,
    render: args => <Bullet {...args} />,
    argTypes: {
        content: {
            control: 'text',
            description: 'Content of the bullet'
        },
        iconType: {
            control: 'select',
            options: allowedTypes,
            description: 'Type of the icon'
        },
        tooltip: {
            control: 'text',
            description: 'Tooltip of the bullet'
        }
    }
};

export const Default = {
    args: {
        content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
        tooltip: ''
    }
};

export const WithIcon = {
    args: {
        content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
        iconType: 'check'
    },
    parameters: {
        controls: {
            exclude: ['tooltip']
        }
    }
};

export const WithTooltip = {
    args: {
        content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
        iconType: 'plus',
        tooltip: 'Aenean vel arcu ac purus mollis sollicitudin. Donec felis tortor, pretiu'
    }
};
