// @appdirect/sfb-theme-components/src/atoms/input-search/stories/InputSearch.stories.js
import React from 'react';
import InputSearch from '../InputSearch';

export default {
    title: 'Atoms/Input Search',
    component: InputSearch
};

export const Default = {
    render: () => <InputSearch />,
    name: 'Default'
};

export const WithSearchButton = {
    render: () => <InputSearch submitLabel="Search" />,
    name: 'With search button'
};

export const CustomSearchIcon = {
    render: () => <InputSearch submitLabel="Search" submitIconUrl="https://via.placeholder.com/30" />,
    name: 'Custom search icon'
};

export const OnlySearchIcon = {
    render: () => <InputSearch submitLabel="Search" submitIconOnly />,
    name: 'Only search icon'
};

export const OnlySearchText = {
    render: () => <InputSearch submitLabel="Search" submitTextOnly />,
    name: 'Only search Text'
};

export const HiddenLabel = {
    render: () => <InputSearch hiddenLabel />,
    name: 'Hidden Label'
};

export const WithoutBorders = {
    render: () => <InputSearch hiddenLabel withBorder={false} />,
    name: 'Without borders'
};

export const CustomStyles = {
    render: () => (
        <InputSearch
            submitLabel="Search"
            borderColor="#ff0000"
            submitBgColor="#008000"
            backgroundColor="#d3d3d3"
            borderRadius="10"
            height="50"
        />
    ),

    name: 'Custom Styles'
};
