// @appdirect/sfb-theme-components/src/atoms/carousel-slide-video/stories/CarouselSlideVideo.stories.js
import React from 'react';
import CarouselSlideVideo from '../CarouselSlideVideo';
import { carouselMockData } from '../../../../mocks/home/carousel';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

const videoSlideMockData = carouselMockData.slides[0];
const dataObj = {
    ...videoSlideMockData.slideData,
    videoUrl: 'https://www.youtube.com/watch?v=oHg5SJYRHA0',
    imageUrl: 'http://placeimg.com/800/480/tech'
};

const getWrappedArgs = args => ({
    ...args,
    ...dataObj
});

export default {
    title: 'Atoms/Carousel Slide/Video',
    component: CarouselSlideVideo,
    argTypes: getArgTypes(CarouselSlideVideo.schema().form),
    render: args => (
        <CarouselSlideVideo
            data={args}
            settings={{
                height: 440
            }}
        />
    ),
    parameters: {
        controls: {
            include: Object.keys(CarouselSlideVideo.schema().form)
        }
    }
};

export const Default = {
    args: getWrappedArgs(getArgs(CarouselSlideVideo.schema().form))
};
