// @appdirect/sfb-theme-components/mocks/cart/cartItems.js
import { cartItemAddons } from './cartItemAddons';

export const cartItemNonStackableNoAddonsMock = [
    {
        pricingPlanId: '7502c355-00d2-4f3c-83bb-aab503edf08b',
        subscriptionId: null,
        serviceConfiguration: null,
        billingConfiguration: null,
        discountCode: null,
        discountId: null,
        discountNumOfBillingCycles: null,
        units: [],
        id: 'b2e62ecb-5110-4c21-a9bb-0a625a0d1972',
        customAttributes: null,
        validated: false,
        customPrices: [],
        selectedPricingPlanId: '7502c355-00d2-4f3c-83bb-aab503edf08b',
        selectedEditionId: 'f07dd3d7-3be1-4990-9227-59d5d15b7487',
        itemDetails: {
            id: '322076',
            uuid: 'e98b3cf4-2d97-4e12-91d4-3845ebfb8283',
            type: 'PRODUCT',
            customIntegration: null,
            stacked: false,
            branding: {
                name: 'Lennys multiple editions no addons',
                vendor: 'Engage19',
                desc: 'Lennys multiple editions no addons',
                iconUrl: 'https://d33na3ni6eqf5j.cloudfront.net/app_resources/322076/thumbs_64/img4186280596906514351.png',
                href: 'https://engage19billing.test.devappdirect.me/api/marketplace/v1/products/322076',
                storefrontUrl: 'https://engage19billing.test.devappdirect.me/apps/322076'
            },
            editions: [
                {
                    id: 'f07dd3d7-3be1-4990-9227-59d5d15b7487',
                    name: 'Edition 3',
                    editionCode: 'edition3',
                    bundleOnly: false,
                    autoUpgradeToPaid: null,
                    freeTrialRequiresBillingInfo: null,
                    revenueModel: 'RECURRING',
                    freeTrialTerms: {
                        freeTrialDuration: 0,
                        freeTrialDurationPeriod: 'DAY',
                        freeTrialGracePeriod: null
                    },
                    pricingPlans: [
                        {
                            id: '7502c355-00d2-4f3c-83bb-aab503edf08b',
                            currency: 'USD',
                            pricingPeriod: 'MONTHLY',
                            pricingSources: [
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'RECURRING_FLAT',
                                    unit: null,
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'FLAT',
                                            salePrice: '30.0000000000',
                                            priceRanges: []
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: '1.0000000000',
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                },
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'INCLUDED',
                                    unit: 'USER',
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'UNIT',
                                            salePrice: null,
                                            priceRanges: [
                                                {
                                                    min: '1.0000000000',
                                                    max: '1.0000000000',
                                                    salePrice: '0.0000000000',
                                                    wholeSalePrice: null
                                                }
                                            ]
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: null,
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                }
                            ],
                            contractConfiguration: null,
                            additionalFees: {
                                terminationFee: null
                            }
                        }
                    ]
                }
            ],
            customAttributes: []
        },
        contractInfo: {
            contractConfiguration: null,
            additionalFees: {
                terminationFee: null
            }
        },
        costs: [
            {
                costId: '6465b811-d17e-446b-b8c8-a53f1b3978d3',
                costTypeCategory: 'RECURRING',
                costType: 'RECURRING_FLAT',
                pricingStrategy: 'FLAT',
                unit: null,
                definition: {
                    price: '30.0000000000',
                    priceRanges: null
                },
                calculations: [
                    {
                        quantity: '1.0000000000',
                        salePrice: '30.0000000000',
                        totalPrice: '30.0000000000',
                        salePriceTaxIncluded: null,
                        totalPriceTaxIncluded: null
                    }
                ],
                costDescription: 'Edition 3'
            },
            {
                costId: null,
                costTypeCategory: 'RECURRING',
                costType: 'INCLUDED',
                pricingStrategy: 'UNIT',
                unit: 'USER',
                definition: {
                    price: null,
                    priceRanges: [
                        {
                            min: '1.0000000000',
                            max: '1.0000000000',
                            price: '0.0000000000'
                        }
                    ]
                },
                calculations: [
                    {
                        quantity: '1.0000000000',
                        salePrice: '0.0000000000',
                        totalPrice: '0.0000000000',
                        salePriceTaxIncluded: null,
                        totalPriceTaxIncluded: null
                    }
                ],
                costDescription: 'Per User Fee Included'
            }
        ],
        externalSource: null,
        detailedCost: {
            costSources: [
                {
                    costId: '6465b811-d17e-446b-b8c8-a53f1b3978d3',
                    costDescription: 'Edition 3',
                    costTypeCategory: 'RECURRING',
                    costType: 'RECURRING_FLAT',
                    unit: null,
                    sources: [
                        {
                            priceType: 'BASE_PRICE',
                            priceLevel: 'NONE',
                            priceEntity: 'NONE',
                            pricingStrategy: 'FLAT',
                            definition: {
                                price: '30.0000000000',
                                priceRanges: null
                            },
                            calculations: [
                                {
                                    quantity: '1.0000000000',
                                    salePrice: '30.0000000000',
                                    totalPrice: '30.0000000000',
                                    salePriceTaxIncluded: null,
                                    totalPriceTaxIncluded: null
                                }
                            ]
                        }
                    ],
                    increment: 1,
                    pricePerIncrement: false
                },
                {
                    costId: null,
                    costDescription: 'Per User Fee Included',
                    costTypeCategory: 'RECURRING',
                    costType: 'INCLUDED',
                    unit: 'USER',
                    sources: [
                        {
                            priceType: 'BASE_PRICE',
                            priceLevel: 'NONE',
                            priceEntity: 'NONE',
                            pricingStrategy: 'UNIT',
                            definition: {
                                price: null,
                                priceRanges: [
                                    {
                                        min: '1.0000000000',
                                        max: '1.0000000000',
                                        price: '0.0000000000'
                                    }
                                ]
                            },
                            calculations: [
                                {
                                    quantity: '1.0000000000',
                                    salePrice: '0.0000000000',
                                    totalPrice: '0.0000000000',
                                    salePriceTaxIncluded: null,
                                    totalPriceTaxIncluded: null
                                }
                            ]
                        }
                    ],
                    increment: null,
                    pricePerIncrement: false
                }
            ]
        },
        payment_schedule: {
            billingCycle: {
                period: 'MONTHLY',
                dayOfMonth: 12
            },
            servicePeriod: {
                start: '2022-05-12',
                end: null
            },
            billingPeriod: {
                start: '2022-05-12',
                end: null
            },
            freeTrialPeriod: null,
            events: [
                {
                    eventType: 'BILLING_EFFECTIVE',
                    date: '2022-05-12'
                },
                {
                    eventType: 'SERVICE_ACTIVATION',
                    date: '2022-05-12'
                }
            ]
        }
    }
];

export const cartItemsMock = [
    {
        pricingPlanId: 'bbf4439f-834d-4fdb-abd1-d15ba9f01fad',
        subscriptionId: null,
        serviceConfiguration: null,
        billingConfiguration: null,
        discountCode: null,
        discountId: null,
        units: [],
        id: '9bbabcd7-4bb0-4d2a-b40b-2d19a5c5f505',
        customAttributes: null,
        validated: false,
        customPrices: [],
        selectedPricingPlanId: 'bbf4439f-834d-4fdb-abd1-d15ba9f01fad',
        selectedEditionId: 'aa574cd3-5ee6-41e2-b537-96887a9a743e',
        itemDetails: {
            id: '293048',
            uuid: '99ea13a6-679d-419f-9d87-d6c25b9026de',
            type: 'PRODUCT',
            customIntegration: null,
            stacked: false,
            branding: {
                name: 'Office 365',
                vendor: 'Microsoft',
                desc: 'Lorem Ipsum dolor sit amet.',
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293048/thumbs_64/image5315487358905934939.png',
                href: 'https://sfbmarketplace.byappdirect.com/api/marketplace/v1/products/293048',
                storefrontUrl: 'https://sfbmarketplace.byappdirect.com/apps/293048'
            },
            editions: [
                {
                    id: 'aa574cd3-5ee6-41e2-b537-96887a9a743e',
                    name: 'Growing',
                    editionCode: 'GROWING-QBO',
                    bundleOnly: false,
                    autoUpgradeToPaid: null,
                    freeTrialRequiresBillingInfo: null,
                    revenueModel: 'RECURRING',
                    freeTrialTerms: {
                        freeTrialDuration: 0,
                        freeTrialDurationPeriod: 'DAY',
                        freeTrialGracePeriod: 0
                    },
                    pricingPlans: [
                        {
                            id: '5b6f2bbf-00ca-415f-94ae-133b8c6d690e',
                            currency: 'USD',
                            pricingPeriod: 'MONTHLY',
                            pricingSources: [
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'RECURRING_FLAT',
                                    unit: null,
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'FLAT',
                                            salePrice: '79.0000000000',
                                            priceRanges: []
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: '1.0000000000',
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                },
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'INCLUDED',
                                    unit: 'USER',
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'UNLIMITED',
                                            salePrice: null,
                                            priceRanges: []
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: null,
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                }
                            ],
                            contractConfiguration: null,
                            additionalFees: {
                                terminationFee: null
                            }
                        }
                    ]
                }
            ],
            customAttributes: []
        },
        contractInfo: {
            contractConfiguration: null,
            additionalFees: {
                terminationFee: null
            }
        },
        costs: [
            {
                costId: 'fb8b7df7-fd2b-4299-91a0-67943d9ffa55',
                costTypeCategory: 'RECURRING',
                costType: 'RECURRING_FLAT',
                pricingStrategy: 'FLAT',
                unit: null,
                definition: {
                    price: '79.0000000000',
                    priceRanges: null
                },
                calculations: [
                    {
                        quantity: '1.0000000000',
                        salePrice: '79.0000000000',
                        totalPrice: '79.0000000000',
                        salePriceTaxIncluded: null,
                        totalPriceTaxIncluded: null
                    }
                ],
                costDescription: 'Growing'
            },
            {
                costId: null,
                costTypeCategory: 'RECURRING',
                costType: 'INCLUDED',
                pricingStrategy: 'UNLIMITED',
                unit: 'USER',
                definition: {
                    price: null,
                    priceRanges: null
                },
                calculations: [
                    {
                        quantity: null,
                        salePrice: '0.0000000000',
                        totalPrice: '0.0000000000',
                        salePriceTaxIncluded: null,
                        totalPriceTaxIncluded: null
                    }
                ],
                costDescription: 'Per User Fee Included'
            }
        ],
        payment_schedule: {
            billingCycle: {
                period: 'MONTHLY',
                dayOfMonth: 26
            },
            servicePeriod: {
                start: '2021-05-26',
                end: null
            },
            billingPeriod: {
                start: '2021-05-26',
                end: null
            },
            freeTrialPeriod: null,
            events: [
                {
                    eventType: 'BILLING_EFFECTIVE',
                    date: '2021-05-26'
                },
                {
                    eventType: 'SERVICE_ACTIVATION',
                    date: '2021-05-26'
                }
            ]
        },
        childItems: cartItemAddons
    },
    // another item to the cart, to test that it is ignored in the just added to cart summary
    {
        pricingPlanId: 'fe56ca29-0d61-40d4-85b6-c6ef5a5e118f',
        subscriptionId: null,
        serviceConfiguration: null,
        billingConfiguration: null,
        discountCode: null,
        discountId: null,
        units: [],
        id: 'e5c87bd8-075e-4c2c-99fe-054f8fdab5a3',
        customAttributes: null,
        validated: false,
        customPrices: [],
        selectedPricingPlanId: 'fe56ca29-0d61-40d4-85b6-c6ef5a5e118f',
        selectedEditionId: '273d600d-fe0a-4077-8d2f-815948c58d62',
        itemDetails: {
            id: 142378,
            uuid: '779adc9e-2ff1-4193-b56e-eb044f260e11',
            type: 'PRODUCT',
            customIntegration: null,
            stacked: false,
            branding: {
                name: 'BaseKit sitebuilder',
                vendor: 'BaseKit',
                desc: 'BaseKit is a website building tool aimed at small businesses. All BaseKit websites, blogs and online stores can be created and updated from a mobile phone, tablet or desktop computer.',
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/142378/thumbs_64/img61982019133247767.png?da2eec94dd3a6a553bae47de42161366',
                href: 'https://sfbmarketplace.byappdirect.com/api/marketplace/v1/products/142378',
                storefrontUrl: 'https://sfbmarketplace.byappdirect.com/apps/142378'
            },
            editions: [
                {
                    id: '273d600d-fe0a-4077-8d2f-815948c58d62',
                    name: 'Trial',
                    editionCode: 1890,
                    bundleOnly: false,
                    autoUpgradeToPaid: null,
                    freeTrialRequiresBillingInfo: null,
                    revenueModel: 'FREE',
                    freeTrialTerms: {
                        freeTrialDuration: 14,
                        freeTrialDurationPeriod: 'DAY',
                        freeTrialGracePeriod: 3
                    },
                    pricingPlans: [
                        {
                            id: 'fe56ca29-0d61-40d4-85b6-c6ef5a5e118f',
                            currency: 'USD',
                            pricingPeriod: 'MONTHLY',
                            pricingSources: [
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'RECURRING_FLAT',
                                    unit: null,
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'FREE',
                                            salePrice: 0,
                                            priceRanges: []
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: 1,
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                },
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'INCLUDED',
                                    unit: 'USER',
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'UNIT',
                                            salePrice: null,
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: 1,
                                                    salePrice: 0,
                                                    wholeSalePrice: null
                                                }
                                            ]
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: null,
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                }
                            ],
                            contractConfiguration: null,
                            additionalFees: {
                                terminationFee: null
                            }
                        }
                    ]
                }
            ],
            customAttributes: []
        },
        contractInfo: {
            contractConfiguration: null,
            additionalFees: {
                terminationFee: null
            }
        },
        costs: [
            {
                costId: '8762f42f-f80c-11e8-8d6d-0a3602cce94f',
                costTypeCategory: 'RECURRING',
                costType: 'RECURRING_FLAT',
                pricingStrategy: 'FREE',
                unit: null,
                definition: {
                    price: 0,
                    priceRanges: null
                },
                calculations: [
                    {
                        quantity: 1,
                        salePrice: 0,
                        totalPrice: 0,
                        salePriceTaxIncluded: null,
                        totalPriceTaxIncluded: null
                    }
                ],
                costDescription: 'Trial'
            },
            {
                costId: null,
                costTypeCategory: 'RECURRING',
                costType: 'INCLUDED',
                pricingStrategy: 'UNIT',
                unit: 'USER',
                definition: {
                    price: null,
                    priceRanges: [
                        {
                            min: 1,
                            max: 1,
                            price: 0
                        }
                    ]
                },
                calculations: [
                    {
                        quantity: 1,
                        salePrice: 0,
                        totalPrice: 0,
                        salePriceTaxIncluded: null,
                        totalPriceTaxIncluded: null
                    }
                ],
                costDescription: 'Per User Fee Included'
            }
        ],
        payment_schedule: {
            billingCycle: {
                period: 'MONTHLY',
                dayOfMonth: 14
            },
            servicePeriod: {
                start: '2021-05-31',
                end: null
            },
            billingPeriod: null,
            freeTrialPeriod: {
                start: '2021-05-31',
                end: '2021-06-14'
            },
            events: [
                {
                    eventType: 'FREE_TRIAL_START',
                    date: '2021-05-31'
                },
                {
                    eventType: 'SERVICE_ACTIVATION',
                    date: '2021-05-31'
                },
                {
                    eventType: 'FREE_TRIAL_EXPIRE',
                    date: '2021-06-14'
                }
            ]
        }
    }
];
