// @appdirect/sfb-theme-components/src/atoms/avatar/stories/Avatar.stories.js
import React from 'react';
import Avatar from '../Avatar';

export default {
    title: 'Atoms/Avatar',
    component: Avatar,
    render: args => <Avatar {...args} />,
    argTypes: {
        name: {
            control: 'text',
            description: 'Description of avatar'
        },
        picture: {
            control: 'select',
            options: ['Wayne Gretzky', 'Sidney Crosby', 'Alex Ovechkin', 'Connor McDavid'],
            mapping: {
                'Wayne Gretzky': 'https://www.hockeydb.com/ihdb/photos/wayne-gretzky-1996-40.jpg',
                'Sidney Crosby': 'https://www.hockeydb.com/ihdb/photos/sidney-crosby-2019-50.jpg',
                'Alex Ovechkin': 'https://www.hockeydb.com/ihdb/photos/alex-ovechkin-2024-49.jpg',
                'Connor McDavid': 'https://www.hockeydb.com/ihdb/photos/connor-mcdavid-2022-41.jpg'
            },
            description: 'The URL of the picture to display'
        },
        size: {
            control: {
                type: 'range',
                min: 25,
                max: 250,
                step: 25
            },
            description: 'The size of the avatar'
        }
    },
    parameters: {
        controls: {
            include: ['name', 'picture', 'size']
        }
    }
};

export const WithPicture = {
    args: {
        name: 'Wayne Gretzky',
        picture: 'Wayne Gretzky',
        size: 100
    }
};

export const WithoutPicture = {
    args: {
        name: 'Wayne Gretzky',
        picture: '',
        size: 100
    }
};
