// @appdirect/sfb-theme-components/src/atoms/pricing/stories/Pricing.stories.js
import React from 'react';
import Pricing from '../Pricing';

export default {
    title: 'Atoms/Pricing',
    component: Pricing
};

export const Default = {
    render: () => <Pricing price={45.11} caption="Starting At" frequency="Year" symbol="$" />,
    name: 'Default'
};

export const FormattedPrice = {
    render: () => <Pricing formattedPrice="$19.00" />,
    name: 'Formatted price'
};

export const WithEmphasis = {
    render: () => <Pricing price={124.22} caption="From" frequency="Month" symbol="$" isEmphasis />,
    name: 'With emphasis'
};

export const WithUnit = {
    render: () => <Pricing price={50.25} unit="user" frequency="month" symbol="$" isEmphasis />,
    name: 'With unit'
};

export const MultipleLines = {
    render: () => (
        <div
            style={{
                display: 'flex'
            }}
        >
            <div
                style={{
                    flexGrow: 1
                }}
            >
                <span
                    style={{
                        display: 'block',
                        fontWeight: 700,
                        marginBottom: '40px'
                    }}
                >
                    Align Left
                </span>
                <Pricing price={56.21} caption="From" symbol="$" frequency="Month" isEmphasis isMultipleLine />
            </div>
            <div>
                <span
                    style={{
                        display: 'block',
                        fontWeight: 700,
                        marginBottom: '40px',
                        textAlign: 'right'
                    }}
                >
                    Align Right
                </span>
                <Pricing
                    price={326.92}
                    caption="Starting At"
                    symbol="$"
                    frequency="Year"
                    align="right"
                    isEmphasis
                    isMultipleLine
                />
            </div>
        </div>
    ),

    name: 'Multiple lines'
};

export const UppercaseFirstLetter = {
    render: () => <Pricing price={50.25} unit="USER" frequency="MONTH" symbol="$" uppercaseFirstLetter />,

    name: 'Uppercase first letter'
};
