// @appdirect/sfb-theme-components/mocks/application/domains/i18n.js
export const i18nMock = {
    domains: 'Domains',
    sectionTitle: 'Search for your domain',
    search: 'Search Domains',
    recommendedTitle: 'Recommended Domains',
    recommendedPlaceholder: 'Domains you search for will appear here',
    add: 'Add',
    added: 'Added',
    addedTitle: 'Added Domains',
    addedPlaceholder: 'Selected domains will appear here',
    checkout: 'Continue Checkout',
    frequency: 'Year',
    remove: 'Remove Domain',
    select: 'Select',
    selected: 'Selected',
    selectedDomains: 'Selected Domains',
    addToCart: 'Add to Cart',
    buyNow: 'Buy Now',
    noResultMessageTitle: 'No results found',
    noResultMessageDescription: 'Try searching for a different domain name',
    skipBuyingDomain: 'Skip buying an associated domain',
    domainFound: '1 domain found',
    domainsFound: '{resultsLength} domains found',
    recommendedDomainName: 'Recommended Domain Name',
    addedDomainName: 'Added Domain Name',
    domainPrice: 'Price',
    domainAction: 'Action',
    selectedDomain: '{domainName} selected',
    selectDomain: 'Select {domainName}',
    removeDomain: 'Remove {domainName}'
};
