// @appdirect/sfb-theme-components/mocks/menu/secondary-nav.js
export const secondaryNavMock = {
    title: 'AppDirect',
    items: [
        {
            type: 'secondary',
            key: 'home',
            href: '/en-US/home',
            label: 'Featured Applications',
            selected: true
        },
        {
            type: 'secondary',
            key: 'listing',
            href: '/en-US/listing',
            label: 'All Applications',
            alternateLabel: 'Browse',
            selected: false
        },
        {
            type: 'secondary',
            href: '/en-US/listing?pl=1013',
            label: 'Test Providers PL',
            selected: false
        },
        {
            type: 'secondary',
            href: '/en-US/listing?pl=566',
            label: 'Last one',
            selected: false
        },
        {
            type: 'secondary',
            href: '/en-US/listing?pl=522',
            label: 'Microsoft',
            selected: false
        },
        {
            type: 'secondary',
            href: '/en-US/listing?pl=266',
            label: 'Core Catalog',
            selected: false
        },
        {
            type: 'secondary',
            href: '/en-US/listing?pl=476',
            label: 'Analytics',
            selected: false
        },
        {
            type: 'secondary',
            href: '/en-US/listing?pl=894',
            label: 'Test Product line',
            selected: false
        },
        {
            type: 'secondary',
            href: '/en-US/listing?pl=526',
            label: "Paul's Favorite",
            selected: false
        }
    ]
};
