// @appdirect/sfb-theme-components/src/atoms/preloader/stories/Preloader.stories.js
import React from 'react';
import Preloader from '../Preloader';

export default {
    title: 'Atoms/Preloader',
    component: Preloader
};

export const Default = {
    render: () => (
        <div
            style={{
                position: 'relative'
            }}
        >
            <div>
                <p>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse commodo egestas neque, sed
                    ultricies nunc convallis at.
                </p>
                <p>
                    Maecenas auctor felis posuere, vehicula tortor ut, venenatis urna. Vivamus hendrerit, eros quis
                    rhoncus molestie, lacus risus tristique orci. Fusce efficitur mi at ligula ornare, quis pretium dui
                    mollis.
                </p>
                <p>
                    Fusce sapien justo, sodales eu nibh eget, finibus fringilla nunc. In suscipit cursus metus et
                    posuere. Quisque iaculis metus eget dolor sagittis. Nulla hendrerit ex quis metus pretium, vel
                    posuere tellus congue.
                </p>
            </div>
            <Preloader isLoading />
        </div>
    ),

    name: 'Default',
    inline: false
};
