// @appdirect/sfb-theme-components/src/atoms/input/stories/Input.stories.js
import React from 'react';
import Input from '../Input';

export default {
    title: 'Atoms/Input',
    component: Input
};

export const InputText = {
    render: () => <Input name="name" />,
    name: 'Input text'
};

export const WithLabel = {
    render: () => <Input name="name" label="Label" />,
    name: 'With label'
};

export const Required = {
    render: () => <Input name="name" label="Label" required />,
    name: 'Required'
};

export const WithPlaceholder = {
    render: () => <Input name="name" label="Label" placeholder="placeholder" />,
    name: 'With placeholder'
};

export const WithDefaultValue = {
    render: () => <Input name="name" label="Label" defaultValue="default value" />,
    name: 'With default value'
};

export const ReadOnly = {
    render: () => <Input name="name" label="Label" defaultValue="default value" readOnly />,
    name: 'Read only'
};

export const WithHiddenLabelA11Y = {
    render: () => <Input name="name" label="Label" hiddenLabel />,
    name: 'With hidden label (a11y)'
};

export const Disabled = {
    render: () => <Input name="name" label="Label" disabled />,
    name: 'Disabled'
};

export const Number = {
    render: () => <Input type="number" name="name" />,
    name: 'Number'
};

export const Email = {
    render: () => <Input type="email" name="name" />,
    name: 'Email'
};

export const WithHint = {
    render: () => <Input label="Label" name="name" hint="Lorem ipsum dolor sit amet, consectetur adipiscing elit." />,

    name: 'With hint'
};

export const PhoneWithAddon = {
    render: () => <Input label="Label" name="name" type="tel" addon="+1" />,
    name: 'Phone with addon'
};

export const WithError = {
    render: () => <Input name="email" label="Email address" type="email" error="Not a valid email address" />,

    name: 'With error'
};
