// @appdirect/sfb-theme-components/src/atoms/checkbox-group/stories/CheckboxGroup.stories.js
import React from 'react';
import CheckboxGroup from '../CheckboxGroup';
import { productFiltersMock } from '../../../../mocks/product/productFilters';

const shallowItems = productFiltersMock.items.find(item => item.shallow).items;
const deepItems = productFiltersMock.items.find(item => !item.shallow).items;

export default {
    title: 'Atoms/CheckboxGroup',
    component: CheckboxGroup
};

export const Shallow = {
    render: () => <CheckboxGroup items={shallowItems} />,
    name: 'Shallow'
};

export const Deep = {
    render: () => <CheckboxGroup items={deepItems} isDisplayChildren />,
    name: 'Deep'
};

export const WithTitle = {
    render: () => <CheckboxGroup items={deepItems} title="Title" />,
    name: 'With Title'
};

export const WithHint = {
    render: () => <CheckboxGroup items={deepItems} hint="Lorem ipsum dolor sit amet, consectetur adipiscing elit." />,

    name: 'With Hint'
};

export const Small = {
    render: () => <CheckboxGroup items={deepItems} checkboxSize="small" />,
    name: 'Small'
};
