// @appdirect/sfb-theme-components/mocks/product/screenshots/index.js
export const screenshotsMock = {
    items: [
        {
            name: 'Create dashboards to monitor your sales pipeline and motivate your sales team.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img1102629510912555107.png',
            imageUrlSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img1102629510912555107.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img1102629510912555107.png'
            },
            imageAlt: 'Dashboard, sales view'
        },
        {
            name: 'Dashboards for customer service let you know immediately if your customers are being served.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img1262432103500464999.png',
            imageUrlSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img1262432103500464999.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img1262432103500464999.png'
            },
            imageAlt: 'Dashboard, service center view'
        },
        {
            name: 'Connect to services like New Relic or Pingdom to manage your technology infrastructure in real time.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img1493562786997021035.png',
            imageUrlSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img1493562786997021035.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img1493562786997021035.png'
            },
            imageAlt: 'Dashboard datacenter performance'
        },
        {
            name: 'Show off your social media reach in your lobby, and check out the buzz around your company at any moment.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img4298206716672378307.png',
            imageUrlSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img4298206716672378307.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img4298206716672378307.png'
            },
            imageAlt: 'Dashboard social media performance'
        },
        {
            name: 'Place KPIs in an executive dashboard and get an overview of the health of your company at a glance.',
            imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img6668021355693451257.png',
            imageUrlSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img6668021355693451257.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/screenshot/img6668021355693451257.png'
            },
            imageAlt: 'Infrastructure tracking data'
        }
    ],
    i18n: {
        title: 'Screenshots',
        sidebarTitle: 'View Screenshots'
    }
};
