// @appdirect/sfb-theme-components/src/components/product-feature-details/stories/FeatureDetails.stories.js
import React from 'react';
import FeatureDetails from '../FeatureDetails';
import { featureDetailsMocks,
    gSuiteFeatureDetailsMock,
    tSheetsFeatureDetailsMock,
    featureDetailsWithHTMLMock } from '../../../../mocks/components/featureDetails';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

export default {
    title: 'Components/Feature Details',
    component: FeatureDetails,
    argTypes: getArgTypes(FeatureDetails.schema().form)
};

export const Default = {
    render: args => <FeatureDetails data={featureDetailsMocks} settings={args} />,
    args: getArgs(FeatureDetails.schema().form)
};

export const WithASingleCharacteristicPerHighlight = {
    render: args => <FeatureDetails data={tSheetsFeatureDetailsMock} settings={args} />,
    args: getArgs(FeatureDetails.schema().form)
};

export const WithMultipleCharacteristicsPerHighlight = {
    render: args => <FeatureDetails data={gSuiteFeatureDetailsMock} settings={args} />,
    args: getArgs(FeatureDetails.schema().form)
};

export const WithHtmlInADescription = {
    render: args => <FeatureDetails data={featureDetailsWithHTMLMock} settings={args} />,
    args: getArgs(FeatureDetails.schema().form)
};
