// @appdirect/sfb-theme-components/src/tools/components/productAdditionalInfo.js
export const hasTags = details => Boolean(details && details.tags && details.tags.length);

export const hasCustomAttributes = customAttributes => Boolean(customAttributes && customAttributes.length);

const hasRelatedContent = relatedContent => Boolean(
    relatedContent && relatedContent.details && hasTags(relatedContent.details)
);

export const hasSidebar = (
    relatedContent,
    customAttributes
) => hasRelatedContent(relatedContent) || hasCustomAttributes(customAttributes);

export const hasSupportInfo = info => {
    const {
        termsUrl,
        description,
        email,
        phone,
        privacyUrl,
        knowledgebaseUrl,
        documentationUrl
    } = info;
    return Boolean(
        info && (termsUrl || description || email || phone || privacyUrl || knowledgebaseUrl || documentationUrl)
    );
};

export const hasContactInfo = (...params) => params.some(param => param);

export const hasAdditionalInfo = data => !!data && (
    hasSupportInfo(data.support) || hasSidebar(data.relatedContent, data.customAttributes)
);
