// @appdirect/sfb-theme-components/src/composite-demos/product-configurator-page/stories/ProductConfiguratorPage.stories.js
import React from 'react';
import { productConfiguratorMock } from '../../../../mocks/product/productConfigurator';
import { footerDataMock } from '../../../../mocks/components/footer';
import { ProductConfiguratorPage } from '../ProductConfiguratorPage';
import ProductConfigurator from '../../../components/product-configurator/ProductConfigurator';
import Footer from '../../../components/footer/Footer';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

export default {
    title: 'Composite Demos/Product Configurator Page',
    component: ProductConfiguratorPage,
    argTypes: getArgTypes(Footer.schema().form),
    parameters: {
        layout: 'fullscreen',
        controls: {
            include: Object.keys(getArgTypes(Footer.schema().form))
        }
    }
};

export const Default = {
    render: args => (
        <ProductConfiguratorPage
            productConfigurator={<ProductConfigurator data={productConfiguratorMock} />}
            footer={<Footer data={footerDataMock} settings={args} />}
        />
    ),
    args: getArgs(Footer.schema().form)
};
