// @appdirect/sfb-theme-components/mocks/bundle/pricing.js
export const pricingMock = {
    editions: [
        {
            productName: 'AppInsights',
            id: 24562,
            fullName: 'AppInsights - Free Trial',
            name: 'Free Trial',
            value: 'Free'
        },
        {
            productName: 'ServiceM8',
            id: 398924,
            fullName: 'ServiceM8 - Premium',
            name: 'Premium',
            value: '$129.00',
            frequency: 'per Month'
        },
        {
            productName: 'PX-EVA Base Platform',
            id: 652311,
            fullName: 'PX-EVA Base Platform - Recurring Edition',
            name: 'Recurring Edition',
            value: '$70.00',
            frequency: 'per Year'
        },
        {
            productName: 'PX-EVA Budgeting Module',
            id: 652315,
            fullName: 'PX-EVA Budgeting Module - Recurring Edition',
            name: 'Recurring Edition',
            value: '$24.00',
            frequency: 'per Year'
        },
        {
            productName: 'PX-EVA Sustainability Module',
            id: 652320,
            fullName: 'PX-EVA Sustainability Module - Recurring Edition',
            name: 'Recurring Edition',
            value: '$24.00',
            frequency: 'per Year'
        },
        {
            productName: 'PX-EVA Real-time Monitoring Module',
            id: 652324,
            fullName: 'PX-EVA Real-time Monitoring Module - Recurring Edition',
            name: 'Recurring Edition',
            value: '$180.00',
            frequency: 'per Year'
        },
        {
            productName: 'PX-EVA Measurement & Verification Module',
            id: 652329,
            fullName: 'PX-EVA Measurement & Verification Module - Recurring Edition',
            name: 'Recurring Edition',
            value: '$24.00',
            frequency: 'per Year'
        },
        {
            productName: 'PX-EVA Peak Load Management Module',
            id: 652333,
            fullName: 'PX-EVA Peak Load Management Module - Recurring Edition',
            name: 'Recurring Edition',
            value: '$24.00',
            frequency: 'per Year'
        },
        {
            productName: 'PX-EVA Automated Utility Data Capture',
            id: 652338,
            fullName: 'PX-EVA Automated Utility Data Capture - Recurring Edition',
            name: 'Recurring Edition',
            value: '$36.00',
            frequency: 'per Year'
        }
    ]
};
