// @appdirect/sfb-theme-components/src/atoms/search-bar-v2/SearchBarV2.jsx
import React from 'react';
import PropTypes from 'prop-types';
import { useForm } from 'react-hook-form';
import InputSearch from '../input-search/InputSearch';
import { createNamespace } from '../../tools/namingTools';
import { imageUploaderShape } from '../../constants/schema/prop-types/image-uploader';
// IMPORTANT: (s)css should be imported after other atoms/components
import './styles/SearchBarV2.scss';

const SearchBarV2 = ({
    n,
    searchValue,
    inputName,
    label,
    onSearch,
    height,
    submitLabel,
    hiddenLabel,
    searchPlaceholderTextColor,
    searchBackgroundColor,
    searchBorderColor,
    searchCornerRadius,
    searchBorderVisible,
    submitIconOnly,
    searchIconColor,
    searchButtonColor,
    searchIcon,
    searchPlaceholder,
    searchBarMinWidth,
    searchBarMaxWidth,
    hasBoxShadow,
    extraClass,
    parentTestId
}) => {
    const { handleSubmit, register } = useForm();

    const styleVariables = {
        'min-width': searchBarMinWidth && `${searchBarMinWidth}px`,
        'max-width': searchBarMaxWidth && `${searchBarMaxWidth}px`
    };

    return (
        <div {...n('wrapper')
            .withVariables(styleVariables)
            .withClass(extraClass)
            .withTestId(parentTestId)
            .props}
        >
            <form {...n('form').props} onSubmit={handleSubmit(onSearch)}>
                <InputSearch
                    name={inputName}
                    label={label}
                    placeholder={searchPlaceholder}
                    hiddenLabel={hiddenLabel}
                    withBorder={searchBorderVisible}
                    borderRadius={searchCornerRadius}
                    borderColor={searchBorderColor}
                    backgroundColor={searchBackgroundColor}
                    searchPlaceholderTextColor={searchPlaceholderTextColor}
                    inputRef={register({})}
                    submitIconOnly={submitIconOnly}
                    submitIconColor={searchIconColor}
                    submitIconUrl={searchIcon?.url}
                    submitBgColor={searchButtonColor}
                    submitLabel={submitLabel}
                    height={height}
                    boxShadow={hasBoxShadow}
                    defaultValue={searchValue}
                    required={false}
                />
            </form>
        </div>
    );
};

SearchBarV2.propTypes = {
    n: PropTypes.func,
    inputName: PropTypes.string,
    label: PropTypes.string,
    searchValue: PropTypes.string,
    onSearch: PropTypes.func,
    height: PropTypes.number,
    submitLabel: PropTypes.string,
    hiddenLabel: PropTypes.bool,
    searchPlaceholderTextColor: PropTypes.string,
    searchBackgroundColor: PropTypes.string,
    searchBorderColor: PropTypes.string,
    searchCornerRadius: PropTypes.number,
    searchBorderVisible: PropTypes.bool,
    submitIconOnly: PropTypes.bool,
    searchIconColor: PropTypes.string,
    searchButtonColor: PropTypes.string,
    searchIcon: PropTypes.shape(imageUploaderShape),
    searchPlaceholder: PropTypes.string,
    searchBarMinWidth: PropTypes.number,
    searchBarMaxWidth: PropTypes.number,
    hasBoxShadow: PropTypes.bool,
    extraClass: PropTypes.string,
    parentTestId: PropTypes.string

};

SearchBarV2.defaultProps = {
    n: createNamespace('SearchBarV2'),
    inputName: 'inputSearch',
    label: 'Search',
    searchValue: null,
    onSearch: () => {},
    height: 40,
    submitLabel: 'Submit',
    hiddenLabel: true,
    searchPlaceholderTextColor: null,
    searchBackgroundColor: '#ffffff',
    searchBorderColor: '#cbcbcb',
    searchCornerRadius: 24,
    searchBorderVisible: true,
    submitIconOnly: true,
    searchIconColor: '#ffffff',
    searchButtonColor: '#0076ce',
    searchIcon: {},
    searchPlaceholder: 'Search',
    searchBarMinWidth: 225,
    searchBarMaxWidth: 500,
    hasBoxShadow: false,
    extraClass: null,
    parentTestId: null
};

export default SearchBarV2;
