// @appdirect/sfb-theme-components/src/constants/prop-types/bundle/items/overview.js
import PropTypes from 'prop-types';

const callToActionsPropTypes = {
    tour: PropTypes.shape({
        label: PropTypes.string,
        alternateLabel: PropTypes.string,
        color: PropTypes.string,
        href: PropTypes.string
    }),
    demo: PropTypes.shape({
        label: PropTypes.string,
        href: PropTypes.string,
        iFrameFriendlyUrl: PropTypes.string
    })
};

export const overviewPropTypes = {
    title: PropTypes.string,
    description: PropTypes.string,
    callToActions: PropTypes.shape(callToActionsPropTypes),
    image: PropTypes.string,
    imageSrcSet: PropTypes.shape({
        default: PropTypes.string
    }),
    showThirdPartyLegalDisclosure: PropTypes.bool,
    documentationUrl: PropTypes.string,
    downloadDocumentationUrl: PropTypes.string,
    pricingUrl: PropTypes.string,
    customers: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        imgSrc: PropTypes.string
    })),
    benefits: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.string,
        description: PropTypes.string,
        descriptionLong: PropTypes.string
    })),
    i18n: PropTypes.shape({
        customers: PropTypes.string,
        featuresAndBenefits: PropTypes.string,
        demo: PropTypes.string,
        overviewImage: PropTypes.string,
        documentation: PropTypes.string,
        availableAddons: PropTypes.string,
        availableBundles: PropTypes.string,
        seePricingPlans: PropTypes.string,
        startingFrom: PropTypes.string
    })
};
