// @appdirect/sfb-theme-components/mocks/components/featuredProduct.js
import { startingPriceFreeMock } from '../product/starting-price';
import { productListingMock } from '../product/productListing';
import { IMAGE_URL } from '../../src/constants/images';

const mockCustomImage = {
    filename: '',
    type: 'IMAGE_UPLOADER',
    url: 'http://placeimg.com/800/480/tech',
    alt: 'custom alt'
};

export const mockProduct = {
    ...productListingMock.items[0],
    name: 'Deputy',
    description: 'Deputy is all-in-one workforce management software that simplifies scheduling, timesheets, tasking and employee communication.',
    overviewImageUrl: IMAGE_URL
};

export const mockProductFree = {
    ...mockProduct,
    startingPrice: startingPriceFreeMock
};

export const featuredProductSettings = {
    product: mockProduct,
    size: 'small',
    tabTitle: 'Featured Product',
    tabBorderRadius: 8,
    tabBackgroundColor: '#474747',
    tabTextColor: '#FFFFFF',
    tabPosition: 'left',
    productTextAlignment: 'topLeft',
    productTitleColor: '#474747',
    productDescriptionColor: '#474747',
    productPricingColor: '#474747',
    productBackgroundColor: '#F5F5F5',
    cardBorderColor: '#FFFFFF',
    cardShadowColor: 'rgba(0,1,0,0.2)',
    cardBorderRadius: 8,
    isShadowVisible: true,
    isPriceVisible: true,
    isButtonVisible: true,
    customProductTitle: null,
    customProductDescription: null,
    buttonLabel: 'Shop now',
    buttonType: 'primary',
    imagePosition: 'left',
    imageBackgroundColor: '#EAEAEA',
    imageSize: 'scaleToFit',
    imageMask: 'noMask',
    containerBackgroundColor: '#EAEAEA'
};

export const featuredProductSettingsWithCustomImage = {
    ...featuredProductSettings,
    customImage: mockCustomImage
};

export const featuredProductSettingsNoImage = {
    ...featuredProductSettings,
    product: {
        ...mockProduct,
        overviewImageUrl: null
    },
    customImage: null
};
