// @appdirect/sfb-theme-components/mocks/components/footer.js
import { localeMock } from '../config/locale';
import { logoMock } from '../config/logo';
import { configMock } from '../config';
import { socialMediasMock } from '../socials/socialMedias';
import { MENUS } from '../../src/constants/menus';

const { FOOTER_MENU, BOTTOM_BAR_MENU } = MENUS;

export const footerDataMock = {
    links: [
        {
            href: 'support@example.com',
            label: 'Contact Us'
        },
        {
            href: 'https://www.example.com/?help_center',
            label: 'Help Center'
        },
        {
            href: 'https://www.example.com/?privacy',
            label: 'Privacy Policy'
        },
        {
            href: 'https://www.example.com/?terms&test=1',
            label: 'Terms & Conditions'
        }
    ],
    support: {
        contactUrl: '',
        email: 'support@example.com',
        phone: '',
        url: 'https://www.example.com/?help_center'
    },
    locale: localeMock,
    marketplaceName: configMock.marketplaceName,
    logo: logoMock,
    i18n: {
        followUs: 'Follow Us',
        menuTitle: 'Footer Menu Navigation',
        navigationTitle: 'Footer navigation',
        languages: 'Choose a language'
    }
};

export const footerMenusMock = {
    [FOOTER_MENU]: {
        menuType: 'MENU_LINK',
        menu: [
            {
                uuid: '50b2a008-d1df-42fa-9bdf-17a8446210d6',
                title: {
                    value: 'Menu 1',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: 'http://test?locale=en-us',
                    isLocalized: true,
                    type: 'INPUT'
                },
                type: 'CUSTOM_LINK',
                openInNewTab: true,
                isVisible: true,
                isDisabled: false,
                children: [
                    {
                        uuid: '89b4d743-b405-43a0-b6ab-9877fcdee455',
                        title: {
                            value: 'Sub-link 1',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: 'http://test?locale=en-us',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: true,
                        isDisabled: false
                    },
                    {
                        uuid: 'e96e3929-2bea-4283-8139-9fee0053487c',
                        title: {
                            value: 'Sub-link 2',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: 'http://test?locale=en-us',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: false,
                        isDisabled: false
                    },
                    {
                        uuid: '03a90d85-ce76-4ba6-bafb-88abdd8e1bc6',
                        title: {
                            value: 'Sub-link 3',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: 'http://test?locale=en-us',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: true,
                        isDisabled: false
                    }
                ]
            },
            {
                uuid: '5c7cf78e-bb0a-4704-b7ba-0d1aa810212a',
                title: {
                    value: 'Menu 2',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: '',
                    isLocalized: true,
                    type: 'INPUT'
                },
                type: 'CUSTOM_LINK',
                openInNewTab: true,
                isVisible: true,
                isDisabled: false
            },
            {
                uuid: '50b2a008-d1df-42fa-9bdf-17a8446210d9',
                title: {
                    value: 'Menu 3',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: 'http://test?locale=en-us',
                    isLocalized: true,
                    type: 'INPUT'
                },
                type: 'CUSTOM_LINK',
                openInNewTab: true,
                isVisible: true,
                isDisabled: false,
                children: [
                    {
                        uuid: '89b4d743-b405-43a0-b6ab-9877fcdee456',
                        title: {
                            value: 'Sub-link 4',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: 'http://test?locale=en-us',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: true,
                        isDisabled: false
                    },
                    {
                        uuid: 'e96e3929-2bea-4283-8139-9fee00534878',
                        title: {
                            value: 'Sub-link 5',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: 'http://test?locale=en-us',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: true,
                        isDisabled: false
                    },
                    {
                        uuid: '03a90d85-ce76-4ba6-bafb-88abdd8e1bc9',
                        title: {
                            value: 'Sub-link 6',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: 'http://test?locale=en-us',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: true,
                        isDisabled: false
                    }
                ]
            },
            {
                uuid: '50b2a008-d1df-42fa-9bdf-17a8446210d3',
                title: {
                    value: 'Menu 4',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: 'http://test?locale=en-us',
                    isLocalized: true,
                    type: 'INPUT'
                },
                type: 'CUSTOM_LINK',
                openInNewTab: true,
                isVisible: true,
                isDisabled: false
            }
        ]
    },
    [BOTTOM_BAR_MENU]: {
        menuType: 'MENU_LINK',
        menu: [
            {
                uuid: '93edf021-a3e3-4686-b686-e7e41e84f61e',
                title: {
                    value: 'Contact Us',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: 'support@example.com',
                    isLocalized: true,
                    type: 'INPUT'
                },
                editorType: 'CONTACT_US',
                type: 'CUSTOM_LINK',
                openInNewTab: true,
                isVisible: true,
                isDisabled: false,
                children: [
                    {
                        uuid: 'b32df1c2-cd55-41f2-b5f3-b7b0e33627f4',
                        title: {
                            value: 'Visible sub-item 1A',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: 'http://test/visible-sub-item-1a?locale=en-us',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: true,
                        isVisible: true,
                        isDisabled: false
                    },
                    {
                        uuid: '5e1b1318-ebfd-4848-aa38-f2ea02b2288c',
                        title: {
                            value: 'Hidden sub-item 1B',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: '',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: false,
                        isDisabled: false
                    },
                    {
                        uuid: '427c705a-61f9-4367-a155-b5ca0c244a0e',
                        title: {
                            value: 'Visible sub-item 1C',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: 'http://test/visible-sub-item-1c?locale=en-us',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: true,
                        isVisible: true,
                        isDisabled: false
                    }
                ]
            },
            {
                uuid: 'f5223632-235f-48df-b096-ea3463b20101',
                title: {
                    value: 'Help Center',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: 'https://www.example.com/?help_center',
                    isLocalized: true,
                    type: 'INPUT'
                },
                type: 'CUSTOM_LINK',
                editorType: 'HELP_CENTER',
                openInNewTab: true,
                isVisible: true,
                isDisabled: false,
                children: [
                    {
                        uuid: 'b7dd0b0a-9c96-4f56-8f17-50dee5583491',
                        title: {
                            value: 'Visible sub-item 2A',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: 'http://test/visible-sub-item-2a?locale=en-us',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: true,
                        isDisabled: false
                    }
                ]
            },
            {
                uuid: 'd5ff2276-79ad-4c76-9c64-6bd5d2dcd008',
                title: {
                    value: 'Privacy Policy',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: 'https://www.example.com/?privacy',
                    isLocalized: true,
                    type: 'INPUT'
                },
                editorType: 'PRIVACY_POLICY',
                type: 'CUSTOM_LINK',
                openInNewTab: false,
                isVisible: false,
                isDisabled: false
            },
            {
                uuid: '2817030d-46b5-42f6-812b-792f12022590',
                title: {
                    value: 'Terms & Conditions',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: 'https://www.example.com/?terms&test=1',
                    isLocalized: true,
                    type: 'INPUT'
                },
                editorType: 'TERMS_AND_CONDITIONS',
                type: 'CUSTOM_LINK',
                openInNewTab: false,
                isVisible: false,
                isDisabled: false,
                children: [
                    {
                        uuid: '9e3b4f0b-f6d3-4ab5-aeae-fa154a5bba1a',
                        title: {
                            value: 'Visible sub-item 4A static',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: '',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: true,
                        isDisabled: false
                    }
                ]
            },
            {
                uuid: 'bbbff771-d186-4ae0-9f66-6abb3c5dc794',
                title: {
                    value: 'Hidden item with all hidden sub-items',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: '',
                    isLocalized: true,
                    type: 'INPUT'
                },
                type: 'CUSTOM_LINK',
                openInNewTab: false,
                isVisible: false,
                isDisabled: false,
                children: [
                    {
                        uuid: '1b156929-6e73-4286-b0d7-8de243541534',
                        title: {
                            value: 'Hidden sub-item 3A',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        link: {
                            value: '',
                            isLocalized: true,
                            type: 'INPUT'
                        },
                        type: 'CUSTOM_LINK',
                        openInNewTab: false,
                        isVisible: false,
                        isDisabled: false
                    }
                ]
            },
            {
                uuid: '6711c944-8733-45ab-beb8-3422bf7d5683',
                title: {
                    value: 'Language',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: '',
                    isLocalized: true,
                    type: 'INPUT'
                },
                editorType: 'LANGUAGE',
                type: 'CUSTOM_LINK',
                openInNewTab: false,
                isVisible: true,
                isDisabled: false
            },
            {
                uuid: '852ddfca-cb29-431b-91a8-c53849bce6aa',
                title: {
                    value: 'Visible Item 3 no sub-item',
                    isLocalized: true,
                    type: 'INPUT'
                },
                link: {
                    value: '',
                    isLocalized: true,
                    type: 'INPUT'
                },
                type: 'CUSTOM_LINK',
                openInNewTab: true,
                isVisible: true,
                isDisabled: false
            }
        ]
    }
};

export const footerSocialsMock = {
    items: socialMediasMock
};
