// @appdirect/sfb-theme-components/mocks/components/leadForm.js
import { merge } from 'lodash'; // eslint-disable-line import/no-extraneous-dependencies
import { leadsMock as baseFormMock } from '../application/leads/index';

export const leadFormMock = merge({}, baseFormMock, {
    form: {
        actionEndpoint: 'https://marketplace.appdirect.com/api/lead/v2/leads?context=ANONYMOUS',
        schema: {
            firstName: {
                isRequired: false,
                validators: []
            },
            lastName: {
                isRequired: false,
                validators: []
            },
            email: {
                isRequired: true
            },
            companyName: {
                isRequired: true
            },
            companySize: {
                isRequired: false,
                validators: []
            },
            country: {
                isRequired: false,
                validators: []
            }
        }
    }
});

export const leadFormV1Mock = merge({}, baseFormMock, {
    form: {
        actionEndpoint: 'https://marketplace.appdirect.com/api/prm/v1/leads?currency=USD&country=US',
        schema: {
            firstName: {
                isRequired: true,
                validators: [
                    'trimmedRequired'
                ]
            },
            lastName: {
                isRequired: true,
                validators: [
                    'trimmedRequired'
                ]
            },
            email: {
                isRequired: true
            },
            companyName: {
                isRequired: true
            },
            companySize: {
                isRequired: true,
                validators: [
                    'required'
                ]
            },
            country: {
                isRequired: true,
                validators: ['required']
            }
        }
    }
});

export const leadFormMockWithoutDefaultValues = {
    ...leadFormMock,
    form: {
        ...leadFormMock.form,
        userAndCompanyInfo: {}
    }
};

export const leadFormMockWithoutDefaultValuesLeadsV1 = {
    ...leadFormV1Mock,
    form: {
        ...leadFormV1Mock.form,
        userAndCompanyInfo: {}
    }
};

export const leadFormMockUS = {
    ...leadFormMock,
    form: {
        ...leadFormMock.form,
        userAndCompanyInfo: {
            ...leadFormMock.form.userAndCompanyInfo,
            city: 'New York City'
        },
        schema: {
            ...leadFormMock.form.schema,
            country: {
                ...leadFormMock.form.schema.country,
                options: leadFormMock.form.schema.country.options.map(_country => ({
                    ..._country,
                    isSelected: _country.value === 'US'
                }))
            },
            state: {
                ...leadFormMock.form.schema.state,
                options: leadFormMock.form.schema.state.options.map(_state => ({
                    ..._state,
                    isSelected: _state.value === 'CA'
                }))
            }
        }
    }
};

export const leadFormMockCanada = {
    ...leadFormMock,
    form: {
        ...leadFormMock.form,
        userAndCompanyInfo: {
            firstName: 'Test',
            lastName: 'User',
            email: 'test.user@example.com',
            companyName: 'AppDirect',
            phone: '514-123-1234',
            city: 'Montreal',
            street1: '123 Fake St.',
            street2: 'suite 123',
            zip: 'H1H 1H1'
        },
        schema: {
            ...leadFormMock.form.schema,
            country: {
                ...leadFormMock.form.schema.country,
                options: leadFormMock.form.schema.country.options.map(_country => ({
                    ..._country,
                    isSelected: _country.value === 'CA'
                }))
            },
            province: {
                ...leadFormMock.form.schema.province,
                options: leadFormMock.form.schema.province.options.map(_province => ({
                    ..._province,
                    isSelected: _province.value === 'QC'
                }))
            }
        }
    }
};

export const leadFormMockAustralia = {
    ...leadFormMock,
    form: {
        ...leadFormMock.form,
        userAndCompanyInfo: {
            firstName: 'Test',
            lastName: 'User',
            email: 'test.user@example.com',
            companyName: 'AppDirect',
            phone: '12345 67890',
            city: 'Sydney',
            street1: '123 Fake St.',
            street2: 'suite 123',
            zip: '2000'
        },
        schema: {
            ...leadFormMock.form.schema,
            country: {
                ...leadFormMock.form.schema.country,
                options: leadFormMock.form.schema.country.options.map(_country => ({
                    ..._country,
                    isSelected: _country.value === 'AU'
                }))
            }
        }
    }
};
