// @appdirect/sfb-theme-components/mocks/components/productViewerDescription.js
import { addonProductsMock } from '../product/addon-products';
import { productOverviewMock } from '../product/overview';
import { startingPriceMock } from '../product/starting-price';
import { screenshotsMock } from '../product/screenshots';
import { youtubeMock } from '../product/overview/callToActions';

export const defaultMock = {
    addonProducts: addonProductsMock,
    overview: productOverviewMock,
    startingPrice: startingPriceMock
};

export const multipleScreenshotsMock = {
    ...defaultMock,
    screenshots: screenshotsMock
};

export const singleScreenshotsMock = {
    ...defaultMock,
    screenshots: {
        items: screenshotsMock.items.slice(0, 1)
    }
};

export const singleScreenshotWithoutOverviewMock = {
    ...defaultMock,
    overview: {
        ...defaultMock.overview,
        image: '',
        imageSrcSet: {}
    },
    screenshots: {
        items: screenshotsMock.items.slice(0, 1)
    }
};

export const singleVideoMock = {
    ...defaultMock,
    screenshots: {},
    overview: {
        ...defaultMock.overview,
        callToActions: {
            ...defaultMock.overview.callToActions,
            demo: youtubeMock
        }
    }
};
