// @appdirect/sfb-theme-components/src/atoms/product-groups-tiles/graphic-style/stories/GraphicStyle.stories.js
import React from 'react';
import GraphicStyle from '../../../../constants/atoms/schema/product-groups-tiles/graphic-style';
import { defaultValues } from '../../../../../mocks/product/productGroupsTiles';
import { getArgTypes, getArgs } from '../../../../tools/controlsHelper';

export default {
    title: 'Atoms/ProductGroupsStyles/GraphicStyle',
    component: GraphicStyle,
    argTypes: {
        customImageUrl: {
            control: 'text',
            description: 'The URL of the custom image to display'
        },
        ...getArgTypes(GraphicStyle.schema().form)
    },
    render: ({ customImageUrl, ...args }) => (
        <GraphicStyle
            {...defaultValues}
            {...args}
            customImage={{ url: customImageUrl }}
        />
    ),
    parameters: {
        controls: {
            include: [...Object.keys(GraphicStyle.schema().form), 'customImageUrl']
        }
    }
};

export const Default = {
    args: {
        customImageUrl: defaultValues.customImage.defaultValue.url,
        ...getArgs(GraphicStyle.schema().form),
        ...defaultValues
    },
    parameters: {
        controls: {
            exclude: ['customImage']
        }
    }
};
