// @appdirect/sfb-theme-components/src/constants/components/schemas/search-bar.js
import { COLORPICKER, DROPDOWN, GROUP_HEADER, IMAGE_UPLOADER, INPUT, RADIO, SLIDER, TOGGLE } from '../../schemaComponentTypes';
import { SPECIAL_CHARACTERS_BLOCKLIST } from '../../regexConstants';
import { SEARCH_BAR_HEIGHTS, SEARCH_BAR_LOCATIONS, SEARCH_BAR_IN_HOME_BAR, SEARCH_BAR_WIDTH } from '../header';
import { MEDIUM } from '../../sizes';

export const searchBarDefaultSettings = {
    searchBorderVisible: true,
    searchCornerRadius: 3,
    searchBarLocation: SEARCH_BAR_IN_HOME_BAR.value,
    searchBarMaxWidth: 500,
    searchBarHeight: MEDIUM.value
};

export const searchBarSchemaForm = {
    visibilityHeader: {
        type: GROUP_HEADER,
        title: 'uieditor.sfbComponent.searchBar.visibilityHeader.title',
        symbol: 'settings'
    },
    searchBarLocation: {
        type: RADIO,
        defaultValue: searchBarDefaultSettings.searchBarLocation,
        options: SEARCH_BAR_LOCATIONS
    },
    settingsHeader: {
        type: GROUP_HEADER,
        title: 'uieditor.sfbComponent.searchBar.settingsHeader.title',
        symbol: 'settings'
    },
    searchBarMaxWidth: {
        title: 'uieditor.sfbComponent.header.searchBar.maxWidth.title',
        type: SLIDER,
        defaultValue: searchBarDefaultSettings.searchBarMaxWidth,
        options: {
            range: true,
            step: 1,
            ...SEARCH_BAR_WIDTH
        }
    },
    searchCornerRadius: {
        title: 'uieditor.sfbComponent.searchBar.cornerRadius.title',
        type: SLIDER,
        required: false,
        defaultValue: searchBarDefaultSettings.searchCornerRadius,
        options: {
            step: 1,
            min: 1,
            max: 24
        }
    },
    searchBarHeight: {
        title: 'uieditor.sfbComponent.header.searchBar.height.title',
        defaultValue: searchBarDefaultSettings.searchBarHeight,
        type: DROPDOWN,
        options: SEARCH_BAR_HEIGHTS
    },
    searchPlaceholder: {
        title: 'uieditor.sfbComponent.searchBar.placeholder.title',
        type: INPUT,
        validation: {
            pattern: SPECIAL_CHARACTERS_BLOCKLIST,
            message: 'uieditor.sfbComponent.input.validation.message'
        }
    },
    searchIcon: {
        title: 'uieditor.sfbComponent.searchBar.searchIcon.title',
        type: IMAGE_UPLOADER,
        placeholder: 'uieditor.sfbComponent.searchBar.searchIcon.placeholder',
        caption: 'uieditor.sfbComponent.header.imageUpload.caption',
        defaultValue: {}
    },
    searchBorderVisible: {
        labelOn: 'uieditor.sfbComponent.searchBar.borderVisible.labelOn',
        labelOff: 'uieditor.sfbComponent.searchBar.borderVisible.labelOff',
        defaultValue: searchBarDefaultSettings.searchBorderVisible,
        type: TOGGLE
    },
    searchBorderColor: {
        title: 'uieditor.sfbComponent.searchBar.borderColor.title',
        type: COLORPICKER,
        conditionalAttributes: {
            isVisible: {
                when: 'searchBorderVisible',
                connector: 'equals',
                value: true
            }
        }
    },
    searchBackgroundColor: {
        title: 'uieditor.sfbComponent.searchBar.backgroundColor.title',
        type: COLORPICKER
    },
    searchButtonColor: {
        title: 'uieditor.sfbComponent.searchBar.buttonColor.title',
        type: COLORPICKER
    },
    searchIconColor: {
        title: 'uieditor.sfbComponent.searchBar.iconColor.title',
        type: COLORPICKER
    }
};
