// @appdirect/sfb-theme-components/mocks/listingProducts/i18n.js
export const i18nMock = {
    allApplications: 'All Applications',
    applySortOptions: 'Apply',
    backToTop: 'Back to top',
    categories: 'Categories',
    compare: 'Compare',
    filters: 'Filters',
    grid: 'Grid',
    hideFilters: 'Hide Filters',
    home: 'Home',
    learnMore: 'Learn More',
    list: 'List',
    next: 'Next',
    previous: 'Previous',
    rating: 'Rating',
    results: 'Results',
    resultsFrom: 'Results from',
    searchLabel: 'Showing 25 Applications',
    selectAppsCompare: 'Select 2-4 products to compare',
    selectToCompare: 'Select to compare',
    showFilters: 'Show Filters',
    soldBy: 'Sold By',
    from: 'From',
    badges: 'Badges',
    sort: 'Sort',
    sortBy: 'Sort By',
    startingAt: 'Starting At',
    title: 'Find Applications | sfbmarketplace',
    titleProduct: 'Find Products | sfbmarketplace',
    viewMode: 'View Mode',
    viewMore: 'View More',
    viewProfile: 'View Profile',
    version: 'Version',
    merchandisingBadges: 'Merchandising Badges',
    breadcrumbs: {
        label: 'Breadcrumbs',
        categories: {
            label: 'Categories'
        },
        attributes: {
            label: 'Attributes'
        },
        productLine: {
            label: 'Product Line'
        }
    }
};
