// @appdirect/sfb-theme-components/src/atoms/grid-container/stories/GridContainer.stories.js
import React from 'react';
import GridContainer from '../GridContainer';

export default {
    title: 'Atoms/GridContainer',
    component: GridContainer
};

export const Default = {
    render: () => (
        <GridContainer>
            <section
                className="test"
                style={{
                    margin: '0 auto',
                    width: 'calc(100% / 12 * 8)'
                }}
            >
                <div
                    style={{
                        padding: '10px',
                        border: '1px solid red'
                    }}
                >
                    <p>Use this css rule on your wrapper to set the number of column.</p>
                    <p>This is an example for 8 columns wide</p>
                    <code>width: calc(100% / 12 \* 8)</code>
                </div>
            </section>
        </GridContainer>
    ),

    name: 'Default'
};

export const ContentSidebar = {
    render: () => (
        <GridContainer>
            <div
                style={{
                    width: 'calc(100% / 12 * 8)'
                }}
            >
                <div
                    style={{
                        padding: '10px',
                        border: '1px solid red'
                    }}
                >
                    <span>Content</span>
                    <code
                        style={{
                            display: 'block',
                            margin: '25px 0 10px'
                        }}
                    >
                        width: calc(100% / 12 \* 8)
                    </code>
                </div>
            </div>
            <div
                style={{
                    width: 'calc(100% / 12 * 4)'
                }}
            >
                <div
                    style={{
                        padding: '10px',
                        border: '1px solid red'
                    }}
                >
                    <span>Sidebar</span>
                    <code
                        style={{
                            display: 'block',
                            margin: '25px 0 10px'
                        }}
                    >
                        width: calc(100% / 12 \* 4)
                    </code>
                </div>
            </div>
        </GridContainer>
    ),

    name: 'Content + sidebar'
};
