// @appdirect/sfb-theme-components/.storybook/preview.js
import React from 'react'
import { themes } from '@storybook/theming';
import * as AllComponents from '../src/components' 
import './styles/storybook.scss';
import '../src/styles/global.scss';
import '../suppress-react-warnings';

export const decorators = [
    (Story, context) => {
        const isContainerFluid =  context.parameters && context.parameters.container && context.parameters.container === 'fluid';
        const containerStyles = isContainerFluid ? { paddingLeft: 0, paddingRight: 0 } : {};
        const containerClass = isContainerFluid ? 'container-fluid': 'container';
        const rowClass = `row${isContainerFluid ? ' no-gutters' : ''}`;

        // on real build, this comes from Webpack
        // for Storybook, has to be attached to window globally for all stories
        window.__ADComponents = AllComponents;

        return (
            <div className={containerClass} style={containerStyles}>
                <div className={rowClass}>
                    <div className="col-12"><Story /></div>
                </div>
            </div>
        );
    }
];
export const parameters = {
    options: {
      storySort: {
        order: ['Intro','Docs', ['Getting Started', 'Customizations'],  'Styles', 'Atoms', 'Components', 'Composite Demos']
      },
    },
    knobs: {
        escapeHTML: false,
        timestamps: true,
    },
    controls: {
        expanded: true,
    },
    darkMode: {
        dark: themes.dark,
        light: themes.normal
  }
  };
