// @appdirect/sfb-theme-components/mocks/socials/socialMedias.js
export const socialMediasMock = [
    {
        label: 'Facebook',
        icon: 'facebook',
        isVisible: true,
        values: {
            accountLink: {
                value: 'https://www.facebook.com'
            },
            uploadIcon: {
                value: {
                    filename: 'facebook',
                    url: 'icon.url'
                }
            }
        }
    },
    {
        label: 'Twitter',
        icon: 'twitter',
        isVisible: true,
        values: {
            accountLink: {
                value: 'https://www.twitter.com'
            },
            uploadIcon: {
                value: {
                    filename: 'twitter',
                    url: 'icon.url'
                }
            }
        }
    },
    {
        label: 'Youtube',
        icon: 'youtube',
        isVisible: true,
        values: {
            accountLink: {
                value: 'https://www.youtube.com'
            },
            uploadIcon: {
                value: {
                    filename: 'youtube',
                    url: 'icon.url'
                }
            }
        }
    },
    {
        label: 'Instagram',
        icon: 'instagram',
        isVisible: true,
        values: {
            accountLink: {
                value: 'https://www.instagram.com'
            },
            uploadIcon: {
                value: {
                    filename: 'instagram',
                    url: 'icon.url'
                }
            }
        }
    },
    {
        label: 'Linkedin',
        icon: 'linkedin',
        isVisible: true,
        values: {
            accountLink: {
                value: 'https://www.linkedin.com'
            },
            uploadIcon: {
                value: {
                    filename: 'linkedin',
                    url: 'icon.url'
                }
            }
        }
    },
    {
        label: 'Pinterest',
        icon: 'pinterest',
        isVisible: false,
        values: {
            accountLink: {
                value: 'https://www.pinterest.com'
            },
            uploadIcon: {
                value: {
                    filename: 'pinterest',
                    url: 'icon.url'
                }
            }
        }
    }
];
