// @appdirect/sfb-theme-components/src/constants/prop-types/application/domains/i18n.js
import PropTypes from 'prop-types';

export const i18nPropTypes = {
    sectionTitle: PropTypes.string,
    search: PropTypes.string,
    recommendedTitle: PropTypes.string,
    recommendedPlaceholder: PropTypes.string,
    add: PropTypes.string,
    added: PropTypes.string,
    addedTitle: PropTypes.string,
    addedPlaceholder: PropTypes.string,
    checkout: PropTypes.string,
    frequency: PropTypes.string,
    remove: PropTypes.string,
    select: PropTypes.string,
    selected: PropTypes.string,
    selectedDomains: PropTypes.string,
    addToCart: PropTypes.string,
    buyNow: PropTypes.string,
    noResultMessageTitle: PropTypes.string,
    noResultMessageDescription: PropTypes.string,
    skipBuyingDomain: PropTypes.string,
    domainFound: PropTypes.string,
    domainsFound: PropTypes.string,
    recommendedDomainName: PropTypes.string,
    addedDomainName: PropTypes.string,
    domainPrice: PropTypes.string,
    domainAction: PropTypes.string,
    selectedDomain: PropTypes.string,
    selectDomain: PropTypes.string,
    removeDomain: PropTypes.string
};
