// @appdirect/sfb-theme-components/src/components/global-alerts/stories/GlobalAlert.stories.js
import React from 'react';
import { GlobalAlerts } from '../GlobalAlerts';
import { productAlertsMock } from '../../../../mocks/product/productAlerts';
import { ALERT_LEAD_ADD_SUCCESS,
    ALERT_QUESTION_ADD_SUCCESS,
    ALERT_QUESTION_ADD_ERROR,
    ALERT_REVIEW_ADD_SUCCESS,
    addAlert } from '../../../tools/alerts';

const QuestionSuccess = () => {
    addAlert(ALERT_QUESTION_ADD_SUCCESS);
    return <GlobalAlerts data={productAlertsMock} persist />;
};

const QuestionError = () => {
    addAlert(ALERT_QUESTION_ADD_ERROR);
    return <GlobalAlerts data={productAlertsMock} persist />;
};

const LeadSuccess = () => {
    addAlert(ALERT_LEAD_ADD_SUCCESS);
    return <GlobalAlerts data={productAlertsMock} persist />;
};

const ReviewSuccess = () => {
    addAlert(ALERT_REVIEW_ADD_SUCCESS);
    return <GlobalAlerts data={productAlertsMock} persist />;
};

const MultipleAlerts = () => {
    addAlert(ALERT_REVIEW_ADD_SUCCESS);
    addAlert(ALERT_QUESTION_ADD_ERROR);
    addAlert(ALERT_LEAD_ADD_SUCCESS);
    return <GlobalAlerts data={productAlertsMock} persist />;
};

export default {
    title: 'Components/Global Alert',
    component: GlobalAlerts
};

export const QuestionSuccessStory = {
    render: () => <QuestionSuccess />,
    name: 'Question success',
    inline: false
};

export const QuestionErrorStory = {
    render: () => <QuestionError />,
    name: 'Question error',
    inline: false
};

export const LeadSuccessStory = {
    render: () => <LeadSuccess />,
    name: 'Lead success',
    inline: false
};

export const ReviewSuccessStory = {
    render: () => <ReviewSuccess />,
    name: 'Review success',
    inline: false
};

export const MultipleAlertsStory = {
    render: () => <MultipleAlerts />,
    name: 'Multiple alerts',
    inline: false,
    height: '250px'
};
