// @appdirect/sfb-theme-components/src/components/product-configurator-v2/stories/ProductConfigurator.stories.js
import React from 'react';
import ProductConfiguratorV2 from '../ProductConfiguratorV2';
import { productConfiguratorMock,
    productConfiguratorNoAddToCartCartMock,
    productConfiguratorBackFromCartMock,
    productConfiguratorNoSelectedEditionsMock,
    productConfiguratorWithoutAddons,
    productConfiguratorUpgradeMock } from '../../../../mocks/product/productConfigurator';

export default {
    title: 'Components (Preview)/Product Configurator V2/Product Configurator V2',
    component: ProductConfiguratorV2
};

export const Default = {
    render: () => <ProductConfiguratorV2 data={productConfiguratorMock} />,
    name: 'Default'
};

export const WithoutAddToCart = {
    render: () => <ProductConfiguratorV2 data={productConfiguratorNoAddToCartCartMock} />,
    name: 'Without add to cart'
};

export const BackFromCart = {
    render: () => <ProductConfiguratorV2 data={productConfiguratorBackFromCartMock} />,
    name: 'Back from cart'
};

export const WithoutSelectedEdition = {
    render: () => <ProductConfiguratorV2 data={productConfiguratorNoSelectedEditionsMock} />,
    name: 'Without selected edition'
};

export const WithoutAddons = {
    render: () => <ProductConfiguratorV2 data={productConfiguratorWithoutAddons} />,
    name: 'Without addons'
};

export const UpgradeButton = {
    render: () => <ProductConfiguratorV2 data={productConfiguratorUpgradeMock} />,
    name: 'Upgrade Button'
};
