// @appdirect/sfb-theme-components/mocks/menu/categories.js
export const categoriesMock = [
    {
        id: 2321,
        name: 'Web Development',
        subCategories: [
            {
                id: 16575,
                name: 'Server Side',
                numTaggedProducts: 58,
                selected: false,
                href: '/en-US/browse/category/web-development/2321/server-side/16575'
            },
            {
                id: 16577,
                name: 'Framework',
                numTaggedProducts: 47,
                selected: false,
                href: '/en-US/browse/category/web-development/2321/framework/16577'
            },
            {
                id: 16578,
                name: 'Database',
                numTaggedProducts: 30,
                selected: false,
                href: '/en-US/browse/category/web-development/2321/database/16578'
            },
            {
                id: 16579,
                name: 'Design',
                numTaggedProducts: 37,
                selected: false,
                href: '/en-US/browse/category/web-development/2321/design/16579'
            },
            {
                id: 16580,
                name: 'test123',
                numTaggedProducts: 19,
                selected: false,
                href: '/en-US/browse/category/web-development/2321/test123/16580'
            },
            {
                id: 16581,
                name: 'SEO',
                numTaggedProducts: 10,
                selected: false,
                href: '/en-US/browse/category/web-development/2321/seo/16581'
            },
            {
                id: 22761,
                name: 'Content Management',
                numTaggedProducts: 8,
                selected: false,
                href: '/en-US/browse/category/web-development/2321/content-management/22761'
            },
            {
                id: 16582,
                name: 'test321',
                numTaggedProducts: 19,
                selected: false,
                href: '/en-US/browse/category/web-development/2321/test123/16580'
            }
        ],
        numTaggedProducts: 548,
        selected: false,
        hrefListing: '/en-US/browse/category/web-development/2321',
        childSelected: false
    },
    {
        id: 2359,
        name: 'Office 365 Small business',
        subCategories: [
            {
                id: 17204,
                name: 'Power Point EDITING',
                numTaggedProducts: 6,
                selected: false,
                hrefListing: '/en-US/browse/category/office-365-small-business/2359/power-point-editing/17204'
            }
        ],
        numTaggedProducts: 110,
        selected: false,
        hrefListing: '/en-US/browse/category/office-365-small-business/2359',
        childSelected: false
    },
    {
        id: 29217,
        name: 'Home Office',
        subCategories: [],
        numTaggedProducts: 9,
        selected: false,
        hrefListing: '/en-US/browse/category/home-office/29217',
        childSelected: false
    },
    {
        id: 42116,
        name: 'Pro',
        subCategories: [],
        numTaggedProducts: 1,
        selected: false,
        hrefListing: '/en-US/browse/category/pro/42116',
        childSelected: false
    },
    {
        id: 42117,
        name: 'Educational',
        subCategories: [],
        numTaggedProducts: 2,
        selected: false,
        hrefListing: '/en-US/browse/category/educational/42117',
        childSelected: false
    },
    {
        id: 42248,
        name: 'Silver',
        subCategories: [],
        numTaggedProducts: 2,
        selected: false,
        hrefListing: '/en-US/browse/category/silver/42248',
        childSelected: false
    }
];
