// @appdirect/sfb-theme-components/mocks/menu/link-section.js
export const linkSectionDataMock = {
    items: [
        {
            linkInfo: [
                {
                    text: 'Manage your account',
                    link: '#'
                },
                {
                    text: 'Change your password',
                    link: '#'
                },
                {
                    text: 'Profile settings',
                    link: '#'
                },
                {
                    text: 'Delete my account',
                    link: '#'
                }
            ],
            header: {
                name: 'Account',
                navigationLabel: 'Account links',
                iconType: 'user'
            }
        },
        {
            linkInfo: [
                {
                    text: 'Browse your invoices',
                    link: '#'
                },
                {
                    text: 'Modify your payment data',
                    link: '#'
                },
                {
                    text: 'Cancel a payment',
                    link: '#'
                }
            ],
            header: {
                name: 'Payment',
                navigationLabel: 'Payment links',
                iconType: 'payment'
            }
        },
        {
            linkInfo: [
                {
                    text: 'Browse your products',
                    link: '#'
                },
                {
                    text: 'Cancel your subscription',
                    link: '#'
                }
            ],
            header: {
                name: 'Products',
                navigationLabel: 'Products links',
                iconType: 'productTile'
            }
        }
    ]
};
