// @appdirect/sfb-theme-components/src/constants/prop-types/bundle/items/index.js
import PropTypes from 'prop-types';
import { overviewPropTypes } from './overview';
import { resourcesPropTypes } from './resources';
import { screenshotsPropTypes } from './screenshots';
import { supportPropTypes } from './support';

const bundlePropTypes = {
    id: PropTypes.number,
    url: PropTypes.string
};

const cardPropTypes = {
    details: PropTypes.string,
    logoSrc: PropTypes.string,
    logoSrcset: PropTypes.shape({
        '1x': PropTypes.string,
        '2x': PropTypes.string,
        default: PropTypes.string
    }),
    merchandisingBadges: PropTypes.arrayOf(PropTypes.shape({
        image: PropTypes.string,
        name: PropTypes.string,
        uuid: PropTypes.string
    })),
    overview: PropTypes.string,
    productRibbon: PropTypes.shape({
        text: PropTypes.string,
        textColor: PropTypes.string,
        ribbonColor: PropTypes.string
    }),
    title: PropTypes.string
};

export const itemPropTypes = {
    bundle: PropTypes.shape(bundlePropTypes),
    card: PropTypes.shape(cardPropTypes),
    customAttributes: PropTypes.arrayOf(PropTypes.shape({
        attributeKey: PropTypes.string,
        label: PropTypes.string,
        type: PropTypes.string,
        value: PropTypes.string
    })),
    edition: PropTypes.shape({
        name: PropTypes.string
    }),
    focused: PropTypes.bool,
    id: PropTypes.number,
    image: PropTypes.string,
    name: PropTypes.string,
    overview: PropTypes.shape(overviewPropTypes),
    resources: PropTypes.shape(resourcesPropTypes),
    screenshots: PropTypes.shape(screenshotsPropTypes),
    support: PropTypes.shape(supportPropTypes),
    tags: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
        type: PropTypes.string,
        url: PropTypes.string
    })),
    url: PropTypes.string
};
