// @appdirect/sfb-theme-components/src/atoms/link/stories/Link.stories.js
import React from 'react';
import Link from '../Link';
import Badge from '../../badge/Badge';
import Icon from '../../icon/Icon';
import { linkMock } from '../../../../mocks/menu/link';
import { THEME_WHITE, THEME_SECONDARY, LINK_THEMES } from '../../../constants/buttons';
import '../styles/Link.story.scss';

const TEXT_SIZES = ['xxLarge', 'xLarge', 'large', 1, 2, 3, 4, 5, 6, 7];
const EMAIL_ADDRESS = 'appdirect@appdirect.com';
const TELEPHONE_NUMBER = '555-555-5555';

export default {
    title: 'Atoms/Link',
    component: Link
};

export const Default = {
    render: () => <Link href={linkMock.url}>Link</Link>,
    name: 'Default'
};

export const WithIcon = {
    render: () => (
        <>
            <div className="component-wrapper">
                <Link icon="caretLeft" iconClass="story-large-icon" href={linkMock.url}>Back</Link>
            </div>
            <div className="component-wrapper">
                <Link icon="caretLeft" href={linkMock.url}>Back</Link>
            </div>
            <div className="component-wrapper">
                <Link icon="caretLeft" iconClass="story-small-icon" href={linkMock.url}>Back</Link>
            </div>
        </>
    ),
    name: 'With Icon'
};

export const WithIconAlignment = {
    render: () => (
        <>
            <div className="component-wrapper">
                <Link icon="caretRight" iconClass="story-large-icon" iconPosition="right" href={linkMock.url}>Next</Link>
            </div>
            <div className="component-wrapper">
                <Link icon="caretRight" iconPosition="right" href={linkMock.url}>Next</Link>
            </div>
            <div className="component-wrapper">
                <Link icon="caretRight" iconClass="story-small-icon" iconPosition="right" href={linkMock.url}>Next</Link>
            </div>
        </>
    ),
    name: 'With Icon Alignment'
};

export const IconOnly = {
    render: () => (
        <>
            <div className="component-wrapper">
                <Link icon="facebook" iconClass="story-large-icon" href={linkMock.url} hiddenText>Facebook</Link>
            </div>
            <div className="component-wrapper">
                <Link icon="facebook" href={linkMock.url} hiddenText>Facebook</Link>
            </div>
            <div className="component-wrapper">
                <Link icon="facebook" iconClass="story-small-icon" href={linkMock.url} hiddenText>Facebook</Link>
            </div>
        </>
    ),
    name: 'Icon Only'
};

export const Sizes = {
    render: () => (
        <>
            {TEXT_SIZES.map(size => (
                <div className={`component-wrapper story_text--${size}`} style={{ lineHeight: '1em' }}>
                    <Link icon="caretRight" iconPosition="right" href={linkMock.url}>{`Size ${size}`}</Link>
                </div>
            ))}
        </>
    ),
    name: 'Sizes'
};

export const ExternalLink = {
    render: () => <Link target="_blank" href={linkMock.url}>External link</Link>,
    name: 'External link'
};

export const CustomIcon = {
    render: () => <Link href={linkMock.url} iconSrc="https://via.placeholder.com/30">External link</Link>,
    name: 'Custom icon'
};

export const Mailto = {
    render: () => (
        <>
            <span>Mail to: </span>
            <Link href={`mailto: ${EMAIL_ADDRESS}`}>{EMAIL_ADDRESS}</Link>
        </>
    ),
    name: 'Mailto'
};

export const Telephone = {
    render: () => (
        <>
            <span>Call: </span>
            <Link href={`tel: ${TELEPHONE_NUMBER}`}>{TELEPHONE_NUMBER}</Link>
        </>
    ),
    name: 'Telephone'
};

export const Anchor = {
    render: () => (
        <>
            <div id="top" style={{ padding: '200px 50px', marginBottom: 50, height: '200vh', backgroundColor: '#FAFAFA' }}>
                Scroll down to view the link
            </div>
            <Link href="#top" iconPosition="right" icon="caretUp">Scroll to top</Link>
        </>
    ),
    name: 'Anchor'
};

export const ThemeLinks = {
    render: () => (
        <>
            {LINK_THEMES.map(theme => (
                <div key={theme} style={{ padding: '5px 10px', background: (theme === THEME_WHITE || theme === THEME_SECONDARY) ? '#000' : 'transparent' }}>
                    <Link theme={theme} href={linkMock.url}>{`Default-${theme}`}</Link>
                </div>
            ))}
        </>
    ),
    name: 'Theme Links'
};

export const WithExtraIconContent = {
    render: () => (
        <>
            <div>
                <h2>Badge</h2>
                <Link
                    icon="cart"
                    href={linkMock.url}
                    iconExtraContent={<Badge className="story-extra-icon story-badge" label={3} />}
                >
                    Cart
                </Link>
            </div>
            <div>
                <h2>Favorite</h2>
                <Link
                    icon="user"
                    href={linkMock.url}
                    iconExtraContent={<Icon className="story-extra-icon story-star" type="star" />}
                >
                    Favorite User
                </Link>
            </div>
        </>
    ),
    name: 'With Extra Icon Content'
};
