// @appdirect/sfb-theme-components/src/atoms/product-groups-tiles/horizontal-style/stories/HorizontalStyle.stories.js
import React from 'react';
import HorizontalStyle from '../../../../constants/atoms/schema/product-groups-tiles/horizontal-style';
import { defaultValues } from '../../../../../mocks/product/productGroupsTiles';
import { getArgTypes, getArgs } from '../../../../tools/controlsHelper';

export default {
    title: 'Atoms/ProductGroupsStyles/HorizontalStyle',
    component: HorizontalStyle,
    argTypes: {
        customImageUrl: {
            control: 'text',
            description: 'The URL of the custom image to display'
        },
        ...getArgTypes(HorizontalStyle.schema().form)
    },
    render: ({ customImageUrl, ...args }) => (
        <HorizontalStyle {...defaultValues} {...args} customImage={{ url: customImageUrl }} />
    ),
    parameters: {
        controls: {
            include: [...Object.keys(HorizontalStyle.schema().form), 'customImageUrl']
        }
    }
};

export const Default = {
    args: {
        customImageUrl: defaultValues.customImage.defaultValue.url,
        ...getArgs(HorizontalStyle.schema().form),
        ...defaultValues
    },
    parameters: {
        controls: {
            exclude: ['customImage']
        }
    }
};
Default.storyName = 'HorizontalStyle';
