// @appdirect/sfb-theme-components/src/atoms/slider/stories/Slider.stories.js
import React from 'react';
import Slider from '../Slider';
import { sliderMockSettings, sliderMockVariousImageSizes } from '../../../../mocks/home/slider';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

const sliderProps = {
    ...sliderMockSettings.settings.productsList,
    buttonLabel: 'See More'
};

const itemList = sliderMockSettings.settings.productsList.items;

export default {
    title: 'Atoms/Slider',
    component: Slider,
    argTypes: getArgTypes(Slider.schema().form),
    parameters: {
        controls: {
            include: Object.keys(Slider.schema().form),
            exclude: ['maxRows']
        }
    },
    args: getArgs(Slider.schema().form)
};

export const Default = {
    render: args => <Slider {...sliderMockSettings.settings.productsList} {...args} />
};

export const DifferentImageSizes = {
    render: args => <Slider {...sliderMockVariousImageSizes.settings.productsList} {...args} />
};

export const ShowMoreButton = {
    render: args => <Slider {...sliderProps} {...args} />
};

export const WithSettings = {
    render: args => <Slider items={itemList} settings={args} />
};
