// @appdirect/sfb-theme-components/mocks/customer/orderConfirmation.js
export const orderConfirmationMockData = {
    links: {
        viewApps: {
            title: 'View my apps',
            url: '/myapps'
        },
        continueShopping: {
            title: 'Continue Shopping',
            url: '/home'
        },
        assignProducts: {
            title: 'Assign products',
            url: '/account/assign/v2/userToApps',
            isVisible: true
        }
    },
    user: {
        customerName: 'Cedric5 Dugas',
        customerEmail: 'cedric.dugas@appdirect.com'
    },
    order: {
        id: '42937',
        uuid: '38bb25c3-8f2e-4ec5-be4a-ed6bb720f919'
    },
    purchaseLink: {
        url: '/account/purchases/38bb25c3-8f2e-4ec5-be4a-ed6bb720f919'
    },
    i18n: {
        customerName: 'Thanks for your purchase, Cedric Dugas.',
        confirmationOrderId: 'Your purchase 42937 is processing.',
        customerPurchase: 'Your purchase',
        customerPurchaseProcessing: 'is processing.',
        customerEmail: 'You will receive an email with a confirmation shortly at cedric.dugas@appdirect.com.'
    }
};
