// @appdirect/sfb-theme-components/src/components/hero-banner/stories/HeroBanner.stories.js
import React from 'react';
import HeroBanner from '../HeroBanner';
import { heroBannerSettings } from '../../../../mocks/components/hero-banner';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

export default {
    title: 'Components/Hero Banner',
    component: HeroBanner,
    argTypes: {
        ...getArgTypes(HeroBanner.schema().form),
        backgroundImageUrl: {
            control: 'text',
            description: 'The URL of the background image'
        }
    },
    parameters: {
        controls: {
            include: ['backgroundImageUrl', ...Object.keys(HeroBanner.schema().form)],
            exclude: ['backgroundImage'],
            sort: 'alpha'
        }
    },
    args: {
        ...getArgs(HeroBanner.schema().form),
        backgroundImageUrl: 'https://placehold.co/400/black/white'
    }
};

export const Default = {
    render: ({ backgroundImageUrl, backgroundImage, ...args }) => (
        <div>
            <HeroBanner
                settings={{
                    ...heroBannerSettings,
                    ...args,
                    backgroundImage: {
                        ...backgroundImage,
                        url: backgroundImageUrl
                    }
                }}
            />
        </div>
    )
};

export const WithSettings = {
    render: ({ backgroundImageUrl, backgroundImage, ...args }) => (
        <div>
            <HeroBanner settings={{ ...args, backgroundImage: { ...backgroundImage, url: backgroundImageUrl } }} />
        </div>
    )
};
