// @appdirect/sfb-theme-components/mocks/home/carousel.js
export const productSlide1 = {
    contentType: 'PRODUCT',
    slideData: {
        selectedProduct: {
            title: 'IBM MaaS360 with Watson™',
            blurb: 'IBM MaaS360 Unified Endpoint Management with Watson',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/overview/img4717967604650738986.png',
            url: '/en-US/apps/185242/ibm-maas360-with-watson'
        }
    }
};

export const productSlide2 = {
    contentType: 'PRODUCT',
    slideData: {
        selectedProduct: {
            title: 'AppInsights',
            blurb: 'Real-time business dashboard solution',
            overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/overview/img5121553238902495902.png',
            url: '/en-US/apps/10030/appinsights'
        }
    }
};

export const textSlideMockData = {
    contentType: 'IMAGE_WITH_TEXT',
    slideData: {
        title: 'Mozy Backup Server',
        text: 'Simple, secure backup for servers',
        image: {
            url: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/48145/overview/img117358683923526362.png'
        },
        slideLink: {
            value: 'http://testappdirect.com'
        },
        buttonLabel: 'Visit product page',
        backgroundColor: '#eee'
    }
};

export const imageSlideMockData = {
    contentType: 'IMAGE_ONLY',
    slideData: {
        title: 'GeoOp',
        text: 'Field service software made easy',
        image: {
            url: 'https://picsum.photos/id/1/800/400'
        },
        textSize: {
            value: 'medium',
            label: 'uieditor.sfbComponent.dropdown.option.medium'
        },
        textColor: '#000000',
        backgroundColor: '#fff',
        imageSizing: 'fill',
        slideLink: '/en-US/apps/17331/geoop'
    }
};
export const videoSlideMockData = {
    slideData: {
        title: 'GeoOp'
    },
    type: 'video-slide',
    contentType: 'video-slide',
    title: 'GeoOp',
    blurb: 'Field service software made easy',
    videoUrl: 'https://www.youtube.com/watch?v=oHg5SJYRHA0',
    imageUrl: 'https://picsum.photos/id/1/800/400'
};

export const carouselMockData = {
    slides: [
        imageSlideMockData,
        textSlideMockData,
        {
            contentType: 'PRODUCT',
            slideData: {
                selectedProduct: {
                    title: 'GoDaddy Domain Registration',
                    blurb: 'GoDaddy Domain Registration',
                    overviewImageUrl: '',
                    url: '/en-US/apps/97079/godaddy-domain-registration'
                }
            }
        },
        productSlide1,
        {
            contentType: 'PRODUCT',
            slideData: {
                selectedProduct: {
                    title: '',
                    blurb: '',
                    overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/overview/img5763238548373303037.png',
                    url: '/en-US/apps/199553/ibm-spss-statistics'
                }
            }
        },
        productSlide2
    ],
    i18n: { learnMore: 'Learn more' }
};
