// @appdirect/sfb-theme-components/src/atoms/textarea/stories/Textarea.stories.js
import React from 'react';
import Textarea from '../Textarea';

export default {
    title: 'Atoms/Textarea',
    component: Textarea
};

export const Default = {
    render: () => <Textarea name="name" label="Label" />,
    name: 'Default'
};

export const WithPlaceholder = {
    render: () => <Textarea name="name" label="Label" placeholder="placeholder" />,
    name: 'With placeholder'
};

export const Required = {
    render: () => <Textarea name="name" label="Label" placeholder="placeholder" required />,
    name: 'Required'
};

export const WithDefaultValue = {
    render: () => <Textarea name="name" label="Label" defaultValue="default value" />,
    name: 'With default value'
};

export const ReadOnly = {
    render: () => <Textarea name="name" label="Label" defaultValue="default value" readOnly />,
    name: 'Read only'
};

export const WithHiddenLabelA11Y = {
    render: () => <Textarea name="name" label="Label" hiddenLabel />,
    name: 'With hidden label (a11y)'
};

export const WithError = {
    render: () => <Textarea name="name" label="Label" error="This is required" />,
    name: 'With error'
};

export const Disabled = {
    render: () => <Textarea name="name" label="Label" disabled />,
    name: 'Disabled'
};
