// @appdirect/sfb-theme-components/mocks/bundle/items.js
export const bundleItemsMock = [
    {
        id: 10030,
        focused: true,
        url: '/en-US/apps/10030/appinsights',
        name: 'AppInsights',
        image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/myAppIcon/img4017668062510137155.png?1ef8d2a6f8f68763d54bc9d0049803d7',
        card: {
            title: 'AppInsights',
            details: 'Real-time business dashboard solution',
            overview: 'AppInsights help you make faster and more informed\nbusiness decisions by letting you view your KPIs on a real-time dashboard, all in\none place.',
            logoSrc: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_112/img3745051225877065498.png?e0df123fedcc5e96c83a248797c76c6a',
            logoSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_112/img3745051225877065498.png?e0df123fedcc5e96c83a248797c76c6a',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_112/img3745051225877065498-2x.png?6a5de3e906301fb37d2cde7ee0a2b930',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_112/img3745051225877065498.png?e0df123fedcc5e96c83a248797c76c6a'
            },
            productRibbon: null,
            merchandisingBadges: [
                {
                    uuid: '57af2145-5543-4fc1-b947-15e08d4575ed',
                    name: 'Single Sign-On',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/57af2145-5543-4fc1-b947-15e08d4575ed/img8223107847585143017.png?289aa01be11853c6ef0f8735471e2ba2'
                },
                {
                    uuid: '08d06719-1027-4952-914c-fe0d3de89d06',
                    name: 'Unified Invoice',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/08d06719-1027-4952-914c-fe0d3de89d06/img5117470399285071356.png?8cf2543bdae1af7a6f287688d07c1d45'
                },
                {
                    uuid: 'f57bd864-c1e3-4eff-af8d-592ee0a5165c',
                    name: 'Access Management',
                    image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/f57bd864-c1e3-4eff-af8d-592ee0a5165c/img765251955674935544.png?ab226874a0f6e57f753b26e96591efe3'
                }
            ]
        },
        customAttributes: [
            {
                type: 'TEXT',
                attributeKey: 'externalID',
                label: 'SKU',
                value: '123123123'
            }
        ],
        edition: {
            name: 'Free Trial'
        },
        overview: {
            title: 'Monitor all your key metrics in one place.',
            description: 'AppInsights dashboards automatically collect the latest data about your company into one place.',
            callToActions: {
                tour: {
                    label: 'Take the Tour',
                    alternateLabel: 'More Features',
                    color: 'primary',
                    href: '/en-US/apps/10030/appinsights/features'
                },
                demo: {
                    label: 'Watch Demo',
                    href: 'https://www.youtube.com/watch?v=s5qnuvKt2Fc',
                    iFrameFriendlyUrl: '//www.youtube.com/embed/s5qnuvKt2Fc'
                }
            },
            image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/overview/img5121553238902495902.png?f752784a1cf710d1d845ab7937ba0962',
            imageSrcSet: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/overview/img5121553238902495902.png?f752784a1cf710d1d845ab7937ba0962'
            },
            showThirdPartyLegalDisclosure: false,
            documentationUrl: 'https://www.appinsights.com/p/resources/users-guide-overview/',
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [
                {
                    name: 'Nike',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/103/img3705877950029557229.png?551ee9e339d132ebe5e834858a24a9b3'
                },
                {
                    name: 'Honda',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/142/img4813558416860680605.png?62441f2afa23d3e5fd3312c25e946e09'
                },
                {
                    name: 'Cheil Worldwide',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/148/img7612482328839562829.png?2dc8842213406c691fc3c5c12f4dbe35'
                },
                {
                    name: 'Vodafone',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/154/img5518808226882979204.png?bcebe0c5954f2f8208c5bd60c35e4c6a'
                },
                {
                    name: 'Yale',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/166/img1021414731794950569.png?dd0aa6f4ae6383ace98d5cddb395e089'
                }
            ],
            benefits: [
                {
                    description: 'Consolidate your KPIs from all the services you use in one place.',
                    descriptionLong: 'Consolidate your KPIs from all the services you use in one place.',
                    title: 'Your key metrics in one place',
                    type: 'Benefit'
                },
                {
                    description: 'Clean and concise visualizations communicate the health of your business quickly.',
                    descriptionLong: 'Clean and concise visualizations communicate the health of your business quickly.',
                    title: 'Be informed at a glance',
                    type: 'Benefit'
                },
                {
                    description: 'Automated data syncing guarantees that everyone is seeing the same numbers, improving communications and unity across your entire organization.',
                    descriptionLong: 'Automated data syncing guarantees that everyone is seeing the same numbers, improving communications and unity across your entire organization.',
                    title: 'Keep everyone on the same page',
                    type: 'Benefit'
                },
                {
                    description: 'Access your dashboards online from anywhere! No software download required, so you\'re always kept up-to-date.',
                    descriptionLong: 'Access your dashboards online from anywhere! No software download required, so you\'re always kept up-to-date.',
                    title: 'Access anywhere',
                    type: 'Benefit'
                }
            ],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation',
                startingFrom: 'Starting From',
                seePricingPlans: 'See Pricing Plans',
                availableAddons: 'Available Add-ons',
                availableBundles: 'Bundle Package Includes'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: [
                        {
                            id: 0,
                            resourceType: 'PDF_LINK',
                            locale: null,
                            name: 'test',
                            link: 'https://www.crowdstrike.com/wp-content/uploads/2020/08/automox-joint-datasheet.pdf',
                            scribdKey: null,
                            scribdId: null,
                            description: '',
                            resellerAccess: false,
                            iFrameFriendlyURL: 'https://www.crowdstrike.com/wp-content/uploads/2020/08/automox-joint-datasheet.pdf'
                        },
                        {
                            id: 1,
                            resourceType: 'PDF_LINK',
                            locale: null,
                            name: 'test3',
                            link: 'https://forms.benefitresource.com/marketing-material/Prospect Flyers/HRA_VEBA_Prospect_Flyer.pdf',
                            scribdKey: null,
                            scribdId: null,
                            description: '',
                            resellerAccess: false,
                            iFrameFriendlyURL: 'https://forms.benefitresource.com/marketing-material/Prospect Flyers/HRA_VEBA_Prospect_Flyer.pdf'
                        },
                        {
                            id: 2,
                            resourceType: 'PDF_LINK',
                            locale: null,
                            name: 'test 2',
                            link: 'https://www.nshealth.ca/sites/nshealth.ca/files/gwl-_claim_form_28health29_new.pdf',
                            scribdKey: null,
                            scribdId: null,
                            description: '',
                            resellerAccess: false,
                            iFrameFriendlyURL: 'https://www.nshealth.ca/sites/nshealth.ca/files/gwl-_claim_form_28health29_new.pdf'
                        },
                        {
                            id: 3,
                            resourceType: 'PDF_LINK',
                            locale: null,
                            name: 'last test',
                            link: 'https://forms.benefitresource.com/marketing-material/Prospect Flyers/HSA_Prospect_Flyer.pdf',
                            scribdKey: null,
                            scribdId: null,
                            description: '',
                            resellerAccess: false,
                            iFrameFriendlyURL: 'https://forms.benefitresource.com/marketing-material/Prospect Flyers/HSA_Prospect_Flyer.pdf'
                        },
                        {
                            id: 4,
                            resourceType: 'PDF_LINK',
                            locale: null,
                            name: 'HRA - uploaded',
                            link: 'https://app-direct-www-cloudfront.s3.amazonaws.com/app_resources/10030/documentation/HRA_VEBA_Prospect_Flyer.pdf',
                            scribdKey: null,
                            scribdId: null,
                            description: '',
                            resellerAccess: false,
                            iFrameFriendlyURL: 'https://app-direct-www-cloudfront.s3.amazonaws.com/app_resources/10030/documentation/HRA_VEBA_Prospect_Flyer.pdf'
                        },
                        {
                            id: 5,
                            resourceType: 'PDF_LINK',
                            locale: null,
                            name: 'HRA - link',
                            link: 'https://forms.benefitresource.com/marketing-material/Prospect Flyers/HRA_VEBA_Prospect_Flyer.pdf',
                            scribdKey: null,
                            scribdId: null,
                            description: '',
                            resellerAccess: false,
                            iFrameFriendlyURL: 'https://forms.benefitresource.com/marketing-material/Prospect Flyers/HRA_VEBA_Prospect_Flyer.pdf'
                        }
                    ]
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [
                {
                    name: 'Create dashboards to monitor your sales pipeline and motivate your sales team.',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1102629510912555107.png?997f1866afb19730631d6d4440cfa7e4',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1102629510912555107.png?997f1866afb19730631d6d4440cfa7e4',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1102629510912555107.png?997f1866afb19730631d6d4440cfa7e4',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1102629510912555107.png?997f1866afb19730631d6d4440cfa7e4'
                    }
                },
                {
                    name: 'Dashboards for customer service let you know immediately if your customers are being served.',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1262432103500464999.png?6e1a0e552681633918e603afb2d573f0',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1262432103500464999.png?6e1a0e552681633918e603afb2d573f0',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1262432103500464999.png?6e1a0e552681633918e603afb2d573f0',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1262432103500464999.png?6e1a0e552681633918e603afb2d573f0'
                    }
                },
                {
                    name: 'Connect to services like New Relic or Pingdom to manage your technology infrastructure in real time.',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1493562786997021035.png?4c1c423408291115b2c4d5067470e869',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1493562786997021035.png?4c1c423408291115b2c4d5067470e869',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1493562786997021035.png?4c1c423408291115b2c4d5067470e869',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img1493562786997021035.png?4c1c423408291115b2c4d5067470e869'
                    }
                },
                {
                    name: 'Show off your social media reach in your lobby, and check out the buzz around your company at any moment.',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img4298206716672378307.png?c769fcb97c1959966cde47db36adf41b',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img4298206716672378307.png?c769fcb97c1959966cde47db36adf41b',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img4298206716672378307.png?c769fcb97c1959966cde47db36adf41b',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img4298206716672378307.png?c769fcb97c1959966cde47db36adf41b'
                    }
                },
                {
                    name: 'Place KPIs in an executive dashboard and get an overview of the health of your company at a glance.',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img6668021355693451257.png?b735814af97545eb908569ac83721f06',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img6668021355693451257.png?b735814af97545eb908569ac83721f06',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img6668021355693451257.png?b735814af97545eb908569ac83721f06',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/screenshot/img6668021355693451257.png?b735814af97545eb908569ac83721f06'
                    }
                }
            ],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: 'https://www.appdirect.com/about/terms-of-service',
            privacyUrl: 'https://www.appdirect.com/about/privacy-policy',
            email: 'appinsights.support@appdirect.com',
            phone: null,
            knowledgebaseUrl: 'https://support.appdirect.com/hc/en-us/categories/115002392247-AppInsights',
            description: 'Have a question or suggestions for improvement? Send us an email and you\'ll definitely hear back from us! We\'re happy to help!',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [
            {
                id: 45285,
                name: 'Freezing Apps',
                type: 'SUB_CATEGORY',
                url: '/en-US/listing?scat=45285'
            }
        ],
        bundle: {
            id: 323839,
            url: '/en-US/bundles/323839/new-bundle/bundledProducts/10030'
        }
    },
    {
        id: 42937,
        focused: false,
        url: '/en-US/apps/42937/appwise',
        name: 'AppWise',
        image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/myAppIcon/img6640948475444359586.png?b00df6f0ba37b4364b967da895ab159c',
        card: {
            title: 'AppWise',
            details: 'Search everything, in one shot',
            overview: 'With Unified Search across over 40+ leading cloud applications, easily find all your emails, documents and tasks in a couple of clicks.',
            logoSrc: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/thumbs_112/img1099429097555068136.png?0b64020ae947c80340ac4f9716c1d01b',
            logoSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/thumbs_112/img1099429097555068136.png?0b64020ae947c80340ac4f9716c1d01b',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/thumbs_112/img1099429097555068136-2x.png?82706de9770108a3d0c5d9f736e08eb5',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/thumbs_112/img1099429097555068136.png?0b64020ae947c80340ac4f9716c1d01b'
            },
            productRibbon: null,
            merchandisingBadges: []
        },
        customAttributes: [],
        edition: {
            name: 'Professional (Annual)'
        },
        overview: {
            title: 'Enterprise Search, in seconds',
            description: 'AppWise is a business productivity tool that provides Unified Search across 40+ leading applications like Salesforce, Gmail, Office 365, Box and Asana.   AppWise has enterprise-grade security built-in from the ground up, and it\'s intuitive enough be setup by end-users in seconds.',
            callToActions: {
                tour: {
                    label: 'Take the Tour',
                    alternateLabel: 'More Features',
                    color: 'primary',
                    href: '/en-US/apps/42937/appwise/features'
                },
                demo: {
                    label: 'Watch Demo',
                    href: 'https://youtu.be/q1BlEQknStM',
                    iFrameFriendlyUrl: '//www.youtube.com/embed/q1BlEQknStM'
                }
            },
            image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/overview/img1301783796595200436.png?18e59c26ffbf1b84953937d376a29fdc',
            imageSrcSet: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/overview/img1301783796595200436.png?18e59c26ffbf1b84953937d376a29fdc',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/overview/img1301783796595200436.png?5890c62dc6fd364c4b6a2d4c9370456f',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/overview/img1301783796595200436-2x.png?8ac98635908c0ea66e0bab71750f3b57'
            },
            showThirdPartyLegalDisclosure: false,
            documentationUrl: 'https://help.appwise.com',
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [
                {
                    name: 'PayPal',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/1/paypal.png'
                },
                {
                    name: 'Google',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/3/img7856022146478847355.png?1051cfcb9c80a581b4e5dab3a283226a'
                },
                {
                    name: 'Allstate',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/4/allstate2.png'
                },
                {
                    name: 'Audi',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/8/audi.png'
                },
                {
                    name: 'The Next Web',
                    imgSrc: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/customer/4294/img1505645661500555267.png?a280487a6db2aa04e0efb46d616d0dc5'
                }
            ],
            benefits: [
                {
                    description: 'Full-text search across 40+ cloud apps. AppWise refines your search as you type, to help you drill into individual apps or look for documents or presentations, wherever they are.',
                    descriptionLong: 'Full-text search across 40+ cloud apps. AppWise refines your search as you type, to help you drill into individual apps or look for documents or presentations, wherever they are.',
                    title: 'Unified search across multiple accounts',
                    type: 'Benefit'
                },
                {
                    description: 'Build and save complex filters from service, content type, date range combinations. Get powerful "proximity searches" used by legal professionals, or simply refine by author, title or document type.',
                    descriptionLong: 'Build and save complex filters from service, content type, date range combinations. Get powerful "proximity searches" used by legal professionals, or simply refine by author, title or document type.',
                    title: 'Get a personal perspective across all your content',
                    type: 'Benefit'
                },
                {
                    description: 'Conveniently preview documents, presentations and more without leaving your search.',
                    descriptionLong: 'Conveniently preview documents, presentations and more without leaving your search.',
                    title: 'Boost productivity by minimizing context switching',
                    type: 'Benefit'
                },
                {
                    description: 'Don\'t worry about missing critical project updates, or new information about customer prospects.   Get pro-actively emailed when your connected accounts are updated about topics you\'re tracking.  You can even cc a group email list or an assistant to feed into company workflows.',
                    descriptionLong: 'Don\'t worry about missing critical project updates, or new information about customer prospects.   Get pro-actively emailed when your connected accounts are updated about topics you\'re tracking.  You can even cc a group email list or an assistant to feed into company workflows.',
                    title: 'Stay on top of developments across your company',
                    type: 'Benefit'
                },
                {
                    description: 'AppWise works seamlessly with your existing security and Identity and Access Management infrastructure.  AppWise is pre-integrated with AppDirect, Okta, OneLogin and support SAML 2.0.',
                    descriptionLong: 'AppWise works seamlessly with your existing security and Identity and Access Management infrastructure.  AppWise is pre-integrated with AppDirect, Okta, OneLogin and support SAML 2.0.',
                    title: 'Secured and integrated with your company',
                    type: 'Benefit'
                },
                {
                    description: 'Many companies don\'t work completely in "the cloud".  AppWise provides an SDK which can be used to connect on-premises and/or proprietary content sources into AppWise so you get the benefits of cloud-based rapid deployment but with a holistic view over your content.',
                    descriptionLong: 'Many companies don\'t work completely in "the cloud".  AppWise provides an SDK which can be used to connect on-premises and/or proprietary content sources into AppWise so you get the benefits of cloud-based rapid deployment but with a holistic view over your content.',
                    title: 'Bridge On-Prem and Cloud environments',
                    type: 'Benefit'
                },
                {
                    description: 'AppWise has a comprehensive REST-based API which can be used to integrate with existing user-experiences, enterprise workflows and mobile applications.',
                    descriptionLong: 'AppWise has a comprehensive REST-based API which can be used to integrate with existing user-experiences, enterprise workflows and mobile applications.',
                    title: 'Enhance existing enterprise workflows',
                    type: 'Benefit'
                },
                {
                    id: 4132314,
                    title: 'Unified Search',
                    description: 'Search all your content across all your accounts, with one, powerful search.  Save time...',
                    descriptionLong: 'Search all your content across all your accounts, with one, powerful search.  Save time gathering information and re-using presentations, documents and templates that have already been created inside your company.',
                    href: '/en-US/apps/42937/appwise/features',
                    linkLabel: 'Learn More',
                    slogan: 'Search Everything, in one shot',
                    type: 'Feature'
                }
            ],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation',
                startingFrom: 'Starting From',
                seePricingPlans: 'See Pricing Plans',
                availableAddons: 'Available Add-ons',
                availableBundles: 'Bundle Package Includes'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [
                {
                    name: 'Preview documents and presentations.',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img6041837174655209846.png?4fbe1c3a9428130c158f2cf686dc4174',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img6041837174655209846.png?4fbe1c3a9428130c158f2cf686dc4174',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img3817492137859166194.png?6888983ec221ac894a442169d6361345',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img6041837174655209846.png?4fbe1c3a9428130c158f2cf686dc4174'
                    }
                },
                {
                    name: 'Create complex filters and apply them in one click.',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img3239441032324518171.png?9e67e353a5d3053f9e6ee358e88905e1',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img3239441032324518171.png?9e67e353a5d3053f9e6ee358e88905e1',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img3239441032324518171.png?9e67e353a5d3053f9e6ee358e88905e1'
                    }
                },
                {
                    name: 'Pre-integrated with 30+ apps &amp; SDK for on-prem apps.',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img4983556615518866077.png?8c20eda45e5f91dfea2b336770e6fdce',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img4983556615518866077.png?8c20eda45e5f91dfea2b336770e6fdce',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/screenshot/img4983556615518866077.png?8c20eda45e5f91dfea2b336770e6fdce'
                    }
                }
            ],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: 'https://www.appdirect.com/about/terms-of-service',
            privacyUrl: 'https://www.appdirect.com/about/privacy-policy',
            email: 'help@appwise.com',
            phone: '+1 415 879 8350',
            knowledgebaseUrl: 'https://help.appwise.com',
            description: '',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [
            {
                id: 67289,
                name: 'Wild Apps',
                type: 'CATEGORY',
                url: '/en-US/listing?cat=67289'
            }
        ],
        bundle: {
            id: 323839,
            url: '/en-US/bundles/323839/new-bundle/bundledProducts/42937'
        }
    },
    {
        id: 132264,
        focused: false,
        url: '/en-US/apps/132264/servicem8',
        name: 'ServiceM8',
        image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/myAppIcon/img6868637964401674449.png?c99277e36e27cb8b3387e94fed58838d',
        card: {
            title: 'ServiceM8',
            details: 'Field Service for Small Business',
            overview: 'ServiceM8 is a cloud\u00adbased solution that helps field service organizations manage their business. ServiceM8 guides users from creating job quotes to invoicing and payment collection, and provides real\u00adtime information on what jobs are active, what quotes have been issued to clients, and where and when employees are at client sites.',
            logoSrc: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/thumbs_112/img4134068526255962732.png?571b7d3cc01faad2f256d7a949720d77',
            logoSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/thumbs_112/img4134068526255962732.png?571b7d3cc01faad2f256d7a949720d77',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/thumbs_112/img4134068526255962732-2x.png?eed74fe4de86f50cac5fd7f4ea7c0dba',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/thumbs_112/img4134068526255962732.png?571b7d3cc01faad2f256d7a949720d77'
            },
            productRibbon: null,
            merchandisingBadges: []
        },
        customAttributes: [],
        edition: {
            name: 'Premium'
        },
        overview: {
            title: 'Powerful mobility solution for field service businesses',
            description: 'ServiceM8 is a cloud-based solution that helps field service organizations manage their business. ServiceM8 enables SMB plumbers, electricians and similar businesses to provide great customer experiences, increase productivity, improve cash flow and win more business.\n\nServiceM8 is an end to end solution that helps from the client’s first call, to quoting, invoicing and taking payment. Providing real-time information on job progress, where and when employees are at client sites and direct integration with major accounting packages.',
            callToActions: {
                tour: {
                    label: 'Take the Tour',
                    alternateLabel: 'More Features',
                    color: 'primary',
                    href: '/en-US/apps/132264/servicem8/features'
                },
                demo: {
                    label: 'Watch Demo',
                    href: 'https://player.vimeo.com/video/283265177',
                    iFrameFriendlyUrl: '//player.vimeo.com/video/283265177'
                }
            },
            image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/overview/img1164992687341006935.png?5b3a4de2d0819cb9f8a0e64124a6c7cb',
            imageSrcSet: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/overview/img1164992687341006935.png?5b3a4de2d0819cb9f8a0e64124a6c7cb'
            },
            showThirdPartyLegalDisclosure: false,
            documentationUrl: '',
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [],
            benefits: [
                {
                    description: 'Create invoices in the field, and process credit card payments through the ServiceM8 mobile app with no external devices required.',
                    descriptionLong: 'Create invoices in the field, and process credit card payments through the ServiceM8 mobile app with no external devices required.',
                    title: 'Get Paid Faster',
                    type: 'Benefit'
                },
                {
                    description: 'Inspire confidence in your clients with professional quotes and invoices. Automatically Notify clients you\'re on-approach to the job, with an estimated time of arrival.',
                    descriptionLong: 'Inspire confidence in your clients with professional quotes and invoices. Automatically Notify clients you\'re on-approach to the job, with an estimated time of arrival.',
                    title: 'Look Professional',
                    type: 'Benefit'
                },
                {
                    description: 'On average businesses using ServiceM8 complete 30% more work in the first 3 months.',
                    descriptionLong: 'On average businesses using ServiceM8 complete 30% more work in the first 3 months.',
                    title: 'Complete More Work',
                    type: 'Benefit'
                },
                {
                    description: 'With professional quotes, better communication and faster service, you’ll win more new work and earn repeat customers.',
                    descriptionLong: 'With professional quotes, better communication and faster service, you’ll win more new work and earn repeat customers.',
                    title: 'Win More Work',
                    type: 'Benefit'
                },
                {
                    id: 4281011,
                    title: 'Field service management',
                    description: 'Powerful job creation, dispatch, navigation, quoting, job documentation and invoicing c...',
                    descriptionLong: 'Powerful job creation, dispatch, navigation, quoting, job documentation and invoicing capability - in the office or from the palm of your hand. \n\nField staff have everything they need to get to the job, complete and record the work, and issue an invoice. They can even accept payment on-site through the app.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                },
                {
                    id: 4281014,
                    title: 'Scheduling',
                    description: 'Easily manage your staff schedules from anywhere. Simply drag jobs onto the desired tim...',
                    descriptionLong: 'Easily manage your staff schedules from anywhere. Simply drag jobs onto the desired time and staff member and they\'ll be instantly notified of their updated schedule.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                },
                {
                    id: 4281032,
                    title: 'Quotes & Invoicing',
                    description: 'Create professional quotes & invoices in seconds, then email or TXT them to your client...',
                    descriptionLong: 'Create professional quotes & invoices in seconds, then email or TXT them to your client while you\'re still on site.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                },
                {
                    id: 4281041,
                    title: 'Mobile',
                    description: 'Impressive communication, navigation, quoting, job management and invoicing capability,...',
                    descriptionLong: 'Impressive communication, navigation, quoting, job management and invoicing capability, tailored for ease of use with your iPhone or iPad.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                },
                {
                    id: 4281038,
                    title: 'Electronic Forms',
                    description: 'ServiceM8\'s Forms have near limitless possibilities. Complete site audits, procedures, ...',
                    descriptionLong: 'ServiceM8\'s Forms have near limitless possibilities. Complete site audits, procedures, and checklists in the field, then immediately turn them into professional PDF reports.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                },
                {
                    id: 4281044,
                    title: 'Online Booking',
                    description: 'Create advanced online booking forms for your business’s services, allowing customers t...',
                    descriptionLong: 'Create advanced online booking forms for your business’s services, allowing customers to get a live quote, select a time, then book a job straight into your schedule, without interacting with your business directly.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                },
                {
                    id: 4281023,
                    title: 'Add-ons',
                    description: 'ServiceM8\'s not just an app, it\'s a platform. Add-ons are additional modules which exte...',
                    descriptionLong: 'ServiceM8\'s not just an app, it\'s a platform. Add-ons are additional modules which extend the functionality of the app in specific areas, or assist certain industries.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                },
                {
                    id: 4281017,
                    title: 'Integrations',
                    description: 'ServiceM8 doesn\'t do everything, so we\'ve integrated with other great products for smal...',
                    descriptionLong: 'ServiceM8 doesn\'t do everything, so we\'ve integrated with other great products for small business, giving you a seamless experience between platforms.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                },
                {
                    id: 4281035,
                    title: 'Reports',
                    description: 'Whether you’re looking for a high-level view of how you\'re tracking or want to get into...',
                    descriptionLong: 'Whether you’re looking for a high-level view of how you\'re tracking or want to get into the nitty-gritty details, get all the information you need lightning fast.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                },
                {
                    id: 4281020,
                    title: 'Security',
                    description: 'Protecting your data is fundamental to everything we do. We have invested in industry-l...',
                    descriptionLong: 'Protecting your data is fundamental to everything we do. We have invested in industry-leading infrastructure and constantly strive to ensure your data is always secure.',
                    href: '/en-US/apps/132264/servicem8/features',
                    linkLabel: 'Learn More',
                    type: 'Feature'
                }
            ],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation',
                startingFrom: 'Starting From',
                seePricingPlans: 'See Pricing Plans',
                availableAddons: 'Available Add-ons',
                availableBundles: 'Bundle Package Includes'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [
                {
                    name: 'The ServiceM8 iOS App',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img1494378612335851883.png?b8e85eb892847fa97eed140851fa24c1',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img1494378612335851883.png?b8e85eb892847fa97eed140851fa24c1',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img1494378612335851883.png?b8e85eb892847fa97eed140851fa24c1',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img1494378612335851883.png?b8e85eb892847fa97eed140851fa24c1'
                    }
                },
                {
                    name: 'The ServiceM8 Dispatch Board',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img3626884849309376074.png?030c4d9efc55e815e518c402819813d9',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img3626884849309376074.png?030c4d9efc55e815e518c402819813d9',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img3626884849309376074.png?030c4d9efc55e815e518c402819813d9',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img3626884849309376074.png?030c4d9efc55e815e518c402819813d9'
                    }
                },
                {
                    name: 'The ServiceM8 Online Schedule',
                    imageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img5936732892310416169.png?eedf4d4781243e050271462e6fe630b4',
                    imageUrlSrcset: {
                        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img5936732892310416169.png?eedf4d4781243e050271462e6fe630b4',
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img5936732892310416169.png?eedf4d4781243e050271462e6fe630b4',
                        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/132264/screenshot/img5936732892310416169.png?eedf4d4781243e050271462e6fe630b4'
                    }
                }
            ],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: 'https://www.servicem8.com/terms-of-service',
            privacyUrl: 'https://www.servicem8.com/privacy-policy',
            email: 'support@servicem8.com',
            phone: null,
            knowledgebaseUrl: 'https://support.servicem8.com/',
            description: 'ServiceM8 has a great range of help articles and videos to help you learn. ServiceM8 offers 24/7 email support and 24/7 chat support.',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [
            {
                id: 67289,
                name: 'Wild Apps',
                type: 'CATEGORY',
                url: '/en-US/listing?cat=67289'
            }
        ],
        bundle: {
            id: 323839,
            url: '/en-US/bundles/323839/new-bundle/bundledProducts/132264'
        }
    },
    {
        id: 329889,
        focused: false,
        url: '/en-US/apps/329889/px-eva-base-platform',
        name: 'PX-EVA Base Platform',
        image: null,
        card: {
            title: 'PX-EVA Base Platform',
            details: 'PX EVA',
            overview: 'PX EVA',
            logoSrc: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png',
            logoSrcset: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263-2x.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png'
            },
            logoAlt: 'Logo of PX-EVA Base Platform',
            productRibbon: null,
            merchandisingBadges: []
        },
        customAttributes: [
            {
                type: 'TEXT',
                attributeKey: 'ERP-Id',
                label: 'ERP Id',
                value: 'VISTA01'
            }
        ],
        edition: {
            name: 'Recurring Edition'
        },
        overview: {
            title: 'Splash Title',
            description: 'Splash Description',
            callToActions: {},
            image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329889/overview/img680213046346428498.png',
            imageSrcSet: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329889/overview/img680213046346428498.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329889/overview/img680213046346428498.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329889/overview/img680213046346428498-2x.png'
            },
            imageAlt: 'Eaton sample budget preview',
            showThirdPartyLegalDisclosure: false,
            documentationUrl: null,
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [],
            benefits: [],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: null,
            privacyUrl: null,
            email: null,
            phone: null,
            knowledgebaseUrl: null,
            description: '',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [],
        bundle: {
            id: 329919,
            url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329889'
        }
    },
    {
        id: 329893,
        focused: false,
        url: '/en-US/apps/329893/px-eva-budgeting-module',
        name: 'PX-EVA Budgeting Module',
        image: null,
        card: {
            title: 'PX-EVA Budgeting Module',
            details: '',
            overview: '',
            logoSrc: null,
            logoSrcset: {},
            productRibbon: null,
            merchandisingBadges: []
        },
        customAttributes: [
            {
                type: 'TEXT',
                attributeKey: 'ERP-Id',
                label: 'ERP Id',
                value: 'VISTA01'
            }
        ],
        edition: {
            name: 'Recurring Edition'
        },
        overview: {
            title: 'Splash Title',
            description: 'Splash Desc',
            callToActions: {},
            image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329893/overview/img4781544673053016721.png',
            imageSrcSet: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329893/overview/img4781544673053016721.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329893/overview/img4781544673053016721.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329893/overview/img4781544673053016721-2x.png'
            },
            imageAlt: 'Eaton sample budget preview',
            showThirdPartyLegalDisclosure: false,
            documentationUrl: null,
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [],
            benefits: [],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: null,
            privacyUrl: null,
            email: null,
            phone: null,
            knowledgebaseUrl: null,
            description: '',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [],
        bundle: {
            id: 329919,
            url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329893'
        }
    },
    {
        id: 329898,
        focused: false,
        url: '/en-US/apps/329898/px-eva-sustainability-module',
        name: 'PX-EVA Sustainability Module',
        image: null,
        card: {
            title: 'PX-EVA Sustainability Module',
            details: '',
            overview: '',
            logoSrc: null,
            logoSrcset: {},
            productRibbon: null,
            merchandisingBadges: []
        },
        customAttributes: [
            {
                type: 'TEXT',
                attributeKey: 'ERP-Id',
                label: 'ERP Id',
                value: 'VISTA01'
            }
        ],
        edition: {
            name: 'Recurring Edition'
        },
        overview: {
            title: 'Splash Title',
            description: 'splash',
            callToActions: {},
            image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329898/overview/img6582364165551198837.png',
            imageSrcSet: {
                default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329898/overview/img6582364165551198837.png',
                '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329898/overview/img6582364165551198837.png',
                '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/329898/overview/img6582364165551198837-2x.png'
            },
            imageAlt: 'Eaton sample budget',
            showThirdPartyLegalDisclosure: false,
            documentationUrl: null,
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [],
            benefits: [],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: null,
            privacyUrl: null,
            email: null,
            phone: null,
            knowledgebaseUrl: null,
            description: '',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [],
        bundle: {
            id: 329919,
            url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329898'
        }
    },
    {
        id: 329902,
        focused: false,
        url: '/en-US/apps/329902/px-eva-real-time-monitoring-module',
        name: 'PX-EVA Real-time Monitoring Module',
        image: null,
        card: {
            title: 'PX-EVA Real-time Monitoring Module',
            details: '',
            overview: '',
            logoSrc: null,
            logoSrcset: {},
            productRibbon: null,
            merchandisingBadges: []
        },
        customAttributes: [],
        edition: {
            name: 'Recurring Edition'
        },
        overview: {
            title: null,
            description: '',
            callToActions: {},
            image: null,
            imageSrcSet: {},
            showThirdPartyLegalDisclosure: false,
            documentationUrl: null,
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [],
            benefits: [],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: null,
            privacyUrl: null,
            email: null,
            phone: null,
            knowledgebaseUrl: null,
            description: '',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [],
        bundle: {
            id: 329919,
            url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329902'
        }
    },
    {
        id: 329907,
        focused: false,
        url: '/en-US/apps/329907/px-eva-measurement-verification-module',
        name: 'PX-EVA Measurement & Verification Module',
        image: null,
        card: {
            title: 'PX-EVA Measurement & Verification Module',
            details: '',
            overview: '',
            logoSrc: null,
            logoSrcset: {},
            productRibbon: null,
            merchandisingBadges: []
        },
        customAttributes: [],
        edition: {
            name: 'Recurring Edition'
        },
        overview: {
            title: null,
            description: '',
            callToActions: {},
            image: null,
            imageSrcSet: {},
            showThirdPartyLegalDisclosure: false,
            documentationUrl: null,
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [],
            benefits: [],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: null,
            privacyUrl: null,
            email: null,
            phone: null,
            knowledgebaseUrl: null,
            description: '',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [],
        bundle: {
            id: 329919,
            url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329907'
        }
    },
    {
        id: 329911,
        focused: false,
        url: '/en-US/apps/329911/px-eva-peak-load-management-module',
        name: 'PX-EVA Peak Load Management Module',
        image: null,
        card: {
            title: 'PX-EVA Peak Load Management Module',
            details: '',
            overview: '',
            logoSrc: null,
            logoSrcset: {},
            productRibbon: null,
            merchandisingBadges: []
        },
        customAttributes: [],
        edition: {
            name: 'Recurring Edition'
        },
        overview: {
            title: null,
            description: '',
            callToActions: {},
            image: null,
            imageSrcSet: {},
            showThirdPartyLegalDisclosure: false,
            documentationUrl: null,
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [],
            benefits: [],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: null,
            privacyUrl: null,
            email: null,
            phone: null,
            knowledgebaseUrl: null,
            description: '',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [],
        bundle: {
            id: 329919,
            url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329911'
        }
    },
    {
        id: 329916,
        focused: false,
        url: '/en-US/apps/329916/px-eva-automated-utility-data-capture',
        name: 'PX-EVA Automated Utility Data Capture',
        image: null,
        card: {
            title: 'PX-EVA Automated Utility Data Capture',
            details: '',
            overview: '',
            logoSrc: null,
            logoSrcset: {},
            productRibbon: null,
            merchandisingBadges: []
        },
        customAttributes: [],
        edition: {
            name: 'Recurring Edition'
        },
        overview: {
            title: null,
            description: '',
            callToActions: {},
            image: null,
            imageSrcSet: {},
            showThirdPartyLegalDisclosure: false,
            documentationUrl: null,
            downloadDocumentationUrl: null,
            pricingUrl: 'undefined/editions',
            customers: [],
            benefits: [],
            i18n: {
                customers: 'Featured Customers',
                featuresAndBenefits: 'Features and Benefits',
                demo: 'Demo',
                overviewImage: 'Overview Image',
                documentation: 'Read Documentation'
            }
        },
        resources: {
            reseller: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            public: {
                videos: {
                    items: []
                },
                documents: {
                    items: []
                }
            },
            i18n: {
                Resources: 'Resources',
                Videos: 'Videos',
                Documents: 'Documents',
                ResellerResources: 'Reseller Resources',
                Download: 'Download'
            }
        },
        screenshots: {
            items: [],
            i18n: {
                title: 'Screenshots',
                sidebarTitle: 'View Screenshots'
            }
        },
        support: {
            termsUrl: null,
            privacyUrl: null,
            email: null,
            phone: null,
            knowledgebaseUrl: null,
            description: '',
            i18n: {
                termsAndConditions: 'Terms & Conditions',
                title: 'Additional Information',
                termsUrl: 'Terms of Service',
                privacyUrl: 'Privacy Policy',
                support: 'Support',
                email: 'Email',
                description: 'Description',
                phone: 'Phone',
                knowledgebaseUrl: 'Knowledge Base'
            }
        },
        tags: [],
        bundle: {
            id: 329919,
            url: '/en-US/bundles/329919/px-eva-product-bundle/bundledProducts/329916'
        }
    }
];
