// @appdirect/sfb-theme-components/src/components/product-description/stories/ProductDescription.stories.js
import React from 'react';
import ProductDescription from '../ProductDescription';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';
import { defaultMock as mockData } from '../../../../mocks/components/productViewerDescription';

const containerStyles = { maxWidth: 600, margin: '0 auto' };

export default {
    title: 'Components/Product Description',
    component: ProductDescription,
    argTypes: getArgTypes(ProductDescription.schema().form)
};

export const Default = {
    render: args => <ProductDescription data={mockData} settings={args} />,
    args: getArgs(ProductDescription.schema().form)
};

export const SmallContainer = {
    render: args => (
        <div style={containerStyles}>
            <ProductDescription data={mockData} settings={args} />
        </div>
    ),
    args: getArgs(ProductDescription.schema().form)
};

export const WithSettings = {
    render: args => (
        <div style={containerStyles}>
            <ProductDescription data={mockData} settings={args} />
        </div>
    ),
    args: getArgs(ProductDescription.schema().form)
};
