// @appdirect/sfb-theme-components/src/atoms/button-page-scroll/stories/ButtonPageScroll.stories.js
import React from 'react';
import ButtonPageScroll from '../ButtonPageScroll';

export default {
    title: 'Atoms/Button Page Scroll',
    component: ButtonPageScroll
};

export const ScrollToTop = {
    render: () => (
        <div
            style={{
                height: '100vh'
            }}
        >
            <p>Please scroll at the bottom to view the button</p>
            <span
                style={{
                    position: 'absolute',
                    bottom: 0
                }}
            >
                <ButtonPageScroll text="Back to top" />
            </span>
        </div>
    ),

    name: 'Scroll to top'
};

export const ScrollToBottom = {
    render: () => (
        <div
            style={{
                marginTop: '50px',
                height: '100vh'
            }}
        >
            <ButtonPageScroll text="Scroll to bottom" direction="bottom" />
        </div>
    ),

    name: 'Scroll to bottom'
};

export const HiddenText = {
    render: () => (
        <div
            style={{
                height: '100vh'
            }}
        >
            <p>Please scroll at the bottom to view the button</p>
            <span
                style={{
                    position: 'absolute',
                    bottom: 0
                }}
            >
                <ButtonPageScroll text="Back to top" hiddenText />
            </span>
        </div>
    ),

    name: 'Hidden text'
};
