// @appdirect/sfb-theme-components/src/atoms/product-edition-v2/list/stories/ProductEditionListV2.stories.js
import React from 'react';
import { editionsMockData as mockData } from '../../../../../mocks/product/productEditions';
import { ProductEditionListComponent as ProductEditionListV2 } from '../ProductEditionListV2';

const editionsData = {
    editions: mockData.items,
    i18n: mockData.i18n,
    callToActions: {
        items: [
            {
                label: 'Buy Now'
            }
        ]
    }
};

export default {
    title: 'Components (Preview)/Product Configurator V2/Atoms/Product Edition List V2',
    component: ProductEditionListV2
};

export const Default = {
    render: () => <ProductEditionListV2 data={editionsData} />,
    name: 'Default'
};

export const Carousel = {
    render: () => (
        <div
            style={{
                maxWidth: 600,
                margin: '0 auto'
            }}
        >
            <ProductEditionListV2
                data={{
                    ...editionsData,
                    editions: editionsData.editions.slice(0, 4)
                }}
                carouselEnabled
            />
        </div>
    ),

    name: 'Carousel'
};

export const ConfiguratorMode = {
    render: () => <ProductEditionListV2 data={editionsData} configuratorMode />,
    name: 'Configurator mode'
};
