// @appdirect/sfb-theme-components/src/components/header/atoms/header-home-bar/stories/HeaderHomeBar.stories.js
import React from 'react';
import { getArgTypes, getArgs } from '../../../../../tools/controlsHelper';
import { headerHomeBarDataMock, headerHomeBarSettingsMock } from '../../../../../../mocks/components/header';
import HeaderHomeBar from '../HeaderHomeBar';
import { configUserLoggedInMock } from '../../../../../../mocks/config';

export default {
    title: 'Global Components/Header/Atoms/Header Home Bar',
    component: HeaderHomeBar,
    argTypes: getArgTypes(HeaderHomeBar.schema().form),
    parameters: {
        layout: 'fullscreen'
    }
};

export const Default = {
    render: args => <HeaderHomeBar data={headerHomeBarDataMock} settings={{ ...headerHomeBarSettingsMock, ...args }} />,
    args: getArgs(HeaderHomeBar.schema().form)
};

export const WithSearchBar = {
    render: args => (
        <HeaderHomeBar
            data={headerHomeBarDataMock}
            settings={{ ...headerHomeBarSettingsMock, ...args }}
            displaySearchBar
        />
    ),
    args: getArgs(HeaderHomeBar.schema().form)
};

export const WithUserLoggedIn = {
    render: args => (
        <HeaderHomeBar
            data={{
                ...headerHomeBarDataMock,
                user: configUserLoggedInMock.user
            }}
            settings={args}
            displaySearchBar
        />
    ),
    args: getArgs(HeaderHomeBar.schema().form)
};
