// @appdirect/sfb-theme-components/src/components/index.js
import '../tools/derivedColorsObserver';

// import global stylesheet for the theme's website
import '../styles/global.scss';

/**
 * the export names here will match the `name` attribute in the themes;
 * this means that any change to existing exports has a high chance of breaking existing themes
 * handle with caution!
 */

export { default as React } from 'react';
export { default as ReactDOM } from 'react-dom';

export { default as AtomsShowcase } from './atoms-showcase/AtomsShowcase';
export { default as Carousel } from './carousel/Carousel';
export { default as CategorySidebar } from './category-sidebar/CategorySidebar';
export { default as ComparableProducts } from './comparable-products/ComparableProducts';
export { default as Faq } from './faq/Faq';
export { default as FeatureDetails } from './product-feature-details/FeatureDetails';
export { default as FeaturedCustomers } from './product-featured-customers/FeaturedCustomers';
export { default as Footer } from './footer/Footer';
export { default as Header } from './header/Header';
export { default as HeaderAnnouncementBar } from './header/atoms/header-announcement-bar/HeaderAnnouncementBar';
export { default as HeaderAuxiliaryBar } from './header/atoms/header-auxiliary-bar/HeaderAuxiliaryBar';
export { default as HeaderHomeBar } from './header/atoms/header-home-bar/HeaderHomeBar';
export { default as HeaderNavigationBar } from './header/atoms/header-navigation-bar/HeaderNavigationBar';
export { default as HeaderSearchBar } from './header/atoms/common/header-search-bar/HeaderSearchBar';
export { default as HeaderCategoriesMenu } from './header/atoms/common/header-categories-menu/HeaderCategoriesMenu';
export { default as GlobalAlerts } from './global-alerts/GlobalAlerts';
export { default as GraphQLAdapter } from './graphql-adapter/GraphQLAdapter';
export { default as LinkSection } from './faq-link-section/LinkSection';
export { default as OrderConfirmation } from './order-confirmation/OrderConfirmation';
export { default as PageOrderableContainer } from './page-orderable-container/PageOrderableContainer';
export { default as ProductAdditionalInfo } from './product-additional-info/AdditionalInfo';
export { default as ProductBanner } from './product-banner/ProductBanner';
export { default as BundleBanner } from './product-banner/BundleBanner';
export { default as ProductBundles } from './product-bundles/ProductBundles';
export { default as ProductBundleViewer } from './product-bundle-viewer/ProductBundleViewer';
export { default as ProductConfigurator } from './product-configurator/ProductConfigurator';
export { default as ProductConfiguratorV2 } from './product-configurator-v2/ProductConfiguratorV2';
export { default as ProductDescription } from './product-viewer-description/ProductViewerDescription';
export { default as ProductLeadForm } from './product-lead-form/LeadForm';
export { default as ProductListing } from './product-listing/ProductListing';
export { default as ProductListingControls } from './product-listing-controls/ProductListingControls';
export { default as ProductImageViewer } from './product-image-viewer/ProductImageViewer';
export { default as ProductQuestions } from './product-questions/ProductQuestions';
export { default as ProductQuestionForm } from './product-question-form/ProductQuestionForm';
export { default as ProductPlan } from './product-plan/ProductPlan';
export { default as DomainApp } from './domain-app/DomainApp';
export { default as ProductAssociatedDomain } from './product-associated-domain/ProductAssociatedDomain';
export { default as ProductPricing } from './product-pricing/ProductPricing';
export { default as ProductResources } from './product-resources/Resources';
export { default as ProductReviewForm } from './product-review-form/ProductReviewForm';
export { default as ProductReviews } from './product-reviews/ProductReviewsAdapter';
export { default as ProductReviewDetail } from './product-review-detail/ProductReviewDetail';
export { default as ProductSimilarApps } from './product-similar-apps/ProductSimilarApps';
export { default as ProductTabMenu } from './product-tabs-menu/TabsMenu';
export { default as ProductVariants } from './product-variants/ProductVariants';
export { default as BundleTabsMenu } from './product-tabs-menu/BundleTabsMenu';
export { default as ProductTopFeatures } from './product-top-features/TopFeatures';
export { default as SecondaryNav } from './secondary-nav/SecondaryNav';
export { default as Slider } from './product-slider/ProductSlider';
export { default as StickyButtonPageScroll } from './sticky-button-page-scroll/StickyButtonPageScroll';
export { default as VolumeDiscountCard } from '../atoms/volume-discount-card/VolumeDiscountCard';
export { default as FeaturedProduct } from './featured-product/FeaturedProduct';
export { default as ProductGroups } from './product-groups/ProductGroups';
export { default as HeroBanner } from './hero-banner/HeroBanner';
