// @appdirect/sfb-theme-components/src/atoms/product-groups-tiles/card-style/stories/CardStyle.stories.js
import React from 'react';
import CardStyle from '../../../../constants/atoms/schema/product-groups-tiles/card-style';
import { defaultValues } from '../../../../../mocks/product/productGroupsTiles';
import { getArgTypes, getArgs } from '../../../../tools/controlsHelper';

export default {
    title: 'Atoms/ProductGroupsStyles/CardStyle',
    component: CardStyle,
    argTypes: {
        customImageUrl: {
            control: 'text',
            description: 'The URL of the custom image to display'
        },
        ...getArgTypes(CardStyle.schema().form)
    },
    render: ({ customImageUrl, ...args }) => (
        <CardStyle {...defaultValues} {...args} customImage={{ url: customImageUrl }} />
    ),
    parameters: {
        controls: {
            include: [...Object.keys(CardStyle.schema().form), 'customImageUrl'],
            exclude: ['customImage']
        }
    }
};

export const Default = {
    args: {
        customImageUrl: defaultValues.customImage.defaultValue.url,
        ...getArgs(CardStyle.schema().form),
        ...defaultValues
    }
};
