// @appdirect/sfb-theme-components/src/composite-demos/full-width/stories/FullWidth.stories.jsx
import React from 'react';

import { TabsMenuComponent } from '../../../components/product-tabs-menu/TabsMenu';
import { FullWidth } from '../FullWidth';
import { navigationMock } from '../../../../mocks/product/navigation';
import { sliderMockSettings } from '../../../../mocks/home/slider';
import { CarouselComponent } from '../../../components/carousel/Carousel';
import { carouselMockData } from '../../../../mocks/home/carousel';
import { ProductSliderComponent } from '../../../components/product-slider/ProductSlider';
import { featuresMock } from '../../../../mocks/product/top-features';
import TopFeatures from '../../../components/product-top-features/TopFeatures';

export default {
    title: 'Composite Demos/Full Width Components',
    component: FullWidth,
    parameters: {
        layout: 'fullscreen',
        container: 'fluid'
    }
};

export const Default = () => {
    const tabs = <TabsMenuComponent data={navigationMock} />;
    const slider = <ProductSliderComponent {...sliderMockSettings} />;
    const topFeatures = <TopFeatures data={featuresMock} />;
    const carousel = <CarouselComponent settings={{ ...carouselMockData, backgroundColor: '#aaa' }} />;
    return <FullWidth tabs={tabs} slider={slider} topFeatures={topFeatures} carousel={carousel} />;
};
