// @appdirect/sfb-theme-components/src/constants/prop-types/listingProducts/i18n.js
import PropTypes from 'prop-types';

export const i18nPropTypes = {
    allApplications: PropTypes.string,
    applySortOptions: PropTypes.string,
    backToTop: PropTypes.string,
    categories: PropTypes.string,
    compare: PropTypes.string,
    filters: PropTypes.string,
    grid: PropTypes.string,
    hideFilters: PropTypes.string,
    home: PropTypes.string,
    learnMore: PropTypes.string,
    list: PropTypes.string,
    next: PropTypes.string,
    previous: PropTypes.string,
    results: PropTypes.string,
    resultsFrom: PropTypes.string,
    searchLabel: PropTypes.string,
    selectAppsCompare: PropTypes.string,
    selectToCompare: PropTypes.string,
    showFilters: PropTypes.string,
    soldBy: PropTypes.string,
    sort: PropTypes.string,
    sortBy: PropTypes.string,
    startingAt: PropTypes.string,
    title: PropTypes.string,
    titleProduct: PropTypes.string,
    viewMode: PropTypes.string,
    viewMore: PropTypes.string,
    viewProfile: PropTypes.string,
    version: PropTypes.string,
    rating: PropTypes.string,
    from: PropTypes.string,
    badges: PropTypes.string,
    merchandisingBadges: PropTypes.string
};
