// @appdirect/sfb-theme-components/src/atoms/select/stories/Select.stories.js
import React from 'react';
import { DARK } from '../../../constants/themeColor';
import { sortingOptionsMock } from '../../../../mocks/product/productListing';
import Select from '../Select';

const { options } = sortingOptionsMock;

export default {
    title: 'Atoms/Select',
    component: Select
};

export const Default = {
    render: () => <Select label="Sorted By" name="select" options={options} />,
    name: 'Default'
};

export const Required = {
    render: () => <Select label="Sorted By" name="select" options={options} required />,
    name: 'Required'
};

export const Inline = {
    render: () => <Select label="Sorted By" name="select" inline options={options} />,
    name: 'Inline'
};

export const SmallHeight = {
    render: () => <Select label="Sorted By" name="select" size="small" options={options} />,
    name: 'Small height'
};

export const HiddenLabel = {
    render: () => <Select label="Sorted By" name="select" options={options} hiddenLabel />,
    name: 'Hidden label'
};

export const ColorTheme = {
    render: () => (
        <div
            style={{
                width: 200,
                padding: '30px',
                backgroundColor: '#F8F8F8'
            }}
        >
            <Select label="Theme dark" name="select" value={options[1].value} theme={DARK.value} options={options} />
        </div>
    ),

    name: 'Color theme'
};

export const WithoutBackground = {
    render: () => (
        <div
            style={{
                width: 200,
                padding: '30px',
                backgroundColor: 'red'
            }}
        >
            <Select
                label="Theme Dark"
                name="select"
                value={options[1].value}
                withBackground={false}
                options={options}
            />
        </div>
    ),

    name: 'Without background'
};

export const Searchable = {
    render: () => <Select label="Sorted By" name="select" isSearchable options={options} />,
    name: 'Searchable'
};

export const ButtonStyles = {
    render: () => <Select label="Sorted By" name="select" type="button" options={options} />,
    name: 'Button styles'
};

export const WithError = {
    render: () => <Select label="Sorted By" name="select" error="This is required" options={options} />,

    name: 'With error'
};

export const CustomWidth = {
    render: () => (
        <div
            style={{
                margin: '10px'
            }}
        >
            <Select label="200px wide" name="select" width={200} options={options} />
        </div>
    ),

    name: 'Custom width'
};
