// @appdirect/sfb-theme-components/src/components/product-variants/atoms/variant-option-list/stories/VariantOptionList.stories.js
import React from 'react';
import VariantOptionList from '../VariantOptionList';
import { disableOption, productVariantsMock } from '../../../../../../mocks/product/product-variants/productVariants';
import { parseProduct } from '../../../../../tools/parsers/productParser';

const variantOption = parseProduct(productVariantsMock).variantOptions[0];
const variantOptionWithDisabled = disableOption(variantOption);

export default {
    title: 'Components (Preview)/Product Variants/Atoms/VariantOptionList',
    component: VariantOptionList
};

export const Default = {
    render: () => <VariantOptionList variantOption={variantOption} />,
    name: 'Default'
};

export const DefaultWithDisabled = {
    render: () => <VariantOptionList variantOption={variantOptionWithDisabled} />,
    name: 'Default with disabled'
};

export const Condensed = {
    render: () => <VariantOptionList variantOption={variantOption} isCondensed />,
    name: 'Condensed'
};

export const CondensedWithDisabled = {
    render: () => <VariantOptionList variantOption={variantOptionWithDisabled} isCondensed />,
    name: 'Condensed with disabled'
};
