// @appdirect/sfb-theme-components/mocks/application/leads/form.js
const stateOptions = [
    {
        label: 'Alabama',
        value: 'AL',
        isSelected: false
    },
    {
        label: 'Alaska',
        value: 'AK',
        isSelected: false
    },
    {
        label: 'American Samoa',
        value: 'AS',
        isSelected: false
    },
    {
        label: 'Arizona',
        value: 'AZ',
        isSelected: false
    },
    {
        label: 'Arkansas',
        value: 'AR',
        isSelected: false
    },
    {
        label: 'Armed Forces Europe',
        value: 'AE',
        isSelected: false
    },
    {
        label: 'Armed Forces Pacific',
        value: 'AP',
        isSelected: false
    },
    {
        label: 'Armed Forces the Americas',
        value: 'AA',
        isSelected: false
    },
    {
        label: 'California',
        value: 'CA',
        isSelected: false
    },
    {
        label: 'Colorado',
        value: 'CO',
        isSelected: false
    },
    {
        label: 'Connecticut',
        value: 'CT',
        isSelected: false
    },
    {
        label: 'Delaware',
        value: 'DE',
        isSelected: false
    },
    {
        label: 'District of Columbia',
        value: 'DC',
        isSelected: false
    },
    {
        label: 'Federated States of Micronesia',
        value: 'FM',
        isSelected: false
    },
    {
        label: 'Florida',
        value: 'FL',
        isSelected: false
    },
    {
        label: 'Georgia',
        value: 'GA',
        isSelected: false
    },
    {
        label: 'Guam',
        value: 'GU',
        isSelected: false
    },
    {
        label: 'Hawaii',
        value: 'HI',
        isSelected: false
    },
    {
        label: 'Idaho',
        value: 'ID',
        isSelected: false
    },
    {
        label: 'Illinois',
        value: 'IL',
        isSelected: false
    },
    {
        label: 'Indiana',
        value: 'IN',
        isSelected: false
    },
    {
        label: 'Iowa',
        value: 'IA',
        isSelected: false
    },
    {
        label: 'Kansas',
        value: 'KS',
        isSelected: false
    },
    {
        label: 'Kentucky',
        value: 'KY',
        isSelected: false
    },
    {
        label: 'Louisiana',
        value: 'LA',
        isSelected: false
    },
    {
        label: 'Maine',
        value: 'ME',
        isSelected: false
    },
    {
        label: 'Marshall Islands',
        value: 'MH',
        isSelected: false
    },
    {
        label: 'Maryland',
        value: 'MD',
        isSelected: false
    },
    {
        label: 'Massachusetts',
        value: 'MA',
        isSelected: false
    },
    {
        label: 'Michigan',
        value: 'MI',
        isSelected: false
    },
    {
        label: 'Minnesota',
        value: 'MN',
        isSelected: false
    },
    {
        label: 'Mississippi',
        value: 'MS',
        isSelected: false
    },
    {
        label: 'Missouri',
        value: 'MO',
        isSelected: false
    },
    {
        label: 'Montana',
        value: 'MT',
        isSelected: false
    },
    {
        label: 'Nebraska',
        value: 'NE',
        isSelected: false
    },
    {
        label: 'Nevada',
        value: 'NV',
        isSelected: false
    },
    {
        label: 'New Hampshire',
        value: 'NH',
        isSelected: false
    },
    {
        label: 'New Jersey',
        value: 'NJ',
        isSelected: false
    },
    {
        label: 'New Mexico',
        value: 'NM',
        isSelected: false
    },
    {
        label: 'New York',
        value: 'NY',
        isSelected: false
    },
    {
        label: 'North Carolina',
        value: 'NC',
        isSelected: false
    },
    {
        label: 'North Dakota',
        value: 'ND',
        isSelected: false
    },
    {
        label: 'Northern Mariana Islands',
        value: 'MP',
        isSelected: false
    },
    {
        label: 'Ohio',
        value: 'OH',
        isSelected: false
    },
    {
        label: 'Oklahoma',
        value: 'OK',
        isSelected: false
    },
    {
        label: 'Oregon',
        value: 'OR',
        isSelected: false
    },
    {
        label: 'Pennsylvania',
        value: 'PA',
        isSelected: false
    },
    {
        label: 'Puerto Rico',
        value: 'PR',
        isSelected: false
    },
    {
        label: 'Rhode Island',
        value: 'RI',
        isSelected: false
    },
    {
        label: 'South Carolina',
        value: 'SC',
        isSelected: false
    },
    {
        label: 'South Dakota',
        value: 'SD',
        isSelected: false
    },
    {
        label: 'Tennessee',
        value: 'TN',
        isSelected: false
    },
    {
        label: 'Texas',
        value: 'TX',
        isSelected: false
    },
    {
        label: 'Utah',
        value: 'UT',
        isSelected: false
    },
    {
        label: 'Vermont',
        value: 'VT',
        isSelected: false
    },
    {
        label: 'Virgin Islands, U.S.',
        value: 'VI',
        isSelected: false
    },
    {
        label: 'Virginia',
        value: 'VA',
        isSelected: false
    },
    {
        label: 'Washington',
        value: 'WA',
        isSelected: false
    },
    {
        label: 'West Virginia',
        value: 'WV',
        isSelected: false
    },
    {
        label: 'Wisconsin',
        value: 'WI',
        isSelected: false
    },
    {
        label: 'Wyoming',
        value: 'WY',
        isSelected: false
    }
];
const provinceOptions = [
    {
        value: 'AB',
        label: 'Alberta',
        isSelected: false
    },
    {
        value: 'BC',
        label: 'British Columbia',
        isSelected: false
    },
    {
        value: 'MB',
        label: 'Manitoba',
        isSelected: false
    },
    {
        value: 'NB',
        label: 'New Brunswick',
        isSelected: false
    },
    {
        value: 'NL',
        label: 'Newfoundland',
        isSelected: false
    },
    {
        value: 'NS',
        label: 'Nova Scotia',
        isSelected: false
    },
    {
        value: 'ON',
        label: 'Ontario',
        isSelected: false
    },
    {
        value: 'PE',
        label: 'Prince Edward Island',
        isSelected: false
    },
    {
        value: 'QC',
        label: 'Quebec',
        isSelected: true
    },
    {
        value: 'SK',
        label: 'Saskatchewan',
        isSelected: false
    },
    {
        value: 'NT',
        label: 'Northwest',
        isSelected: false
    },
    {
        value: 'NU',
        label: 'Nunavut',
        isSelected: false
    },
    {
        value: 'YT',
        label: 'Yukon',
        isSelected: false
    }
];
const countryOptions = [
    {
        value: 'US',
        value_iso3: 'USA',
        phoneCode: '1',
        label: 'United States',
        isSelected: false
    },
    {
        value: 'CA',
        value_iso3: 'CAN',
        phoneCode: '1',
        label: 'Canada',
        isSelected: true
    },
    {
        value: 'AF',
        value_iso3: 'AFG',
        phoneCode: '93',
        label: 'Afghanistan',
        isSelected: false
    },
    {
        value: 'AX',
        value_iso3: 'ALA',
        phoneCode: '358',
        label: 'Åland Islands',
        isSelected: false
    },
    {
        value: 'AL',
        value_iso3: 'ALB',
        phoneCode: '355',
        label: 'Albania',
        isSelected: false
    },
    {
        value: 'DZ',
        value_iso3: 'DZA',
        phoneCode: '213',
        label: 'Algeria',
        isSelected: false
    },
    {
        value: 'AS',
        value_iso3: 'ASM',
        phoneCode: '1',
        label: 'American Samoa',
        isSelected: false
    },
    {
        value: 'AD',
        value_iso3: 'AND',
        phoneCode: '376',
        label: 'Andorra',
        isSelected: false
    },
    {
        value: 'AO',
        value_iso3: 'AGO',
        phoneCode: '244',
        label: 'Angola',
        isSelected: false
    },
    {
        value: 'AI',
        value_iso3: 'AIA',
        phoneCode: '1',
        label: 'Anguilla',
        isSelected: false
    },
    {
        value: 'AQ',
        value_iso3: 'ATA',
        phoneCode: '0',
        label: 'Antarctica',
        isSelected: false
    },
    {
        value: 'AG',
        value_iso3: 'ATG',
        phoneCode: '1',
        label: 'Antigua and Barbuda',
        isSelected: false
    },
    {
        value: 'AR',
        value_iso3: 'ARG',
        phoneCode: '54',
        label: 'Argentina',
        isSelected: false
    },
    {
        value: 'AM',
        value_iso3: 'ARM',
        phoneCode: '374',
        label: 'Armenia',
        isSelected: false
    },
    {
        value: 'AW',
        value_iso3: 'ABW',
        phoneCode: '297',
        label: 'Aruba',
        isSelected: false
    },
    {
        value: 'AU',
        value_iso3: 'AUS',
        phoneCode: '61',
        label: 'Australia',
        isSelected: false
    },
    {
        value: 'AT',
        value_iso3: 'AUT',
        phoneCode: '43',
        label: 'Austria',
        isSelected: false
    },
    {
        value: 'AZ',
        value_iso3: 'AZE',
        phoneCode: '994',
        label: 'Azerbaijan',
        isSelected: false
    },
    {
        value: 'BS',
        value_iso3: 'BHS',
        phoneCode: '1',
        label: 'Bahamas',
        isSelected: false
    },
    {
        value: 'BH',
        value_iso3: 'BHR',
        phoneCode: '973',
        label: 'Bahrain',
        isSelected: false
    },
    {
        value: 'BD',
        value_iso3: 'BGD',
        phoneCode: '880',
        label: 'Bangladesh',
        isSelected: false
    },
    {
        value: 'BB',
        value_iso3: 'BRB',
        phoneCode: '1',
        label: 'Barbados',
        isSelected: false
    },
    {
        value: 'BY',
        value_iso3: 'BLR',
        phoneCode: '375',
        label: 'Belarus',
        isSelected: false
    },
    {
        value: 'BE',
        value_iso3: 'BEL',
        phoneCode: '32',
        label: 'Belgium',
        isSelected: false
    },
    {
        value: 'BZ',
        value_iso3: 'BLZ',
        phoneCode: '501',
        label: 'Belize',
        isSelected: false
    },
    {
        value: 'BJ',
        value_iso3: 'BEN',
        phoneCode: '229',
        label: 'Benin',
        isSelected: false
    },
    {
        value: 'BM',
        value_iso3: 'BMU',
        phoneCode: '1',
        label: 'Bermuda',
        isSelected: false
    },
    {
        value: 'BT',
        value_iso3: 'BTN',
        phoneCode: '975',
        label: 'Bhutan',
        isSelected: false
    },
    {
        value: 'BO',
        value_iso3: 'BOL',
        phoneCode: '591',
        label: 'Bolivia, Plurinational State of',
        isSelected: false
    },
    {
        value: 'BQ',
        value_iso3: 'BES',
        phoneCode: '599',
        label: 'Bonaire, Sint Eustatius and Saba',
        isSelected: false
    },
    {
        value: 'BA',
        value_iso3: 'BIH',
        phoneCode: '387',
        label: 'Bosnia and Herzegovina',
        isSelected: false
    },
    {
        value: 'BW',
        value_iso3: 'BWA',
        phoneCode: '267',
        label: 'Botswana',
        isSelected: false
    },
    {
        value: 'BV',
        value_iso3: 'BVT',
        phoneCode: '0',
        label: 'Bouvet Island',
        isSelected: false
    },
    {
        value: 'BR',
        value_iso3: 'BRA',
        phoneCode: '55',
        label: 'Brazil',
        isSelected: false
    },
    {
        value: 'IO',
        value_iso3: 'IOT',
        phoneCode: '246',
        label: 'British Indian Ocean Territory',
        isSelected: false
    },
    {
        value: 'BN',
        value_iso3: 'BRN',
        phoneCode: '673',
        label: 'Brunei Darussalam',
        isSelected: false
    },
    {
        value: 'BG',
        value_iso3: 'BGR',
        phoneCode: '359',
        label: 'Bulgaria',
        isSelected: false
    },
    {
        value: 'BF',
        value_iso3: 'BFA',
        phoneCode: '226',
        label: 'Burkina Faso',
        isSelected: false
    },
    {
        value: 'BI',
        value_iso3: 'BDI',
        phoneCode: '257',
        label: 'Burundi',
        isSelected: false
    },
    {
        value: 'KH',
        value_iso3: 'KHM',
        phoneCode: '855',
        label: 'Cambodia',
        isSelected: false
    },
    {
        value: 'CM',
        value_iso3: 'CMR',
        phoneCode: '237',
        label: 'Cameroon',
        isSelected: false
    },
    {
        value: 'CV',
        value_iso3: 'CPV',
        phoneCode: '238',
        label: 'Cape Verde',
        isSelected: false
    },
    {
        value: 'KY',
        value_iso3: 'CYM',
        phoneCode: '1',
        label: 'Cayman Islands',
        isSelected: false
    },
    {
        value: 'CF',
        value_iso3: 'CAF',
        phoneCode: '236',
        label: 'Central African Republic',
        isSelected: false
    },
    {
        value: 'TD',
        value_iso3: 'TCD',
        phoneCode: '235',
        label: 'Chad',
        isSelected: false
    },
    {
        value: 'CL',
        value_iso3: 'CHL',
        phoneCode: '56',
        label: 'Chile',
        isSelected: false
    },
    {
        value: 'CN',
        value_iso3: 'CHN',
        phoneCode: '86',
        label: 'China',
        isSelected: false
    },
    {
        value: 'CX',
        value_iso3: 'CXR',
        phoneCode: '213',
        label: 'Christmas Island',
        isSelected: false
    },
    {
        value: 'CC',
        value_iso3: 'CCK',
        phoneCode: '61',
        label: 'Cocos (Keeling) Islands',
        isSelected: false
    },
    {
        value: 'CO',
        value_iso3: 'COL',
        phoneCode: '57',
        label: 'Colombia',
        isSelected: false
    },
    {
        value: 'KM',
        value_iso3: 'COM',
        phoneCode: '269',
        label: 'Comoros',
        isSelected: false
    },
    {
        value: 'CG',
        value_iso3: 'COG',
        phoneCode: '242',
        label: 'Congo',
        isSelected: false
    },
    {
        value: 'CD',
        value_iso3: 'COD',
        phoneCode: '243',
        label: 'Congo, the Democratic Republic of the',
        isSelected: false
    },
    {
        value: 'CK',
        value_iso3: 'COK',
        phoneCode: '682',
        label: 'Cook Islands',
        isSelected: false
    },
    {
        value: 'CR',
        value_iso3: 'CRI',
        phoneCode: '506',
        label: 'Costa Rica',
        isSelected: false
    },
    {
        value: 'CI',
        value_iso3: 'CIV',
        phoneCode: '225',
        label: 'Côte d\'Ivoire',
        isSelected: false
    },
    {
        value: 'HR',
        value_iso3: 'HRV',
        phoneCode: '385',
        label: 'Croatia',
        isSelected: false
    },
    {
        value: 'CU',
        value_iso3: 'CUB',
        phoneCode: '53',
        label: 'Cuba',
        isSelected: false
    },
    {
        value: 'CW',
        value_iso3: 'CUW',
        phoneCode: '599',
        label: 'Curaçao',
        isSelected: false
    },
    {
        value: 'CY',
        value_iso3: 'CYP',
        phoneCode: '357',
        label: 'Cyprus',
        isSelected: false
    },
    {
        value: 'CZ',
        value_iso3: 'CZE',
        phoneCode: '420',
        label: 'Czech Republic',
        isSelected: false
    },
    {
        value: 'DK',
        value_iso3: 'DNK',
        phoneCode: '45',
        label: 'Denmark',
        isSelected: false
    },
    {
        value: 'DJ',
        value_iso3: 'DJI',
        phoneCode: '253',
        label: 'Djibouti',
        isSelected: false
    },
    {
        value: 'DM',
        value_iso3: 'DMA',
        phoneCode: '1',
        label: 'Dominica',
        isSelected: false
    },
    {
        value: 'DO',
        value_iso3: 'DOM',
        phoneCode: '1',
        label: 'Dominican Republic',
        isSelected: false
    },
    {
        value: 'EC',
        value_iso3: 'ECU',
        phoneCode: '593',
        label: 'Ecuador',
        isSelected: false
    },
    {
        value: 'EG',
        value_iso3: 'EGY',
        phoneCode: '20',
        label: 'Egypt',
        isSelected: false
    },
    {
        value: 'SV',
        value_iso3: 'SLV',
        phoneCode: '503',
        label: 'El Salvador',
        isSelected: false
    },
    {
        value: 'GQ',
        value_iso3: 'GNQ',
        phoneCode: '240',
        label: 'Equatorial Guinea',
        isSelected: false
    },
    {
        value: 'ER',
        value_iso3: 'ERI',
        phoneCode: '291',
        label: 'Eritrea',
        isSelected: false
    },
    {
        value: 'EE',
        value_iso3: 'EST',
        phoneCode: '372',
        label: 'Estonia',
        isSelected: false
    },
    {
        value: 'ET',
        value_iso3: 'ETH',
        phoneCode: '251',
        label: 'Ethiopia',
        isSelected: false
    },
    {
        value: 'FK',
        value_iso3: 'FLK',
        phoneCode: '500',
        label: 'Falkland Islands (Malvinas)',
        isSelected: false
    },
    {
        value: 'FO',
        value_iso3: 'FRO',
        phoneCode: '298',
        label: 'Faroe Islands',
        isSelected: false
    },
    {
        value: 'FJ',
        value_iso3: 'FJI',
        phoneCode: '679',
        label: 'Fiji',
        isSelected: false
    },
    {
        value: 'FI',
        value_iso3: 'FIN',
        phoneCode: '358',
        label: 'Finland',
        isSelected: false
    },
    {
        value: 'FR',
        value_iso3: 'FRA',
        phoneCode: '33',
        label: 'France',
        isSelected: false
    },
    {
        value: 'GF',
        value_iso3: 'GUF',
        phoneCode: '594',
        label: 'French Guiana',
        isSelected: false
    },
    {
        value: 'PF',
        value_iso3: 'PYF',
        phoneCode: '689',
        label: 'French Polynesia',
        isSelected: false
    },
    {
        value: 'TF',
        value_iso3: 'ATF',
        phoneCode: '0',
        label: 'French Southern Territories',
        isSelected: false
    },
    {
        value: 'GA',
        value_iso3: 'GAB',
        phoneCode: '241',
        label: 'Gabon',
        isSelected: false
    },
    {
        value: 'GM',
        value_iso3: 'GMB',
        phoneCode: '220',
        label: 'Gambia',
        isSelected: false
    },
    {
        value: 'GE',
        value_iso3: 'GEO',
        phoneCode: '995',
        label: 'Georgia',
        isSelected: false
    },
    {
        value: 'DE',
        value_iso3: 'DEU',
        phoneCode: '49',
        label: 'Germany',
        isSelected: false
    },
    {
        value: 'GH',
        value_iso3: 'GHA',
        phoneCode: '233',
        label: 'Ghana',
        isSelected: false
    },
    {
        value: 'GI',
        value_iso3: 'GIB',
        phoneCode: '350',
        label: 'Gibraltar',
        isSelected: false
    },
    {
        value: 'GR',
        value_iso3: 'GRC',
        phoneCode: '30',
        label: 'Greece',
        isSelected: false
    },
    {
        value: 'GL',
        value_iso3: 'GRL',
        phoneCode: '299',
        label: 'Greenland',
        isSelected: false
    },
    {
        value: 'GD',
        value_iso3: 'GRD',
        phoneCode: '1',
        label: 'Grenada',
        isSelected: false
    },
    {
        value: 'GP',
        value_iso3: 'GLP',
        phoneCode: '590',
        label: 'Guadeloupe',
        isSelected: false
    },
    {
        value: 'GU',
        value_iso3: 'GUM',
        phoneCode: '1',
        label: 'Guam',
        isSelected: false
    },
    {
        value: 'GT',
        value_iso3: 'GTM',
        phoneCode: '502',
        label: 'Guatemala',
        isSelected: false
    },
    {
        value: 'GG',
        value_iso3: 'GGY',
        phoneCode: '44',
        label: 'Guernsey',
        isSelected: false
    },
    {
        value: 'GN',
        value_iso3: 'GIN',
        phoneCode: '224',
        label: 'Guinea',
        isSelected: false
    },
    {
        value: 'GW',
        value_iso3: 'GNB',
        phoneCode: '245',
        label: 'Guinea-Bissau',
        isSelected: false
    },
    {
        value: 'GY',
        value_iso3: 'GUY',
        phoneCode: '592',
        label: 'Guyana',
        isSelected: false
    },
    {
        value: 'HT',
        value_iso3: 'HTI',
        phoneCode: '509',
        label: 'Haiti',
        isSelected: false
    },
    {
        value: 'HM',
        value_iso3: 'HMD',
        phoneCode: '0',
        label: 'Heard Island and McDonald Islands',
        isSelected: false
    },
    {
        value: 'VA',
        value_iso3: 'VAT',
        phoneCode: '39',
        label: 'Holy See (Vatican City State)',
        isSelected: false
    },
    {
        value: 'HN',
        value_iso3: 'HND',
        phoneCode: '504',
        label: 'Honduras',
        isSelected: false
    },
    {
        value: 'HK',
        value_iso3: 'HKG',
        phoneCode: '852',
        label: 'Hong Kong',
        isSelected: false
    },
    {
        value: 'HU',
        value_iso3: 'HUN',
        phoneCode: '36',
        label: 'Hungary',
        isSelected: false
    },
    {
        value: 'IS',
        value_iso3: 'ISL',
        phoneCode: '354',
        label: 'Iceland',
        isSelected: false
    },
    {
        value: 'IN',
        value_iso3: 'IND',
        phoneCode: '91',
        label: 'India',
        isSelected: false
    },
    {
        value: 'ID',
        value_iso3: 'IDN',
        phoneCode: '62',
        label: 'Indonesia',
        isSelected: false
    },
    {
        value: 'IR',
        value_iso3: 'IRN',
        phoneCode: '98',
        label: 'Iran, Islamic Republic of',
        isSelected: false
    },
    {
        value: 'IQ',
        value_iso3: 'IRQ',
        phoneCode: '964',
        label: 'Iraq',
        isSelected: false
    },
    {
        value: 'IE',
        value_iso3: 'IRL',
        phoneCode: '353',
        label: 'Ireland',
        isSelected: false
    },
    {
        value: 'IM',
        value_iso3: 'IMN',
        phoneCode: '44',
        label: 'Isle of Man',
        isSelected: false
    },
    {
        value: 'IL',
        value_iso3: 'ISR',
        phoneCode: '952',
        label: 'Israel',
        isSelected: false
    },
    {
        value: 'IT',
        value_iso3: 'ITA',
        phoneCode: '39',
        label: 'Italy',
        isSelected: false
    },
    {
        value: 'JM',
        value_iso3: 'JAM',
        phoneCode: '1',
        label: 'Jamaica',
        isSelected: false
    },
    {
        value: 'JP',
        value_iso3: 'JPN',
        phoneCode: '81',
        label: 'Japan',
        isSelected: false
    },
    {
        value: 'JE',
        value_iso3: 'JEY',
        phoneCode: '44',
        label: 'Jersey',
        isSelected: false
    },
    {
        value: 'JO',
        value_iso3: 'JOR',
        phoneCode: '962',
        label: 'Jordan',
        isSelected: false
    },
    {
        value: 'KZ',
        value_iso3: 'KAZ',
        phoneCode: '7',
        label: 'Kazakhstan',
        isSelected: false
    },
    {
        value: 'KE',
        value_iso3: 'KEN',
        phoneCode: '254',
        label: 'Kenya',
        isSelected: false
    },
    {
        value: 'KI',
        value_iso3: 'KIR',
        phoneCode: '686',
        label: 'Kiribati',
        isSelected: false
    },
    {
        value: 'KP',
        value_iso3: 'PRK',
        phoneCode: '850',
        label: 'Korea, Democratic People\'s Republic of',
        isSelected: false
    },
    {
        value: 'KR',
        value_iso3: 'KOR',
        phoneCode: '82',
        label: 'Korea, Republic of',
        isSelected: false
    },
    {
        value: 'KW',
        value_iso3: 'KWT',
        phoneCode: '965',
        label: 'Kuwait',
        isSelected: false
    },
    {
        value: 'KG',
        value_iso3: 'KGZ',
        phoneCode: '996',
        label: 'Kyrgyzstan',
        isSelected: false
    },
    {
        value: 'LA',
        value_iso3: 'LAO',
        phoneCode: '856',
        label: 'Lao People\'s Democratic Republic',
        isSelected: false
    },
    {
        value: 'LV',
        value_iso3: 'LVA',
        phoneCode: '371',
        label: 'Latvia',
        isSelected: false
    },
    {
        value: 'LB',
        value_iso3: 'LBN',
        phoneCode: '961',
        label: 'Lebanon',
        isSelected: false
    },
    {
        value: 'LS',
        value_iso3: 'LSO',
        phoneCode: '266',
        label: 'Lesotho',
        isSelected: false
    },
    {
        value: 'LR',
        value_iso3: 'LBR',
        phoneCode: '231',
        label: 'Liberia',
        isSelected: false
    },
    {
        value: 'LY',
        value_iso3: 'LBY',
        phoneCode: '218',
        label: 'Libya',
        isSelected: false
    },
    {
        value: 'LI',
        value_iso3: 'LIE',
        phoneCode: '423',
        label: 'Liechtenstein',
        isSelected: false
    },
    {
        value: 'LT',
        value_iso3: 'LTU',
        phoneCode: '370',
        label: 'Lithuania',
        isSelected: false
    },
    {
        value: 'LU',
        value_iso3: 'LUX',
        phoneCode: '352',
        label: 'Luxembourg',
        isSelected: false
    },
    {
        value: 'MO',
        value_iso3: 'MAC',
        phoneCode: '853',
        label: 'Macao',
        isSelected: false
    },
    {
        value: 'MK',
        value_iso3: 'MKD',
        phoneCode: '389',
        label: 'Macedonia, the former Yugoslav Republic of',
        isSelected: false
    },
    {
        value: 'MG',
        value_iso3: 'MDG',
        phoneCode: '261',
        label: 'Madagascar',
        isSelected: false
    },
    {
        value: 'MW',
        value_iso3: 'MWI',
        phoneCode: '265',
        label: 'Malawi',
        isSelected: false
    },
    {
        value: 'MY',
        value_iso3: 'MYS',
        phoneCode: '60',
        label: 'Malaysia',
        isSelected: false
    },
    {
        value: 'MV',
        value_iso3: 'MDV',
        phoneCode: '960',
        label: 'Maldives',
        isSelected: false
    },
    {
        value: 'ML',
        value_iso3: 'MLI',
        phoneCode: '223',
        label: 'Mali',
        isSelected: false
    },
    {
        value: 'MT',
        value_iso3: 'MLT',
        phoneCode: '356',
        label: 'Malta',
        isSelected: false
    },
    {
        value: 'MH',
        value_iso3: 'MHL',
        phoneCode: '692',
        label: 'Marshall Islands',
        isSelected: false
    },
    {
        value: 'MQ',
        value_iso3: 'MTQ',
        phoneCode: '596',
        label: 'Martinique',
        isSelected: false
    },
    {
        value: 'MR',
        value_iso3: 'MRT',
        phoneCode: '222',
        label: 'Mauritania',
        isSelected: false
    },
    {
        value: 'MU',
        value_iso3: 'MUS',
        phoneCode: '230',
        label: 'Mauritius',
        isSelected: false
    },
    {
        value: 'YT',
        value_iso3: 'MYT',
        phoneCode: '262',
        label: 'Mayotte',
        isSelected: false
    },
    {
        value: 'MX',
        value_iso3: 'MEX',
        phoneCode: '52',
        label: 'Mexico',
        isSelected: false
    },
    {
        value: 'FM',
        value_iso3: 'FSM',
        phoneCode: '691',
        label: 'Micronesia, Federated States of',
        isSelected: false
    },
    {
        value: 'MD',
        value_iso3: 'MDA',
        phoneCode: '373',
        label: 'Moldova, Republic of',
        isSelected: false
    },
    {
        value: 'MC',
        value_iso3: 'MCO',
        phoneCode: '377',
        label: 'Monaco',
        isSelected: false
    },
    {
        value: 'MN',
        value_iso3: 'MNG',
        phoneCode: '976',
        label: 'Mongolia',
        isSelected: false
    },
    {
        value: 'ME',
        value_iso3: 'MNE',
        phoneCode: '382',
        label: 'Montenegro',
        isSelected: false
    },
    {
        value: 'MS',
        value_iso3: 'MSR',
        phoneCode: '1',
        label: 'Montserrat',
        isSelected: false
    },
    {
        value: 'MA',
        value_iso3: 'MAR',
        phoneCode: '212',
        label: 'Morocco',
        isSelected: false
    },
    {
        value: 'MZ',
        value_iso3: 'MOZ',
        phoneCode: '258',
        label: 'Mozambique',
        isSelected: false
    },
    {
        value: 'MM',
        value_iso3: 'MMR',
        phoneCode: '95',
        label: 'Myanmar',
        isSelected: false
    },
    {
        value: 'NA',
        value_iso3: 'NAM',
        phoneCode: '264',
        label: 'Namibia',
        isSelected: false
    },
    {
        value: 'NR',
        value_iso3: 'NRU',
        phoneCode: '674',
        label: 'Nauru',
        isSelected: false
    },
    {
        value: 'NP',
        value_iso3: 'NPL',
        phoneCode: '977',
        label: 'Nepal',
        isSelected: false
    },
    {
        value: 'NL',
        value_iso3: 'NLD',
        phoneCode: '31',
        label: 'Netherlands',
        isSelected: false
    },
    {
        value: 'NC',
        value_iso3: 'NCL',
        phoneCode: '687',
        label: 'New Caledonia',
        isSelected: false
    },
    {
        value: 'NZ',
        value_iso3: 'NZL',
        phoneCode: '64',
        label: 'New Zealand',
        isSelected: false
    },
    {
        value: 'NI',
        value_iso3: 'NIC',
        phoneCode: '505',
        label: 'Nicaragua',
        isSelected: false
    },
    {
        value: 'NE',
        value_iso3: 'NER',
        phoneCode: '227',
        label: 'Niger',
        isSelected: false
    },
    {
        value: 'NG',
        value_iso3: 'NGA',
        phoneCode: '234',
        label: 'Nigeria',
        isSelected: false
    },
    {
        value: 'NU',
        value_iso3: 'NIU',
        phoneCode: '683',
        label: 'Niue',
        isSelected: false
    },
    {
        value: 'NF',
        value_iso3: 'NFK',
        phoneCode: '672',
        label: 'Norfolk Island',
        isSelected: false
    },
    {
        value: 'MP',
        value_iso3: 'MNP',
        phoneCode: '1',
        label: 'Northern Mariana Islands',
        isSelected: false
    },
    {
        value: 'NO',
        value_iso3: 'NOR',
        phoneCode: '47',
        label: 'Norway',
        isSelected: false
    },
    {
        value: 'OM',
        value_iso3: 'OMN',
        phoneCode: '968',
        label: 'Oman',
        isSelected: false
    },
    {
        value: 'PK',
        value_iso3: 'PAK',
        phoneCode: '92',
        label: 'Pakistan',
        isSelected: false
    },
    {
        value: 'PW',
        value_iso3: 'PLW',
        phoneCode: '680',
        label: 'Palau',
        isSelected: false
    },
    {
        value: 'PS',
        value_iso3: 'PSE',
        phoneCode: '970',
        label: 'Palestine, State of',
        isSelected: false
    },
    {
        value: 'PA',
        value_iso3: 'PAN',
        phoneCode: '507',
        label: 'Panama',
        isSelected: false
    },
    {
        value: 'PG',
        value_iso3: 'PNG',
        phoneCode: '675',
        label: 'Papua New Guinea',
        isSelected: false
    },
    {
        value: 'PY',
        value_iso3: 'PRY',
        phoneCode: '595',
        label: 'Paraguay',
        isSelected: false
    },
    {
        value: 'PE',
        value_iso3: 'PER',
        phoneCode: '51',
        label: 'Peru',
        isSelected: false
    },
    {
        value: 'PH',
        value_iso3: 'PHL',
        phoneCode: '63',
        label: 'Philippines',
        isSelected: false
    },
    {
        value: 'PN',
        value_iso3: 'PCN',
        phoneCode: '0',
        label: 'Pitcairn',
        isSelected: false
    },
    {
        value: 'PL',
        value_iso3: 'POL',
        phoneCode: '48',
        label: 'Poland',
        isSelected: false
    },
    {
        value: 'PT',
        value_iso3: 'PRT',
        phoneCode: '351',
        label: 'Portugal',
        isSelected: false
    },
    {
        value: 'PR',
        value_iso3: 'PRI',
        phoneCode: '1',
        label: 'Puerto Rico',
        isSelected: false
    },
    {
        value: 'QA',
        value_iso3: 'QAT',
        phoneCode: '974',
        label: 'Qatar',
        isSelected: false
    },
    {
        value: 'RE',
        value_iso3: 'REU',
        phoneCode: '262',
        label: 'Réunion',
        isSelected: false
    },
    {
        value: 'RO',
        value_iso3: 'ROU',
        phoneCode: '40',
        label: 'Romania',
        isSelected: false
    },
    {
        value: 'RU',
        value_iso3: 'RUS',
        phoneCode: '7',
        label: 'Russian Federation',
        isSelected: false
    },
    {
        value: 'RW',
        value_iso3: 'RWA',
        phoneCode: '250',
        label: 'Rwanda',
        isSelected: false
    },
    {
        value: 'BL',
        value_iso3: 'BLM',
        phoneCode: '590',
        label: 'Saint Barthélemy',
        isSelected: false
    },
    {
        value: 'SH',
        value_iso3: 'SHN',
        phoneCode: '290',
        label: 'Saint Helena, Ascension and Tristan da Cunha',
        isSelected: false
    },
    {
        value: 'KN',
        value_iso3: 'KNA',
        phoneCode: '1',
        label: 'Saint Kitts and Nevis',
        isSelected: false
    },
    {
        value: 'LC',
        value_iso3: 'LCA',
        phoneCode: '1',
        label: 'Saint Lucia',
        isSelected: false
    },
    {
        value: 'MF',
        value_iso3: 'MAF',
        phoneCode: '590',
        label: 'Saint Martin (French part)',
        isSelected: false
    },
    {
        value: 'PM',
        value_iso3: 'SPM',
        phoneCode: '508',
        label: 'Saint Pierre and Miquelon',
        isSelected: false
    },
    {
        value: 'VC',
        value_iso3: 'VCT',
        phoneCode: '1',
        label: 'Saint Vincent and the Grenadines',
        isSelected: false
    },
    {
        value: 'WS',
        value_iso3: 'WSM',
        phoneCode: '685',
        label: 'Samoa',
        isSelected: false
    },
    {
        value: 'SM',
        value_iso3: 'SMR',
        phoneCode: '378',
        label: 'San Marino',
        isSelected: false
    },
    {
        value: 'ST',
        value_iso3: 'STP',
        phoneCode: '239',
        label: 'Sao Tome and Principe',
        isSelected: false
    },
    {
        value: 'SA',
        value_iso3: 'SAU',
        phoneCode: '966',
        label: 'Saudi Arabia',
        isSelected: false
    },
    {
        value: 'SN',
        value_iso3: 'SEN',
        phoneCode: '221',
        label: 'Senegal',
        isSelected: false
    },
    {
        value: 'RS',
        value_iso3: 'SRB',
        phoneCode: '381',
        label: 'Serbia',
        isSelected: false
    },
    {
        value: 'SC',
        value_iso3: 'SYC',
        phoneCode: '248',
        label: 'Seychelles',
        isSelected: false
    },
    {
        value: 'SL',
        value_iso3: 'SLE',
        phoneCode: '232',
        label: 'Sierra Leone',
        isSelected: false
    },
    {
        value: 'SG',
        value_iso3: 'SGP',
        phoneCode: '65',
        label: 'Singapore',
        isSelected: false
    },
    {
        value: 'SX',
        value_iso3: 'SXM',
        phoneCode: '1',
        label: 'Sint Maarten (Dutch part)',
        isSelected: false
    },
    {
        value: 'SK',
        value_iso3: 'SVK',
        phoneCode: '421',
        label: 'Slovakia',
        isSelected: false
    },
    {
        value: 'SI',
        value_iso3: 'SVN',
        phoneCode: '386',
        label: 'Slovenia',
        isSelected: false
    },
    {
        value: 'SB',
        value_iso3: 'SLB',
        phoneCode: '677',
        label: 'Solomon Islands',
        isSelected: false
    },
    {
        value: 'SO',
        value_iso3: 'SOM',
        phoneCode: '252',
        label: 'Somalia',
        isSelected: false
    },
    {
        value: 'ZA',
        value_iso3: 'ZAF',
        phoneCode: '27',
        label: 'South Africa',
        isSelected: false
    },
    {
        value: 'GS',
        value_iso3: 'SGS',
        phoneCode: '0',
        label: 'South Georgia and the South Sandwich Islands',
        isSelected: false
    },
    {
        value: 'SS',
        value_iso3: 'SSD',
        phoneCode: '211',
        label: 'South Sudan',
        isSelected: false
    },
    {
        value: 'ES',
        value_iso3: 'ESP',
        phoneCode: '34',
        label: 'Spain',
        isSelected: false
    },
    {
        value: 'LK',
        value_iso3: 'LKA',
        phoneCode: '94',
        label: 'Sri Lanka',
        isSelected: false
    },
    {
        value: 'SD',
        value_iso3: 'SDN',
        phoneCode: '249',
        label: 'Sudan',
        isSelected: false
    },
    {
        value: 'SR',
        value_iso3: 'SUR',
        phoneCode: '597',
        label: 'Suriname',
        isSelected: false
    },
    {
        value: 'SJ',
        value_iso3: 'SJM',
        phoneCode: '47',
        label: 'Svalbard and Jan Mayen',
        isSelected: false
    },
    {
        value: 'SZ',
        value_iso3: 'SWZ',
        phoneCode: '268',
        label: 'Swaziland',
        isSelected: false
    },
    {
        value: 'SE',
        value_iso3: 'SWE',
        phoneCode: '46',
        label: 'Sweden',
        isSelected: false
    },
    {
        value: 'CH',
        value_iso3: 'CHE',
        phoneCode: '41',
        label: 'Switzerland',
        isSelected: false
    },
    {
        value: 'SY',
        value_iso3: 'SYR',
        phoneCode: '963',
        label: 'Syrian Arab Republic',
        isSelected: false
    },
    {
        value: 'TW',
        value_iso3: 'TWN',
        phoneCode: '886',
        label: 'Taiwan, Province of China',
        isSelected: false
    },
    {
        value: 'TJ',
        value_iso3: 'TJK',
        phoneCode: '992',
        label: 'Tajikistan',
        isSelected: false
    },
    {
        value: 'TZ',
        value_iso3: 'TZA',
        phoneCode: '255',
        label: 'Tanzania, United Republic of',
        isSelected: false
    },
    {
        value: 'TH',
        value_iso3: 'THA',
        phoneCode: '66',
        label: 'Thailand',
        isSelected: false
    },
    {
        value: 'TL',
        value_iso3: 'TLS',
        phoneCode: '670',
        label: 'Timor-Leste',
        isSelected: false
    },
    {
        value: 'TG',
        value_iso3: 'TGO',
        phoneCode: '228',
        label: 'Togo',
        isSelected: false
    },
    {
        value: 'TK',
        value_iso3: 'TKL',
        phoneCode: '690',
        label: 'Tokelau',
        isSelected: false
    },
    {
        value: 'TO',
        value_iso3: 'TON',
        phoneCode: '676',
        label: 'Tonga',
        isSelected: false
    },
    {
        value: 'TT',
        value_iso3: 'TTO',
        phoneCode: '1',
        label: 'Trinidad and Tobago',
        isSelected: false
    },
    {
        value: 'TN',
        value_iso3: 'TUN',
        phoneCode: '216',
        label: 'Tunisia',
        isSelected: false
    },
    {
        value: 'TR',
        value_iso3: 'TUR',
        phoneCode: '90',
        label: 'Turkey',
        isSelected: false
    },
    {
        value: 'TM',
        value_iso3: 'TKM',
        phoneCode: '993',
        label: 'Turkmenistan',
        isSelected: false
    },
    {
        value: 'TC',
        value_iso3: 'TCA',
        phoneCode: '1',
        label: 'Turks and Caicos Islands',
        isSelected: false
    },
    {
        value: 'TV',
        value_iso3: 'TUV',
        phoneCode: '688',
        label: 'Tuvalu',
        isSelected: false
    },
    {
        value: 'UG',
        value_iso3: 'UGA',
        phoneCode: '256',
        label: 'Uganda',
        isSelected: false
    },
    {
        value: 'UA',
        value_iso3: 'UKR',
        phoneCode: '380',
        label: 'Ukraine',
        isSelected: false
    },
    {
        value: 'AE',
        value_iso3: 'ARE',
        phoneCode: '971',
        label: 'United Arab Emirates',
        isSelected: false
    },
    {
        value: 'GB',
        value_iso3: 'GBR',
        phoneCode: '44',
        label: 'United Kingdom',
        isSelected: false
    },
    {
        value: 'UM',
        value_iso3: 'UMI',
        phoneCode: '0',
        label: 'United States Minor Outlying Islands',
        isSelected: false
    },
    {
        value: 'UY',
        value_iso3: 'URY',
        phoneCode: '598',
        label: 'Uruguay',
        isSelected: false
    },
    {
        value: 'UZ',
        value_iso3: 'UZB',
        phoneCode: '998',
        label: 'Uzbekistan',
        isSelected: false
    },
    {
        value: 'VU',
        value_iso3: 'VUT',
        phoneCode: '678',
        label: 'Vanuatu',
        isSelected: false
    },
    {
        value: 'VE',
        value_iso3: 'VEN',
        phoneCode: '58',
        label: 'Venezuela, Bolivarian Republic of',
        isSelected: false
    },
    {
        value: 'VN',
        value_iso3: 'VNM',
        phoneCode: '84',
        label: 'Viet Nam',
        isSelected: false
    },
    {
        value: 'VG',
        value_iso3: 'VGB',
        phoneCode: '1',
        label: 'Virgin Islands, British',
        isSelected: false
    },
    {
        value: 'VI',
        value_iso3: 'VIR',
        phoneCode: '1',
        label: 'Virgin Islands, U.S.',
        isSelected: false
    },
    {
        value: 'WF',
        value_iso3: 'WLF',
        phoneCode: '681',
        label: 'Wallis and Futuna',
        isSelected: false
    },
    {
        value: 'EH',
        value_iso3: 'ESH',
        phoneCode: '212',
        label: 'Western Sahara',
        isSelected: false
    },
    {
        value: 'YE',
        value_iso3: 'YEM',
        phoneCode: '967',
        label: 'Yemen',
        isSelected: false
    },
    {
        value: 'ZM',
        value_iso3: 'ZMB',
        phoneCode: '260',
        label: 'Zambia',
        isSelected: false
    },
    {
        value: 'ZW',
        value_iso3: 'ZWE',
        phoneCode: '263',
        label: 'Zimbabwe',
        isSelected: false
    }
];
const companySizeOptions = [
    {
        value: 'RANGE_1_9',
        label: '1 – 9'
    },
    {
        value: 'RANGE_10_24',
        label: '10 – 24'
    },
    {
        value: 'RANGE_25_49',
        label: '25 – 49'
    },
    {
        value: 'RANGE_50_249',
        label: '50 – 249'
    },
    {
        value: 'RANGE_250_499',
        label: '250 – 499'
    },
    {
        value: 'RANGE_500_999',
        label: '500 – 999'
    },
    {
        value: 'RANGE_1000_4999',
        label: '1000 – 4999'
    },
    {
        value: 'RANGE_5000_PLUS',
        label: '5000 +'
    }
];

export const formMock = {
    actionEndpoint: 'https://sfbmarketplace.byappdirect.com/api/prm/v1/leads?currency=USD&country=US',
    productUrl: '/en-US/apps/10030/appinsights',
    otherData: {
        application: {
            id: 10030
        },
        source: 'CUSTOMER',
        target: 'VENDOR'
    },
    userAndCompanyInfo: {
        firstName: 'John',
        lastName: 'Doe',
        email: 'test@test.ca',
        companyName: 'sfbmarketplace',
        city: 'Montreal',
        street1: '1234 Fake St.',
        phone: '415-123-1234',
        street2: 'apt. 1234',
        zip: 'H0H0H0'
    },
    schema: {
        firstName: {
            name: 'firstName',
            type: 'Text',
            title: 'First Name',
            isRequired: true,
            validators: [
                'trimmedRequired'
            ],
            editorAttrs: {
                maxlength: 50
            }
        },
        lastName: {
            name: 'lastName',
            type: 'Text',
            title: 'Last Name',
            isRequired: true,
            validators: [
                'trimmedRequired'
            ],
            editorAttrs: {
                maxlength: 50
            }
        },
        email: {
            name: 'email',
            type: 'Text',
            title: 'Email',
            isRequired: true,
            editorAttrs: {
                placeholder: 'email@address.com',
                maxlength: 50
            },
            validators: [
                'trimmedRequired',
                'email'
            ]
        },
        companyName: {
            name: 'companyName',
            title: 'Company Name',
            validators: [
                'trimmedRequired'
            ],
            isRequired: true,
            inline: true,
            editorAttrs: {
                maxlength: 50
            }
        },
        companySize: {
            name: 'companySize',
            title: 'Company Size',
            type: 'StyledSelect',
            isRequired: true,
            validators: [
                'required'
            ],
            inline: true,
            includeEmptyOption: true,
            options: companySizeOptions
        },
        street1: {
            name: 'street1',
            title: 'Street',
            type: 'Text',
            isRequired: false,
            inline: true,
            editorAttrs: {
                placeholder: '1234 Main St.',
                maxlength: 50
            },
            validators: []
        },
        street2: {
            name: 'street2',
            title: 'Suite',
            type: 'Text',
            inline: true,
            editorAttrs: {
                placeholder: 'Suite 100',
                maxlength: 50
            }
        },
        city: {
            name: 'city',
            title: 'City',
            type: 'Text',
            inline: true,
            isRequired: false,
            editorAttrs: {
                maxlength: 50
            },
            validators: []
        },
        state: {
            name: 'state',
            title: 'State',
            type: 'Select',
            inline: true,
            includeEmptyOption: true,
            emptyOptionLabel: 'Select',
            toggledBy: 'country',
            visibleOn: 'USA',
            isRequired: false,
            options: stateOptions,
            validators: []
        },
        province: {
            name: 'province',
            title: 'Province',
            type: 'Select',
            inline: true,
            includeEmptyOption: true,
            emptyOptionLabel: 'Select',
            isRequired: false,
            toggledBy: 'country',
            visibleOn: 'CAN',
            options: provinceOptions,
            validators: []
        },
        zip: {
            name: 'zip',
            title: 'Zip Code',
            type: 'Text',
            inline: true,
            isRequired: false,
            editorAttrs: {
                maxlength: 50
            },
            validators: []
        },
        country: {
            name: 'country',
            title: 'Country',
            type: 'Select',
            iso3: true,
            includeEmptyOption: true,
            isRequired: true,
            validators: [
                'required'
            ],
            options: countryOptions
        },
        phone: {
            name: 'phone',
            type: 'Text',
            title: 'Phone',
            isRequired: false,
            countryCodeValidationField: 'country',
            validators: []
        },
        notes: {
            name: 'comment',
            title: 'Notes',
            type: 'TextArea',
            editorAttrs: {
                maxlength: 400,
                placeholder: 'Please specify any details'
            }
        },
        custom: []
    }
};
