// @appdirect/sfb-theme-components/src/components/product-viewer-description/stories/ProductViewerDescription.stories.js
import React from 'react';
import ProductViewerDescription from '../ProductViewerDescription';
import { multipleScreenshotsMock,
    singleScreenshotsMock,
    defaultMock as overviewImageMock,
    singleVideoMock,
    singleScreenshotWithoutOverviewMock } from '../../../../mocks/components/productViewerDescription';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

export default {
    title: 'Components/Product Viewer Description',
    component: ProductViewerDescription,
    argTypes: getArgTypes(ProductViewerDescription.schema().form),
    parameters: {
        controls: {
            include: Object.keys(getArgTypes(ProductViewerDescription.schema().form))
        }
    }
};

export const Default = {
    render: args => <ProductViewerDescription data={multipleScreenshotsMock} settings={args} />,
    args: getArgs(ProductViewerDescription.schema().form)
};

export const SingleScreenshot = {
    render: args => <ProductViewerDescription data={singleScreenshotWithoutOverviewMock} settings={args} />,
    args: getArgs(ProductViewerDescription.schema().form)
};

export const SingleScreenshotWithOverview = {
    render: args => <ProductViewerDescription data={singleScreenshotsMock} settings={args} />,
    args: getArgs(ProductViewerDescription.schema().form)
};

export const SingleVideoWithOverview = {
    render: args => <ProductViewerDescription data={singleVideoMock} settings={args} />,
    args: getArgs(ProductViewerDescription.schema().form)
};

export const WithOnlyOverviewImage = {
    render: args => <ProductViewerDescription data={overviewImageMock} settings={args} />,
    args: getArgs(ProductViewerDescription.schema().form)
};

export const WithSettings = {
    render: args => <ProductViewerDescription data={multipleScreenshotsMock} settings={args} />,
    args: getArgs(ProductViewerDescription.schema().form)
};
