// @appdirect/sfb-theme-components/src/components/product-lead-form/stories/LeadForm.stories.js
import React from 'react';
import LeadForm from '../LeadForm';
import { leadFormMockWithoutDefaultValues,
    leadFormMockWithoutDefaultValuesLeadsV1,
    leadFormMockUS,
    leadFormMockCanada,
    leadFormMockAustralia } from '../../../../mocks/components/leadForm';
import { leadFormWithCustomAttributes } from '../../../../mocks/form/leadForm';

export default {
    title: 'Components/Product Lead Form',
    component: LeadForm
};

export const DefaultUs = {
    render: () => <LeadForm data={leadFormMockUS} />,
    name: 'Default - US'
};

export const Canada = {
    render: () => <LeadForm data={leadFormMockCanada} />,
    name: 'Canada'
};

export const Australia = {
    render: () => <LeadForm data={leadFormMockAustralia} />,
    name: 'Australia'
};

export const NoDefaultValuesMinimumRequiredForLeadsV2 = {
    render: () => <LeadForm data={leadFormMockWithoutDefaultValues} />,
    name: 'No default values - minimum required for Leads V2'
};

export const NoDefaultValuesMinimumRequiredForLeadsV1 = {
    render: () => <LeadForm data={leadFormMockWithoutDefaultValuesLeadsV1} />,
    name: 'No default values - minimum required for Leads V1'
};

export const CustomAttributes = {
    render: () => <LeadForm data={leadFormWithCustomAttributes} />,
    name: 'Custom attributes'
};
