// @appdirect/sfb-theme-components/mocks/application/resources/index.js
import { i18nMock } from './i18n';

export const mockPublicDocument = {
    id: 0,
    description: 'Deputy, your complete staff management toolkit.',
    iFrameFriendlyURL: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/documentation/564218_en.pdf',
    link: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/18480/documentation/564218_en.pdf',
    name: 'Deputy flyer',
    resellerAccess: false,
    resourceType: 'PDF_UPLOAD',
    scribdId: null,
    scribdKey: null
};

export const mockPublicVideo = {
    id: 1,
    description: 'An overview of Deputy.',
    iFrameFriendlyURL: '//www.youtube.com/embed/D4nTROArtcg',
    link: 'https://www.youtube.com/watch?v=D4nTROArtcg&feature=youtu.be',
    name: 'Deputy video',
    resellerAccess: false,
    resourceType: 'VIDEO',
    scribdId: null,
    scribdKey: null
};

export const mockResellerDocument = {
    ...mockPublicDocument,
    id: 2,
    resellerAccess: true,
    name: 'Deputy flyer - reseller only'
};

export const mockResellerVideo = {
    ...mockPublicVideo,
    id: 3,
    resellerAccess: true,
    name: 'Deputy video - reseller only'
};

export const resourcesMock = {
    public: {
        documents: {
            items: [mockPublicDocument]
        },
        videos: {
            items: [mockPublicVideo]
        }
    },
    reseller: {
        documents: {
            items: [mockResellerDocument]
        },
        videos: {
            items: [mockResellerVideo]
        }
    },
    i18n: i18nMock
};

export const emptyResources = {
    documents: {
        items: []
    },
    videos: {
        items: []
    }
};
