// @appdirect/sfb-theme-components/mocks/listingFilters/filters.js
export const filtersMock = [
    {
        shallow: true,
        id: 'attributes',
        text: 'Attributes',
        items: [
            {
                queryKey: 'filter',
                id: 'FREE',
                text: 'Free',
                url: '/en-US/listing?filter=FREE',
                selected: false
            },
            {
                queryKey: 'filter',
                id: 'DISCOUNT',
                text: 'Discount Available',
                url: '/en-US/listing?filter=DISCOUNT',
                selected: false
            },
            {
                queryKey: 'filter',
                id: 'FREE_TRIAL',
                text: 'Free Trial Available',
                url: '/en-US/listing?filter=FREE_TRIAL',
                selected: false
            },
            {
                queryKey: 'type',
                id: 'BUNDLE',
                text: 'Bundle',
                url: '/en-US/listing?type=BUNDLE',
                selected: false
            }
        ]
    },
    {
        shallow: false,
        id: 'categories',
        text: 'Categories',
        items: [
            {
                queryKey: 'cat',
                id: 67285,
                text: 'Cool Apps',
                url: '/en-US/listing?cat=67285',
                selected: false,
                items: [
                    {
                        queryKey: 'scat',
                        id: 45281,
                        text: 'Chilly Apps',
                        url: '/en-US/listing?scat=45281',
                        selected: false
                    },
                    {
                        queryKey: 'scat',
                        id: 45285,
                        text: 'Freezing Apps',
                        url: '/en-US/listing?scat=45285',
                        selected: false
                    }
                ]
            },
            {
                queryKey: 'cat',
                id: 67289,
                text: 'Wild Apps',
                url: '/en-US/listing?cat=67289',
                selected: false,
                items: []
            }
        ]
    }
];

export const responseFiltersMock = [
    {
        id: 'categories',
        items: [
            {
                selected: true,
                text: 'analytics',
                id: 99849,
                url: '/en-US/listing?cat=99849&cat=99855&attr=16815&attr=19400&order=POPULARITY&scat=57417'
            }
        ]
    }
];
