// @appdirect/sfb-theme-components/src/atoms/badge/stories/Badge.stories.js
import React from 'react';
import Badge from '../Badge';
import '../styles/Badge.story.scss';

export default {
    title: 'Atoms/Badge',
    component: Badge,
    render: args => <Badge {...args} />,
    argTypes: {
        label: {
            control: 'text',
            description: 'Label of the badge'
        },
        type: {
            control: 'select',
            options: ['success', 'emphasis', 'pending', 'error'],
            description: 'Type of the badge'
        },
        className: {
            control: 'select',
            options: ['test-class-one', 'test-class-two'],
            description: 'Custom CSS class name of the badge'
        }
    }
};

const excludeClassName = {
    controls: {
        exclude: ['className']
    }
};

const excludeType = {
    controls: {
        exclude: ['type']
    }
};

export const Default = {
    args: {
        label: 'This is a default Badge'
    },
    parameters: { ...excludeClassName }
};

export const Success = {
    args: {
        label: 'Success',
        type: 'success'
    },
    parameters: { ...excludeClassName }
};

export const Emphasis = {
    args: {
        label: 'Emphasis',
        type: 'emphasis'
    },
    parameters: { ...excludeClassName }
};

export const Pending = {
    args: {
        label: 'Pending',
        type: 'pending'
    },
    parameters: { ...excludeClassName }
};

export const Error = {
    args: {
        label: 'Error',
        type: 'error'
    },
    parameters: { ...excludeClassName }
};

export const WithClass = {
    args: {
        label: 'This is a badge with a CSS class applied',
        className: 'test-class-one'
    },
    parameters: { ...excludeType }
};
