// @appdirect/sfb-theme-components/src/atoms/Error/stories/Error.stories.js
/* eslint-disable react/no-danger */
import React from 'react';
import Error from '../Error';

export default {
    title: 'Atoms/Error',
    component: Error,
    render: args => (
        <Error {...args} />
    ),
    argTypes: {
        errorText: {
            control: 'select',
            options: ['TEXT', 'JSX'],
            mapping: {
                TEXT: 'This is a text error message',
                JSX: (
                    <>
                        The page could not load because of a server error. Wait, then refresh the page.
                        <br />
                        If this error reappears, contact support.
                    </>
                )
            },
            description: 'Error message (can be plain text or JSX)'
        },
        errorIcon: {
            control: 'select',
            options: ['alert', 'caution']
        },
        links: {
            control: 'object'
        }
    },
    args: {
        title: 'Internal Server Error (Error 500)',
        errorText: 'JSX',
        errorIcon: 'alert'
    }
};

export const WithLinks = {
    render: ({ links, ...args }) => {
        const mappedLinks = links.map(link => (
            <a key={link.key} target="_blank" rel="noopener noreferrer" href={link.href}>{link.text}</a>
        ));
        return (
            <Error
                {...args}
                links={mappedLinks}
            />
        );
    },
    args: {
        links: [
            { key: 'invoices', href: 'https://example.org', text: 'View all invoices' },
            { key: 'homepage', href: 'https://example.org', text: 'Go to homepage' }
        ]
    }
};

export const WithoutLinks = {
    parameters: {
        controls: {
            exclude: ['links']
        }
    }
};
