// @appdirect/sfb-theme-components/src/constants/prop-types/listingProducts/items.js
import PropTypes from 'prop-types';

import { ratingsSummaryShape } from '../common';

export const itemsPropTypes = {
    id: PropTypes.number,
    name: PropTypes.string,
    blurb: PropTypes.string,
    overview: PropTypes.string,
    version: PropTypes.string,
    vendorName: PropTypes.string,
    startingPrice: PropTypes.shape({
        formattedFullPrice: PropTypes.string,
        formattedPrice: PropTypes.string,
        price: PropTypes.string,
        priceCaption: PropTypes.string,
        frequency: PropTypes.shape({
            label: PropTypes.string,
            shortLabel: PropTypes.string
        })
    }),
    rating: PropTypes.number,
    numRatings: PropTypes.number,
    ratingCaption: PropTypes.string,
    url: PropTypes.string,
    hidePricings: PropTypes.bool,
    iconUrl: PropTypes.string,
    iconSrcset: PropTypes.shape({
        default: PropTypes.string,
        '1x': PropTypes.string,
        '2x': PropTypes.string
    }),
    iconAlt: PropTypes.string,
    badges: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string,
            label: PropTypes.string,
            type: PropTypes.string
        })
    ),
    productRibbon: PropTypes.shape({
        text: PropTypes.string,
        textColor: PropTypes.string,
        ribbonColor: PropTypes.string
    }),
    merchandisingBadges: PropTypes.arrayOf(
        PropTypes.shape({
            uuid: PropTypes.string,
            name: PropTypes.string,
            image: PropTypes.string
        })
    ),
    ratingsSummary: PropTypes.shape(ratingsSummaryShape),
    useCheckoutForSpecialPricingMessage: PropTypes.bool
};
