// @appdirect/sfb-theme-components/mocks/cart/cartItemAddons.js
export const cartItemAddons = [
    {
        pricingPlanId: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d4',
        subscriptionId: null,
        serviceConfiguration: null,
        billingConfiguration: null,
        discountCode: null,
        discountId: null,
        units: [],
        id: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d4',
        customAttributes: null,
        validated: false,
        customPrices: [],
        selectedPricingPlanId: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d4',
        selectedEditionId: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d4',
        itemDetails: {
            id: '293048',
            uuid: '99ea13a6-679d-419f-9d87-d6c25b9026de',
            type: 'PRODUCT',
            customIntegration: null,
            stacked: false,
            branding: {
                name: 'Addon # 1',
                vendor: 'Microsoft',
                desc: 'Lorem Ipsum dolor sit amet.',
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293048/thumbs_64/image5315487358905934939.png',
                href: 'https://sfbmarketplace.byappdirect.com/api/marketplace/v1/products/293048',
                storefrontUrl: 'https://sfbmarketplace.byappdirect.com/apps/293048'
            },
            editions: [
                {
                    id: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d4',
                    name: 'Growing',
                    editionCode: 'GROWING-QBO',
                    bundleOnly: false,
                    autoUpgradeToPaid: null,
                    freeTrialRequiresBillingInfo: null,
                    revenueModel: 'RECURRING',
                    freeTrialTerms: {
                        freeTrialDuration: 0,
                        freeTrialDurationPeriod: 'DAY',
                        freeTrialGracePeriod: 0
                    },
                    pricingPlans: [
                        {
                            id: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d4',
                            currency: 'USD',
                            pricingPeriod: 'MONTHLY',
                            pricingSources: [
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'RECURRING_FLAT',
                                    unit: null,
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'FLAT',
                                            salePrice: '79.0000000000',
                                            priceRanges: []
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: '1.0000000000',
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                },
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'INCLUDED',
                                    unit: 'USER',
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'UNLIMITED',
                                            salePrice: null,
                                            priceRanges: []
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: null,
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                }
                            ],
                            contractConfiguration: null,
                            additionalFees: {
                                terminationFee: null
                            }
                        }
                    ]
                }
            ],
            customAttributes: []
        }
    },
    {
        pricingPlanId: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d5',
        subscriptionId: null,
        serviceConfiguration: null,
        billingConfiguration: null,
        discountCode: null,
        discountId: null,
        units: [],
        id: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d5',
        customAttributes: null,
        validated: false,
        customPrices: [],
        selectedPricingPlanId: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d5',
        selectedEditionId: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d5',
        itemDetails: {
            id: '293048',
            uuid: '99ea13a6-679d-419f-9d87-d6c25b9026de',
            type: 'PRODUCT',
            customIntegration: null,
            stacked: false,
            branding: {
                name: 'Addon # 2',
                vendor: 'Microsoft',
                desc: 'Lorem Ipsum dolor sit amet.',
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/293048/thumbs_64/image5315487358905934939.png',
                href: 'https://sfbmarketplace.byappdirect.com/api/marketplace/v1/products/293048',
                storefrontUrl: 'https://sfbmarketplace.byappdirect.com/apps/293048'
            },
            editions: [
                {
                    id: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d5',
                    name: 'Growing',
                    editionCode: 'GROWING-QBO',
                    bundleOnly: false,
                    autoUpgradeToPaid: null,
                    freeTrialRequiresBillingInfo: null,
                    revenueModel: 'RECURRING',
                    freeTrialTerms: {
                        freeTrialDuration: 0,
                        freeTrialDurationPeriod: 'DAY',
                        freeTrialGracePeriod: 0
                    },
                    pricingPlans: [
                        {
                            id: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d5',
                            currency: 'USD',
                            pricingPeriod: 'MONTHLY',
                            pricingSources: [
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'RECURRING_FLAT',
                                    unit: null,
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'FLAT',
                                            salePrice: '79.0000000000',
                                            priceRanges: []
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: '1.0000000000',
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                },
                                {
                                    costTypeCategory: 'RECURRING',
                                    costType: 'INCLUDED',
                                    unit: 'USER',
                                    sources: [
                                        {
                                            priceType: 'BASE_PRICE',
                                            pricingStrategy: 'UNLIMITED',
                                            salePrice: null,
                                            priceRanges: []
                                        }
                                    ],
                                    unitAllowsDecimals: null,
                                    increment: null,
                                    allowsMeteredUsage: null,
                                    displaysPricePerIncrement: false
                                }
                            ],
                            contractConfiguration: null,
                            additionalFees: {
                                terminationFee: null
                            }
                        }
                    ]
                }
            ],
            customAttributes: []
        }
    }
];
