// @appdirect/sfb-theme-components/mocks/cart/cartItemNonStackable.js
export const cartItemNonStackableWithAddonsMock = [
    {
        pricingPlanId: '76068349-1098-4904-a91e-a038f8b3a9a0',
        id: 'cca9c0a4-46f7-48b2-9c5d-6a44bd301b80',
        selectedPricingPlanId: '76068349-1098-4904-a91e-a038f8b3a9a0',
        selectedEditionId: 'caa842d4-d333-4b31-a793-c4d9f507a83a',
        itemDetails: {
            uuid: '50f70715-0dce-4cb2-8287-3f7329dec2aa'
        },
        childItems: [
            {
                id: '39b8eac7-e776-45cd-ac16-ed1bac20e8eb'
            }
        ]
    }
];

export const cartWithNonStackableWithAddonsMock = {
    id: '60ae9302f53d703ba2d57580',
    status: 'ACTIVE',
    buyerType: 'CUSTOMER',
    currency: 'USD',
    items: cartItemNonStackableWithAddonsMock,
    itemsCount: cartItemNonStackableWithAddonsMock.length,
    associations: [
        {
            parentItemId: 'cca9c0a4-46f7-48b2-9c5d-6a44bd301b80',
            childItemIds: [
                '39b8eac7-e776-45cd-ac16-ed1bac20e8eb'
            ],
            type: 'ADDON'
        }
    ]
};
