// @appdirect/sfb-theme-components/mocks/home/products.js
export const productMock = {
    id: 1001,
    name: 'Google Pixelbook',
    blurb: 'The high performance Chromebook',
    url: '/en-US/apps/241879/google-pixelbook',
    hidePricings: false,
    rating: 4.25,
    ratingsSummary: {
        title: 'Rating',
        caption: '4.25/5 Based on 59 reviews'
    },
    numRatings: 59,
    iconUrl:
        'https://d33na3ni6eqf5j.cloudfront.net/app_resources/241879/thumbs_64/image1966147977002842893.png',
    iconSrcset: {
        default: 'https://d33na3ni6eqf5j.cloudfront.net/app_resources/241879/thumbs_64/image1966147977002842893.png',
        '2x': 'https://d33na3ni6eqf5j.cloudfront.net/app_resources/241879/thumbs_64/image1966147977002842893-2x.png',
        '1x': 'https://d33na3ni6eqf5j.cloudfront.net/app_resources/241879/thumbs_64/image1966147977002842893.png'
    },
    startingPrice: {
        formattedFullPrice: '$999.00',
        priceCaption: 'Starting At'
    }
};

export const productMockWithoutIcon = {
    ...productMock,
    iconUrl: null,
    iconSrcset: {}
};

export const productWithBadgesMock = {
    ...productMock,
    badges: [
        {
            id: 'FREE_TRIAL',
            label: 'Free Trial',
            type: 'success'
        }
    ],
    productRibbon: {
        text: 'CORE APP',
        textColor: '#000000',
        ribbonColor: '#FFFFFF'
    },
    merchandisingBadges: [
        {
            uuid: '76eb89e4-def9-4322-9a10-5a8d9f78df89',
            name: 'iOS Compatible',
            image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/76eb89e4-def9-4322-9a10-5a8d9f78df89/img4805831920228314845.png?12294cbcc5425724b161137570276e41'
        },
        {
            uuid: '4e5a0070-28de-11e8-bd43-0aea64b20f8f',
            name: 'Monkey',
            image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/4e5a0070-28de-11e8-bd43-0aea64b20f8f/img5143663096997530867.png?23ad4994e6947eed95fc58810c363fb1'
        },
        {
            uuid: 'dsf87g87y-64e854t65-65d4rg-6s84rdg68',
            name: 'Single Sign-On',
            image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/ba3dc7bc-b3ca-4d45-90dd-6a0a083ab872/img1508692810840088346.png?a98c1e4ae512ec64db825b0d31435b08'
        },
        {
            uuid: 'er98gt89-654dfb684354gf-65sd4f-6sd85f4',
            name: 'Access Management',
            image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/0af0e01c-5c06-4bbb-9da1-214428362bdf/img7972380461864714752.png?0a566604eb8930fa19c2275dddec67fd'
        },
        {
            uuid: '08d06719-1027-4952-914c-fe0d3de89d06',
            name: 'Unified Invoice',
            image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/08d06719-1027-4952-914c-fe0d3de89d06/img5117470399285071356.png?8cf2543bdae1af7a6f287688d07c1d45'
        },
        {
            uuid: '76eb89e4-def9-4322-9a10-5a8d9f78df85',
            name: 'iOS Compatible v2',
            image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/76eb89e4-def9-4322-9a10-5a8d9f78df89/img4805831920228314845.png?12294cbcc5425724b161137570276e41'
        }
    ]
};
