// @appdirect/sfb-theme-components/mocks/home/slider.js
import { sliderProductsMock } from '../product/slider';
import { i18nMock } from '../productLists/i18n';

export const sliderMockData = {
    data: {
        i18n: i18nMock
    }
};

export const sliderMockSettings = {
    settings: {
        sliderTitle: 'Slider title',
        productsList: {
            url: 'https://seeItAll.test',
            items: sliderProductsMock,
            contentType: 'dynamic'
        }
    }
};

export const sliderMockSettingsNoQuickview = {
    settings: {
        ...sliderMockSettings.settings,
        quickviewOptions: []
    }
};

export const sliderMockVariousImageSizes = {
    settings: {
        sliderTitle: 'Slider title',
        productsList: {
            url: 'https://seeItAll.test',
            items: [
                {
                    ...sliderProductsMock[0],
                    name: 'Portrait image',
                    iconSrcset: {
                        // portrait image
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/4ebb95b2-83b2-41fb-b3c8-f1a2a81ce0ca/0f79fea4-1b46-4910-944b-24be067c82f4.jpg'
                    }
                },
                {
                    ...sliderProductsMock[1],
                    name: 'Landscape image',
                    iconSrcset: {
                        // landscape image
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/f1614e0c-17a9-45ba-9c7f-23b1d75d5ea6/444dd67b-cec7-414e-8ac1-3ac8847f6501.jpg'
                    }
                },
                {
                    ...sliderProductsMock[2],
                    name: 'Square image',
                    iconSrcset: {
                        // square image
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/67dd74f5-3fb1-4f1a-aa1a-2e0d82b6c569/a91affc3-ed9e-41f1-a68f-7dc231febcb5.jpg'
                    }
                },
                {
                    ...sliderProductsMock[3],
                    name: 'Small icon',
                    iconSrcset: {
                        // small icon
                        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/dd2c721d-e6fe-4646-86cf-b9dbd3776875/f6015b89-22b7-4938-85d3-4dd1023073ff.png'
                    }
                },
                ...sliderProductsMock.slice(4)
            ],
            contentType: 'dynamic'
        }
    }
};
