// @appdirect/sfb-theme-components/mocks/product/related-content/pricing.js
export const pricingMock = {
    editions: [
        {
            name: 'Free Trial',
            value: '',
            href: '/apps/10030/editions'
        },
        {
            name: 'Monitor Yearly Plan',
            value: '$1,068.00',
            frequency: 'YEARLY',
            formattedFrequency: 'per year',
            href: '/apps/10030/editions'
        },
        {
            name: 'Report Yearly Plan',
            value: '$2,148.00',
            frequency: 'YEARLY',
            formattedFrequency: 'per year',
            href: '/apps/10030/editions'
        },
        {
            name: 'Engage Yearly Plan',
            value: '$4,308.00',
            frequency: 'YEARLY',
            formattedFrequency: 'per year',
            href: '/apps/10030/editions'
        },
        {
            name: 'Monitor Monthly Plan',
            value: '$99.00',
            frequency: 'MONTHLY',
            formattedFrequency: 'per month',
            href: '/apps/10030/editions'
        },
        {
            name: 'Report Monthly Plan',
            value: '$199.00',
            frequency: 'MONTHLY',
            formattedFrequency: 'per month',
            href: '/apps/10030/editions'
        },
        {
            name: 'Engage Monthly Plan',
            value: '$399.00',
            frequency: 'MONTHLY',
            formattedFrequency: 'per month',
            href: '/apps/10030/editions'
        }
    ]
};
