// @appdirect/sfb-theme-components/src/components/product-variants/stories/ProductVariants.stories.js
import React from 'react';
import ProductVariants, { ProductVariantsComponent } from '../ProductVariants';
import { productVariantsMock,
    productVariantsMockWithoutDefaults } from '../../../../mocks/product/product-variants/productVariants';

export default {
    title: 'Components (Preview)/Product Variants/ProductVariants',
    component: ProductVariants
};

export const Default = {
    render: () => <ProductVariantsComponent data={productVariantsMock} />,
    name: 'Default'
};

export const WithoutDefaultValues = {
    render: () => <ProductVariantsComponent data={productVariantsMockWithoutDefaults} />,
    name: 'Without default values'
};

export const LoadingData = {
    render: () => (
        <ProductVariantsComponent
            data={{
                ...productVariantsMock,
                isLoadingVariantsData: true
            }}
        />
    ),

    name: 'Loading data'
};

export const WithError = {
    render: () => (
        <ProductVariantsComponent
            data={{
                ...productVariantsMock,
                hasErrorsLoadingVariantsData: {}
            }}
        />
    ),

    name: 'With error'
};
