// @appdirect/sfb-theme-components/mocks/product/summary/index.js
import { merchandisingBadgesMock } from './merchandisingBadges';
import { i18nSummaryMock } from './i18n';

export const summaryMock = {
    title: 'AppInsights',
    details: 'Real-time business dashboard solution',
    overview: 'AppInsights help you make faster and more informed\nbusiness decisions by letting you view your KPIs on a real-time dashboard, all in\none place.',
    logoSrc: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_112/img3745051225877065498.png?e0df123fedcc5e96c83a248797c76c6a',
    logoSrcset: {
        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_112/img3745051225877065498.png?e0df123fedcc5e96c83a248797c76c6a',
        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_112/img3745051225877065498-2x.png?6a5de3e906301fb37d2cde7ee0a2b930',
        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/10030/thumbs_112/img3745051225877065498.png?e0df123fedcc5e96c83a248797c76c6a'
    },
    logoAlt: 'The logo of AppInsights',
    productRibbon: {
        text: 'Core App',
        textColor: '#FFFFFF',
        ribbonColor: '#D42020'
    },
    merchandisingBadges: merchandisingBadgesMock,
    i18n: i18nSummaryMock
};
