// @appdirect/sfb-theme-components/src/atoms/tile/stories/Tile.stories.js
import React from 'react';
import { productMock, productWithBadgesMock, productMockWithoutIcon } from '../../../../mocks/home/products';
import Tile from '../Tile';
import Grid from '../../grid/Grid';

export default {
    title: 'Atoms/Tile',
    component: Tile,
    argTypes: {
        layout: {
            control: 'radio',
            options: ['Classic', 'Square', 'Horizontal', 'Tall', 'Graphic'],
            mapping: {
                Classic: 'classic',
                Square: 'square',
                Horizontal: 'horizontal',
                Tall: 'tall',
                Graphic: 'graphic'
            },
            description: 'The layout of the tile'
        }
    },
    parameters: {
        controls: {
            include: ['layout']
        }
    },
    args: {
        layout: 'Classic'
    }
};

export const Default = {
    render: args => (
        <Grid maxCol={4}>
            <Tile data={productMock} settings={args} />
        </Grid>
    )
};

export const WithPlaceholderImage = {
    render: args => (
        <Grid maxCol={4}>
            <Tile data={productMockWithoutIcon} settings={args} />
        </Grid>
    )
};

export const WithBadges = {
    render: args => (
        <Grid maxCol={6}>
            <Tile data={productWithBadgesMock} settings={args} />
        </Grid>
    )
};

export const WithSettings = {
    render: args => (
        <Grid maxCol={4}>
            <Tile data={productWithBadgesMock} settings={args} />
        </Grid>
    )
};
