// @appdirect/sfb-theme-components/src/constants/components/prop-types/addon.js
import PropTypes from 'prop-types';

export const addonShape = PropTypes.shape({
    items: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        edition: PropTypes.arrayOf(PropTypes.shape({
            plans: PropTypes.arrayOf(PropTypes.shape({
                costs: PropTypes.shape({
                    unitPrice: PropTypes.arrayOf(PropTypes.shape({
                        usageRange: PropTypes.arrayOf(PropTypes.shape({
                            formattedFullPriceElements: PropTypes.arrayOf(PropTypes.string)
                        }))
                    }))
                }),
                contractLength: PropTypes.number,
                formattedContract: PropTypes.oneOfType([
                    PropTypes.string,
                    PropTypes.bool
                ])
            })),
            bullets: PropTypes.arrayOf(PropTypes.shape({
                content: PropTypes.string
            }))
        }))
    })),
    i18n: PropTypes.shape({
        added: PropTypes.string,
        addons: PropTypes.string,
        addToOrder: PropTypes.string,
        audience: PropTypes.string,
        availableAddons: PropTypes.string,
        inclusions: PropTypes.string,
        seePricingPlans: PropTypes.string,
        selectAddonsTitle: PropTypes.string,
        startingFrom: PropTypes.string
    })
});
