// @appdirect/sfb-theme-components/src/atoms/product-quick-view/stories/ProductQuickView.stories.js
import React from 'react';
import { LAYOUT_NO_IMG,
    LAYOUT_DEFAULT,
    LAYOUT_DETAILED,
    LAYOUT_2COLUMNS } from '../../../constants/components/schemas/product-quickview';
import ProductQuickView from '../ProductQuickView';

// use the same mock as product slider, as QuickView is child of slider
import { sliderProductsMock } from '../../../../mocks/product/slider';

const productData = sliderProductsMock[0];

const layoutOptions = {
    Default: LAYOUT_DEFAULT,
    Detailed: LAYOUT_DETAILED,
    '2 Columns': LAYOUT_2COLUMNS,
    'No Image': LAYOUT_NO_IMG
};

export default {
    title: 'Atoms/ProductQuickView',
    component: ProductQuickView,
    argTypes: {
        viewMode: {
            description: 'Controls the layout of the ProductQuickView',
            control: 'select',
            options: Object.keys(layoutOptions),
            mapping: layoutOptions
        }
    },
    render: args => (
        <ProductQuickView
            data={productData}
            settings={{
                quickviewViewMode: args.viewMode
            }}
        />
    ),
    parameters: {
        controls: {
            include: ['viewMode']
        }
    }
};

export const Default = {
    args: {
        viewMode: Object.keys(layoutOptions)[0]
    }
};

export const Detailed = {
    args: {
        viewMode: Object.keys(layoutOptions)[1]
    }
};

export const TwoColumns = {
    args: {
        viewMode: Object.keys(layoutOptions)[2]
    }
};

export const NoImage = {
    args: {
        viewMode: Object.keys(layoutOptions)[3]
    }
};

export const WithSettings = {
    args: {
        viewMode: Object.keys(layoutOptions)[0]
    }
};
