// @appdirect/sfb-theme-components/src/constants/buttons.js
// Types
export const TYPE_BUTTON = 'button';
export const TYPE_SUBMIT = 'submit';
export const TYPE_RESET = 'reset';

// sizes
export const SIZE_DEFAULT = 'default';

// Themes
export const THEME_PRIMARY = 'primary';
export const THEME_SECONDARY = 'secondary';
export const THEME_EMPHASIS = 'emphasis';
export const THEME_EMPHASIS_OUTLINE = 'emphasis-outline';
export const THEME_EMPHASIS_ALT = 'emphasisAlt';
export const THEME_EMPHASIS_ALT_OUTLINE = 'emphasisAlt-outline';
export const THEME_WHITE = 'white';
export const THEME_CARD = 'card';

// Link target
export const TARGET_BLANK = '_blank';

// attributes
export const ATTR_REFER_URL = 'data-refer-url';
export const REL_NOOPENER = 'noopener';

export const BUTTON_THEMES = [
    THEME_PRIMARY,
    THEME_SECONDARY,
    THEME_EMPHASIS,
    THEME_EMPHASIS_OUTLINE,
    THEME_EMPHASIS_ALT,
    THEME_EMPHASIS_ALT_OUTLINE,
    THEME_WHITE,
    THEME_CARD
];

export const LINK_THEMES = [
    THEME_PRIMARY,
    THEME_SECONDARY,
    THEME_EMPHASIS,
    THEME_EMPHASIS_ALT,
    THEME_WHITE
];
