// @appdirect/sfb-theme-components/src/namespaces/profile/index.js
export * from '../global';
export { default as FeatureDetails } from '../../components/product-feature-details/FeatureDetails';
export { default as FeaturedCustomers } from '../../components/product-featured-customers/FeaturedCustomers';
export { default as GraphQLAdapter } from '../../components/graphql-adapter/GraphQLAdapter';
export { default as LinkSection } from '../../components/faq-link-section/LinkSection';
export { default as ProductAdditionalInfo } from '../../components/product-additional-info/AdditionalInfo';
export { default as ProductBanner } from '../../components/product-banner/ProductBanner';
export { default as ProductConfigurator } from '../../components/product-configurator/ProductConfigurator';
export { default as ProductConfiguratorV2 } from '../../components/product-configurator-v2/ProductConfiguratorV2';
export { default as ProductDescription } from '../../components/product-viewer-description/ProductViewerDescription';
export { default as ProductLeadForm } from '../../components/product-lead-form/LeadForm';
export { default as ProductImageViewer } from '../../components/product-image-viewer/ProductImageViewer';
export { default as ProductQuestions } from '../../components/product-questions/ProductQuestions';
export { default as ProductQuestionForm } from '../../components/product-question-form/ProductQuestionForm';
export { default as ProductPlan } from '../../components/product-plan/ProductPlan';
export { default as DomainApp } from '../../components/domain-app/DomainApp';
export { default as ProductAssociatedDomain } from '../../components/product-associated-domain/ProductAssociatedDomain';
export { default as ProductPricing } from '../../components/product-pricing/ProductPricing';
export { default as ProductResources } from '../../components/product-resources/Resources';
export { default as ProductReviewForm } from '../../components/product-review-form/ProductReviewForm';
export { default as ProductReviews } from '../../components/product-reviews/ProductReviewsAdapter';
export { default as ProductReviewDetail } from '../../components/product-review-detail/ProductReviewDetail';
export { default as ProductSimilarApps } from '../../components/product-similar-apps/ProductSimilarApps';
export { default as ProductTabMenu } from '../../components/product-tabs-menu/TabsMenu';
export { default as ProductVariants } from '../../components/product-variants/ProductVariants';
export { default as BundleTabsMenu } from '../../components/product-tabs-menu/BundleTabsMenu';
export { default as ProductTopFeatures } from '../../components/product-top-features/TopFeatures';
export { default as VolumeDiscountCard } from '../../atoms/volume-discount-card/VolumeDiscountCard';
export { default as PageOrderableContainer } from '../../components/page-orderable-container/ProfileOrderableContainer';
