// @appdirect/sfb-theme-components/mocks/product/description/addons.js
export const addonsWithDiscountsMock = [{
    id: 334561,
    visible: true,
    parentEditionId: 658567,
    name: 'Add-on 1',
    edition: [{
        name: 'Recurring Add-On',
        id: 660049,
        primary: true,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$5.00',
                        formattedFullPrice: '$5.00 / User / Month',
                        formattedFullPriceElements: [
                            '$5.00',
                            '/ User',
                            '/ Month'
                        ],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 10,
                            value: 5,
                            originalValue: 10,
                            discountValue: 5,
                            editionStartingPrice: 5,
                            formatted: {
                                originalPrice: '$10.00',
                                originalFullPrice: '$10.00 / User / Month',
                                originalFullPriceElements: [
                                    '$10.00',
                                    '/ User',
                                    '/ Month'
                                ],
                                discount: '-$5.00',
                                discountOff: '$5.00 off per User',
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: {
                description: 'Black Friday',
                percentage: null,
                startDate: null,
                expirationDate: null,
                numOfBillingCycles: null,
                availableRedemptions: null,
                amount: {
                    USD: 5
                },
                pricingUnit: 'USER'
            },
            uuid: 'c8d5b10e-dd85-47a7-86d3-cf4dc5c64183'
        }],
        bullets: [{
            content: 'Custom Tables and Advanced Statistics Add-on provides easy to use drag and drop interactive tables exportable to Microsoft/PDF',
            tooltip: null
        }, {
            content: 'You can access a variety of additional techniques such as; Non-linear, logistic, 2-stage least squares regression, Generalized linear modeling and survival analysis',
            tooltip: null
        }, {
            content: 'Additionally, Bayesian Statistics now available',
            tooltip: null
        }],
        callToAction: {
            isPrimary: false,
            isSecondary: true
        },
        audience: [
            {
                content: 'Test target audience Son english'
            }
        ],
        items: [
            {
                content: '0 Users'
            }
        ],
        invisible: false,
        leadGen: false,
        revenueType: 'RECURRING'
    }]
}, {
    id: 334563,
    visible: true,
    parentEditionId: 658567,
    name: 'Add-on 3',
    edition: [{
        name: 'Recurring Add-On',
        id: 660055,
        primary: true,
        plans: [{
            costs: {
                unitPrice: [
                    {
                        id: 'USER',
                        usageRange: [
                            {
                                formattedPrice: '$30.00',
                                formattedFullPrice: '$30.00 / User / Month',
                                formattedFullPriceElements: [
                                    '$30.00',
                                    '/ User',
                                    '/ Month'
                                ],
                                unit: {
                                    min: 1,
                                    max: 100,
                                    label: 'User',
                                    dependency: '',
                                    id: 'USER',
                                    limitLabel: 'For 1-100 Users'
                                },
                                frequency: {
                                    shortLabel: 'Mo',
                                    label: 'Month',
                                    id: 'MONTHLY'
                                },
                                price: {
                                    symbol: '$',
                                    currency: 'USD',
                                    amount: 50,
                                    value: 30,
                                    originalValue: 50,
                                    discountValue: 20,
                                    editionStartingPrice: 130,
                                    formatted: {
                                        originalPrice: '$50.00',
                                        originalFullPrice: '$50.00 / User / Month',
                                        originalFullPriceElements: [
                                            '$50.00',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        discount: '-$20.00',
                                        discountOff: '$20.00 off per User',
                                        meteredUsage: 'For 1-100 Users'
                                    }
                                }
                            }
                        ]
                    }
                ],
                flatPrice: {
                    formattedPrice: '$100.00',
                    formattedFullPrice: '$100.00 / Month',
                    formattedFullPriceElements: [
                        '$100.00',
                        null,
                        '/ Month'
                    ],
                    unit: {
                        min: 0,
                        max: null,
                        dependency: '',
                        id: 'NOT_APPLICABLE'
                    },
                    frequency: {
                        shortLabel: 'Mo',
                        label: 'Month',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: 'USD',
                        amount: 100,
                        value: 100,
                        originalValue: 100,
                        discountValue: 0,
                        editionStartingPrice: 150,
                        formatted: {
                            originalPrice: '$100.00',
                            originalFullPrice: '$100.00 / Month',
                            originalFullPriceElements: [
                                '$100.00',
                                null,
                                '/ Month'
                            ]
                        }
                    }
                }
            },
            setupFees: [
                {
                    formattedPrice: '$30.00',
                    formattedFullPrice: '$30.00 Setup fee',
                    formattedFullPriceElements: [
                        '$30.00',
                        'Setup fee',
                        ''
                    ],
                    unit: {
                        min: 0,
                        max: null,
                        label: 'Setup fee',
                        dependency: '',
                        id: 'ONE_TIME_SETUP'
                    },
                    frequency: {
                        shortLabel: '',
                        label: '',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: 'USD',
                        amount: 30,
                        value: 30,
                        originalValue: 30,
                        discountValue: 0,
                        editionStartingPrice: 150,
                        formatted: {
                            originalPrice: '$30.00',
                            originalFullPrice: '$30.00 Setup fee',
                            originalFullPriceElements: [
                                '$30.00',
                                'Setup fee',
                                ''
                            ]
                        }
                    }
                }
            ],
            contractFees: [
                {
                    formattedPrice: '$40.00',
                    formattedFullPrice: '$40.00 Contract Fee',
                    formattedFullPriceElements: [
                        '$40.00',
                        'Contract Fee',
                        ''
                    ],
                    unit: {
                        min: 0,
                        max: null,
                        label: 'Contract Fee',
                        dependency: '',
                        id: 'CONTRACT_FEE'
                    },
                    frequency: {
                        shortLabel: '',
                        label: '',
                        id: 'MONTHLY'
                    },
                    price: {
                        symbol: '$',
                        currency: 'USD',
                        amount: 40,
                        value: 40,
                        originalValue: 40,
                        discountValue: 0,
                        editionStartingPrice: 150,
                        formatted: {
                            originalPrice: '$40.00',
                            originalFullPrice: '$40.00 Contract Fee',
                            originalFullPriceElements: [
                                '$40.00',
                                'Contract Fee',
                                ''
                            ]
                        }
                    }
                }
            ],
            contractLength: 12,
            formattedContract: '12 month contract',
            discount: {
                description: 'discount!',
                percentage: null,
                startDate: null,
                expirationDate: null,
                numOfBillingCycles: null,
                availableRedemptions: null,
                amount: {
                    USD: 20
                },
                pricingUnit: 'USER'
            },
            uuid: '48a37f7c-4030-4d40-8c87-5262966ffd94'
        }],
        bullets: [{
            content: 'Custom Tables and Advanced Statistics Add-on provides easy to use drag and drop interactive tables exportable to Microsoft/PDF',
            tooltip: null
        }, {
            content: 'You can access a variety of additional techniques such as; Non-linear, logistic, 2-stage least squares regression, Generalized linear modeling and survival analysis',
            tooltip: null
        }, {
            content: 'Additionally, Bayesian Statistics now available',
            tooltip: null
        }],
        callToAction: {
            isPrimary: false,
            isSecondary: true
        },
        audience: [
            {
                content: 'Test target audience Son english'
            }
        ],
        items: [
            {
                content: '0 Users'
            }
        ],
        invisible: false,
        leadGen: false,
        revenueType: 'RECURRING'
    }]
}, {
    id: 334566,
    visible: true,
    name: 'Add-on 4',
    edition: [
        {
            name: 'Recurring Add-On',
            id: 660054,
            primary: true,
            plans: [
                {
                    costs: {
                        unitPrice: [
                            {
                                id: 'USER',
                                usageRange: [
                                    {
                                        formattedPrice: '$9.50',
                                        formattedFullPrice: '$9.50 / User / Month',
                                        formattedFullPriceElements: [
                                            '$9.50',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        unit: {
                                            min: 1,
                                            max: null,
                                            label: 'User',
                                            dependency: '',
                                            id: 'USER',
                                            limitLabel: 'For 1 User And Above'
                                        },
                                        frequency: {
                                            shortLabel: 'Mo',
                                            label: 'Month',
                                            id: 'MONTHLY'
                                        },
                                        price: {
                                            symbol: '$',
                                            currency: 'USD',
                                            amount: 10,
                                            value: 9.5,
                                            originalValue: 10,
                                            discountValue: 0.5,
                                            editionStartingPrice: 9.5,
                                            formatted: {
                                                originalPrice: '$10.00',
                                                originalFullPrice: '$10.00 / User / Month',
                                                originalFullPriceElements: [
                                                    '$10.00',
                                                    '/ User',
                                                    '/ Month'
                                                ],
                                                discount: '-$0.50',
                                                discountOff: '5% off per User',
                                                billingCycles: 'on the first 2 payments.',
                                                discountExpirationNotice: '5% off per User on the first 2 payments.',
                                                discountExpirationDate: 'This discount is available for redemption through Oct 10, 2021.',
                                                discountUses: 'Only 2 redemption(s) remaining.',
                                                meteredUsage: 'For 1 User And Above'
                                            }
                                        }
                                    }
                                ]
                            }
                        ]
                    },
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: {
                        discountNotice: '5% off per User on the first 2 payments.',
                        discountAvailability: 'This discount is available for redemption through Oct 10, 2021.',
                        discountUses: 'Only 2 redemption(s) remaining.',
                        description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
                        percentage: 0.05,
                        startDate: 1631592000000,
                        expirationDate: 1633838400000,
                        numOfBillingCycles: 2,
                        availableRedemptions: 2,
                        amount: null,
                        pricingUnit: 'USER'
                    },
                    uuid: '174bd803-0ea0-498a-82a4-b734e372e87d'
                }
            ],
            bullets: [{
                content: 'Custom Tables and Advanced Statistics Add-on provides easy to use drag and drop interactive tables exportable to Microsoft/PDF',
                tooltip: null
            }, {
                content: 'You can access a variety of additional techniques such as; Non-linear, logistic, 2-stage least squares regression, Generalized linear modeling and survival analysis',
                tooltip: null
            }, {
                content: 'Additionally, Bayesian Statistics now available',
                tooltip: null
            }],
            callToAction: {
                isPrimary: false,
                isSecondary: true
            },
            audience: [],
            items: [
                {
                    content: '0 Users'
                }
            ],
            invisible: false,
            leadGen: false,
            revenueType: 'RECURRING'
        }
    ]
}, {
    id: 3902,
    visible: true,
    name: 'Exchange Online Advanced Threat Protection',
    edition: [
        {
            name: 'Exchange Online Advanced Threat Protection',
            id: 9234,
            primary: true,
            plans: [
                {
                    costs: {
                        unitPrice: [
                            {
                                id: 'USER',
                                usageRange: [
                                    {
                                        formattedPrice: '$10.00',
                                        formattedFullPrice: '$10.00 / User / Month',
                                        formattedFullPriceElements: [
                                            '$10.00',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        unit: {
                                            min: 1,
                                            max: null,
                                            label: 'User',
                                            dependency: '',
                                            id: 'USER',
                                            limitLabel: 'For 1 User And Above'
                                        },
                                        frequency: {
                                            shortLabel: 'Mo',
                                            label: 'Month',
                                            id: 'MONTHLY'
                                        },
                                        price: {
                                            symbol: '$',
                                            currency: 'USD',
                                            amount: 10,
                                            value: 10,
                                            originalValue: 10,
                                            discountValue: 0,
                                            editionStartingPrice: 10,
                                            formatted: {
                                                originalPrice: '$10.00',
                                                originalFullPrice: '$10.00 / User / Month',
                                                originalFullPriceElements: [
                                                    '$10.00',
                                                    '/ User',
                                                    '/ Month'
                                                ],
                                                meteredUsage: 'For 1 User And Above'
                                            }
                                        }
                                    }
                                ]
                            }
                        ]
                    },
                    setupFees: [
                        {
                            formattedPrice: '$5.00',
                            formattedFullPrice: '$5.00 Setup fee',
                            formattedFullPriceElements: [
                                '$5.00',
                                'Setup fee',
                                ''
                            ],
                            unit: {
                                min: 0,
                                max: null,
                                label: 'Setup fee',
                                dependency: '',
                                id: 'ONE_TIME_SETUP'
                            },
                            frequency: {
                                shortLabel: '',
                                label: '',
                                id: 'MONTHLY'
                            },
                            price: {
                                symbol: '$',
                                currency: 'USD',
                                amount: 10,
                                value: 5,
                                originalValue: 10,
                                discountValue: 5,
                                editionStartingPrice: 5,
                                formatted: {
                                    originalPrice: '$10.00',
                                    originalFullPrice: '$10.00 Setup fee',
                                    originalFullPriceElements: [
                                        '$10.00',
                                        'Setup fee',
                                        ''
                                    ],
                                    discount: '-$5.00',
                                    discountOff: '$5.00 off'
                                }
                            }
                        }
                    ],
                    contractFees: [{
                        formattedPrice: '$5.00',
                        formattedFullPrice: '$5.00 Contract fee',
                        formattedFullPriceElements: [
                            '$5.00',
                            'Contract fee',
                            ''
                        ],
                        unit: {
                            min: 0,
                            max: null,
                            label: 'Contract Fee',
                            dependency: '',
                            id: 'CONTRACT_FEE'
                        },
                        frequency: {
                            shortLabel: '',
                            label: '',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 10,
                            value: 5,
                            originalValue: 10,
                            discountValue: 5,
                            editionStartingPrice: 150,
                            formatted: {
                                originalPrice: '$10.00',
                                originalFullPrice: '$10.00 Contract Fee',
                                originalFullPriceElements: [
                                    '$10.00',
                                    'Contract Fee',
                                    ''
                                ],
                                discount: '-$5.00',
                                discountOff: '$5.00 off'
                            }
                        }
                    }],
                    contractLength: null,
                    formattedContract: false,
                    discount: {
                        description: 'black friday',
                        percentage: null,
                        startDate: null,
                        expirationDate: null,
                        numOfBillingCycles: null,
                        availableRedemptions: null,
                        amount: {
                            USD: 5
                        },
                        pricingUnit: null
                    },
                    uuid: '8f0348a1-cd4a-46fc-b0eb-b49995e678d7'
                }
            ],
            bullets: [{
                content: 'Custom Tables and Advanced Statistics Add-on provides easy to use drag and drop interactive tables exportable to Microsoft/PDF',
                tooltip: null
            }, {
                content: 'You can access a variety of additional techniques such as; Non-linear, logistic, 2-stage least squares regression, Generalized linear modeling and survival analysis',
                tooltip: null
            }, {
                content: 'Additionally, Bayesian Statistics now available',
                tooltip: null
            }],
            callToAction: {
                isPrimary: false,
                isSecondary: true
            },
            audience: [],
            items: [
                {
                    content: '0 Users'
                }
            ],
            invisible: false,
            leadGen: false,
            revenueType: 'RECURRING'
        }
    ]
}];

export const addonsMock = [{
    name: 'IBM SPSS Statistics - Custom Tables & Advanced Statistics',
    id: 3,
    edition: [{
        name: 'Custom',
        id: 432928,
        primary: true,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$65.00',
                        formattedFullPrice: '$65.00 / User / Month',
                        formattedFullPriceElements: ['$65.00', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 65,
                            value: 65,
                            originalValue: 65,
                            discountValue: 0,
                            editionStartingPrice: 65,
                            formatted: {
                                originalPrice: '$65.00',
                                originalFullPrice: '$65.00 / User / Month',
                                originalFullPriceElements: ['$65.00', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [{
                formattedPrice: '$7.00',
                formattedFullPrice: '$7.00 Setup fee',
                formattedFullPriceElements: ['$7.00', 'Setup fee', ''],
                unit: { min: 0, max: null, label: 'Setup fee', dependency: '', id: 'ONE_TIME_SETUP' },
                frequency: { shortLabel: '', label: '', id: 'YEARLY' },
                price: {
                    symbol: '$',
                    currency: 'USD',
                    value: 7,
                    originalValue: 7,
                    discountValue: 0,
                    editionStartingPrice: null,
                    formatted: {
                        originalPrice: '$7.00',
                        originalFullPrice: '$7.00 Setup fee',
                        originalFullPriceElements: ['$7.00', 'Setup fee', '']
                    }
                }
            }],
            contractFees: [{
                formattedPrice: '$9.00',
                formattedFullPrice: '$9.00 Contract Fee',
                formattedFullPriceElements: ['$9.00', 'Contract Fee', ''],
                unit: { min: 0, max: null, label: 'Contract Fee', dependency: '', id: 'CONTRACT_FEE' },
                frequency: { shortLabel: '', label: '', id: 'MONTHLY' },
                price: {
                    symbol: '$',
                    currency: 'USD',
                    amount: 9,
                    value: 9,
                    originalValue: 9,
                    discountValue: 0,
                    editionStartingPrice: null,
                    formatted: {
                        originalPrice: '$9.00',
                        originalFullPrice: '$9.00 Contract Fee',
                        originalFullPriceElements: ['$9.00', 'Contract Fee', '']
                    }
                }
            }],
            contractLength: 12,
            formattedContract: '12 month contract',
            discount: false,
            uuid: 'b3d319d1-60fd-4fc3-bd86-3b09129fc27d'
        },
        {
            costs: {
                unitPrice: [{
                    id: 'SMS_TEXT_MESSAGE',
                    usageRange: [{
                        formattedPrice: '$0.01',
                        formattedFullPrice: '$0.01 / SMS Text Message / Month',
                        formattedFullPriceElements: ['$0.01', '/ SMS Text Message', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'SMS Text Message',
                            dependency: '',
                            id: 'SMS_TEXT_MESSAGE',
                            limitLabel: 'For 1 SMS Text Message And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 0.01,
                            value: 0.01,
                            originalValue: 0.01,
                            discountValue: 0,
                            editionStartingPrice: 0.01,
                            formatted: {
                                originalPrice: '$0.01',
                                originalFullPrice: '$0.01 / SMS Text Message / Month',
                                originalFullPriceElements: ['$0.01', '/ SMS Text Message', '/ Month'],
                                meteredUsage: 'For 1 SMS Text Message And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [{
                formattedPrice: 'Free',
                formattedFullPrice: 'Free Setup fee',
                formattedFullPriceElements: ['Free', 'Setup fee', ''],
                unit: { min: 0, max: null, label: 'Setup fee', dependency: '', id: 'ONE_TIME_SETUP' },
                frequency: { shortLabel: '', label: '', id: 'YEARLY' },
                price: {
                    symbol: '$',
                    currency: 'USD',
                    value: null,
                    originalValue: null,
                    discountValue: 0,
                    editionStartingPrice: null,
                    formatted: {
                        originalPrice: '$0.00',
                        originalFullPrice: '$0.00 Setup fee',
                        originalFullPriceElements: ['$0.00', 'Setup fee', '']
                    }
                }
            }],
            contractFees: [],
            contractLength: 12,
            formattedContract: '12 month contract',
            discount: false,
            uuid: 'b3d319d1-60fd-4fc3-bd86-3b09129fc27d'
        }],
        bullets: [{
            content: 'Custom Tables and Advanced Statistics Add-on provides easy to use drag and drop interactive tables exportable to Microsoft/PDF',
            tooltip: null
        }, {
            content: 'You can access a variety of additional techniques such as; Non-linear, logistic, 2-stage least squares regression, Generalized linear modeling and survival analysis',
            tooltip: null
        }, {
            content: 'Additionally, Bayesian Statistics now available',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=199553&type=BUY&addonId=199557',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [{
            content: 'Professional office suite with 30GB storage'
        }, {
            content: 'Most small and medium businesses'
        }],
        items: [{
            content: '0 Users'
        }],
        invisible: false,
        leadGen: false
    }]
}, {
    name: 'IBM SPSS Statistics - Forecasting & Decision Trees',
    parentEditionId: 404624,
    id: 1,
    edition: [{
        name: 'IBM',
        id: 432924,
        primary: true,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$79.00',
                        formattedFullPrice: '$79.00 / User / Month',
                        formattedFullPriceElements: ['$79.00', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 79,
                            value: 79,
                            originalValue: 79,
                            discountValue: 0,
                            editionStartingPrice: 79,
                            formatted: {
                                originalPrice: '$79.00',
                                originalFullPrice: '$79.00 / User / Month',
                                originalFullPriceElements: ['$79.00', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d4'
        }],
        bullets: [{
            content: 'Forecasting and Decision Trees Add-on provides ARIMA and exponential smoothing forecasting capabilities',
            tooltip: null
        }, {
            content: 'Classification and decision trees based on four established tree-growing algorithms are also available',
            tooltip: null
        }, {
            content: 'You can also create neural network predictive models as well as RFM analysis to test marketing campaigns',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=199553&type=BUY&addonId=199561',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [],
        items: [{
            content: '0 Users'
        }],
        invisible: false,
        leadGen: false
    }]
}, {
    name: 'IBM SPSS Statistics - Complex Sampling & Testing',
    id: 2,
    edition: [{
        name: 'Complex',
        id: 432772,
        primary: true,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$92.00',
                        formattedFullPrice: '$92.00 / User / Month',
                        formattedFullPriceElements: ['$92.00', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 92,
                            value: 92,
                            originalValue: 92,
                            discountValue: 0,
                            editionStartingPrice: 92,
                            formatted: {
                                originalPrice: '$92.00',
                                originalFullPrice: '$92.00 / User / Month',
                                originalFullPriceElements: ['$92.00', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: 'd34ebf01-62fd-428a-8ca2-a85eedfaeca8'
        }],
        bullets: [{
            content: 'Complex Sampling and Testing Add-on provides functionality for small sample sizes, missing data , and complex sampling',
            tooltip: null
        }, {
            content: 'You can access regression with optimal scaling including lasso and elastic net',
            tooltip: null
        }, {
            content: 'Additional features include; categorical principal components analysis, multidimensional scaling and unfolding, and multiple correspondence analysis',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=199553&type=BUY&addonId=199565',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [],
        items: [{
            content: '0 Users'
        }],
        invisible: false,
        leadGen: false
    }]
}, {
    name: 'Test Addon Number 4 to see spacing',
    parentEditionId: 404624,
    id: 4,
    edition: [{
        name: 'Test Addon Number 4 to see spacing',
        id: 432924,
        primary: true,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$79.00',
                        formattedFullPrice: '$79.00 / User / Month',
                        formattedFullPriceElements: ['$79.00', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 79,
                            value: 79,
                            originalValue: 79,
                            discountValue: 0,
                            editionStartingPrice: 79,
                            formatted: {
                                originalPrice: '$79.00',
                                originalFullPrice: '$79.00 / User / Month',
                                originalFullPriceElements: ['$79.00', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d4'
        }],
        bullets: [{
            content: 'Forecasting and Decision Trees Add-on provides ARIMA and exponential smoothing forecasting capabilities',
            tooltip: null
        }, {
            content: 'Classification and decision trees based on four established tree-growing algorithms are also available',
            tooltip: null
        }, {
            content: 'You can also create neural network predictive models as well as RFM analysis to test marketing campaigns',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=199553&type=BUY&addonId=199561',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [],
        items: [{
            content: '0 Users'
        }],
        invisible: false,
        leadGen: false
    }]
}, {
    name: 'One Team, One Dream Example addon #4',
    edition: [{
        name: 'AppDirect add0n for unlimited power',
        id: 432924,
        primary: true,
        plans: [{
            costs: {
                unitPrice: [{
                    id: 'USER',
                    usageRange: [{
                        formattedPrice: '$79.00',
                        formattedFullPrice: '$79.00 / User / Month',
                        formattedFullPriceElements: ['$79.00', '/ User', '/ Month'],
                        unit: {
                            min: 1,
                            max: null,
                            label: 'User',
                            dependency: '',
                            id: 'USER',
                            limitLabel: 'For 1 User And Above'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 79,
                            value: 79,
                            originalValue: 79,
                            discountValue: 0,
                            editionStartingPrice: 79,
                            formatted: {
                                originalPrice: '$79.00',
                                originalFullPrice: '$79.00 / User / Month',
                                originalFullPriceElements: ['$79.00', '/ User', '/ Month'],
                                meteredUsage: 'For 1 User And Above'
                            }
                        }
                    }]
                }]
            },
            setupFees: [],
            contractFees: [],
            contractLength: null,
            formattedContract: false,
            discount: false,
            uuid: '46c6dbe3-15c2-4abb-ac52-58d916b4e7d4'
        }],
        bullets: [{
            content: 'Forecasting and Decision Trees Add-on provides ARIMA and exponential smoothing forecasting capabilities',
            tooltip: null
        }, {
            content: 'Classification and decision trees based on four established tree-growing algorithms are also available',
            tooltip: null
        }, {
            content: 'You can also create neural network predictive models as well as RFM analysis to test marketing campaigns',
            tooltip: null
        }],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            enabled: true,
            url: 'https://marketplace.appdirect.com/api/internal/storefront/v1/cta?productId=199553&type=BUY&addonId=199561',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [],
        items: [{
            content: '0 Users'
        }],
        invisible: false,
        leadGen: false
    }]
}, {
    id: 158629,
    visible: true,
    name: 'JH Addon V2',
    edition: [
        {
            name: 'Recurring Add-On',
            id: 281449,
            primary: true,
            plans: [
                {
                    costs: {
                        unitPrice: [
                            {
                                id: 'USER',
                                usageRange: [
                                    {
                                        formattedPrice: '$10.00',
                                        formattedFullPrice: '$10.00 / User / Month',
                                        formattedFullPriceElements: [
                                            '$10.00',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        unit: {
                                            min: 2,
                                            max: null,
                                            label: 'User',
                                            dependency: '',
                                            id: 'USER',
                                            limitLabel: 'For 1 User And Above'
                                        },
                                        frequency: {
                                            shortLabel: 'Mo',
                                            label: 'Month',
                                            id: 'MONTHLY'
                                        },
                                        price: {
                                            symbol: '$',
                                            currency: 'USD',
                                            amount: 10,
                                            value: 10,
                                            originalValue: 10,
                                            discountValue: 0,
                                            editionStartingPrice: 20,
                                            formatted: {
                                                originalPrice: '$10.00',
                                                originalFullPrice: '$10.00 / User / Month',
                                                originalFullPriceElements: [
                                                    '$10.00',
                                                    '/ User',
                                                    '/ Month'
                                                ],
                                                meteredUsage: 'For 1 User And Above'
                                            }
                                        }
                                    }
                                ]
                            }
                        ],
                        flatPrice: {
                            formattedPrice: '$10.00',
                            formattedFullPrice: '$10.00 / Month',
                            formattedFullPriceElements: [
                                '$10.00',
                                null,
                                '/ Month'
                            ],
                            unit: {
                                min: 0,
                                max: null,
                                dependency: '',
                                id: 'NOT_APPLICABLE'
                            },
                            frequency: {
                                shortLabel: 'Mo',
                                label: 'Month',
                                id: 'MONTHLY'
                            },
                            price: {
                                symbol: '$',
                                currency: 'USD',
                                amount: 10,
                                value: 10,
                                originalValue: 10,
                                discountValue: 0,
                                editionStartingPrice: 20,
                                formatted: {
                                    originalPrice: '$10.00',
                                    originalFullPrice: '$10.00 / Month',
                                    originalFullPriceElements: [
                                        '$10.00',
                                        null,
                                        '/ Month'
                                    ]
                                }
                            }
                        }
                    },
                    setupFees: [
                        {
                            formattedPrice: '$7.00',
                            formattedFullPrice: '$7.00 Setup fee',
                            formattedFullPriceElements: [
                                '$7.00',
                                'Setup fee',
                                ''
                            ],
                            unit: {
                                min: 0,
                                max: null,
                                label: 'Setup fee',
                                dependency: '',
                                id: 'ONE_TIME_SETUP'
                            },
                            frequency: {
                                shortLabel: '',
                                label: '',
                                id: 'MONTHLY'
                            },
                            price: {
                                symbol: '$',
                                currency: 'USD',
                                amount: 7,
                                value: 7,
                                originalValue: 7,
                                discountValue: 0,
                                editionStartingPrice: 20,
                                formatted: {
                                    originalPrice: '$7.00',
                                    originalFullPrice: '$7.00 Setup fee',
                                    originalFullPriceElements: [
                                        '$7.00',
                                        'Setup fee',
                                        ''
                                    ]
                                }
                            }
                        }
                    ],
                    contractFees: [
                        {
                            formattedPrice: '$9.00',
                            formattedFullPrice: '$9.00 Contract Fee',
                            formattedFullPriceElements: [
                                '$9.00',
                                'Contract Fee',
                                ''
                            ],
                            unit: {
                                min: 0,
                                max: null,
                                label: 'Contract Fee',
                                dependency: '',
                                id: 'CONTRACT_FEE'
                            },
                            frequency: {
                                shortLabel: '',
                                label: '',
                                id: 'MONTHLY'
                            },
                            price: {
                                symbol: '$',
                                currency: 'USD',
                                amount: 9,
                                value: 9,
                                originalValue: 9,
                                discountValue: 0,
                                editionStartingPrice: 20,
                                formatted: {
                                    originalPrice: '$9.00',
                                    originalFullPrice: '$9.00 Contract Fee',
                                    originalFullPriceElements: [
                                        '$9.00',
                                        'Contract Fee',
                                        ''
                                    ]
                                }
                            }
                        }
                    ],
                    contractLength: 1,
                    formattedContract: '1 month contract',
                    discount: false,
                    uuid: '0dee2234-88ae-4284-982c-1ad61c714911'
                }
            ],
            bullets: [{
                content: 'Custom Tables and Advanced Statistics Add-on provides easy to use drag and drop interactive tables exportable to Microsoft/PDF',
                tooltip: null
            }, {
                content: 'You can access a variety of additional techniques such as; Non-linear, logistic, 2-stage least squares regression, Generalized linear modeling and survival analysis',
                tooltip: null
            }, {
                content: 'Additionally, Bayesian Statistics now available',
                tooltip: null
            }],
            callToAction: {
                actionType: 'NO_ACTION'
            },
            audience: [],
            items: [
                {
                    content: '0 Users'
                }
            ],
            invisible: false,
            leadGen: false,
            revenueType: 'RECURRING'
        }
    ]
}, {
    id: 334566,
    visible: true,
    name: 'Add-on 2',
    edition: [
        {
            name: 'Recurring Add-On',
            id: 660054,
            primary: true,
            plans: [
                {
                    costs: {
                        unitPrice: [
                            {
                                id: 'USER',
                                usageRange: [
                                    {
                                        formattedPrice: '$10.00',
                                        formattedFullPrice: '$10.00 / User / Month',
                                        formattedFullPriceElements: [
                                            '$10.00',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        unit: {
                                            min: 1,
                                            max: null,
                                            label: 'User',
                                            dependency: '',
                                            id: 'USER',
                                            limitLabel: 'For 1 User And Above'
                                        },
                                        frequency: {
                                            shortLabel: 'Mo',
                                            label: 'Month',
                                            id: 'MONTHLY'
                                        },
                                        price: {
                                            symbol: '$',
                                            currency: 'USD',
                                            amount: 10,
                                            value: 10,
                                            originalValue: 10,
                                            discountValue: 0,
                                            editionStartingPrice: 10,
                                            formatted: {
                                                originalPrice: '$10.00',
                                                originalFullPrice: '$10.00 / User / Month',
                                                originalFullPriceElements: [
                                                    '$10.00',
                                                    '/ User',
                                                    '/ Month'
                                                ],
                                                meteredUsage: 'For 1 User And Above'
                                            }
                                        }
                                    }
                                ]
                            }
                        ]
                    },
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: '174bd803-0ea0-498a-82a4-b734e372e87d'
                }
            ],
            bullets: [{
                content: 'Custom Tables and Advanced Statistics Add-on provides easy to use drag and drop interactive tables exportable to Microsoft/PDF',
                tooltip: null
            }, {
                content: 'You can access a variety of additional techniques such as; Non-linear, logistic, 2-stage least squares regression, Generalized linear modeling and survival analysis',
                tooltip: null
            }, {
                content: 'Additionally, Bayesian Statistics now available',
                tooltip: null
            }],
            callToAction: {
                isPrimary: false,
                isSecondary: true
            },
            audience: [],
            items: [
                {
                    content: '0 Users'
                }
            ],
            invisible: false,
            leadGen: false,
            revenueType: 'RECURRING'
        }
    ]
}, ...addonsWithDiscountsMock];

export const addonsMockWithAdditionLimitations = [
    {
        ...addonsMock[0],
        parentEditionId: 28155,
        parentEditionIds: [28155, 28158]
    },
    {
        ...addonsMock[1],
        parentEditionId: 28158,
        parentEditionIds: [28158]
    }
];
