// @appdirect/sfb-theme-components/src/constants/components/schemas/hero-banner.js
import { searchBarSchemaForm } from './search-bar';
import { CONTENT_WIDTH, CONTENT_WIDTH_IMAGE_STYLE, SEARCH_BAR_MIN_WIDTH, IMAGE_STYLE } from '../hero-banner';
import { COLORPICKER, IMAGE_UPLOADER, SLIDER, GROUP_HEADER } from '../../schemaComponentTypes';

export const conditionalImageStyleAttributes = {
    isVisible: {
        when: 'style',
        connector: 'equals',
        value: IMAGE_STYLE
    }
};

export const conditionalNotImageStyleAttributes = {
    isVisible: {
        when: 'style',
        connector: 'notEquals',
        value: IMAGE_STYLE
    }
};

export const heroBannerSearchBarSchemaForm = themeColors => ({
    searchBarHeading: {
        type: GROUP_HEADER,
        title: 'uieditor.sfbComponent.heroBanner.searchBarHeading.title',
        symbol: 'style'
    },
    searchBarWidth: {
        title: 'uieditor.sfbComponent.heroBanner.searchBarWidth.title',
        type: SLIDER,
        defaultValue: 521,
        options: {
            range: true,
            step: 1,
            min: SEARCH_BAR_MIN_WIDTH,
            max: CONTENT_WIDTH
        },
        conditionalAttributes: conditionalNotImageStyleAttributes
    },
    searchBarWidthImageStyle: {
        title: 'uieditor.sfbComponent.heroBanner.searchBarWidthImageStyle.title',
        type: SLIDER,
        defaultValue: CONTENT_WIDTH_IMAGE_STYLE,
        options: {
            range: true,
            step: 1,
            min: SEARCH_BAR_MIN_WIDTH,
            max: CONTENT_WIDTH_IMAGE_STYLE
        },
        conditionalAttributes: conditionalImageStyleAttributes
    },
    searchCornerRadius: {
        ...searchBarSchemaForm.searchCornerRadius,
        title: 'uieditor.sfbComponent.heroBanner.searchCornerRadius.title'
    },
    searchPlaceholder: {
        ...searchBarSchemaForm.searchPlaceholder,
        isLocalized: true,
        title: 'uieditor.sfbComponent.heroBanner.searchPlaceholder.title'
    },
    searchIcon: {
        title: 'uieditor.sfbComponent.heroBanner.searchIcon.title',
        type: IMAGE_UPLOADER,
        placeholder: 'uieditor.sfbComponent.heroBanner.searchIcon.placeholder',
        caption: 'uieditor.sfbComponent.heroBanner.searchIcon.caption',
        defaultValue: {
            filename: '',
            type: IMAGE_UPLOADER,
            url: '',
            alt: ''
        }
    },
    searchPlaceholderTextColor: {
        title: 'uieditor.sfbComponent.heroBanner.searchPlaceholderTextColor.title',
        type: COLORPICKER,
        defaultValue: 'rgba(0,0,0,0.5)'
    },
    searchBorderColor: {
        title: 'uieditor.sfbComponent.heroBanner.searchBorderColor.title',
        type: COLORPICKER,
        defaultValue: '#cacaca'
    },
    searchBackgroundColor: {
        title: 'uieditor.sfbComponent.heroBanner.searchBackgroundColor.title',
        type: COLORPICKER,
        defaultValue: '#fff'
    },
    searchButtonColor: {
        title: 'uieditor.sfbComponent.heroBanner.searchButtonColor.title',
        type: COLORPICKER,
        defaultValue: themeColors.primaryColor
    },
    searchIconColor: {
        title: 'uieditor.sfbComponent.heroBanner.searchIconColor.title',
        type: COLORPICKER,
        defaultValue: '#fff'
    }
});
