// @appdirect/sfb-theme-components/src/atoms/slick-slider/stories/SlickSlider.stories.js
import React from 'react';
import SlickSlider from '../SlickSlider';
import { sliderProductsMock } from '../../../../mocks/product/slider';

const renderSlides = item => (
    <div>
        <div style={{ padding: '0 40px' }}>
            <h1>{item.name}</h1>
            <div>{item.description}</div>
        </div>
    </div>
);

export default {
    title: 'Atoms/Slick Slider',
    component: SlickSlider
};

export const Default = {
    render: () => (
        <div
            style={{
                textAlign: 'center'
            }}
        >
            <SlickSlider>{sliderProductsMock.map(renderSlides)}</SlickSlider>
        </div>
    ),

    name: 'Default'
};

export const NoArrows = {
    render: () => (
        <div
            style={{
                textAlign: 'center'
            }}
        >
            <SlickSlider
                settings={{
                    arrows: false
                }}
            >
                {sliderProductsMock.map(renderSlides)}
            </SlickSlider>
        </div>
    ),

    name: 'No arrows'
};

export const NoDots = {
    render: () => (
        <div
            style={{
                textAlign: 'center'
            }}
        >
            <SlickSlider
                settings={{
                    dots: false
                }}
            >
                {sliderProductsMock.map(renderSlides)}
            </SlickSlider>
        </div>
    ),

    name: 'No dots'
};
