// @appdirect/sfb-theme-components/src/atoms/product-edition-v2/card/stories/ProductEditionCardV2.stories.js
import React from 'react';
import { editionsMockData, editionWithDiscountedUnit } from '../../../../../mocks/product/productEditions';
import ProductEditionCardV2 from '../ProductEditionCardV2';

export default {
    title: 'Components (Preview)/Product Configurator V2/Atoms/Product Edition Card V2',
    component: ProductEditionCardV2
};

export const Default = {
    render: () => <ProductEditionCardV2 data={editionsMockData.items[0]} />,
    name: 'Default'
};

export const PopularEdition = {
    render: () => <ProductEditionCardV2 data={editionsMockData.items[1]} />,
    name: 'Popular edition'
};

export const PremiumEdition = {
    render: () => <ProductEditionCardV2 data={editionsMockData.items[2]} />,
    name: 'Premium edition'
};

export const Selected = {
    render: () => <ProductEditionCardV2 data={editionsMockData.items[0]} isSelected />,
    name: 'Selected'
};

export const DiscountedUnit = {
    render: () => <ProductEditionCardV2 data={editionWithDiscountedUnit} />,
    name: 'Discounted Unit'
};
