// @appdirect/sfb-theme-components/src/atoms/product-edition/card/stories/ProductEditionCard.stories.js
import React from 'react';
import { editionsMockData,
    editionWithDiscountedUnit,
    editionsSkipConfiguratorMockData } from '../../../../../mocks/product/productEditions';
import ProductEditionCard from '../ProductEditionCard';

export default {
    title: 'Atoms/Product Edition Card',
    component: ProductEditionCard
};

export const Default = {
    render: () => <ProductEditionCard data={editionsMockData.items[0]} />,
    name: 'Default'
};

export const PopularEdition = {
    render: () => <ProductEditionCard data={editionsMockData.items[1]} />,
    name: 'Popular edition'
};

export const PremiumEdition = {
    render: () => <ProductEditionCard data={editionsMockData.items[2]} />,
    name: 'Premium edition'
};

export const Selected = {
    render: () => <ProductEditionCard data={editionsMockData.items[0]} isSelected configuratorMode />,
    name: 'Selected'
};

export const AddedToCart = {
    render: () => <ProductEditionCard data={editionsSkipConfiguratorMockData[0]} isAddedToCart nonStackableProduct />,

    name: 'Added to Cart'
};

export const DiscountedUnit = {
    render: () => <ProductEditionCard data={editionWithDiscountedUnit} />,
    name: 'Discounted Unit'
};
