// @appdirect/sfb-theme-components/mocks/product/overview/index.js
import { callToActionsMock } from './callToActions';
import { customersMock } from './customers';
import { benefitsMock } from './benefits';
import { i18nMock } from './i18n';

export const productOverviewMock = {
    title: 'Enterprise Search, in seconds',
    description: 'AppWise is a business productivity tool that provides Unified Search across 40+ leading applications like Salesforce, Gmail, Office 365, Box and Asana. AppWise has enterprise-grade security built-in from the ground up, and it\'s intuitive enough be setup by end-users in seconds.',
    callToActions: callToActionsMock,
    image: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/overview/img1301783796595200436.png?18e59c26ffbf1b84953937d376a29fdc',
    imageSrcSet: {
        default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/overview/img1301783796595200436.png?18e59c26ffbf1b84953937d376a29fdc',
        '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/overview/img1301783796595200436.png?5890c62dc6fd364c4b6a2d4c9370456f',
        '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/customizations/SFBMARKETPLACE/42937/overview/img1301783796595200436-2x.png?8ac98635908c0ea66e0bab71750f3b57'
    },
    showThirdPartyLegalDisclosure: false,
    documentationUrl: 'https://help.appwise.com',
    downloadDocumentationUrl: null,
    pricingUrl: '/en-US/apps/42937/appwise/editions',
    customers: customersMock,
    benefits: benefitsMock,
    i18n: i18nMock
};
