// @appdirect/sfb-theme-components/src/tools/videos/vimeoHelpers.js
import { makeRequest, METHOD_GET } from '../httpRequest';

export const PROVIDER = 'vimeo';
export const THUMBNAIL_LARGE = 'large';
export const THUMBNAIL_MEDIUM = 'medium';
export const THUMBNAIL_SMALL = 'small';
export const REGEX_VIDEO_ID = /(?:http:|https:|)\/\/(?:player.|www.)?vimeo\.com\/(?:video\/|embed\/|watch\?\S*v=|v\/)?(\d*)/;

export const getVimeoMetaData = async videoId => {
    if (!videoId) {
        return null;
    }

    const requestUrl = `http://vimeo.com/api/v2/video/${videoId}.json`;

    return makeRequest(
        METHOD_GET,
        requestUrl
    ).then(response => response && JSON.parse(response)).catch(error => {
        console.error('getVimeoMetaData :: ', error);

        return null;
    });
};

export const getVimeoThumbnail = async (videoId, size = THUMBNAIL_MEDIUM) => {
    if (!videoId) {
        return null;
    }

    const metaData = await getVimeoMetaData(videoId);
    const thumbnail = metaData && metaData.length ? metaData[0][`thumbnail_${size}`] : null;

    return thumbnail;
};

export const getVimeoVideoId = url => {
    const matchesArray = url && url.match(REGEX_VIDEO_ID);

    return matchesArray && matchesArray.length ? matchesArray[1] : null;
};
