// @appdirect/sfb-theme-components/mocks/product/additionalInfo.js
export const additionalInfoMock = {
    support: {
        termsUrl: 'https://gsuite.google.com/intl/en_in/terms/2013/1/premier_terms.html',
        privacyUrl: 'https://gsuite.google.com/security/?secure-by-design_activeEl=data-centers',
        email: 'support@appdirect.com',
        phone: '(877) 404-2777',
        knowledgebaseUrl: 'https://support.google.com/a',
        description: 'We provide a variety of ways to contact our Customer Success team, below. Community Forum: community. dreamfactory.com. Call 1-888-399-DREAM(3732) Option 2 Email: support@dreamfactrory.com Support hours are open from 8 AM to 8 PM EST Monday trhough Friday',
        i18n: {
            termsAndConditions: 'Terms & Conditions',
            termsUrl: 'Terms of Service',
            privacyUrl: 'Privacy Policy',
            support: 'Support',
            email: 'Email',
            description: 'Description',
            phone: 'Phone',
            knowledgebaseUrl: 'Knowledge Base'
        }
    },
    relatedContent: {
        resources: {
            data: [
                {
                    label: 'View Screenshots',
                    icon: 'screenshot',
                    count: 3,
                    href: '/apps/577/screenshots'
                },
                {
                    label: 'Watch Demo',
                    icon: 'play',
                    href: 'https://youtu.be/EZmXEFGFeNA'
                },
                {
                    label: 'Read Documentation',
                    icon: 'file',
                    href: 'https://gsuite.google.com/learning-center/'
                }
            ],
            i18n: {
                resources: 'Resources',
                resellerResources: 'Reseller Resources',
                videos: 'Videos',
                documents: 'Documents',
                notPublic: 'These resources are not public and are only visible to resellers.',
                download: 'Download'
            }
        },
        details: {
            reviews: {
                title: 'Rating',
                rating: 5,
                caption: '5/5 Based on 1 reviews',
                numRatings: 1
            },
            tags: [
                {
                    title: 'Language availability',
                    items: [
                        {
                            label: 'English (US)',
                            href: '/en-US/listing?attr=4606'
                        },
                        {
                            label: 'Chinese',
                            href: '/en-US/listing?attr=4610'
                        },
                        {
                            label: 'Danish',
                            href: '/en-US/listing?attr=4618'
                        },
                        {
                            label: 'French',
                            href: '/en-US/listing?attr=4626'
                        },
                        {
                            label: 'German',
                            href: '/en-US/listing?attr=4638'
                        },
                        {
                            label: 'Italian',
                            href: '/en-US/listing?attr=4642'
                        },
                        {
                            label: 'Polish',
                            href: '/en-US/listing?attr=4646'
                        },
                        {
                            label: 'Portuguese',
                            href: '/en-US/listing?attr=4650'
                        },
                        {
                            label: 'Spanish (European)',
                            href: '/en-US/listing?attr=4658'
                        },
                        {
                            label: 'Swedish',
                            href: '/en-US/listing?attr=4662'
                        },
                        {
                            label: 'English (AU)',
                            href: '/en-US/listing?attr=4674'
                        },
                        {
                            label: 'English (UK)',
                            href: '/en-US/listing?attr=4678'
                        },
                        {
                            label: 'Spanish (Mexican)',
                            href: '/en-US/listing?attr=4682'
                        },
                        {
                            label: 'Dutch',
                            href: '/en-US/listing?attr=4686'
                        },
                        {
                            label: 'Mandarin',
                            href: '/en-US/listing?attr=4690'
                        }
                    ]
                },
                {
                    title: 'Office Productivity',
                    items: [
                        {
                            label: 'Office Tools',
                            href: '/en-US/listing?scat=2258'
                        }
                    ]
                },
                {
                    title: 'Supported Devices',
                    items: [
                        {
                            label: 'Android',
                            href: '/en-US/listing?attr=811'
                        },
                        {
                            label: 'iPhone',
                            href: '/en-US/listing?attr=815'
                        },
                        {
                            label: 'Web-based',
                            href: '/en-US/listing?attr=835'
                        },
                        {
                            label: 'Windows Phone',
                            href: '/en-US/listing?attr=843'
                        }
                    ]
                },
                {
                    title: 'Catalog',
                    items: [
                        {
                            label: 'Core',
                            href: '/en-US/listing?attr=4320'
                        }
                    ]
                }
            ],
            industries: [],
            vendor: {
                title: 'Developer',
                items: [
                    {
                        label: 'Google',
                        href: '/en-US/listing?vendor=0d0208f2-15de-4c1a-8e79-9ce6c3b99301'
                    }
                ]
            },
            platforms: false
        },
        pricing: {
            editions: [
                {
                    name: 'G Suite Basic - No Contract',
                    value: '$6.00',
                    frequency: 'MONTHLY',
                    formattedFrequency: 'per month',
                    href: '/apps/577/editions'
                },
                {
                    name: 'G Suite Basic',
                    value: '$6.00',
                    frequency: 'MONTHLY',
                    formattedFrequency: 'per month',
                    href: '/apps/577/editions'
                },
                {
                    name: 'G Suite Business - No Contract',
                    value: '$12.00',
                    frequency: 'MONTHLY',
                    formattedFrequency: 'per month',
                    href: '/apps/577/editions'
                },
                {
                    name: 'G Suite Business',
                    value: '$12.00',
                    frequency: 'MONTHLY',
                    formattedFrequency: 'per month',
                    href: '/apps/577/editions'
                },
                {
                    name: 'G Suite Enterprise - No Contract',
                    value: '$25.00',
                    frequency: 'MONTHLY',
                    formattedFrequency: 'per month',
                    href: '/apps/577/editions'
                },
                {
                    name: 'G Suite Enterprise',
                    value: '$25.00',
                    frequency: 'MONTHLY',
                    formattedFrequency: 'per month',
                    href: '/apps/577/editions'
                },
                {
                    name: 'G Suite Basic - Annual Plan',
                    value: '$72.00',
                    frequency: 'YEARLY',
                    formattedFrequency: 'per year',
                    href: '/apps/577/editions'
                },
                {
                    name: 'G Suite Business - Annual Plan',
                    value: '$144.00',
                    frequency: 'YEARLY',
                    formattedFrequency: 'per year',
                    href: '/apps/577/editions'
                },
                {
                    name: 'G Suite Enterprise - Annual Plan',
                    value: '$300.00',
                    frequency: 'YEARLY',
                    formattedFrequency: 'per year',
                    href: '/apps/577/editions'
                }
            ]
        },
        related: [],
        resellers: [],
        i18n: {
            details: 'Details',
            pricing: 'Pricing',
            related: 'Related Products',
            resellers: 'Authorized Resellers'
        }
    }
};
