// @appdirect/sfb-theme-components/mocks/product/configuratorSelectedAddons.js
export const selectedAddonsMock = {
    configurator: {
        selectedEdition: {
            name: 'G Suite Business - No Contract',
            id: '404632',
            primary: false,
            plans: [{
                costs: {
                    unitPrice: [{
                        id: 'USER',
                        usageRange: [{
                            formattedPrice: '$12.00',
                            formattedFullPrice: '$12.00 / User / Month',
                            formattedFullPriceElements: [
                                '$12.00',
                                '/ User',
                                '/ Month'
                            ],
                            unit: {
                                min: 1,
                                max: null,
                                label: 'User',
                                dependency: '',
                                id: 'USER',
                                limitLabel: 'For 1 User And Above'
                            },
                            frequency: {
                                shortLabel: 'Mo',
                                label: 'Month',
                                id: 'MONTHLY'
                            },
                            price: {
                                symbol: '$',
                                currency: 'USD',
                                amount: 12,
                                value: 12,
                                originalValue: 12,
                                discountValue: 0,
                                editionStartingPrice: 12,
                                formatted: {
                                    originalPrice: '$12.00',
                                    originalFullPrice: '$12.00 / User / Month',
                                    originalFullPriceElements: [
                                        '$12.00',
                                        '/ User',
                                        '/ Month'
                                    ],
                                    meteredUsage: 'For 1 User And Above'
                                }
                            }
                        }]
                    }]
                },
                setupFees: [],
                contractFees: [],
                contractLength: null,
                formattedContract: false,
                discount: false,
                uuid: '4e6237b7-7a74-4ee5-b050-aaafed2e44e9'
            }],
            bullets: [{
                content: 'Unlimited cloud storage (or 1TB per user if fewer than 5 users)',
                tooltip: null
            }, {
                content: 'Smart search across G Suite with Cloud Search',
                tooltip: null
            }, {
                content: 'Archive and set retention policies for emails and chats',
                tooltip: null
            }, {
                content: 'eDiscovery for emails, chats, and files',
                tooltip: null
            }, {
                content: 'Audit reports to track user activity',
                tooltip: null
            }, {
                content: '*This is a monthly-billed plan, with no contract.',
                tooltip: null
            }],
            callToAction: {
                isPrimary: false,
                isSecondary: true
            },
            audience: [{
                content: 'Enhanced office suite with unlimited storage and archiving'
            }],
            items: [],
            invisible: false,
            leadGen: false
        },
        selectedAddons: [{
            name: 'Google Vault',
            edition: [{
                name: 'Google Vault',
                id: '395056',
                primary: true,
                plans: [{
                    costs: {
                        unitPrice: [{
                            id: 'USER',
                            usageRange: [{
                                formattedPrice: '$4.17',
                                formattedFullPrice: '$4.17 / User / Month',
                                formattedFullPriceElements: [
                                    '$4.17',
                                    '/ User',
                                    '/ Month'
                                ],
                                unit: {
                                    min: 1,
                                    max: null,
                                    label: 'User',
                                    dependency: '',
                                    id: 'USER',
                                    limitLabel: 'For 1 User And Above'
                                },
                                frequency: {
                                    shortLabel: 'Mo',
                                    label: 'Month',
                                    id: 'MONTHLY'
                                },
                                price: {
                                    symbol: '$',
                                    currency: 'USD',
                                    amount: 4.17,
                                    value: 4.17,
                                    originalValue: 4.17,
                                    discountValue: 0,
                                    editionStartingPrice: 4.17,
                                    formatted: {
                                        originalPrice: '$4.17',
                                        originalFullPrice: '$4.17 / User / Month',
                                        originalFullPriceElements: [
                                            '$4.17',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        meteredUsage: 'For 1 User And Above'
                                    }
                                }
                            }]
                        }]
                    },
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: 'a3c690ec-082b-47a0-bf30-029b0a881df8'
                }, {
                    costs: {
                        unitPrice: [{
                            id: 'USER',
                            usageRange: [{
                                formattedPrice: '$50.00',
                                formattedFullPrice: '$50.00 / User / Year',
                                formattedFullPriceElements: [
                                    '$50.00',
                                    '/ User',
                                    '/ Year'
                                ],
                                unit: {
                                    min: 1,
                                    max: null,
                                    label: 'User',
                                    dependency: '',
                                    id: 'USER',
                                    limitLabel: 'For 1 User And Above'
                                },
                                frequency: {
                                    shortLabel: 'Yr',
                                    label: 'Year',
                                    id: 'YEARLY'
                                },
                                price: {
                                    symbol: '$',
                                    currency: 'USD',
                                    amount: 50,
                                    value: 50,
                                    originalValue: 50,
                                    discountValue: 0,
                                    editionStartingPrice: 50,
                                    formatted: {
                                        originalPrice: '$50.00',
                                        originalFullPrice: '$50.00 / User / Year',
                                        originalFullPriceElements: [
                                            '$50.00',
                                            '/ User',
                                            '/ Year'
                                        ],
                                        meteredUsage: 'For 1 User And Above'
                                    }
                                }
                            }]
                        }]
                    },
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: '5887a3e8-6cdf-44ec-9055-b2761dd88217'
                }],
                bullets: [{
                    content: 'Manage, retain, search and export your organization’s email and on-the-record chats. Vault is included with G Suite\'s Business and Enterprise editions.',
                    tooltip: null
                }, {
                    content: 'Control how long your email messages and on-the-record chats are retained. Specify policies for your entire domain or based on organizational units, date ranges and specific terms.',
                    tooltip: null
                }, {
                    content: 'Find and retrieve valuable information, even from closed accounts. It’s a smart way to gather what your legal team needs and prevent data loss from employee turnover.',
                    tooltip: null
                }],
                callToAction: {
                    isPrimary: false,
                    isSecondary: true
                },
                audience: [],
                items: [],
                invisible: false,
                leadGen: false
            }]
        }, {
            name: 'Google Drive storage upgrade - 50 GB',
            edition: [{
                name: 'Google Drive storage upgrade - 50 GB',
                id: '395072',
                primary: true,
                plans: [{
                    costs: {
                        unitPrice: [{
                            id: 'USER',
                            usageRange: [{
                                formattedPrice: '$6.25',
                                formattedFullPrice: '$6.25 / User / Month',
                                formattedFullPriceElements: [
                                    '$6.25',
                                    '/ User',
                                    '/ Month'
                                ],
                                unit: {
                                    min: 1,
                                    max: null,
                                    label: 'User',
                                    dependency: '',
                                    id: 'USER',
                                    limitLabel: 'For 1 User And Above'
                                },
                                frequency: {
                                    shortLabel: 'Mo',
                                    label: 'Month',
                                    id: 'MONTHLY'
                                },
                                price: {
                                    symbol: '$',
                                    currency: 'USD',
                                    amount: 6.25,
                                    value: 6.25,
                                    originalValue: 6.25,
                                    discountValue: 0,
                                    editionStartingPrice: 6.25,
                                    formatted: {
                                        originalPrice: '$6.25',
                                        originalFullPrice: '$6.25 / User / Month',
                                        originalFullPriceElements: [
                                            '$6.25',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        meteredUsage: 'For 1 User And Above'
                                    }
                                }
                            }]
                        }]
                    },
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: '46075d33-fd69-40c5-9ec5-748c0f832f34'
                }],
                bullets: [{
                    content: 'Your storage is shared across Google Drive, Gmail, and Google Photos',
                    tooltip: null
                }, {
                    content: 'Get unlimited storage with G Suite Business or Enterprise plans',
                    tooltip: null
                }],
                callToAction: {
                    isPrimary: false,
                    isSecondary: true
                },
                audience: [],
                items: [],
                invisible: false,
                leadGen: false
            }]
        }],
        backAction: {
            label: 'Back to Office 365',
            url: '/en-US/apps/577/office/editions'
        },
        i18n: {
            selectedEdition: 'Selected Edition',
            selectedAddons: 'Selected Add-ons',
            startingAt: 'Starting at'
        },
        productId: 'productId'
    },
    callToActions: {
        items: [{
            actionType: 'BUY_NOW',
            enabled: true,
            popup: {},
            isPrimary: true,
            label: 'Buy Now',
            tooltip: '',
            url: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=1&type=BUY_NOW',
            urlCreator: {
                base: 'http: //marketplace.com/api/internal/storefront/v1/cta',
                actionType: 'BUY_NOW',
                editionParam: 'editionId',
                addonParam: 'addonId'
            }
        }]
    }
};

export const selectedAddonsMultiButtonMock = {
    configurator: {
        selectedEdition: {
            name: 'G Suite Business - No Contract',
            id: '404632',
            primary: false,
            plans: [{
                costs: {
                    unitPrice: [{
                        id: 'USER',
                        usageRange: [{
                            formattedPrice: '$12.00',
                            formattedFullPrice: '$12.00 / User / Month',
                            formattedFullPriceElements: [
                                '$12.00',
                                '/ User',
                                '/ Month'
                            ],
                            unit: {
                                min: 1,
                                max: null,
                                label: 'User',
                                dependency: '',
                                id: 'USER',
                                limitLabel: 'For 1 User And Above'
                            },
                            frequency: {
                                shortLabel: 'Mo',
                                label: 'Month',
                                id: 'MONTHLY'
                            },
                            price: {
                                symbol: '$',
                                currency: 'USD',
                                amount: 12,
                                value: 12,
                                originalValue: 12,
                                discountValue: 0,
                                editionStartingPrice: 12,
                                formatted: {
                                    originalPrice: '$12.00',
                                    originalFullPrice: '$12.00 / User / Month',
                                    originalFullPriceElements: [
                                        '$12.00',
                                        '/ User',
                                        '/ Month'
                                    ],
                                    meteredUsage: 'For 1 User And Above'
                                }
                            }
                        }]
                    }]
                },
                setupFees: [],
                contractFees: [],
                contractLength: null,
                formattedContract: false,
                discount: false,
                uuid: '4e6237b7-7a74-4ee5-b050-aaafed2e44e9'
            }],
            bullets: [{
                content: 'Unlimited cloud storage (or 1TB per user if fewer than 5 users)',
                tooltip: null
            }, {
                content: 'Smart search across G Suite with Cloud Search',
                tooltip: null
            }, {
                content: 'Archive and set retention policies for emails and chats',
                tooltip: null
            }, {
                content: 'eDiscovery for emails, chats, and files',
                tooltip: null
            }, {
                content: 'Audit reports to track user activity',
                tooltip: null
            }, {
                content: '*This is a monthly-billed plan, with no contract.',
                tooltip: null
            }],
            callToAction: {
                isPrimary: false,
                isSecondary: true
            },
            audience: [{
                content: 'Enhanced office suite with unlimited storage and archiving'
            }],
            items: [],
            invisible: false,
            leadGen: false
        },
        selectedAddons: [{
            name: 'Google Vault',
            edition: [{
                name: 'Google Vault',
                id: '395056',
                primary: true,
                plans: [{
                    costs: {
                        unitPrice: [{
                            id: 'USER',
                            usageRange: [{
                                formattedPrice: '$4.17',
                                formattedFullPrice: '$4.17 / User / Month',
                                formattedFullPriceElements: [
                                    '$4.17',
                                    '/ User',
                                    '/ Month'
                                ],
                                unit: {
                                    min: 1,
                                    max: null,
                                    label: 'User',
                                    dependency: '',
                                    id: 'USER',
                                    limitLabel: 'For 1 User And Above'
                                },
                                frequency: {
                                    shortLabel: 'Mo',
                                    label: 'Month',
                                    id: 'MONTHLY'
                                },
                                price: {
                                    symbol: '$',
                                    currency: 'USD',
                                    amount: 4.17,
                                    value: 4.17,
                                    originalValue: 4.17,
                                    discountValue: 0,
                                    editionStartingPrice: 4.17,
                                    formatted: {
                                        originalPrice: '$4.17',
                                        originalFullPrice: '$4.17 / User / Month',
                                        originalFullPriceElements: [
                                            '$4.17',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        meteredUsage: 'For 1 User And Above'
                                    }
                                }
                            }]
                        }]
                    },
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: 'a3c690ec-082b-47a0-bf30-029b0a881df8'
                }, {
                    costs: {
                        unitPrice: [{
                            id: 'USER',
                            usageRange: [{
                                formattedPrice: '$50.00',
                                formattedFullPrice: '$50.00 / User / Year',
                                formattedFullPriceElements: [
                                    '$50.00',
                                    '/ User',
                                    '/ Year'
                                ],
                                unit: {
                                    min: 1,
                                    max: null,
                                    label: 'User',
                                    dependency: '',
                                    id: 'USER',
                                    limitLabel: 'For 1 User And Above'
                                },
                                frequency: {
                                    shortLabel: 'Yr',
                                    label: 'Year',
                                    id: 'YEARLY'
                                },
                                price: {
                                    symbol: '$',
                                    currency: 'USD',
                                    amount: 50,
                                    value: 50,
                                    originalValue: 50,
                                    discountValue: 0,
                                    editionStartingPrice: 50,
                                    formatted: {
                                        originalPrice: '$50.00',
                                        originalFullPrice: '$50.00 / User / Year',
                                        originalFullPriceElements: [
                                            '$50.00',
                                            '/ User',
                                            '/ Year'
                                        ],
                                        meteredUsage: 'For 1 User And Above'
                                    }
                                }
                            }]
                        }]
                    },
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: '5887a3e8-6cdf-44ec-9055-b2761dd88217'
                }],
                bullets: [{
                    content: 'Manage, retain, search and export your organization’s email and on-the-record chats. Vault is included with G Suite\'s Business and Enterprise editions.',
                    tooltip: null
                }, {
                    content: 'Control how long your email messages and on-the-record chats are retained. Specify policies for your entire domain or based on organizational units, date ranges and specific terms.',
                    tooltip: null
                }, {
                    content: 'Find and retrieve valuable information, even from closed accounts. It’s a smart way to gather what your legal team needs and prevent data loss from employee turnover.',
                    tooltip: null
                }],
                callToAction: {
                    isPrimary: false,
                    isSecondary: true
                },
                audience: [],
                items: [],
                invisible: false,
                leadGen: false
            }]
        }, {
            name: 'Google Drive storage upgrade - 50 GB',
            edition: [{
                name: 'Google Drive storage upgrade - 50 GB',
                id: '395072',
                primary: true,
                plans: [{
                    costs: {
                        unitPrice: [{
                            id: 'USER',
                            usageRange: [{
                                formattedPrice: '$6.25',
                                formattedFullPrice: '$6.25 / User / Month',
                                formattedFullPriceElements: [
                                    '$6.25',
                                    '/ User',
                                    '/ Month'
                                ],
                                unit: {
                                    min: 1,
                                    max: null,
                                    label: 'User',
                                    dependency: '',
                                    id: 'USER',
                                    limitLabel: 'For 1 User And Above'
                                },
                                frequency: {
                                    shortLabel: 'Mo',
                                    label: 'Month',
                                    id: 'MONTHLY'
                                },
                                price: {
                                    symbol: '$',
                                    currency: 'USD',
                                    amount: 6.25,
                                    value: 6.25,
                                    originalValue: 6.25,
                                    discountValue: 0,
                                    editionStartingPrice: 6.25,
                                    formatted: {
                                        originalPrice: '$6.25',
                                        originalFullPrice: '$6.25 / User / Month',
                                        originalFullPriceElements: [
                                            '$6.25',
                                            '/ User',
                                            '/ Month'
                                        ],
                                        meteredUsage: 'For 1 User And Above'
                                    }
                                }
                            }]
                        }]
                    },
                    setupFees: [],
                    contractFees: [],
                    contractLength: null,
                    formattedContract: false,
                    discount: false,
                    uuid: '46075d33-fd69-40c5-9ec5-748c0f832f34'
                }],
                bullets: [{
                    content: 'Your storage is shared across Google Drive, Gmail, and Google Photos',
                    tooltip: null
                }, {
                    content: 'Get unlimited storage with G Suite Business or Enterprise plans',
                    tooltip: null
                }],
                callToAction: {
                    isPrimary: false,
                    isSecondary: true
                },
                audience: [],
                items: [],
                invisible: false,
                leadGen: false
            }]
        }],
        backAction: {
            label: 'Back to Office 365',
            url: '/en-US/apps/577/office/editions'
        },
        i18n: {
            selectedEdition: 'Selected Edition',
            selectedAddons: 'Selected Add-ons',
            startingAt: 'Starting at'
        }
    },
    callToActions: {
        items: [
            {
                actionType: 'BUY_NOW',
                enabled: true,
                popup: {},
                isPrimary: true,
                label: 'Buy Now',
                tooltip: '',
                url: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=1&type=BUY_NOW',
                urlCreator: {
                    base: 'http: //marketplace.com/api/internal/storefront/v1/cta',
                    actionType: 'BUY_NOW',
                    editionParam: 'editionId',
                    addonParam: 'addonId'
                }
            },
            {
                label: 'Manage App',
                tooltip: '',
                isPrimary: false,
                enabled: true,
                url: 'https://www.google.ca',
                actionType: 'MANAGE_APP',
                popup: {}
            },
            {
                label: 'Save Changes',
                tooltip: '',
                isPrimary: false,
                enabled: true,
                url: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=273407&type=UPDATE_CART_ITEM&editionId=493724&cartItemId=4cf04569-3cc7-44ee-b823-b34f4a840b6c',
                actionType: 'UPDATE_CART_ITEM',
                popup: {},
                cartItemId: '4cf04569-3cc7-44ee-b823-b34f4a840b6c',
                urlCreator: {
                    base: 'http: //marketplace.com/api/internal/storefront/v1/cta?productId=273407&type=UPDATE_CART_ITEM&cartItemId=4cf04569-3cc7-44ee-b823-b34f4a840b6c',
                    actionType: 'UPDATE_CART_ITEM',
                    editionParam: 'eppid',
                    addonParam: 'appid'
                }
            }
        ]
    }
};
