// @appdirect/sfb-theme-components/mocks/product/faq.js
export const featuresMock = {
    faqGroups: {
        items: [
            {
                name: 'FAQs',

                faqs: [
                    {
                        answer: '"In order to purchase an application, please visit the Application Marketplace and click on the profile of the application you would like to purchase. From the product profile, you can either select a free trial or drill down to the Editions and Pricing to select a subscription level that best suits your needs."',
                        question: 'What is AppDirect marketplace?'
                    },
                    {
                        answer: 'In order to list your product with AppDirect, please visit the AppDirect Developer Portal to learn about our API and click-on “Integrate Now” to begin your integration. Please reach out to us at developer@appdirect.com should you have any additional questions.',
                        question: 'As an ISV, how do I list my product with AppDirect?'
                    },
                    {
                        answer: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
                        question: 'Can I purchase an application through the AppDirect marketplace?'
                    },
                    {
                        answer: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
                        question: 'What is App direct?'
                    }
                ]
            },
            {
                name: 'General Questions',
                faqs: [
                    {
                        question: 'What is the difference between AppDirect and other application marketplaces?',
                        answer: 'AppDirect allows users to find applications to suit their Small Business needs in a curated environment with applications broken out by function and industry. The business marketplace allows ISVs to distribute their applications to a broad set of end-users both through AppDirect’s marketplace and through our numerous channel partnerships. In addition, through ISV integration with AppDirect, end-users can manage their cloud solutions directly via the marketplace allowing administrators to quickly provision and deprovision users, upgrade and downgrade subscriptions and cancel subscriptions.'
                    },
                    {
                        question: 'How do I sign-up for AppDirect?',
                        answer: 'In order to sign-up for AppDirect, please click Sign-Up on the AppDirect Homepage where you will be prompted to enter in your email address. You will then receive an email regarding sign-up instructions.'
                    },
                    {
                        question: 'How do I purchase an application through the AppDirect marketplace?',
                        answer: 'In order to purchase an application, please visit the Application Marketplace and click on the profile of the application you would like to purchase. From the product profile, you can either select a free trial or drill down to the Editions and Pricing to select a subscription level that best suits your needs.'
                    }

                ]
            },
            {
                name: 'ISV Questions',
                faqs: [
                    {
                        question: 'As an ISV, why should I list my application with AppDirect?',
                        answer: 'The AppDirect business marketplace allows ISVs to distribute their applications to a broad set of end-users both through AppDirect’s marketplace and through our numerous channel partnerships. With one integration, ISVs are able to access AppDirect’s network of channel partnerships.'
                    },
                    {
                        question: 'As an ISV, how do I sell my product with AppDirect?',
                        answer: 'In order to list your product with AppDirect, please visit the AppDirect Developer Portal to learn about our API and click-on “Integrate Now” to begin your integration. Please reach out to us at developer@appdirect.com should you have any additional questions.'
                    },
                    {
                        question: 'Does AppDirect have an API? Where can I find the documentation to this API?',
                        answer: 'AppDirect uses a RESTful API and Open ID to allow for Single Sign-On and real time subscription and access management. Orders, invoices and payments are handled by AppDirect and no billing integration work is necessary. To view our API documentation, please visit http://info.appdirect.com/developer/docs/getting-started/introduction.'
                    }
                ]
            },
            {
                name: 'Support',
                faqs: [
                    {
                        question: 'Where can I talk to someone within AppDirect Support?',
                        answer: 'If you have any questions, please feel free to access the Help Center near the bottom of the Marketplace or contact us toll free at (777) 404-APPS.'
                    }
                ]
            }
        ],
        i18n: {
            title: 'Frequently asked questions',
            navTitle: 'Categories'
        }
    }
};
