// @appdirect/sfb-theme-components/mocks/product/navigation.js
import { productBannerMock } from './productBanner';
import { callToActionsReferable, callToActionsBuy } from './callToActions';
import { i18nProductNavigationMock } from './navigation/i18n';

export const navigationMock = {
    ...productBannerMock,
    navigation: {
        items: [
            {
                anchor: 'product-section-reviews',
                href: '/en-US/apps/10030/appinsights/reviews',
                label: 'Reviews',
                selected: false,
                value: 'reviews'
            },
            {
                anchor: 'product-section-overview',
                href: '/en-US/apps/10030/appinsights/overview',
                label: 'Overview',
                selected: true,
                value: 'overview'
            },
            {
                anchor: 'product-section-features',
                href: '/en-US/apps/10030/appinsights/features',
                label: 'Features',
                selected: false,
                value: 'features'
            }
        ],
        i18n: i18nProductNavigationMock
    },
    rootConfig: {
        canRouteToV2: true
    }
};

export const navigationMockReferableCta = {
    ...productBannerMock,
    navigation: {
        items: [
            {
                anchor: 'product-section-reviews',
                href: '/en-US/apps/10030/appinsights/reviews',
                label: 'Reviews',
                selected: false,
                value: 'reviews'
            },
            {
                anchor: 'product-section-overview',
                href: '/en-US/apps/10030/appinsights/overview',
                label: 'Overview',
                selected: true,
                value: 'overview'
            },
            {
                anchor: 'product-section-features',
                href: '/en-US/apps/10030/appinsights/features',
                label: 'Features',
                selected: false,
                value: 'features'
            }
        ],
        i18n: i18nProductNavigationMock
    },
    config: {
        referUrl: 'https://appdirect.com'
    },
    callToActions: callToActionsReferable.callToActions,
    rootConfig: {
        canRouteToV2: true
    }
};

export const navigationMockBuyNowCta = {
    ...productBannerMock,
    navigation: {
        items: [
            {
                anchor: 'product-section-reviews',
                href: '/en-US/apps/10030/appinsights/reviews',
                label: 'Reviews',
                selected: false,
                value: 'reviews'
            },
            {
                anchor: 'product-section-overview',
                href: '/en-US/apps/10030/appinsights/overview',
                label: 'Overview',
                selected: true,
                value: 'overview'
            },
            {
                anchor: 'product-section-features',
                href: '/en-US/apps/10030/appinsights/features',
                label: 'Features',
                selected: false,
                value: 'features'
            }
        ],
        i18n: i18nProductNavigationMock
    },
    config: {
        referUrl: 'https://appdirect.com'
    },
    callToActions: callToActionsBuy.callToActions,
    rootConfig: {
        canRouteToV2: true
    }
};
