// @appdirect/sfb-theme-components/mocks/product/productFilters.js
// mock window.location.search
export const searchStringMock = '?scat=2237&cat=428&cat=3203&platform=606755a8-86bd-409f-ba26-40605456cf6c&attr=926&filter=DISCOUNT&order=ALPHABETICAL&q=asd';

export const productFiltersMock = {
    items: [{
        shallow: true,
        id: 'attributes',
        text: 'Attributes',
        items: [{
            queryKey: 'filter',
            id: 'FEATURED',
            text: 'Featured Apps',
            url: '/en-US/listing?filter=FEATURED',
            selected: false
        }, {
            queryKey: 'filter',
            id: 'FREE',
            text: 'Free',
            url: '/en-US/listing?filter=FREE',
            selected: false
        }, {
            queryKey: 'filter',
            id: 'DISCOUNT',
            text: 'Discount Available',
            url: '/en-US/listing?filter=DISCOUNT',
            selected: false
        }, {
            queryKey: 'filter',
            id: 'FREE_TRIAL',
            text: 'Free Trial Available',
            url: '/en-US/listing?filter=FREE_TRIAL',
            selected: false
        }]
    }, {
        shallow: false,
        id: 'categories',
        text: 'Categories',
        items: [{
            queryKey: 'cat',
            id: 422,
            text: 'Sales',
            url: '/en-US/listing?cat=422',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 2221,
                text: 'Configure Price Quote',
                url: '/en-US/listing?scat=2221',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2222,
                text: 'Contact Management',
                url: '/en-US/listing?scat=2222',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2223,
                text: 'Customer Relationship Management',
                url: '/en-US/listing?scat=2223',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2224,
                text: 'Lead Management',
                url: '/en-US/listing?scat=2224',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2225,
                text: 'Predictive Lead Scoring',
                url: '/en-US/listing?scat=2225',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2226,
                text: 'Quoting',
                url: '/en-US/listing?scat=2226',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2227,
                text: 'Sales Commission',
                url: '/en-US/listing?scat=2227',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2228,
                text: 'Sales Enablement',
                url: '/en-US/listing?scat=2228',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2229,
                text: 'Sales Force Automation',
                url: '/en-US/listing?scat=2229',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2268,
                text: 'Sales Forecasting & Analytics',
                url: '/en-US/listing?scat=2268',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2298,
                text: 'Sales Management',
                url: '/en-US/listing?scat=2298',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 423,
            text: 'IT Management',
            url: '/en-US/listing?cat=423',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 2230,
                text: 'Email Hosting',
                url: '/en-US/listing?scat=2230',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2231,
                text: 'Security',
                url: '/en-US/listing?scat=2231',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2233,
                text: 'Domain Name Management',
                url: '/en-US/listing?scat=2233',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3531,
                text: 'Application Builder',
                url: '/en-US/listing?scat=3531',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3535,
                text: 'Application Monitoring & Performance',
                url: '/en-US/listing?scat=3535',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3539,
                text: 'Bug & Issue Tracking',
                url: '/en-US/listing?scat=3539',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3543,
                text: 'Data Management',
                url: '/en-US/listing?scat=3543',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3547,
                text: 'Electronic Data Capture',
                url: '/en-US/listing?scat=3547',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3551,
                text: 'IT, Server, & Network Monitoring',
                url: '/en-US/listing?scat=3551',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3555,
                text: 'System Administration',
                url: '/en-US/listing?scat=3555',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3559,
                text: 'IT Asset Management',
                url: '/en-US/listing?scat=3559',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 424,
            text: 'Online Backup & Storage',
            url: '/en-US/listing?cat=424',
            selected: false,
            items: []
        }, {
            queryKey: 'cat',
            id: 425,
            text: 'Customer Management',
            url: '/en-US/listing?cat=425',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 2234,
                text: 'Customer Relationship Management',
                url: '/en-US/listing?scat=2234',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2235,
                text: 'Call Management',
                url: '/en-US/listing?scat=2235',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2236,
                text: 'Live Chat',
                url: '/en-US/listing?scat=2236',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2237,
                text: 'Scheduling & Appointments',
                url: '/en-US/listing?scat=2237',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3563,
                text: 'Customer Engagement',
                url: '/en-US/listing?scat=3563',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3567,
                text: 'Customer Loyalty and Gamification',
                url: '/en-US/listing?scat=3567',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3571,
                text: 'Point of Sale',
                url: '/en-US/listing?scat=3571',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3575,
                text: 'Social CRM',
                url: '/en-US/listing?scat=3575',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3579,
                text: 'Survey',
                url: '/en-US/listing?scat=3579',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 426,
            text: 'Human Resources',
            url: '/en-US/listing?cat=426',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 2238,
                text: 'Time Tracking',
                url: '/en-US/listing?scat=2238',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2239,
                text: 'Timesheets & Vacation Management',
                url: '/en-US/listing?scat=2239',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2240,
                text: 'Recruitment',
                url: '/en-US/listing?scat=2240',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2241,
                text: 'Employee Directory',
                url: '/en-US/listing?scat=2241',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2242,
                text: 'Performance Management',
                url: '/en-US/listing?scat=2242',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3631,
                text: 'Talent Management',
                url: '/en-US/listing?scat=3631',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3635,
                text: 'Training',
                url: '/en-US/listing?scat=3635',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3639,
                text: 'Learning Management System',
                url: '/en-US/listing?scat=3639',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3643,
                text: 'Workforce Management',
                url: '/en-US/listing?scat=3643',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 427,
            text: 'Finance',
            url: '/en-US/listing?cat=427',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 2243,
                text: 'Accounting',
                url: '/en-US/listing?scat=2243',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2244,
                text: 'Billing & Invoicing',
                url: '/en-US/listing?scat=2244',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2245,
                text: 'Expense Management',
                url: '/en-US/listing?scat=2245',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2246,
                text: 'Financial Analytics & Reporting',
                url: '/en-US/listing?scat=2246',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2247,
                text: 'Payroll Management',
                url: '/en-US/listing?scat=2247',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3607,
                text: 'Accounts Payable',
                url: '/en-US/listing?scat=3607',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3611,
                text: 'Accounts Receivable',
                url: '/en-US/listing?scat=3611',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3615,
                text: 'Budgeting',
                url: '/en-US/listing?scat=3615',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3619,
                text: 'Compliance',
                url: '/en-US/listing?scat=3619',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3623,
                text: 'Payment Management',
                url: '/en-US/listing?scat=3623',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3627,
                text: 'Purchasing',
                url: '/en-US/listing?scat=3627',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 428,
            text: 'Developer Tools',
            url: '/en-US/listing?cat=428',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 2248,
                text: 'Cloud Security and Monitoring',
                url: '/en-US/listing?scat=2248',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2249,
                text: 'Performance Management',
                url: '/en-US/listing?scat=2249',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2250,
                text: 'Mobile Development',
                url: '/en-US/listing?scat=2250',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2251,
                text: 'User Management',
                url: '/en-US/listing?scat=2251',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2252,
                text: 'Email Delivery',
                url: '/en-US/listing?scat=2252',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2253,
                text: 'Development and Test Tools',
                url: '/en-US/listing?scat=2253',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2254,
                text: 'Cloud Databases',
                url: '/en-US/listing?scat=2254',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2255,
                text: 'Cloud Deployment & Management',
                url: '/en-US/listing?scat=2255',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2917,
                text: 'Open Source',
                url: '/en-US/listing?scat=2917',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 429,
            text: 'Office Productivity',
            url: '/en-US/listing?cat=429',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 2256,
                text: 'Project Management',
                url: '/en-US/listing?scat=2256',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2257,
                text: 'File sharing & Document Management',
                url: '/en-US/listing?scat=2257',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2258,
                text: 'Office Tools',
                url: '/en-US/listing?scat=2258',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2259,
                text: 'Meeting Management',
                url: '/en-US/listing?scat=2259',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2260,
                text: 'Email',
                url: '/en-US/listing?scat=2260',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2266,
                text: 'Surveys',
                url: '/en-US/listing?scat=2266',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2267,
                text: 'Multimedia',
                url: '/en-US/listing?scat=2267',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 430,
            text: 'Communication',
            url: '/en-US/listing?cat=430',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 2261,
                text: 'Conferencing',
                url: '/en-US/listing?scat=2261',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2262,
                text: 'Email',
                url: '/en-US/listing?scat=2262',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2263,
                text: 'Call Management',
                url: '/en-US/listing?scat=2263',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2264,
                text: 'Live Chat',
                url: '/en-US/listing?scat=2264',
                selected: false
            }, {
                queryKey: 'scat',
                id: 2265,
                text: 'Social Media',
                url: '/en-US/listing?scat=2265',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3647,
                text: 'Telephony',
                url: '/en-US/listing?scat=3647',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3651,
                text: 'Mobile',
                url: '/en-US/listing?scat=3651',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3655,
                text: 'VoIP',
                url: '/en-US/listing?scat=3655',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 997,
            text: 'BI & Analytics',
            url: '/en-US/listing?cat=997',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 2377,
                text: 'Analytics',
                url: '/en-US/listing?scat=2377',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3515,
                text: 'Business Intelligence',
                url: '/en-US/listing?scat=3515',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3519,
                text: 'Predictive Analytics',
                url: '/en-US/listing?scat=3519',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3523,
                text: 'Reporting & Dashboard',
                url: '/en-US/listing?scat=3523',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3527,
                text: 'Web & Mobile Analytics',
                url: '/en-US/listing?scat=3527',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 3203,
            text: 'Customer Service',
            url: '/en-US/listing?cat=3203',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 3583,
                text: 'Call Center',
                url: '/en-US/listing?scat=3583',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3587,
                text: 'Feedback and Reviews Management',
                url: '/en-US/listing?scat=3587',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3591,
                text: 'Help Desk & Ticketing',
                url: '/en-US/listing?scat=3591',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3595,
                text: 'IT Service Management',
                url: '/en-US/listing?scat=3595',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3599,
                text: 'Live Chat',
                url: '/en-US/listing?scat=3599',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3603,
                text: 'Remote Support',
                url: '/en-US/listing?scat=3603',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 3207,
            text: 'Marketing',
            url: '/en-US/listing?cat=3207',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 3659,
                text: 'Advertising',
                url: '/en-US/listing?scat=3659',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3663,
                text: 'Brand Management',
                url: '/en-US/listing?scat=3663',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3667,
                text: 'Campaign Management',
                url: '/en-US/listing?scat=3667',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3671,
                text: 'Content Marketing',
                url: '/en-US/listing?scat=3671',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3675,
                text: 'Email Marketing',
                url: '/en-US/listing?scat=3675',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3679,
                text: 'Marketing Analytics',
                url: '/en-US/listing?scat=3679',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3683,
                text: 'Marketing Automation',
                url: '/en-US/listing?scat=3683',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3687,
                text: 'Mobile Marketing',
                url: '/en-US/listing?scat=3687',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3691,
                text: 'Personalization and Behavioral Targeting',
                url: '/en-US/listing?scat=3691',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3695,
                text: 'Social Media Marketing',
                url: '/en-US/listing?scat=3695',
                selected: false
            }]
        }, {
            queryKey: 'cat',
            id: 3211,
            text: 'Operations',
            url: '/en-US/listing?cat=3211',
            selected: false,
            items: [{
                queryKey: 'scat',
                id: 3699,
                text: 'Business Management',
                url: '/en-US/listing?scat=3699',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3703,
                text: 'Business Workflow',
                url: '/en-US/listing?scat=3703',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3707,
                text: 'Field Activity & Service Management',
                url: '/en-US/listing?scat=3707',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3711,
                text: 'Inventory Management',
                url: '/en-US/listing?scat=3711',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3715,
                text: 'Order Management',
                url: '/en-US/listing?scat=3715',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3719,
                text: 'Scheduling',
                url: '/en-US/listing?scat=3719',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3723,
                text: 'Service Dispatch',
                url: '/en-US/listing?scat=3723',
                selected: false
            }, {
                queryKey: 'scat',
                id: 3727,
                text: 'Supply Chain Management',
                url: '/en-US/listing?scat=3727',
                selected: false
            }]
        }]
    }, {
        shallow: true,
        id: 'platforms',
        text: 'Platforms',
        items: [{
            queryKey: 'platform',
            id: '0b260e8c-f23b-4cfb-982d-fd0670f0ddc1',
            text: 'Android',
            url: '/en-US/listing?platform=0b260e8c-f23b-4cfb-982d-fd0670f0ddc1',
            selected: false
        }, {
            queryKey: 'platform',
            id: '606755a8-86bd-409f-ba26-40605456cf6c',
            text: 'iOS',
            url: '/en-US/listing?platform=606755a8-86bd-409f-ba26-40605456cf6c',
            selected: false
        }]
    }, {
        shallow: true,
        id: 232,
        text: 'Private Cloud',
        items: [{
            queryKey: 'attr',
            id: 517,
            text: 'Private Cloud',
            url: '/en-US/listing?attr=517',
            selected: false
        }, {
            queryKey: 'attr',
            id: 520,
            text: 'Public Cloud',
            url: '/en-US/listing?attr=520',
            selected: false
        }]
    }, {
        shallow: true,
        id: 374,
        text: 'Integration',
        items: [{
            queryKey: 'attr',
            id: 920,
            text: 'Single Sign-On',
            url: '/en-US/listing?attr=920',
            selected: false
        }, {
            queryKey: 'attr',
            id: 926,
            text: 'Data Visualization',
            url: '/en-US/listing?attr=926',
            selected: false
        }, {
            queryKey: 'attr',
            id: 932,
            text: 'Buyable',
            url: '/en-US/listing?attr=932',
            selected: false
        }, {
            queryKey: 'attr',
            id: 938,
            text: 'Importable',
            url: '/en-US/listing?attr=938',
            selected: false
        }, {
            queryKey: 'attr',
            id: 941,
            text: 'User Provisioning',
            url: '/en-US/listing?attr=941',
            selected: false
        }, {
            queryKey: 'attr',
            id: 2000,
            text: 'Search',
            url: '/en-US/listing?attr=2000',
            selected: false
        }]
    }, {
        shallow: true,
        id: 1276,
        text: 'Catalog',
        items: [{
            queryKey: 'attr',
            id: 4320,
            text: 'Core',
            url: '/en-US/listing?attr=4320',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4334,
            text: 'Mobility',
            url: '/en-US/listing?attr=4334',
            selected: false
        }]
    }, {
        shallow: true,
        id: 1350,
        text: 'Language availability',
        items: [{
            queryKey: 'attr',
            id: 4606,
            text: 'English (US)',
            url: '/en-US/listing?attr=4606',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4610,
            text: 'Chinese',
            url: '/en-US/listing?attr=4610',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4618,
            text: 'Danish',
            url: '/en-US/listing?attr=4618',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4626,
            text: 'French',
            url: '/en-US/listing?attr=4626',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4630,
            text: 'Japanese',
            url: '/en-US/listing?attr=4630',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4634,
            text: 'Finnish',
            url: '/en-US/listing?attr=4634',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4638,
            text: 'German',
            url: '/en-US/listing?attr=4638',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4642,
            text: 'Italian',
            url: '/en-US/listing?attr=4642',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4646,
            text: 'Polish',
            url: '/en-US/listing?attr=4646',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4650,
            text: 'Portuguese',
            url: '/en-US/listing?attr=4650',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4654,
            text: 'Russian',
            url: '/en-US/listing?attr=4654',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4658,
            text: 'Spanish (European)',
            url: '/en-US/listing?attr=4658',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4662,
            text: 'Swedish',
            url: '/en-US/listing?attr=4662',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4666,
            text: 'Turkish',
            url: '/en-US/listing?attr=4666',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4670,
            text: 'Czech',
            url: '/en-US/listing?attr=4670',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4674,
            text: 'English (AU)',
            url: '/en-US/listing?attr=4674',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4678,
            text: 'English (UK)',
            url: '/en-US/listing?attr=4678',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4682,
            text: 'Spanish (Mexican)',
            url: '/en-US/listing?attr=4682',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4686,
            text: 'Dutch',
            url: '/en-US/listing?attr=4686',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4690,
            text: 'Mandarin',
            url: '/en-US/listing?attr=4690',
            selected: false
        }, {
            queryKey: 'attr',
            id: 4694,
            text: 'Greek',
            url: '/en-US/listing?attr=4694',
            selected: false
        }]
    }],
    i18n: {
        filters: 'Filters',
        showFilters: 'Show Filters',
        hideFilters: 'Hide Filters',
        clearFilters: 'Clear Filters',
        ratings: 'Ratings',
        unrated: 'Unrated',
        done: 'Done'
    }
};
