// @appdirect/sfb-theme-components/mocks/product/productGroupsTiles.js
import { getArgs } from '../../src/tools/controlsHelper';
import CardStyle from '../../src/constants/atoms/schema/product-groups-tiles/card-style';
import { IMAGE_URL } from '../../src/constants/images';

export const defaultValues = {
    customImage: {
        defaultValue: {
            url: IMAGE_URL
        }
    },
    upperTitle: 'Upper title',
    title: 'Title',
    description: 'Description...',
    buttonLabel: 'Button Text'
};

export const productGroupsTilesMocks = {
    data: [{
        ...getArgs(CardStyle.schema().form),
        key: '1a',
        cardBorderColor: 'red',
        link: 'https://www.google.com',
        customLink: 'https://www.customLink1.com',
        ...defaultValues
    }, {
        ...getArgs(CardStyle.schema().form),
        key: '2b',
        cardBorderColor: 'blue',
        link: 'https://www.google.com',
        customLink: 'https://www.customLink2.com',
        ...defaultValues
    }, {
        ...getArgs(CardStyle.schema().form),
        key: '3c',
        cardBorderColor: 'green',
        link: 'https://www.google.com',
        customLink: 'https://www.customLink3.com',
        ...defaultValues
    }, {
        ...getArgs(CardStyle.schema().form),
        key: '4d',
        cardBorderColor: 'yellow',
        link: 'https://www.google.com',
        customLink: 'https://www.customLink4.com',
        ...defaultValues
    }, {
        ...getArgs(CardStyle.schema().form),
        key: '5e',
        cardBorderColor: 'pink',
        link: 'https://www.google.com',
        customLink: 'https://www.customLink5.com',
        ...defaultValues
    }, {
        ...getArgs(CardStyle.schema().form),
        key: '6f',
        cardBorderColor: 'orange',
        link: 'https://www.google.com',
        customLink: 'https://www.customLink6.com',
        ...defaultValues
    }]
};
