// @appdirect/sfb-theme-components/mocks/product/productReviewsGraphQL.js
export const productReviewsGraphQLMock = {
    data: {
        reviews: {
            nodes: [{
                reviewAuthor: {
                    company: {
                        type: 'EDUCATION',
                        name: 'AppDirect'
                    },
                    name: 'bob0'
                },
                reviewTags: [{
                    description: 'Invitation from G2. This reviewer was offered a nominal gift card as thank you for completing this review.',
                    label: 'Invitation from G2',
                    type: 'SYNDICATOR_REVIEW_SOURCE'
                }],
                reviewUrl: 'https://www.g2.com/products/slack/reviews/slack-review-5298626',
                reviewLastModified: '2020-10-29T14:45:26.58Z',
                syndication: {
                    externalProvider: 'G2'
                },
                id: '62bbaa0c-22ec-4ff8-a608-f3e6df59192c',
                language: 'fr',
                providerReviewId: '0a',
                rating: 4,
                sections: [
                    {
                        content: 'I love how easily accessible Microsoft 365 is. It can be accessed anywhere on the go, and on any platform. It is extremely easy to share documents among coworkers, where we are all able to edit it in real time.',
                        title: 'What is good?'
                    },
                    {
                        content: 'I prefer the layout of the original Microsoft Office. Microsoft 365',
                        title: 'What do you dislike?'
                    },
                    {
                        content: 'Microsoft 365 has helped my department stay better connected with one another, and given us the ability to easily share documents and reports amongst one another.',
                        title: 'What problems are you solving with the product? What benefits have you realized?'
                    }
                ],
                title: 'Microsoft 365 is a great tool for any office environment'
            }, {
                reviewAuthor: {
                    company: {
                        name: 'AppDirect'
                    },
                    name: 'bob1'
                },
                reviewTags: [{
                    description: 'Verified Current User',
                    label: 'Verified Current User',
                    type: 'SYNDICATOR_REVIEW_SOURCE'
                }],
                reviewUrl: 'http://dsada.com',
                reviewLastModified: '2020-10-29T14:45:26.6Z',
                syndication: {
                    externalProvider: 'Provider Syndication Name'
                },
                id: '56d7b896-b03b-4a0e-b673-fb15351db86c',
                language: 'fr',
                providerReviewId: '1a',
                rating: 4,
                sections: [
                    {
                        content: 'Hook up your SQL DB no problems.',
                        title: 'What is good?'
                    }
                ],
                title: 'Hook up your SQL DB no problems'
            }],
            pageInfo: {
                startCursor: '0',
                endCursor: '1',
                hasNextPage: true
            },
            totalCount: 2
        }
    }
};
