// @appdirect/sfb-theme-components/mocks/product/productSimilarApps.js
export const similarAppsMockData = {
    recommendations: {
        items: [
            {
                id: 473,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/473/thumbs_64/img993721259211857590.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/473/thumbs_64/img993721259211857590.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/473/thumbs_112/img4507210906661651287.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/473/overview/UGwrIe5UKx7fbv98.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: '61f3504c-ae52-11e3-ac06-22000a1dd446',
                description: 'With WORKetc, manage every aspect of your business from organizing contacts and gathering leads to collaborating on projects and sending out the final bill.  Gmail & Outlook integration.  Native Android, iPhone and mobile web apps.',
                overview: 'WORKetc combines CRM, project management, billing, sales, help desk and\nmore with a powerful automation engine to drive your small business to\nthe next level.',
                vendorName: 'WORKetc CRM',
                showRating: true,
                hidePricings: false,
                free: false,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: true,
                discounted: false,
                pending: false,
                name: 'WORKetc CRM + Projects + Billing',
                blurb: 'Complete small business management platform.',
                url: '/en-US/apps/473/worketc-crm-projects-billing',
                startingPrice: {
                    formattedFullPrice: '$195.00/Mo',
                    formattedPrice: '$195.00',
                    price: '195.00',
                    priceCaption: 'Starting At',
                    frequency: {
                        label: 'Month',
                        shortLabel: 'Mo',
                        id: 'MONTHLY'
                    },
                    symbol: '$',
                    currency: 'USD'
                },
                tags: [
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2256,
                                name: 'Project Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    }
                ],
                rating: 5,
                ratingsSummary: {
                    title: 'Rating',
                    caption: '5/5 Based on 3 reviews'
                },
                numRatings: 3,
                badges: [
                    {
                        id: 'FREE_TRIAL',
                        label: 'Free Trial',
                        type: 'success'
                    }
                ]
            },
            {
                id: 484,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/484/thumbs_64/img1770749016292536480.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/484/thumbs_64/img1770749016292536480.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/484/thumbs_112/img1631325241161507357.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/484/overview/iEtlD63l2mnUeNNb.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: '61f36078-ae52-11e3-ac06-22000a1dd446',
                description: 'MindQuilt connects your questions with answers, documents, or people',
                overview: 'MindQuilt provides a question and answer platform for your company or organization. With MindQuilt, you have access to your entire organization\'s knowledge base.',
                vendorName: 'MindQuilt',
                showRating: true,
                hidePricings: false,
                free: false,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: true,
                discounted: false,
                pending: false,
                name: 'MindQuilt',
                blurb: 'Connects People with Answers',
                url: '/en-US/apps/484/mindquilt',
                startingPrice: {
                    formattedFullPrice: '$25.00/Mo',
                    formattedPrice: '$25.00',
                    price: '25.00',
                    priceCaption: 'Starting At',
                    frequency: {
                        label: 'Month',
                        shortLabel: 'Mo',
                        id: 'MONTHLY'
                    },
                    symbol: '$',
                    currency: 'USD'
                },
                tags: [
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2257,
                                name: 'File sharing & Document Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 2258,
                                name: 'Office Tools',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 374,
                        name: 'Integration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 920,
                                name: 'Single Sign-On',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 932,
                                name: 'Buyable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 938,
                                name: 'Importable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 941,
                                name: 'User Provisioning',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 327,
                        name: 'Supported Devices',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 811,
                                name: 'Android',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 827,
                                name: 'Mac',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 835,
                                name: 'Web-based',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 839,
                                name: 'Windows',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    }
                ],
                rating: 5,
                ratingsSummary: {
                    title: 'Rating',
                    caption: '5/5 Based on 2 reviews'
                },
                numRatings: 2,
                badges: [
                    {
                        id: 'FREE_TRIAL',
                        label: 'Free Trial',
                        type: 'success'
                    }
                ]
            },
            {
                id: 544,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/544/thumbs_64/img7826440015608931882.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/544/thumbs_64/img7826440015608931882.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/544/thumbs_112/img5194893520778687861.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/544/overview/fH2VJAnSYFWkEpfY.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: '61f3c1da-ae52-11e3-ac06-22000a1dd446',
                description: 'The Galaxie Broadband Player is a simple and cost-effective music solution for businesses. A no-hassle, professional music service for commercial use.',
                overview: 'A turn-key, music service for businesses of all kinds and sizes. 45 channels of commercial free music, 24/7.  Galaxie offers a wide variety of music styles and genres for any mood or any situation.  Use Galaxie in your office, your store, your hotel, your restaurant, your place of business.',
                vendorName: 'Stingray Digital',
                showRating: true,
                hidePricings: false,
                free: false,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: false,
                discounted: false,
                pending: false,
                name: 'Galaxie',
                blurb: 'Online, continuous music service',
                url: '/en-US/apps/544/galaxie',
                startingPrice: {
                    formattedFullPrice: '$4.99/Mo',
                    formattedPrice: '$4.99',
                    price: '4.99',
                    priceCaption: 'Starting At',
                    frequency: {
                        label: 'Month',
                        shortLabel: 'Mo',
                        id: 'MONTHLY'
                    },
                    symbol: '$',
                    currency: 'USD'
                },
                tags: [
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 374,
                        name: 'Integration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 920,
                                name: 'Single Sign-On',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 932,
                                name: 'Buyable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 938,
                                name: 'Importable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 941,
                                name: 'User Provisioning',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 327,
                        name: 'Supported Devices',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 811,
                                name: 'Android',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 827,
                                name: 'Mac',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 835,
                                name: 'Web-based',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 839,
                                name: 'Windows',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2267,
                                name: 'Multimedia',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    }
                ],
                rating: 5,
                ratingsSummary: {
                    title: 'Rating',
                    caption: '5/5 Based on 2 reviews'
                },
                numRatings: 2,
                badges: []
            },
            {
                id: 3137,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3137/thumbs_64/img3422786702231527732.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3137/thumbs_64/img3422786702231527732.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3137/thumbs_112/img1698187633467694257.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3137/overview/img7629493808768431412.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: '62053096-ae52-11e3-ac06-22000a1dd446',
                description: 'Forget bookmarks and copy pasting. With Memonic, preparing for your next meeting is a breeze. Note-taking made easy.',
                overview: 'First in Knowledge Curation; Ultimate client information archive; Collaborate in small teams; Integrate in existing solutions',
                vendorName: 'Squirro',
                showRating: true,
                hidePricings: false,
                free: false,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: true,
                discounted: false,
                pending: false,
                name: 'Memonic',
                blurb: 'Note-taking made easy.',
                url: '/en-US/apps/3137/memonic',
                startingPrice: {
                    formattedFullPrice: '$28.00/yr',
                    formattedPrice: '$28.00',
                    price: '28.00',
                    priceCaption: 'Starting At',
                    frequency: {
                        label: 'Year',
                        shortLabel: 'yr',
                        id: 'YEARLY'
                    },
                    symbol: '$',
                    currency: 'USD'
                },
                tags: [
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2258,
                                name: 'Office Tools',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 2259,
                                name: 'Meeting Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 327,
                        name: 'Supported Devices',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 835,
                                name: 'Web-based',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 374,
                        name: 'Integration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 920,
                                name: 'Single Sign-On',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 932,
                                name: 'Buyable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 941,
                                name: 'User Provisioning',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    }
                ],
                rating: 5,
                ratingCaption: '5/5 Based on 1 review',
                numRatings: 1,
                badges: [
                    {
                        id: 'FREE_TRIAL',
                        label: 'Free Trial',
                        type: 'success'
                    }
                ]
            },
            {
                id: 3346,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3346/thumbs_64/img8565224848194976503.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3346/thumbs_64/img8565224848194976503.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3346/thumbs_112/img6329933794560761443.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3346/overview/img4350626711219796221.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: '62067b90-ae52-11e3-ac06-22000a1dd446',
                description: 'TimO - Travel Expense System is a comprehensive an affordable expense tracking System for managing and tracking employee travel expenses.',
                overview: 'This cloud application allows you to track your national and international travel expenses, attach scanned bills and documents and create reporting\'s of travel costs.',
                vendorName: 'Timo-Time Management Office GmbH (SU & Test)',
                showRating: true,
                hidePricings: false,
                free: false,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: true,
                discounted: false,
                pending: false,
                name: 'TimO - Travel Expense System',
                blurb: 'Powerful travel expense tracking and reporting system',
                url: '/en-US/apps/3346/timo---travel-expense-system',
                startingPrice: {
                    formattedFullPrice: '$6.49/Mo',
                    formattedPrice: '$6.49',
                    price: '6.49',
                    priceCaption: 'Starting At',
                    frequency: {
                        label: 'Month',
                        shortLabel: 'Mo',
                        id: 'MONTHLY'
                    },
                    symbol: '$',
                    currency: 'USD'
                },
                tags: [
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2258,
                                name: 'Office Tools',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 327,
                        name: 'Supported Devices',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 811,
                                name: 'Android',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 827,
                                name: 'Mac',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 835,
                                name: 'Web-based',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 839,
                                name: 'Windows',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 374,
                        name: 'Integration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 920,
                                name: 'Single Sign-On',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 932,
                                name: 'Buyable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 941,
                                name: 'User Provisioning',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 427,
                        name: 'Finance',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2245,
                                name: 'Expense Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    }
                ],
                rating: 5,
                ratingCaption: '5/5 Based on 1 review',
                numRatings: 1,
                badges: [
                    {
                        id: 'FREE_TRIAL',
                        label: 'Free Trial',
                        type: 'success'
                    }
                ]
            },
            {
                id: 3374,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3374/thumbs_64/img8161460257882887607.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3374/thumbs_64/img8161460257882887607.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3374/thumbs_112/img9151496744797752500.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/3374/overview/img6542859018406473323.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: '6206a93a-ae52-11e3-ac06-22000a1dd446',
                description: 'TimO is a web-based project management tool with a powerful Gantt Chart,\nCRM, DMS Workflow and billing and automatic time/cost control. It\ncombines project management, CRM, DMS, project time tracking (incl.\nattendance time), travel expense management, issue tracking system and\nmore. This easy to use tool helps small, medium and large companies\nsince 1999 across 50 countries.',
                overview: 'TimO is a web-based project management tool with a powerful Gantt Chart,\nCRM, DMS, Workflow and Project billing and automatic time/cost control.\nIt combines project management, CRM, DMS, project time tracking (incl.\nattendance time), travel expense management, issue tracking system and\nmore. This easy to use tool helps small, medium and large companies\nsince 1999 across 50 countries.',
                vendorName: 'Timo-Time Management Office GmbH (SU & Test)',
                showRating: true,
                hidePricings: false,
                free: false,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: true,
                discounted: false,
                pending: false,
                name: 'TimO Project Management Complete',
                blurb: 'An easy and powerful project management and billing software',
                url: '/en-US/apps/3374/timo-project-management-complete',
                startingPrice: {
                    formattedFullPrice: '$9.99/Mo',
                    formattedPrice: '$9.99',
                    price: '9.99',
                    priceCaption: 'Starting At',
                    frequency: {
                        label: 'Month',
                        shortLabel: 'Mo',
                        id: 'MONTHLY'
                    },
                    symbol: '$',
                    currency: 'USD'
                },
                tags: [
                    {
                        type: 'CATEGORY',
                        id: 427,
                        name: 'Finance',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2244,
                                name: 'Billing & Invoicing',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 2245,
                                name: 'Expense Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 425,
                        name: 'Customer Management',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2234,
                                name: 'Customer Relationship Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 422,
                        name: 'Sales',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2223,
                                name: 'Customer Relationship Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 2225,
                                name: 'Predictive Lead Scoring',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 327,
                        name: 'Supported Devices',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 835,
                                name: 'Web-based',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 374,
                        name: 'Integration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 920,
                                name: 'Single Sign-On',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 932,
                                name: 'Buyable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 941,
                                name: 'User Provisioning',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 426,
                        name: 'Human Resources',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2238,
                                name: 'Time Tracking',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 2239,
                                name: 'Timesheets & Vacation Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2256,
                                name: 'Project Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    }
                ],
                rating: 5,
                ratingCaption: '5/5 Based on 1 review',
                numRatings: 1,
                badges: [
                    {
                        id: 'FREE_TRIAL',
                        label: 'Free Trial',
                        type: 'success'
                    }
                ]
            },
            {
                id: 10030,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
                    '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669-2x.png',
                    '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/overview/img5121553238902495902.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: 'a14a89d1-e62e-4138-b579-2c2a6f79fce7',
                description: 'AppInsights dashboards automatically collect the latest data about your company into one place.',
                overview: 'AppInsights help you make faster and more informed\nbusiness decisions by letting you view your KPIs on a real-time dashboard, all in\none place.',
                vendorName: 'AppInsights',
                showRating: true,
                hidePricings: false,
                free: false,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: true,
                discounted: false,
                pending: false,
                name: 'AppInsights',
                blurb: 'Real-time business dashboard solution',
                url: '/en-US/apps/10030/appinsights',
                startingPrice: {
                    formattedFullPrice: 'Free',
                    formattedPrice: 'Free',
                    price: 'Free',
                    priceCaption: '',
                    frequency: {
                        label: '',
                        shortLabel: '',
                        id: ''
                    }
                },
                tags: [
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2258,
                                name: 'Office Tools',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 1350,
                        name: 'Language availability',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 4606,
                                name: 'English (US)',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: 'Currently available or Core Application Catalog only.'
                    }
                ],
                rating: 5,
                ratingCaption: '5/5 Based on 1 review',
                numRatings: 1,
                badges: [
                    {
                        id: 'FREE_TRIAL',
                        label: 'Free Trial',
                        type: 'success'
                    }
                ]
            },
            {
                id: 42937,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_64/img63732497944352803.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_64/img63732497944352803.png',
                    '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_64/img63732497944352803-2x.png',
                    '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_64/img63732497944352803.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_112/img1099429097555068136.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/overview/img1301783796595200436.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: '622a27d2-09ac-46d1-87e2-3ffa764e8bce',
                description: 'AppWise is a business productivity tool that provides Unified Search across 40+ leading applications like Salesforce, Gmail, Office 365, Box and Asana. AppWise has enterprise-grade security built-in from the ground up, and it\'s intuitive enough be setup by end-users in seconds.',
                overview: 'With Unified Search across over 40+ leading cloud applications, easily find all your emails, documents and tasks in a couple of clicks.',
                vendorName: 'Xendo, Inc.',
                showRating: true,
                hidePricings: false,
                free: true,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: true,
                discounted: false,
                pending: false,
                name: 'AppWise',
                blurb: 'Search everything, in one shot',
                url: '/en-US/apps/42937/appwise',
                startingPrice: {
                    formattedFullPrice: 'Free',
                    formattedPrice: 'Free',
                    price: 'Free',
                    priceCaption: '',
                    frequency: {
                        label: '',
                        shortLabel: '',
                        id: ''
                    }
                },
                tags: [
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 232,
                        name: 'Private Cloud',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 517,
                                name: 'Private Cloud',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 520,
                                name: 'Public Cloud',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 422,
                        name: 'Sales',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2223,
                                name: 'Customer Relationship Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 3203,
                        name: 'Customer Service',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 3591,
                                name: 'Help Desk & Ticketing',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 1350,
                        name: 'Language availability',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 4606,
                                name: 'English (US)',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: 'Currently available or Core Application Catalog only.'
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 327,
                        name: 'Supported Devices',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 811,
                                name: 'Android',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 815,
                                name: 'iPhone',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 819,
                                name: 'iPad',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 823,
                                name: 'Linux',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 827,
                                name: 'Mac',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 831,
                                name: 'Mobile Web App',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 835,
                                name: 'Web-based',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 839,
                                name: 'Windows',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 843,
                                name: 'Windows Phone',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 3207,
                        name: 'Marketing',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 374,
                        name: 'Integration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 920,
                                name: 'Single Sign-On',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 941,
                                name: 'User Provisioning',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    }
                ],
                rating: 5,
                ratingsSummary: {
                    title: 'Rating',
                    caption: '5/5 Based on 139 reviews'
                },
                numRatings: 139,
                badges: []
            },
            {
                id: 53185,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/53185/thumbs_64/img1596724994380261585.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/53185/thumbs_64/img1596724994380261585.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/53185/thumbs_112/img5514891912140543937.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/53185/overview/img112226600830447841.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: 'ddce318a-5e78-4c50-9efd-e9ef922e7d39',
                description: 'NextMinute is the app that your business cant do without. It will save you time and money by helping to better manage your jobs and organise your business, your teams and your operations. \n\nYoull spend less time on paperwork and more time on the job. Designed with trade and field services in mind, NextMinute is an easy-to-use tool that allows you to plan, schedule, communicate, track time, quote, order, monitor, and invoice on the go, anytime, through your mobile, tablet or laptop.',
                overview: 'NextMinute is essential online job management software that allows you to run your mobile work-force more efficiently.',
                vendorName: 'NextMinute',
                showRating: true,
                hidePricings: false,
                free: false,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: true,
                discounted: false,
                pending: false,
                name: 'NextMinute',
                blurb: 'Workforce Job Management',
                url: '/en-US/apps/53185/nextminute',
                startingPrice: {
                    formattedFullPrice: '$28.00/Mo',
                    formattedPrice: '$28.00',
                    price: '28.00',
                    priceCaption: 'Starting At',
                    frequency: {
                        label: 'Month',
                        shortLabel: 'Mo',
                        id: 'MONTHLY'
                    },
                    symbol: '$',
                    currency: 'USD'
                },
                tags: [
                    {
                        type: 'CATEGORY',
                        id: 430,
                        name: 'Communication',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 3651,
                                name: 'Mobile',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 374,
                        name: 'Integration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 920,
                                name: 'Single Sign-On',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 926,
                                name: 'Data Visualization',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 932,
                                name: 'Buyable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 941,
                                name: 'User Provisioning',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 422,
                        name: 'Sales',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2222,
                                name: 'Contact Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 2226,
                                name: 'Quoting',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 425,
                        name: 'Customer Management',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2234,
                                name: 'Customer Relationship Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 2237,
                                name: 'Scheduling & Appointments',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 424,
                        name: 'Online Backup & Storage',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 3211,
                        name: 'Operations',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 3699,
                                name: 'Business Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 3703,
                                name: 'Business Workflow',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 3707,
                                name: 'Field Activity & Service Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 3719,
                                name: 'Scheduling',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 3723,
                                name: 'Service Dispatch',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 327,
                        name: 'Supported Devices',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 811,
                                name: 'Android',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 815,
                                name: 'iPhone',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 819,
                                name: 'iPad',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 823,
                                name: 'Linux',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 827,
                                name: 'Mac',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 831,
                                name: 'Mobile Web App',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 835,
                                name: 'Web-based',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 839,
                                name: 'Windows',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 843,
                                name: 'Windows Phone',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2256,
                                name: 'Project Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 2259,
                                name: 'Meeting Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 997,
                        name: 'BI & Analytics',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 3515,
                                name: 'Business Intelligence',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'SUB_CATEGORY',
                                id: 3523,
                                name: 'Reporting & Dashboard',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    }
                ],
                rating: 5,
                ratingsSummary: {
                    title: 'Rating',
                    caption: '5/5 Based on 4 reviews'
                },
                numRatings: 4,
                badges: [
                    {
                        id: 'FREE_TRIAL',
                        label: 'Free Trial',
                        type: 'success'
                    }
                ]
            },
            {
                id: 422,
                iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/422/thumbs_64/img4493880142979521424.png',
                iconSrcset: {
                    default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/422/thumbs_64/img4493880142979521424.png'
                },
                profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/422/thumbs_112/img3873821909190703346.png',
                overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/422/overview/img9092496391968404953.png',
                productRibbon: null,
                merchandisingBadges: [],
                uuid: '61f2febc-ae52-11e3-ac06-22000a1dd446',
                description: 'BootStrapToday is The Next Generation Software Project Management Tool. It adds intelligence, integration and automation to the software development process and makes development team highly productive.',
                overview: 'BootStrapToday is The Next Generation Software Project Management Tool. It adds intelligence, integration and automation to the software development process and makes development team highly productive.',
                vendorName: 'Sensible Softwares',
                showRating: true,
                hidePricings: false,
                free: false,
                referable: false,
                buyable: true,
                freeTrialOrEditionPresent: true,
                discounted: false,
                pending: false,
                name: 'BootStrapToday',
                blurb: 'Intelligent Workspace for Smart Teams',
                url: '/en-US/apps/422/bootstraptoday',
                startingPrice: {
                    formattedFullPrice: '$9.00/Mo',
                    formattedPrice: '$9.00',
                    price: '9.00',
                    priceCaption: 'Starting At',
                    frequency: {
                        label: 'Month',
                        shortLabel: 'Mo',
                        id: 'MONTHLY'
                    },
                    symbol: '$',
                    currency: 'USD'
                },
                tags: [
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 374,
                        name: 'Integration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 920,
                                name: 'Single Sign-On',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 932,
                                name: 'Buyable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 938,
                                name: 'Importable',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            },
                            {
                                type: 'ATTRIBUTE',
                                id: 941,
                                name: 'User Provisioning',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE_GROUP',
                        id: 327,
                        name: 'Supported Devices',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'ATTRIBUTE',
                                id: 835,
                                name: 'Web-based',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 428,
                        name: 'Developer Tools',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2253,
                                name: 'Development and Test Tools',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    },
                    {
                        type: 'CATEGORY',
                        id: 429,
                        name: 'Office Productivity',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: [
                            {
                                type: 'SUB_CATEGORY',
                                id: 2256,
                                name: 'Project Management',
                                badge: null,
                                showBadge: false,
                                showOnNavigation: true,
                                children: null,
                                description: null
                            }
                        ],
                        description: null
                    }
                ],
                rating: 4,
                ratingsSummary: {
                    title: 'Rating',
                    caption: '4/5 Based on 4 reviews'
                },
                numRatings: 4,
                badges: [
                    {
                        id: 'FREE_TRIAL',
                        label: 'Free Trial',
                        type: 'success'
                    }
                ]
            }
        ],
        viewAllLink: '/en-US/listing?scat=2258',
        i18n: {
            title: 'Similar Apps',
            viewAll: 'View All'
        }
    }
};
