// @appdirect/sfb-theme-components/mocks/product/productReviewForm.js
import { productCardMock } from './productCard';

export const productReviewsMock = {
    summary: productCardMock,
    actionEndpoint: 'https://testmarketplace.appdirect.com/',
    productUrl: '/product',
    i18n: {
        labels: {
            edit: 'Edit',
            delete: 'Delete',
            cancel: 'Cancel',
            close: 'Close the write a review form',
            confirm: 'Confirm',
            submit: 'Submit Review',
            myReview: 'My review',
            myRating: 'My rating',
            signup: 'Sign Up to Write a Review',
            writeReview: 'Write a review',
            reviews: 'Reviews',
            reviewDetails: 'Review details',
            description: 'Description',
            title: 'Title',
            singleStar: ' star',
            multiStars: ' stars',
            headlineLabel: 'Write a headline',
            headlinePlaceholder: 'Review headline',
            headlineDescription: 'Write a short headline sharing the most important things to know.',
            ratingLabel: 'Select your rating',
            ratingLevel1: 'Poor',
            ratingLevel2: 'Fair',
            ratingLevel3: 'Average',
            ratingLevel4: 'Good',
            ratingLevel5: 'Excellent',
            reviewLabel: 'Write your review',
            reviewPlaceholder: 'Review description',
            reviewDescription: 'What did you like most or least about this product?'
        },
        messages: {
            success: {
                postEntry: 'Thank you for the review, it will be posted shortly'
            },
            error: {
                postEntry: 'You must log in to write a review.',
                textareaMaxLength: 'Must be at most 2048 characters',
                fieldRequired: 'Required',
                mustBeBetween: 'Must be between 1 and 5',
                ratingRequired: 'Error: Please select a rating option for your review'
            }
        }
    }
};
