// @appdirect/sfb-theme-components/mocks/product/sideBar.js
export const sideBarMock = {
    tags: [
        {
            title: 'Group Title',
            items: [
                {
                    id: 4626,
                    label: 'French',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4626'
                },
                {
                    id: 4638,
                    label: 'German',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4638'
                },
                {
                    id: 4642,
                    label: 'Italian',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4642'
                },
                {
                    id: 4650,
                    label: 'Portuguese',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4650'
                },
                {
                    id: 4674,
                    label: 'English (AU)',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4674'
                },
                {
                    id: 4678,
                    label: 'English (UK)',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4678'
                },
                {
                    id: 4682,
                    label: 'Spanish (Mexican)',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4682'
                },
                {
                    id: 4690,
                    label: 'Mandarin',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4690'
                }
            ]
        },
        {
            title: 'integration',
            items: [
                {
                    id: 4626,
                    label: 'Single Sign On',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4626'
                },
                {
                    id: 4638,
                    label: 'Data Visualization',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4638'
                },
                {
                    id: 4642,
                    label: 'Importable',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4642'
                },
                {
                    id: 4650,
                    label: 'User Provisioning',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4650'
                }
            ]
        },
        {
            title: 'something',
            items: [
                {
                    id: 4626,
                    label: 'Single Sign On',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4626'
                },
                {
                    id: 4638,
                    label: 'Data Visualization',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4638'
                },
                {
                    id: 4642,
                    label: 'Importable',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4642'
                },
                {
                    id: 4650,
                    label: 'User Provisioning',
                    type: 'ATTRIBUTE',
                    href: '/en-US/listing?attr=4650'
                }
            ]
        }
    ]
};
