// @appdirect/sfb-theme-components/mocks/product/slider.js
export const sliderProductsMock = [
    {
        id: 10030,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/overview/img5121553238902495902.png',
        productRibbon: null,
        merchandisingBadges: [
            {
                uuid: '76eb89e4-def9-4322-9a10-5a8d9f78df89',
                name: 'iOS Compatible',
                image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/76eb89e4-def9-4322-9a10-5a8d9f78df89/img4805831920228314845.png?12294cbcc5425724b161137570276e41'
            },
            {
                uuid: '4e5a0070-28de-11e8-bd43-0aea64b20f8f',
                name: 'Monkey',
                image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/4e5a0070-28de-11e8-bd43-0aea64b20f8f/img5143663096997530867.png?23ad4994e6947eed95fc58810c363fb1'
            },
            {
                uuid: 'dsf87g87y-64e854t65-65d4rg-6s84rdg68',
                name: 'Single Sign-On',
                image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/ba3dc7bc-b3ca-4d45-90dd-6a0a083ab872/img1508692810840088346.png?a98c1e4ae512ec64db825b0d31435b08'
            },
            {
                uuid: 'er98gt89-654dfb684354gf-65sd4f-6sd85f4',
                name: 'Access Management',
                image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/0af0e01c-5c06-4bbb-9da1-214428362bdf/img7972380461864714752.png?0a566604eb8930fa19c2275dddec67fd'
            },
            {
                uuid: '08d06719-1027-4952-914c-fe0d3de89d06',
                name: 'Unified Invoice',
                image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/08d06719-1027-4952-914c-fe0d3de89d06/img5117470399285071356.png?8cf2543bdae1af7a6f287688d07c1d45'
            },
            {
                uuid: '76eb89e4-def9-4322-9a10-5a8d9f78df85',
                name: 'iOS Compatible v2',
                image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/76eb89e4-def9-4322-9a10-5a8d9f78df89/img4805831920228314845.png?12294cbcc5425724b161137570276e41'
            }
        ],
        uuid: 'a14a89d1-e62e-4138-b579-2c2a6f79fce7',
        description: 'AppInsights dashboards automatically collect the latest data about your company into one place.',
        overview: 'AppInsights help you make faster and more informed\nbusiness decisions by letting you view your KPIs on a real-time dashboard, all in\none place.',
        vendorName: 'AppInsights',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'AppInsights 1',
        blurb: 'Real-time business dashboard solution',
        url: '/en-US/apps/10030/appinsights',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'CATEGORY',
                id: 429,
                name: 'Office Productivity',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2258,
                        name: 'Office Tools',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            }
        ],
        rating: 5,
        ratingCaption: '5/5 Based on 1 review',
        numRatings: 1,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 17331,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_112/img232707846245687698.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/overview/img8616184248634727728.png',
        productRibbon: {
            text: 'CORE APP',
            textColor: '#000000',
            ribbonColor: '#FFFFFF'
        },
        merchandisingBadges: [
            {
                uuid: '76eb89e4-def9-4322-9a10-5a8d9f78df89',
                name: 'iOS Compatible',
                image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/76eb89e4-def9-4322-9a10-5a8d9f78df89/img4805831920228314845.png?12294cbcc5425724b161137570276e41'
            },
            {
                uuid: '4e5a0070-28de-11e8-bd43-0aea64b20f8f',
                name: 'Monkey',
                image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/4e5a0070-28de-11e8-bd43-0aea64b20f8f/img5143663096997530867.png?23ad4994e6947eed95fc58810c363fb1'
            },
            {
                uuid: 'dsf87g87y-64e854t65-65d4rg-6s84rdg68',
                name: 'Single Sign-On',
                image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/ba3dc7bc-b3ca-4d45-90dd-6a0a083ab872/img1508692810840088346.png?a98c1e4ae512ec64db825b0d31435b08'
            }
        ],
        uuid: '70c364fc-b558-49b5-a901-2891efa2dd6b',
        description: "Field service management all wrapped into a low cost, easy to use service with premium support. It's wherever you and your team are and can be used on any device.",
        overview: "Job management all wrapped into a low cost, easy to use service with premium support. It's wherever you and your team are and can be used on any device. \n+More Billable Hours – with smart GPS scheduling\n+Win Business Faster – create the quote, have it accepted whilst in the field\n+Complete visibility – update job sheets, attach signatures, track time, add photos and files \n+Right First Time – actual parts, actual time, invoice and take payment on the fly",
        vendorName: 'GeoOp',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'GeoOp 2',
        blurb: 'Field service software made easy',
        url: '/en-US/apps/17331/geoop',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4674,
                        name: 'English (AU)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4678,
                        name: 'English (UK)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 374,
                name: 'Integration',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 938,
                        name: 'Importable',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 3211,
                name: 'Operations',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 3707,
                        name: 'Field Activity & Service Management',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3719,
                        name: 'Scheduling',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 327,
                name: 'Supported Devices',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 811,
                        name: 'Android',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 815,
                        name: 'iPhone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 819,
                        name: 'iPad',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 827,
                        name: 'Mac',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 831,
                        name: 'Mobile Web App',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 835,
                        name: 'Web-based',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 839,
                        name: 'Windows',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 843,
                        name: 'Windows Phone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1276,
                name: 'Catalog',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4320,
                        name: 'Core',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4334,
                        name: 'Mobility',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            }
        ],
        rating: 4.4,
        ratingsSummary: {
            title: 'Rating',
            caption: '4.4/5 Based on 27 reviews'
        },
        numRatings: 27,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 185242,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img1209424854243715667.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img1209424854243715667.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img2549482514628400999-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img2549482514628400999.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_112/img6079421240100494806.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/overview/image7431663851268253158.png',
        productRibbon: {
            text: 'CORE APP',
            textColor: '#000000',
            ribbonColor: '#FFFFFF'
        },
        merchandisingBadges: [
            {
                uuid: '76eb89e4-def9-4322-9a10-5a8d9f78df89',
                name: 'iOS Compatible',
                image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/76eb89e4-def9-4322-9a10-5a8d9f78df89/img4805831920228314845.png?12294cbcc5425724b161137570276e41'
            },
            {
                uuid: '4e5a0070-28de-11e8-bd43-0aea64b20f8f',
                name: 'Monkey',
                image: 'https://d33na3ni6eqf5j.cloudfront.net/common_resources/badge/4e5a0070-28de-11e8-bd43-0aea64b20f8f/img5143663096997530867.png?23ad4994e6947eed95fc58810c363fb1'
            },
            {
                uuid: 'dsf87g87y-64e854t65-65d4rg-6s84rdg68',
                name: 'Single Sign-On',
                image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/ba3dc7bc-b3ca-4d45-90dd-6a0a083ab872/img1508692810840088346.png?a98c1e4ae512ec64db825b0d31435b08'
            },
            {
                uuid: 'er98gt89-654dfb684354gf-65sd4f-6sd85f4',
                name: 'Access Management',
                image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/0af0e01c-5c06-4bbb-9da1-214428362bdf/img7972380461864714752.png?0a566604eb8930fa19c2275dddec67fd'
            },
            {
                uuid: '08d06719-1027-4952-914c-fe0d3de89d06',
                name: 'Unified Invoice',
                image: 'https://d3bql97l1ytoxn.cloudfront.net/common_resources/badge/08d06719-1027-4952-914c-fe0d3de89d06/img5117470399285071356.png?8cf2543bdae1af7a6f287688d07c1d45'
            }
        ],
        uuid: 'db93a98c-715f-4e76-96e4-a3d278fd866e',
        description: 'IBM MaaS360 with Watson™ unifies desktop and mobile devices with an unified endpoint management (UEM) solution powered by cognitive insights.',
        overview: 'IBM MaaS360 with Watson™ unifies desktop and mobile devices with an unified endpoint management (UEM) solution powered by cognitive insights.',
        vendorName: 'IBM Corporation (prod-dev)',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'IBM MaaS360 with Watson™ 3',
        blurb: 'IBM MaaS360 Unified Endpoint Management with Watson',
        url: '/en-US/apps/185242/ibm-maas360-with-watson',
        startingPrice: {
            formattedFullPrice: '$0.00/Mo',
            formattedPrice: '$0.00',
            price: '0.00',
            priceCaption: 'Starting At',
            frequency: {
                label: 'Month',
                shortLabel: 'Mo',
                id: 'MONTHLY'
            },
            symbol: '$',
            currency: 'USD'
        },
        tags: [
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4674,
                        name: 'English (AU)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4678,
                        name: 'English (UK)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 327,
                name: 'Supported Devices',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 811,
                        name: 'Android',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 815,
                        name: 'iPhone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 819,
                        name: 'iPad',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 827,
                        name: 'Mac',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 831,
                        name: 'Mobile Web App',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 835,
                        name: 'Web-based',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 839,
                        name: 'Windows',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 843,
                        name: 'Windows Phone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 374,
                name: 'Integration',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 920,
                        name: 'Single Sign-On',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 926,
                        name: 'Data Visualization',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 932,
                        name: 'Buyable',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 941,
                        name: 'User Provisioning',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 423,
                name: 'IT Management',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2231,
                        name: 'Security',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3555,
                        name: 'System Administration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3559,
                        name: 'IT Asset Management',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1276,
                name: 'Catalog',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4320,
                        name: 'Core',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 997,
                name: 'BI & Analytics',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 3515,
                        name: 'Business Intelligence',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            }
        ],
        rating: 3.8,
        ratingsSummary: {
            title: 'Rating',
            caption: '3.8/5 Based on 52 reviews'
        },
        numRatings: 52,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 199553,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img6288868949990734409.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img6288868949990734409.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img8355631705174868774-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img8355631705174868774.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_112/img7829688944535979615.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/overview/img5763238548373303037.png',
        productRibbon: {
            text: 'CORE APP',
            textColor: '#000000',
            ribbonColor: '#FFFFFF'
        },
        merchandisingBadges: [],
        uuid: '6748efdf-df3c-4927-89b0-a99d6a68d970',
        description: 'IBM SPSS Statistics is leading statistical software that helps you quickly and easily find new insights in your data.',
        overview: 'IBM SPSS Statistics is leading statistical software that helps you quickly and easily find new insights in your data.',
        vendorName: 'IBM Corporation (prod-dev)',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'IBM SPSS Statistics 4',
        blurb: 'Apply advanced statistical analysis to your toughest business problems',
        url: '/en-US/apps/199553/ibm-spss-statistics',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 33,
                name: 'App Collections',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 75,
                        name: 'Staff Picks',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 374,
                name: 'Integration',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 920,
                        name: 'Single Sign-On',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 926,
                        name: 'Data Visualization',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 932,
                        name: 'Buyable',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 941,
                        name: 'User Provisioning',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 997,
                name: 'BI & Analytics',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2377,
                        name: 'Analytics',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3515,
                        name: 'Business Intelligence',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3519,
                        name: 'Predictive Analytics',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1276,
                name: 'Catalog',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4320,
                        name: 'Core',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            }
        ],
        rating: null,
        ratingCaption: '',
        numRatings: null,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 10031,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/overview/img5121553238902495902.png',
        productRibbon: null,
        merchandisingBadges: [],
        uuid: 'a14a89d1-e62e-4138-b579-2c2a6f79fce7',
        description: 'AppInsights dashboards automatically collect the latest data about your company into one place.',
        overview: 'AppInsights help you make faster and more informed\nbusiness decisions by letting you view your KPIs on a real-time dashboard, all in\none place.',
        vendorName: 'AppInsights',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'AppInsights 5',
        blurb: 'Real-time business dashboard solution',
        url: '/en-US/apps/10030/appinsights',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'CATEGORY',
                id: 429,
                name: 'Office Productivity',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2258,
                        name: 'Office Tools',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            }
        ],
        rating: 5,
        ratingCaption: '5/5 Based on 1 review',
        numRatings: 1,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 17332,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_112/img232707846245687698.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/overview/img8616184248634727728.png',
        productRibbon: {
            text: 'CORE APP',
            textColor: '#000000',
            ribbonColor: '#FFFFFF'
        },
        merchandisingBadges: [],
        uuid: '70c364fc-b558-49b5-a901-2891efa2dd6b',
        description: "Field service management all wrapped into a low cost, easy to use service with premium support. It's wherever you and your team are and can be used on any device.",
        overview: "Job management all wrapped into a low cost, easy to use service with premium support. It's wherever you and your team are and can be used on any device. \n+More Billable Hours – with smart GPS scheduling\n+Win Business Faster – create the quote, have it accepted whilst in the field\n+Complete visibility – update job sheets, attach signatures, track time, add photos and files \n+Right First Time – actual parts, actual time, invoice and take payment on the fly",
        vendorName: 'GeoOp',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'GeoOp 6',
        blurb: 'Field service software made easy',
        url: '/en-US/apps/17331/geoop',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4674,
                        name: 'English (AU)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4678,
                        name: 'English (UK)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 374,
                name: 'Integration',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 938,
                        name: 'Importable',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 3211,
                name: 'Operations',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 3707,
                        name: 'Field Activity & Service Management',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3719,
                        name: 'Scheduling',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 327,
                name: 'Supported Devices',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 811,
                        name: 'Android',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 815,
                        name: 'iPhone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 819,
                        name: 'iPad',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 827,
                        name: 'Mac',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 831,
                        name: 'Mobile Web App',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 835,
                        name: 'Web-based',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 839,
                        name: 'Windows',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 843,
                        name: 'Windows Phone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1276,
                name: 'Catalog',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4320,
                        name: 'Core',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4334,
                        name: 'Mobility',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            }
        ],
        rating: 4.4,
        ratingsSummary: {
            title: 'Rating',
            caption: '4.4/5 Based on 27 reviews'
        },
        numRatings: 27,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 185243,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img1209424854243715667.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img1209424854243715667.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img2549482514628400999-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img2549482514628400999.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_112/img6079421240100494806.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/overview/image7431663851268253158.png',
        productRibbon: {
            text: 'CORE APP',
            textColor: '#000000',
            ribbonColor: '#FFFFFF'
        },
        merchandisingBadges: [],
        uuid: 'db93a98c-715f-4e76-96e4-a3d278fd866e',
        description: 'IBM MaaS360 with Watson™ unifies desktop and mobile devices with an unified endpoint management (UEM) solution powered by cognitive insights.',
        overview: 'IBM MaaS360 with Watson™ unifies desktop and mobile devices with an unified endpoint management (UEM) solution powered by cognitive insights.',
        vendorName: 'IBM Corporation (prod-dev)',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'IBM MaaS360 with Watson™ 7',
        blurb: 'IBM MaaS360 Unified Endpoint Management with Watson',
        url: '/en-US/apps/185242/ibm-maas360-with-watson',
        startingPrice: {
            formattedFullPrice: '$0.00/Mo',
            formattedPrice: '$0.00',
            price: '0.00',
            priceCaption: 'Starting At',
            frequency: {
                label: 'Month',
                shortLabel: 'Mo',
                id: 'MONTHLY'
            },
            symbol: '$',
            currency: 'USD'
        },
        tags: [
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4674,
                        name: 'English (AU)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4678,
                        name: 'English (UK)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 327,
                name: 'Supported Devices',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 811,
                        name: 'Android',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 815,
                        name: 'iPhone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 819,
                        name: 'iPad',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 827,
                        name: 'Mac',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 831,
                        name: 'Mobile Web App',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 835,
                        name: 'Web-based',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 839,
                        name: 'Windows',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 843,
                        name: 'Windows Phone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 374,
                name: 'Integration',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 920,
                        name: 'Single Sign-On',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 926,
                        name: 'Data Visualization',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 932,
                        name: 'Buyable',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 941,
                        name: 'User Provisioning',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 423,
                name: 'IT Management',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2231,
                        name: 'Security',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3555,
                        name: 'System Administration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3559,
                        name: 'IT Asset Management',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1276,
                name: 'Catalog',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4320,
                        name: 'Core',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 997,
                name: 'BI & Analytics',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 3515,
                        name: 'Business Intelligence',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            }
        ],
        rating: 3.8,
        ratingsSummary: {
            title: 'Rating',
            caption: '3.8/5 Based on 52 reviews'
        },
        numRatings: 52,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 199554,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img6288868949990734409.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img6288868949990734409.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img8355631705174868774-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img8355631705174868774.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_112/img7829688944535979615.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/overview/img5763238548373303037.png',
        productRibbon: {
            text: 'CORE APP',
            textColor: '#000000',
            ribbonColor: '#FFFFFF'
        },
        merchandisingBadges: [],
        uuid: '6748efdf-df3c-4927-89b0-a99d6a68d970',
        description: 'IBM SPSS Statistics is leading statistical software that helps you quickly and easily find new insights in your data.',
        overview: 'IBM SPSS Statistics is leading statistical software that helps you quickly and easily find new insights in your data.',
        vendorName: 'IBM Corporation (prod-dev)',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'IBM SPSS Statistics 8',
        blurb: 'Apply advanced statistical analysis to your toughest business problems',
        url: '/en-US/apps/199553/ibm-spss-statistics',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 33,
                name: 'App Collections',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 75,
                        name: 'Staff Picks',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 374,
                name: 'Integration',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 920,
                        name: 'Single Sign-On',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 926,
                        name: 'Data Visualization',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 932,
                        name: 'Buyable',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 941,
                        name: 'User Provisioning',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 997,
                name: 'BI & Analytics',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2377,
                        name: 'Analytics',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3515,
                        name: 'Business Intelligence',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3519,
                        name: 'Predictive Analytics',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1276,
                name: 'Catalog',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4320,
                        name: 'Core',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            }
        ],
        rating: null,
        ratingCaption: '',
        numRatings: null,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 10032,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/overview/img5121553238902495902.png',
        productRibbon: null,
        merchandisingBadges: [],
        uuid: 'a14a89d1-e62e-4138-b579-2c2a6f79fce7',
        description: 'AppInsights dashboards automatically collect the latest data about your company into one place.',
        overview: 'AppInsights help you make faster and more informed\nbusiness decisions by letting you view your KPIs on a real-time dashboard, all in\none place.',
        vendorName: 'AppInsights',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'AppInsights 9',
        blurb: 'Real-time business dashboard solution',
        url: '/en-US/apps/10030/appinsights',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'CATEGORY',
                id: 429,
                name: 'Office Productivity',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2258,
                        name: 'Office Tools',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            }
        ],
        rating: 5,
        ratingCaption: '5/5 Based on 1 review',
        numRatings: 1,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 17333,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_64/img2561248845350105123.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/thumbs_112/img232707846245687698.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/17331/overview/img8616184248634727728.png',
        productRibbon: {
            text: 'CORE APP',
            textColor: '#000000',
            ribbonColor: '#FFFFFF'
        },
        merchandisingBadges: [],
        uuid: '70c364fc-b558-49b5-a901-2891efa2dd6b',
        description: "Field service management all wrapped into a low cost, easy to use service with premium support. It's wherever you and your team are and can be used on any device.",
        overview: "Job management all wrapped into a low cost, easy to use service with premium support. It's wherever you and your team are and can be used on any device. \n+More Billable Hours – with smart GPS scheduling\n+Win Business Faster – create the quote, have it accepted whilst in the field\n+Complete visibility – update job sheets, attach signatures, track time, add photos and files \n+Right First Time – actual parts, actual time, invoice and take payment on the fly",
        vendorName: 'GeoOp',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'GeoOp 10',
        blurb: 'Field service software made easy',
        url: '/en-US/apps/17331/geoop',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4674,
                        name: 'English (AU)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4678,
                        name: 'English (UK)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 374,
                name: 'Integration',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 938,
                        name: 'Importable',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 3211,
                name: 'Operations',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 3707,
                        name: 'Field Activity & Service Management',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3719,
                        name: 'Scheduling',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 327,
                name: 'Supported Devices',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 811,
                        name: 'Android',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 815,
                        name: 'iPhone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 819,
                        name: 'iPad',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 827,
                        name: 'Mac',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 831,
                        name: 'Mobile Web App',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 835,
                        name: 'Web-based',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 839,
                        name: 'Windows',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 843,
                        name: 'Windows Phone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1276,
                name: 'Catalog',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4320,
                        name: 'Core',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4334,
                        name: 'Mobility',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            }
        ],
        rating: 4.4,
        ratingsSummary: {
            title: 'Rating',
            caption: '4.4/5 Based on 27 reviews'
        },
        numRatings: 27,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 185244,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img1209424854243715667.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img1209424854243715667.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img2549482514628400999-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_64/img2549482514628400999.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/thumbs_112/img6079421240100494806.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/185242/overview/image7431663851268253158.png',
        productRibbon: {
            text: 'CORE APP',
            textColor: '#000000',
            ribbonColor: '#FFFFFF'
        },
        merchandisingBadges: [],
        uuid: 'db93a98c-715f-4e76-96e4-a3d278fd866e',
        description: 'IBM MaaS360 with Watson™ unifies desktop and mobile devices with an unified endpoint management (UEM) solution powered by cognitive insights.',
        overview: 'IBM MaaS360 with Watson™ unifies desktop and mobile devices with an unified endpoint management (UEM) solution powered by cognitive insights.',
        vendorName: 'IBM Corporation (prod-dev)',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'IBM MaaS360 with Watson™ 11',
        blurb: 'IBM MaaS360 Unified Endpoint Management with Watson',
        url: '/en-US/apps/185242/ibm-maas360-with-watson',
        startingPrice: {
            formattedFullPrice: '$0.00/Mo',
            formattedPrice: '$0.00',
            price: '0.00',
            priceCaption: 'Starting At',
            frequency: {
                label: 'Month',
                shortLabel: 'Mo',
                id: 'MONTHLY'
            },
            symbol: '$',
            currency: 'USD'
        },
        tags: [
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4674,
                        name: 'English (AU)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 4678,
                        name: 'English (UK)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 327,
                name: 'Supported Devices',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 811,
                        name: 'Android',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 815,
                        name: 'iPhone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 819,
                        name: 'iPad',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 827,
                        name: 'Mac',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 831,
                        name: 'Mobile Web App',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 835,
                        name: 'Web-based',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 839,
                        name: 'Windows',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 843,
                        name: 'Windows Phone',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 374,
                name: 'Integration',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 920,
                        name: 'Single Sign-On',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 926,
                        name: 'Data Visualization',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 932,
                        name: 'Buyable',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 941,
                        name: 'User Provisioning',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 423,
                name: 'IT Management',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2231,
                        name: 'Security',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3555,
                        name: 'System Administration',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3559,
                        name: 'IT Asset Management',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1276,
                name: 'Catalog',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4320,
                        name: 'Core',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 997,
                name: 'BI & Analytics',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 3515,
                        name: 'Business Intelligence',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            }
        ],
        rating: 3.8,
        ratingsSummary: {
            title: 'Rating',
            caption: '3.8/5 Based on 52 reviews'
        },
        numRatings: 52,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 199555,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img6288868949990734409.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img6288868949990734409.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img8355631705174868774-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_64/img8355631705174868774.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/thumbs_112/img7829688944535979615.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/199553/overview/img5763238548373303037.png',
        productRibbon: {
            text: 'CORE APP',
            textColor: '#000000',
            ribbonColor: '#FFFFFF'
        },
        merchandisingBadges: [],
        uuid: '6748efdf-df3c-4927-89b0-a99d6a68d970',
        description: 'IBM SPSS Statistics is leading statistical software that helps you quickly and easily find new insights in your data.',
        overview: 'IBM SPSS Statistics is leading statistical software that helps you quickly and easily find new insights in your data.',
        vendorName: 'IBM Corporation (prod-dev)',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'IBM SPSS Statistics 12',
        blurb: 'Apply advanced statistical analysis to your toughest business problems',
        url: '/en-US/apps/199553/ibm-spss-statistics',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 33,
                name: 'App Collections',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 75,
                        name: 'Staff Picks',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 374,
                name: 'Integration',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 920,
                        name: 'Single Sign-On',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 926,
                        name: 'Data Visualization',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 932,
                        name: 'Buyable',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'ATTRIBUTE',
                        id: 941,
                        name: 'User Provisioning',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'CATEGORY',
                id: 997,
                name: 'BI & Analytics',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2377,
                        name: 'Analytics',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3515,
                        name: 'Business Intelligence',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    },
                    {
                        type: 'SUB_CATEGORY',
                        id: 3519,
                        name: 'Predictive Analytics',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1276,
                name: 'Catalog',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4320,
                        name: 'Core',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            }
        ],
        rating: null,
        ratingCaption: '',
        numRatings: null,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    },
    {
        id: 10033,
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
        iconSrcset: {
            default: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png',
            '2x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669-2x.png',
            '1x': 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_64/img822415822151000669.png'
        },
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/thumbs_112/img2505072429437208263.png',
        overviewImageUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/10030/overview/img5121553238902495902.png',
        productRibbon: null,
        merchandisingBadges: [],
        uuid: 'a14a89d1-e62e-4138-b579-2c2a6f79fce7',
        description: 'AppInsights dashboards automatically collect the latest data about your company into one place.',
        overview: 'AppInsights help you make faster and more informed\nbusiness decisions by letting you view your KPIs on a real-time dashboard, all in\none place.',
        vendorName: 'AppInsights',
        showRating: true,
        hidePricings: false,
        free: false,
        referable: false,
        buyable: true,
        freeTrialOrEditionPresent: true,
        discounted: false,
        pending: false,
        name: 'AppInsights 13',
        blurb: 'Real-time business dashboard solution',
        url: '/en-US/apps/10030/appinsights',
        startingPrice: {
            formattedFullPrice: 'Free',
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            {
                type: 'CATEGORY',
                id: 429,
                name: 'Office Productivity',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'SUB_CATEGORY',
                        id: 2258,
                        name: 'Office Tools',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: null
            },
            {
                type: 'ATTRIBUTE_GROUP',
                id: 1350,
                name: 'Language availability',
                badge: null,
                showBadge: false,
                showOnNavigation: true,
                children: [
                    {
                        type: 'ATTRIBUTE',
                        id: 4606,
                        name: 'English (US)',
                        badge: null,
                        showBadge: false,
                        showOnNavigation: true,
                        children: null,
                        description: null
                    }
                ],
                description: 'Currently available or Core Application Catalog only.'
            }
        ],
        rating: 5,
        ratingCaption: '5/5 Based on 1 review',
        numRatings: 1,
        badges: [
            {
                id: 'FREE_TRIAL',
                label: 'Free Trial',
                type: 'success'
            }
        ]
    }
];
