// @appdirect/sfb-theme-components/mocks/product/volumePricing.js
const mockVolumeDiscountCardI18n = {
    title: '{vendor} VIP Pricing',
    content: 'This product is eligible for volume based discounts.',
    information: 'Based on the total number of licences you own. Number of users can be selected at checkout.',
    linkName: 'See {vendor} VIP Pricing'
};

const mockContractDurationI18n = {
    daily: '{0} day contract',
    monthly: '{0} month contract',
    yearly: '{0} year contract'
};

const mockUnits = {
    USER: {
        code: 'USER',
        meaning: 'Per User',
        label: 'User',
        unitsLabel: 'Users',
        editable: true,
        decimals: false
    }
};

export const mockPricingData = [{
    pricingDuration: 'MONTHLY',
    contract: {
        minimumServiceLength: 36
    },
    price: {
        priceBookName: 'Adobe VIP 1',
        priceBookDescription: 'some description - Discounted price for level 1 customers',
        priceLevel: '1-9',
        priceLevelDescription: 'some description - Price for quantity 1 to 9',
        unitFees: [
            {
                unit: 'USER',
                pricingStrategy: 'UNIT',
                priceRanges: [
                    {
                        min: 1,
                        max: null,
                        origialPrice: 100.0000000000,
                        percentage: false,
                        updatedPrice: null,
                        finalPrice: 100.0000000000
                    }
                ]
            }
        ]
    }
}];

const editionsDataStore = [
    {
        productEditionRefId: 'd9d0836a-a647-44ed-9c27-30ecf899b8e5',
        name: 'Edition 1'
    },
    {
        productEditionRefId: 'd9d0836a-a647-44ed-9c27-30ecf899b8f6',
        name: 'Edition 2'
    },
    {
        productEditionRefId: 'd9d0836a-a647-44ed-9c27-30ecf899b8e7',
        name: 'Edition 3'
    },
    {
        productEditionRefId: 'd9d0836a-a647-44ed-9c27-30ecf899b8e8',
        name: 'Edition 4'
    }
];

export const mockEditionsI18nData = {
    currency: 'USD',
    locale: 'en-US',
    volumeDiscountPricing: mockVolumeDiscountCardI18n,
    pricingDurationEnum: {
        MONTHLY: {
            code: 'MONTHLY',
            meaning: 'monthly',
            label: 'Month',
            shortLabel: 'mo',
            durationsLabel: 'Months',
            unitMultiplier: 1,
            lengthUnit: 'MONTHS',
            frequencyLabel: 'Monthly'
        }
    },
    contractDuration: mockContractDurationI18n,
    units: mockUnits,
    editionsDataStore
};

export const mockVolumePricingV2 = {
    marketplaceProduct: {
        id: '0b82724b-02db-4ada-9566-85329050631b',
        name: 'TestFirst',
        type: 'WEB_APP',
        externalSource: 'Adobe',
        editions: [
            {
                id: 'd9d0836a-a647-44ed-9c27-30ecf899b8e5',
                code: 'RECURRING',
                version: 'WORKING',
                pricing: {
                    id: 'd9d0836a-a647-44ed-9c27-30ecf899b8f5',
                    pricingPlans: [
                        {
                            id: 'd2ca4d46-64ce-47d3-9402-792465fea916',
                            pricingDuration: 'MONTHLY',
                            contract: {
                                minimumServiceLength: null
                            },
                            externalSourcePrices: [
                                {
                                    priceBookName: 'Adobe VIP 1',
                                    priceBookDescription: 'some description - Discounted price for level 1 customers',
                                    priceLevel: '1-9',
                                    priceLevelDescription: 'some description - Price for quantity 1 to 9',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: null,
                                                    finalPrice: 100.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 2',
                                    priceBookDescription: 'some description - Discounted price for level 2 customers',
                                    priceLevel: '10-49',
                                    priceLevelDescription: 'some description - Price for quantity 10 to 49',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -10.0000000000,
                                                    finalPrice: 90.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 3',
                                    priceBookDescription: 'some description - Discounted price for level 3 customers',
                                    priceLevel: '50-99',
                                    priceLevelDescription: 'some description - Price for quantity 50 to 99',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -20.0000000000,
                                                    finalPrice: 80.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 4',
                                    priceBookDescription: 'some description - Discounted price for level 4 customers',
                                    priceLevel: '100+',
                                    priceLevelDescription: 'some description - Price for quantity more than 100 users',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -30.0000000000,
                                                    finalPrice: 70.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            id: 'd2ca4d46-64ce-47d3-9402-792465fea917',
                            pricingDuration: 'MONTHLY',
                            contract: {
                                minimumServiceLength: 36
                            },
                            externalSourcePrices: [
                                {
                                    priceBookName: 'Adobe VIP 2',
                                    priceBookDescription: 'some description - Discounted price for level 2 customers',
                                    priceLevel: '10-49',
                                    priceLevelDescription: 'some description - Price for quantity 10 to 49',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 80.0000000000,
                                                    finalPrice: 80.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 3',
                                    priceBookDescription: 'some description - Discounted price for level 3 customers',
                                    priceLevel: '50-99',
                                    priceLevelDescription: 'some description - Price for quantity 50 to 99',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 70.0000000000,
                                                    finalPrice: 70.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 4',
                                    priceBookDescription: 'some description - Discounted price for level 4 customers',
                                    priceLevel: '100+',
                                    priceLevelDescription: 'some description - Price for quantity more than 100 users',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 60.0000000000,
                                                    finalPrice: 60.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            },
            {
                id: 'd9d0836a-a647-44ed-9c27-30ecf899b8f6',
                code: 'RECURRING',
                version: 'WORKING',
                pricing: {
                    id: 'd9d0836a-a647-44ed-9c27-30ecf899b8e6',
                    pricingPlans: [
                        {
                            id: 'd2ca4d46-64ce-47d3-9402-792465fea918',
                            pricingDuration: 'MONTHLY',
                            contract: {
                                minimumServiceLength: null
                            },
                            externalSourcePrices: [
                                {
                                    priceBookName: 'Adobe VIP 1',
                                    priceBookDescription: 'some description - Discounted price for level 1 customers',
                                    priceLevel: '1-9',
                                    priceLevelDescription: 'some description - Price for quantity 1 to 9',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: null,
                                                    finalPrice: 100.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 2',
                                    priceBookDescription: 'some description - Discounted price for level 2 customers',
                                    priceLevel: '10-49',
                                    priceLevelDescription: 'some description - Price for quantity 10 to 49',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -10.0000000000,
                                                    finalPrice: 90.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 3',
                                    priceBookDescription: 'some description - Discounted price for level 3 customers',
                                    priceLevel: '50-99',
                                    priceLevelDescription: 'some description - Price for quantity 50 to 99',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -20.0000000000,
                                                    finalPrice: 80.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 4',
                                    priceBookDescription: 'some description - Discounted price for level 4 customers',
                                    priceLevel: '100+',
                                    priceLevelDescription: 'some description - Price for quantity more than 100 users',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -30.0000000000,
                                                    finalPrice: 70.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            id: 'd2ca4d46-64ce-47d3-9402-792465fea919',
                            pricingDuration: 'MONTHLY',
                            contract: {
                                minimumServiceLength: 36
                            },
                            externalSourcePrices: [
                                {
                                    priceBookName: 'Adobe VIP 2',
                                    priceBookDescription: 'some description - Discounted price for level 2 customers',
                                    priceLevel: '10-49',
                                    priceLevelDescription: 'some description - Price for quantity 10 to 49',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 80.0000000000,
                                                    finalPrice: 80.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 3',
                                    priceBookDescription: 'some description - Discounted price for level 3 customers',
                                    priceLevel: '50-99',
                                    priceLevelDescription: 'some description - Price for quantity 50 to 99',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 70.0000000000,
                                                    finalPrice: 70.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 4',
                                    priceBookDescription: 'some description - Discounted price for level 4 customers',
                                    priceLevel: '100+',
                                    priceLevelDescription: 'some description - Price for quantity more than 100 users',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 60.0000000000,
                                                    finalPrice: 60.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            },
            {
                id: 'd9d0836a-a647-44ed-9c27-30ecf899b8e7',
                code: 'RECURRING',
                version: 'WORKING',
                pricing: {
                    id: 'd9d0836a-a647-44ed-9c27-30ecf899b8e7',
                    pricingPlans: [
                        {
                            id: 'd2ca4d46-64ce-47d3-9402-792465fea920',
                            pricingDuration: 'MONTHLY',
                            contract: {
                                minimumServiceLength: null
                            },
                            externalSourcePrices: [
                                {
                                    priceBookName: 'Adobe VIP 1',
                                    priceBookDescription: 'some description - Discounted price for level 1 customers',
                                    priceLevel: '1-9',
                                    priceLevelDescription: 'some description - Price for quantity 1 to 9',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: null,
                                                    finalPrice: 100.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 2',
                                    priceBookDescription: 'some description - Discounted price for level 2 customers',
                                    priceLevel: '10-49',
                                    priceLevelDescription: 'some description - Price for quantity 10 to 49',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -10.0000000000,
                                                    finalPrice: 90.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 3',
                                    priceBookDescription: 'some description - Discounted price for level 3 customers',
                                    priceLevel: '50-99',
                                    priceLevelDescription: 'some description - Price for quantity 50 to 99',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -20.0000000000,
                                                    finalPrice: 80.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 4',
                                    priceBookDescription: 'some description - Discounted price for level 4 customers',
                                    priceLevel: '100+',
                                    priceLevelDescription: 'some description - Price for quantity more than 100 users',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -30.0000000000,
                                                    finalPrice: 70.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            id: 'd2ca4d46-64ce-47d3-9402-792465fea921',
                            pricingDuration: 'MONTHLY',
                            contract: {
                                minimumServiceLength: 36
                            },
                            externalSourcePrices: [
                                {
                                    priceBookName: 'Adobe VIP 2',
                                    priceBookDescription: 'some description - Discounted price for level 2 customers',
                                    priceLevel: '10-49',
                                    priceLevelDescription: 'some description - Price for quantity 10 to 49',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 80.0000000000,
                                                    finalPrice: 80.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 3',
                                    priceBookDescription: 'some description - Discounted price for level 3 customers',
                                    priceLevel: '50-99',
                                    priceLevelDescription: 'some description - Price for quantity 50 to 99',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 70.0000000000,
                                                    finalPrice: 70.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 4',
                                    priceBookDescription: 'some description - Discounted price for level 4 customers',
                                    priceLevel: '100+',
                                    priceLevelDescription: 'some description - Price for quantity more than 100 users',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 60.0000000000,
                                                    finalPrice: 60.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            },
            {
                id: 'd9d0836a-a647-44ed-9c27-30ecf899b8e8',
                code: 'RECURRING',
                version: 'WORKING',
                pricing: {
                    id: 'd9d0836a-a647-44ed-9c27-30ecf899b8e8',
                    pricingPlans: [
                        {
                            id: 'd2ca4d46-64ce-47d3-9402-792465fea922',
                            pricingDuration: 'MONTHLY',
                            contract: {
                                minimumServiceLength: null
                            },
                            externalSourcePrices: [
                                {
                                    priceBookName: 'Adobe VIP 1',
                                    priceBookDescription: 'some description - Discounted price for level 1 customers',
                                    priceLevel: '1-9',
                                    priceLevelDescription: 'some description - Price for quantity 1 to 9',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: null,
                                                    finalPrice: 100.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 2',
                                    priceBookDescription: 'some description - Discounted price for level 2 customers',
                                    priceLevel: '10-49',
                                    priceLevelDescription: 'some description - Price for quantity 10 to 49',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -10.0000000000,
                                                    finalPrice: 90.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 3',
                                    priceBookDescription: 'some description - Discounted price for level 3 customers',
                                    priceLevel: '50-99',
                                    priceLevelDescription: 'some description - Price for quantity 50 to 99',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -20.0000000000,
                                                    finalPrice: 80.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 4',
                                    priceBookDescription: 'some description - Discounted price for level 4 customers',
                                    priceLevel: '100+',
                                    priceLevelDescription: 'some description - Price for quantity more than 100 users',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: true,
                                                    updatedPrice: -30.0000000000,
                                                    finalPrice: 70.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            id: 'd2ca4d46-64ce-47d3-9402-792465fea923',
                            pricingDuration: 'MONTHLY',
                            contract: {
                                minimumServiceLength: 36
                            },
                            externalSourcePrices: [
                                {
                                    priceBookName: 'Adobe VIP 2',
                                    priceBookDescription: 'some description - Discounted price for level 2 customers',
                                    priceLevel: '10-49',
                                    priceLevelDescription: 'some description - Price for quantity 10 to 49',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 80.0000000000,
                                                    finalPrice: 80.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 3',
                                    priceBookDescription: 'some description - Discounted price for level 3 customers',
                                    priceLevel: '50-99',
                                    priceLevelDescription: 'some description - Price for quantity 50 to 99',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 70.0000000000,
                                                    finalPrice: 70.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                },
                                {
                                    priceBookName: 'Adobe VIP 4',
                                    priceBookDescription: 'some description - Discounted price for level 4 customers',
                                    priceLevel: '100+',
                                    priceLevelDescription: 'some description - Price for quantity more than 100 users',
                                    unitFees: [
                                        {
                                            unit: 'USER',
                                            pricingStrategy: 'UNIT',
                                            priceRanges: [
                                                {
                                                    min: 1,
                                                    max: null,
                                                    origialPrice: 100.0000000000,
                                                    percentage: false,
                                                    updatedPrice: 60.0000000000,
                                                    finalPrice: 60.0000000000
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            }
        ]
    }
};
