// @appdirect/sfb-theme-components/src/constants/components/header.js
import { LEFT, RIGHT, CENTER } from '../alignment';
import { SMALL, MEDIUM, LARGE } from '../sizes';

export const NAVIGATION_TEXT_ALIGN = [LEFT, CENTER, RIGHT];
export const NAVIGATION_TEXT_SIZE = [SMALL, MEDIUM, LARGE];

export const DEFAULT_TEXT_COLOR = '#474747';
export const SEARCH_BAR_WIDTH = {
    min: 225,
    max: 650
};

export const SEARCH_BAR_IN_HOME_BAR = {
    value: 'home_bar',
    label: 'uieditor.sfbComponent.header.searchBar.location.homeBar.label'
};
export const SEARCH_BAR_IN_NAV_BAR = {
    value: 'navigation_bar',
    label: 'uieditor.sfbComponent.header.searchBar.location.navigationBar.label'
};
export const NO_SEARCH_BAR = {
    value: 'none',
    label: 'uieditor.sfbComponent.header.searchBar.location.none.label'
};
export const SEARCH_BAR_LOCATIONS = [
    SEARCH_BAR_IN_HOME_BAR,
    SEARCH_BAR_IN_NAV_BAR,
    NO_SEARCH_BAR
];

export const SEARCH_BAR_HEIGHTS = [SMALL, MEDIUM, LARGE];
export const SEARCH_BAR_HEIGHTS_VALUES = {
    small: 32,
    medium: 40,
    large: 48
};

export const LOGO_TARGET_HOMEPAGE = {
    value: 'home_page',
    label: 'uieditor.sfbComponent.header.homeBar.logoTargetType.home.label'
};
export const LOGO_TARGET_CUSTOM_URL = {
    value: 'custom_url',
    label: 'uieditor.sfbComponent.header.homeBar.logoTargetType.custom.label'
};
export const LOGO_TARGET_TYPES = [LOGO_TARGET_HOMEPAGE, LOGO_TARGET_CUSTOM_URL];

export const USER_BRANCH_OPEN = 'user_branch';

export const CART_OVERLAY_TRANSITION_DURATION = 500;
