// @appdirect/sfb-theme-components/src/constants/components/tile.js
export const CLASSIC = 'classic';
export const SQUARE = 'square';
export const TALL = 'tall';
export const GRAPHIC = 'graphic';
export const HORIZONTAL = 'horizontal';
export const CARD = 'card';

export const CLASSIC_TILE = {
    value: CLASSIC,
    label: 'uieditor.sfbComponent.slider.layout.option.classic'
};

export const SQUARE_TILE = {
    value: SQUARE,
    label: 'uieditor.sfbComponent.slider.layout.option.square'
};

export const HORIZONTAL_TILE = {
    value: HORIZONTAL,
    label: 'uieditor.sfbComponent.slider.layout.option.horizontal'
};

export const TALL_TILE = {
    value: TALL,
    label: 'uieditor.sfbComponent.slider.layout.option.tall'
};

export const GRAPHIC_TILE = {
    value: GRAPHIC,
    label: 'uieditor.sfbComponent.slider.layout.option.graphic'
};

export const CARD_TILE = {
    value: CARD,
    label: 'uieditor.sfbComponent.slider.layout.option.card'
};

export const TILE_TYPES = [
    CLASSIC_TILE,
    SQUARE_TILE,
    HORIZONTAL_TILE,
    TALL_TILE,
    GRAPHIC_TILE
];

export const TILE_MAX_MERCHANDISING_BADGES = 4;
