// @appdirect/sfb-theme-components/src/components/comparable-products/atoms/compare-product-header/CompareProductHeader.jsx
import React from 'react';
import PropTypes from 'prop-types';
import { createNamespace } from '../../../../tools/namingTools';
import { MAX_COMPARE_ITEMS_LENGTH } from '../../../../constants/components/compare';
// IMPORTANT: (s)css should be imported after other atoms/components
import './styles/CompareProductHeader.scss';

const n = createNamespace('CompareProductHeader');

export const CompareProductHeader = ({
    label,
    nItems,
    i18n
}) => (
    <div {...n('wrapper').props}>
        <h1 {...n('title').props}>{label}</h1>
        <span {...n('total').props}>{i18n.nbselected.replace('{0}', nItems).replace('{1}', MAX_COMPARE_ITEMS_LENGTH)}</span>
    </div>
);

CompareProductHeader.propTypes = {
    label: PropTypes.string,
    nItems: PropTypes.number,
    i18n: PropTypes.shape({
        compareApplications: PropTypes.string,
        description: PropTypes.string,
        nbselected: PropTypes.string,
        showDifferences: PropTypes.string
    }).isRequired
};

CompareProductHeader.defaultProps = {
    label: null,
    nItems: 0
};
