// @appdirect/sfb-theme-components/mocks/config/user.js
export const userLoggedOutMock = {
    isLoggedIn: false,
    roles: [],
    customUserAttributes: {},
    customerContractEditionUuids: [],
    ownedPartners: [],
    segments: {
        items: []
    }
};

export const userLoggedInMock = {
    name: 'John Doe',
    isLoggedIn: true,
    email: 'john.doe@sfbmarketplace.byappdirect.com',
    family_name: 'Doe',
    given_name: 'John',
    roles: [
        'DISABLE_COMPANY_USER_ACCOUNT',
        'MANAGE_SCHEDULED_REPORTS',
        'CREATE_REPORTS',
        'DELETE_REPORTS',
        'ROLES_MANAGEMENT',
        'CREATE_QUOTE',
        'REPORTS_CHANNEL_ADMIN',
        'CHANNEL_PRODUCT_SUPPORT',
        'SET_OPPORTUNITY_PROVISIONING_DATE',
        'PURCHASE_ON_BEHALF_OF_ALL_USERS',
        'LICENSES_MANAGEMENT_ACCESS',
        'CHANNEL_SUPPORT',
        'SALES_SUPPORT',
        'MANAGE_REPORT_TEMPLATES',
        'CREATE_REPORT_TEMPLATES',
        'MIGRATION_COMPANY_ACCESS',
        'CANCEL_SUBSCRIPTION',
        'DEVELOPER',
        'BILLING_ADMIN',
        'MIGRATION_CHANNEL_ACCESS',
        'USER',
        'QUOTE_SET_RETAIL_PRICE',
        'SET_OPPORTUNITY_BILLING_DATE',
        'DISABLE_COMPANY',
        'JOIN_COMPANY',
        'SYS_ADMIN',
        'CHANNEL_ADMIN',
        'ENABLE_DOMAIN_VERIFICATION'
    ],
    userUuid: '06e7f152-a15e-4f85-a27f-fc48771f7d0b',
    hasCustomerContract: false,
    locale: 'en_US',
    attributes: {
        HAS_CUSTOMER_CONTRACT: 'false'
    },
    companyUuid: '1df65377-7269-4368-b671-1d129d966b4e',
    currency: 'USD',
    customUserAttributes: {},
    customerContractEditionUuids: [],
    ownedPartners: [],
    segments: {
        items: []
    },
    companyOnlyAdminsCanBuy: false
};

export const userWithUserRoleMock = {
    name: 'John Doe',
    isLoggedIn: true,
    email: 'john.doe@sfbmarketplace.byappdirect.com',
    family_name: 'Doe',
    given_name: 'John',
    roles: [
        'USER'
    ],
    userUuid: '06e7f152-a15e-4f85-a27f-fc48771f7d0b',
    hasCustomerContract: false,
    locale: 'en_US',
    attributes: {
        HAS_CUSTOMER_CONTRACT: 'false'
    },
    companyUuid: '1df65377-7269-4368-b671-1d129d966b4e',
    currency: 'USD',
    customUserAttributes: {},
    customerContractEditionUuids: [],
    ownedPartners: [],
    segments: {
        items: []
    },
    companyOnlyAdminsCanBuy: false
};
