// @appdirect/sfb-theme-components/src/atoms/swatch/stories/Swatch.stories.js
import React from 'react';
import Swatch from '../Swatch';

const swatchColors = [
    {
        id: 'pink',
        name: 'Pink',
        value: '#fbe2dd'
    },
    {
        id: 'blue',
        name: 'Blue',
        value: '#437790'
    },
    {
        id: 'midnight',
        name: 'Midnight',
        value: '#42474c'
    },
    {
        id: 'starlight',
        name: 'Starlight',
        value: '#faf7f4'
    },
    {
        id: 'red',
        name: 'Red',
        value: '#c72433'
    }
];

const disabledSwatchColors = swatchColors.map(color => ({ ...color, isDisabled: true }));

const rgbSwatchColors = [
    {
        id: 'pink',
        name: 'Pink',
        value: 'rgb(238, 130, 238)'
    },
    {
        id: 'blue',
        name: 'Blue',
        value: 'rgb(0, 0, 255)'
    },
    {
        id: 'violet',
        name: 'violet',
        value: 'rgb(106, 90, 205)'
    },
    {
        id: 'midnight',
        name: 'Midnight',
        value: '#42474c'
    }
];

const invalidSwatchColors = [
    {
        id: undefined,
        name: undefined,
        value: undefined
    },
    {
        id: null,
        name: null,
        value: null
    },
    {
        id: 123,
        name: 123,
        value: 123
    },
    {
        id: 'starlight',
        name: 'Starlight',
        value: 'Starlight'
    },
    {
        id: '',
        name: '',
        value: ''
    },
    {
        id: 'lennyo',
        name: 'lennyo',
        value: '#lennyo'
    },
    {
        id: 'lennyo',
        name: 'lennyo',
        value: 'rgb("hi", "hello", "hey")'
    }
];

export default {
    title: 'Atoms (Preview)/Swatch',
    component: Swatch
};

export const Default = {
    render: () => <Swatch swatchColors={swatchColors} />,
    name: 'Default'
};

export const ColorSelected = {
    render: () => <Swatch swatchColors={swatchColors} selectedColorId="blue" />,
    name: 'Color selected'
};

export const ColorDisabled = {
    render: () => <Swatch swatchColors={disabledSwatchColors} />,
    name: 'Color disabled'
};

export const RgbColors = {
    render: () => <Swatch swatchColors={rgbSwatchColors} />,
    name: 'RGB colors'
};

export const InvalidColors = {
    render: () => <Swatch swatchColors={invalidSwatchColors} />,
    name: 'Invalid colors'
};
