// @appdirect/sfb-theme-components/src/atoms/video-player/stories/VideoPlayer.stories.js
import React from 'react';
import { youtubeMock, vimeoMock } from '../../../../mocks/product/overview/callToActions';
import VideoPlayer from '../VideoPlayer';

export default {
    title: 'Atoms/Video Player',
    component: VideoPlayer,
    argTypes: {
        title: {
            control: 'text',
            description: 'The title of the video player'
        },
        src: {
            control: 'text',
            description: 'The source of the video player'
        }
    },
    render: args => (
        <div
            style={{
                width: 800,
                height: 450
            }}
        >
            <VideoPlayer {...args} />
        </div>
    ),
    parameters: {
        controls: {
            include: ['title', 'src']
        }
    }
};

export const YouTube = {
    args: {
        title: youtubeMock.label,
        src: youtubeMock.iFrameFriendlyUrl
    },
    name: 'YouTube'
};

export const Vimeo = {
    args: {
        title: vimeoMock.label,
        src: vimeoMock.iFrameFriendlyUrl
    }
};
