// @appdirect/sfb-theme-components/src/constants/components/carousel.js
export const defaultSlide = [{
    contentType: 'IMAGE_ONLY',
    slideData: {
        backgroundColor: '#F4F4F4',
        image: {},
        imageSizing: 'fill',
        slideLink: null,
        slideName: '',
        text: '',
        textColor: '#fff',
        textOverlay: false,
        textSize: 'small',
        title: '',
        titleColor: null,
        titleSize: null,
        textAlignment: null,
        overlayColor: null
    }
}];

export const DELAY_OPTIONS = [
    { value: 3, label: 'uieditor.sfbComponent.delay.seconds.three' },
    { value: 4, label: 'uieditor.sfbComponent.delay.seconds.four' },
    { value: 5, label: 'uieditor.sfbComponent.delay.seconds.five' },
    { value: 7, label: 'uieditor.sfbComponent.delay.seconds.seven' },
    { value: 10, label: 'uieditor.sfbComponent.delay.seconds.ten' },
    { value: 15, label: 'uieditor.sfbComponent.delay.seconds.fifteen' }
];

// FIXME: SFB-4203 - allow passing param to translation
export const HEIGHT_OPTIONS = [
    { value: 400, label: 'Small 400px' },
    { value: 555, label: 'Medium 555px' },
    { value: 660, label: 'Large 660px' }
];

export const COLOR_DARK = 'uieditor.sfbComponent.carousel.caretsColorTheme.dark';
export const COLOR_LIGHT = 'uieditor.sfbComponent.carousel.caretsColorTheme.light';
export const CARETS_COLOR_OPTIONS = [
    { value: COLOR_DARK, label: COLOR_DARK },
    { value: COLOR_LIGHT, label: COLOR_LIGHT }
];

// on mobile, the height of carousel is a smaller proportion of the desired height
export const MOBILE_HEIGHT_RATIO = 0.75;
