// @appdirect/sfb-theme-components/src/constants/components/footer.js
import { SMALL, MEDIUM, LARGE } from '../sizes';
import { LEFT, RIGHT, TOP_LEFT, TOP_RIGHT, BOTTOM_LEFT, BOTTOM_RIGHT } from '../alignment';
import { THEME_PRIMARY, THEME_SECONDARY } from '../buttons';
import { INPUT, IMAGE_UPLOADER } from '../schemaComponentTypes';
import { URL_VALIDATION } from '../regexConstants';
import { socialMediasMock } from '../../../mocks/socials/socialMedias';

export const DEFAULT_LOGO_SIZE = 125;

export const CLASSIC_LAYOUT = 'classic';
export const NEWSLETTER_SOCIAL_LAYOUT = 'newsletter_social';
export const SITEMAP_LAYOUT = 'sitemap';
export const FEATURE_BAR_LAYOUT = 'feature_bar';

export const FOOTER_LAYOUT_OPTIONS = [
    { value: CLASSIC_LAYOUT, label: 'uieditor.sfbComponent.footer.style.option.classic' },
    { value: NEWSLETTER_SOCIAL_LAYOUT, label: 'uieditor.sfbComponent.footer.style.option.newsletter_social' },
    { value: SITEMAP_LAYOUT, label: 'uieditor.sfbComponent.footer.style.option.sitemap' }
    // FIXME: https://appdirect.jira.com/browse/MP-34811
    // { value: FEATURE_BAR_LAYOUT, label: 'uieditor.sfbComponent.footer.style.option.feature_bar' }
];
export const FOOTER_NEWSLETTER_BUTTON_STYLE_OPTIONS = [
    { value: THEME_PRIMARY, label: 'uieditor.sfbComponent.footer.newsletterButtonStyle.option.primary' },
    { value: THEME_SECONDARY, label: 'uieditor.sfbComponent.footer.newsletterButtonStyle.option.secondary' }
];
export const FOOTER_SIZES_OPTIONS = [SMALL, MEDIUM, LARGE];
export const FOOTER_NEWSLETTER_POSITION_OPTIONS = [LEFT, RIGHT];
export const FOOTER_NEWSLETTER_LOGO_POSITION_OPTIONS = [LEFT, RIGHT];
export const FOOTER_LOGO_POSITION_OPTIONS = [TOP_LEFT, BOTTOM_LEFT, TOP_RIGHT, BOTTOM_RIGHT];
export const FOOTER_SOCIAL_ICON_POSITION_OPTIONS = [LEFT, RIGHT];

export const FOOTER_PADDING_SMALL = 25;
export const FOOTER_PADDING_MEDIUM = 50;
export const FOOTER_PADDING_LARGE = 80;

export const DEFAULT_SOCIAL_MEDIAS = {
    items: socialMediasMock,
    prompts: {
        delete: {
            title: 'uieditor.sfbComponent.footer.socialIcons.delete.title',
            description: 'uieditor.sfbComponent.footer.socialIcons.delete.message',
            submitButtonLabel: 'uieditor.sfbComponent.footer.socialIcons.delete.account',
            cancelButtonLabel: 'uieditor.cancel.button'
        }
    },
    form: {
        accountLink: {
            title: 'uieditor.sfbComponent.footer.socialIcons.accountLink.title',
            placeholder: 'uieditor.sfbComponent.footer.socialIcons.accountLink.placeholder',
            defaultValue: '',
            type: INPUT,
            required: false,
            validation: {
                pattern: URL_VALIDATION,
                message: 'uieditor.sfbComponent.input.validation.message'
            }
        },
        uploadIcon: {
            title: 'uieditor.sfbComponent.footer.socialIcons.uploadIcon.title',
            caption: 'uieditor.editorProperties.globalStyles.logo.logoImage.caption',
            placeholder: 'uieditor.sfbComponent.footer.socialIcons.uploadIcon.placeholder',
            buttonText: 'uieditor.sfbComponent.footer.socialIcons.uploadIcon.buttonText',
            defaultValue: '',
            type: IMAGE_UPLOADER,
            required: false
        }
    },
    contextMenuOptions: ['SHOW_HIDE']
};
