// @appdirect/sfb-theme-components/src/constants/components/hero-banner.js
import { SMALL, MEDIUM, LARGE, XSMALL_255, SMALL_280, MEDIUM_350, LARGE_466,
    XLARGE_600 } from '../sizes';
import { LEFT, RIGHT, CENTER, TOP, MIDDLE, BOTTOM, TOP_LEFT, TOP_CENTER, TOP_RIGHT, MIDDLE_LEFT,
    MIDDLE_CENTER, MIDDLE_RIGHT, BOTTOM_LEFT, BOTTOM_CENTER, BOTTOM_RIGHT } from '../alignment';
import { PRIMARY, SECONDARY, LINK } from './schemas/button-types';
import { SCALE_TO_FILL, SCALE_TO_FIT } from './schemas/image-sizes';

export const CONTENT_WIDTH = 632;
export const CONTENT_WIDTH_IMAGE_STYLE = 412; // since image takes half the page
export const SEARCH_BAR_MIN_WIDTH = 225;

export const HERO_STYLE = 'hero';
export const IMAGE_STYLE = 'image';
export const VIDEO_STYLE = 'video';

export const HERO_STYLE_OPTION = {
    value: HERO_STYLE,
    label: 'uieditor.sfbComponent.heroBanner.layout.option.hero'
};

export const IMAGE_STYLE_OPTION = {
    value: IMAGE_STYLE,
    label: 'uieditor.sfbComponent.heroBanner.layout.option.image'
};

export const VIDEO_STYLE_OPTION = {
    value: VIDEO_STYLE,
    label: 'uieditor.sfbComponent.heroBanner.layout.option.video'
};

// FIXME: VIDEO will be implemented later SFB-3984
export const HERO_BANNER_STYLE_OPTIONS = [HERO_STYLE_OPTION, IMAGE_STYLE_OPTION];
export const HERO_BANNER_STYLE_OPTIONS_VALUES = HERO_BANNER_STYLE_OPTIONS.map(option => option.value);

export const SIZES_OPTIONS = [XSMALL_255, SMALL_280, MEDIUM_350, LARGE_466, XLARGE_600];
export const SIZES_OPTIONS_VALUES = SIZES_OPTIONS.map(option => option.value);

export const SCALE_TO_FIT_PAGE = {
    value: 'scaleToFitPage',
    label: 'uieditor.sfbComponent.imageSize.option.scaleToFitPage'
};

export const SCALE_TO_FIT_SCREEN = {
    value: 'scaleToFitScreen',
    label: 'uieditor.sfbComponent.imageSize.option.scaleToFitScreen'
};

export const IMAGE_SCALE_TYPES = [SCALE_TO_FIT_PAGE, SCALE_TO_FIT_SCREEN];
export const IMAGE_SCALE_TYPES_VALUES = IMAGE_SCALE_TYPES.map(option => option.value);

export const TEXT_ALIGN_OPTIONS = [
    LEFT, CENTER, RIGHT
];
export const TEXT_ALIGN_OPTIONS_VALUES = TEXT_ALIGN_OPTIONS.map(option => option.value);

export const TEXT_POSITION_OPTIONS = [
    TOP_LEFT, TOP_CENTER, TOP_RIGHT,
    MIDDLE_LEFT, MIDDLE_CENTER, MIDDLE_RIGHT,
    BOTTOM_LEFT, BOTTOM_CENTER, BOTTOM_RIGHT
];

export const TEXT_SIZE_OPTIONS = [SMALL, MEDIUM, LARGE];
export const TEXT_SIZE_OPTIONS_VALUES = TEXT_SIZE_OPTIONS.map(option => option.value);

export const BUTTON_TYPES = [PRIMARY, SECONDARY, LINK];
export const BUTTON_TYPES_VALUES = BUTTON_TYPES.map(option => option.value);

export const FOREGROUND_IMAGE_ALIGN_OPTIONS = [
    LEFT,
    RIGHT
];

export const IMAGE_STYLE_TEXT_POSITION_OPTIONS = [
    TOP,
    MIDDLE,
    BOTTOM
];

export const FOREGROUND_IMAGE_SIZE_TYPES = [SCALE_TO_FILL, SCALE_TO_FIT];
