// @appdirect/sfb-theme-components/src/constants/components/product-image-viewer.js
export const MAX_THUMBNAILS_VISIBLE = 4;
export const LAYOUT_VERTICAL = 'vertical-thumbnails';
export const LAYOUT_HORIZONTAL = 'horizontal-thumbnails';
export const LAYOUT_TITLES = 'titles';
export const LAYOUT_CARETS = 'carets';

export const LAYOUT_OPTIONS = [
    {
        label: 'uieditor.sfbComponent.productImageViewer.layout.option.horizontalThumbnails',
        value: LAYOUT_HORIZONTAL
    },
    // temporarily disabled due to it causing issues with content rendering on the page
    // {
    //     label: 'uieditor.sfbComponent.productImageViewer.layout.option.verticalThumbnails',
    //     value: LAYOUT_VERTICAL
    // },
    {
        label: 'uieditor.sfbComponent.productImageViewer.layout.option.slideTitles',
        value: LAYOUT_TITLES
    },
    {
        label: 'uieditor.sfbComponent.productImageViewer.layout.option.caretsOnly',
        value: LAYOUT_CARETS
    }
];

export const NEXT = 'next';
export const PREVIOUS = 'previous';
export const ARROW_DIRECTIONS = [NEXT, PREVIOUS];

export const ARROW_DEFAULT_PROPS = {
    i18nWithDefaults: {
        previous: 'Previous',
        next: 'Next'
    },
    caretOnly: false,
    currentSlide: 0,
    slideCount: 0
};
