// @appdirect/sfb-theme-components/src/components/order-confirmation/stories/OrderConfirmation.stories.js
import React from 'react';
import { orderConfirmationMockData } from '../../../../mocks/customer/orderConfirmation';
import OrderConfirmation from '../OrderConfirmation';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

export default {
    title: 'Components/Order Confirmations',
    component: OrderConfirmation,
    argTypes: {
        ...getArgTypes(OrderConfirmation.schema().form),
        iconUrl: {
            control: 'text',
            description: 'The URL of the icon'
        },
        viewAppsIconUrl: {
            control: 'text',
            description: 'The URL of the view apps icon'
        },
        assignProductsIconUrl: {
            control: 'text',
            description: 'The URL of the assign products icon'
        },
        continueShoppingIconUrl: {
            control: 'text',
            description: 'The URL of the continue shopping icon'
        }
    },
    render: ({
        iconUrl,
        viewAppsIconUrl,
        assignProductsIconUrl,
        continueShoppingIconUrl,
        ...args
    }) => (
        <OrderConfirmation
            data={orderConfirmationMockData}
            settings={{
                ...args,
                icon: { url: iconUrl },
                viewAppsIcon: { url: viewAppsIconUrl },
                assignProductsIcon: { url: assignProductsIconUrl },
                continueShoppingIcon: { url: continueShoppingIconUrl }
            }}
        />
    ),
    parameters: {
        controls: {
            include: [...Object.keys(OrderConfirmation.schema().form), 'iconUrl', 'viewAppsIconUrl', 'viewAppsText', 'assignProductsIconUrl', 'continueShoppingIconUrl'],
            exclude: ['icon', 'viewAppsIcon', 'viewApps', 'assignProductsIcon', 'continueShoppingIcon']
        }
    }
};

export const Default = {
    args: {
        ...getArgs(OrderConfirmation.schema().form),
        viewAppsIconUrl: 'https://placehold.co/100/black/white',
        assignProductsIconUrl: 'https://placehold.co/100/black/white',
        continueShoppingIconUrl: 'https://placehold.co/100/black/white'
    }
};

export const WithSettings = {
    args: {
        ...getArgs(OrderConfirmation.schema().form),
        viewAppsIconUrl: 'https://placehold.co/100/black/white',
        assignProductsIconUrl: 'https://placehold.co/100/black/white',
        continueShoppingIconUrl: 'https://placehold.co/100/black/white'
    }
};
