// @appdirect/sfb-theme-components/mocks/application/editions/items.js
export const itemsMock = [
    {
        name: 'Starter',
        id: 579438,
        primary: true,
        plans: [
            {
                costs: {
                    unitPrice: [
                        {
                            id: 'SMS_TEXT_MESSAGE',
                            usageRange: [
                                {
                                    formattedPrice: 'Free',
                                    formattedFullPrice: 'Free / SMS Text Message / Month',
                                    formattedFullPriceElements: [
                                        'Free',
                                        '/ SMS Text Message',
                                        '/ Month'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        label: 'SMS Text Message',
                                        dependency: '',
                                        id: 'SMS_TEXT_MESSAGE'
                                    },
                                    frequency: {
                                        shortLabel: 'Mo',
                                        label: 'Month',
                                        id: 'MONTHLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 0,
                                        value: 0,
                                        originalValue: 0,
                                        discountValue: 0,
                                        editionStartingPrice: 25,
                                        formatted: {
                                            originalPrice: '$0.00',
                                            originalFullPrice: '$0.00 / SMS Text Message / Month',
                                            originalFullPriceElements: [
                                                '$0.00',
                                                '/ SMS Text Message',
                                                '/ Month'
                                            ]
                                        }
                                    }
                                }
                            ]
                        }
                    ],
                    flatPrice: {
                        formattedPrice: '$25.00',
                        formattedFullPrice: '$25.00 / Month',
                        formattedFullPriceElements: [
                            '$25.00',
                            null,
                            '/ Month'
                        ],
                        unit: {
                            min: 0,
                            max: null,
                            dependency: '',
                            id: 'NOT_APPLICABLE'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 25,
                            value: 25,
                            originalValue: 25,
                            discountValue: 0,
                            editionStartingPrice: 25,
                            formatted: {
                                originalPrice: '$25.00',
                                originalFullPrice: '$25.00 / Month',
                                originalFullPriceElements: [
                                    '$25.00',
                                    null,
                                    '/ Month'
                                ]
                            }
                        }
                    }
                },
                setupFees: [],
                contractFees: [],
                contractLength: null,
                formattedContract: false,
                discount: false,
                uuid: 'abf5cfbe-00e7-4849-bd1c-6e964ff6d0f2'
            }
        ],
        bullets: [],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: true,
            eppid: 'abf5cfbe-00e7-4849-bd1c-6e964ff6d0f2',
            enabled: true,
            url: 'https://sfbmarketplace.byappdirect.com/api/internal/storefront/v1/cta?productId=293052&type=BUY&editionId=579438&eppid=abf5cfbe-00e7-4849-bd1c-6e964ff6d0f2',
            actionType: 'BUY',
            popup: {},
            isSecondary: false
        },
        audience: [
            {
                content: 'New businesses'
            }
        ],
        items: [
            {
                content: 'Unlimited Users'
            },
            {
                content: '50 Credits'
            },
            {
                content: '50 SMS Text Messages'
            }
        ],
        invisible: false,
        leadGen: false,
        revenueType: 'RECURRING'
    },
    {
        name: 'Growing',
        id: 579441,
        primary: false,
        plans: [
            {
                costs: {
                    unitPrice: [
                        {
                            id: 'SMS_TEXT_MESSAGE',
                            usageRange: [
                                {
                                    formattedPrice: 'Free',
                                    formattedFullPrice: 'Free / SMS Text Message / Month',
                                    formattedFullPriceElements: [
                                        'Free',
                                        '/ SMS Text Message',
                                        '/ Month'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        label: 'SMS Text Message',
                                        dependency: '',
                                        id: 'SMS_TEXT_MESSAGE'
                                    },
                                    frequency: {
                                        shortLabel: 'Mo',
                                        label: 'Month',
                                        id: 'MONTHLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 0,
                                        value: 0,
                                        originalValue: 0,
                                        discountValue: 0,
                                        editionStartingPrice: 69,
                                        formatted: {
                                            originalPrice: '$0.00',
                                            originalFullPrice: '$0.00 / SMS Text Message / Month',
                                            originalFullPriceElements: [
                                                '$0.00',
                                                '/ SMS Text Message',
                                                '/ Month'
                                            ]
                                        }
                                    }
                                }
                            ]
                        }
                    ],
                    flatPrice: {
                        formattedPrice: '$69.00',
                        formattedFullPrice: '$69.00 / Month',
                        formattedFullPriceElements: [
                            '$69.00',
                            null,
                            '/ Month'
                        ],
                        unit: {
                            min: 0,
                            max: null,
                            dependency: '',
                            id: 'NOT_APPLICABLE'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 69,
                            value: 69,
                            originalValue: 69,
                            discountValue: 0,
                            editionStartingPrice: 69,
                            formatted: {
                                originalPrice: '$69.00',
                                originalFullPrice: '$69.00 / Month',
                                originalFullPriceElements: [
                                    '$69.00',
                                    null,
                                    '/ Month'
                                ]
                            }
                        }
                    }
                },
                setupFees: [],
                contractFees: [],
                contractLength: null,
                formattedContract: false,
                discount: false,
                uuid: 'd71d8b5a-c112-46b2-9bdd-d6c4e75f939d'
            }
        ],
        bullets: [
            {
                content: 'Forms Addon',
                tooltip: 'Create site audits, procedures, and checklists in the field, then immediately turn them into professional PDF reports in the field.'
            },
            {
                content: 'Email Inbox Addon',
                tooltip: 'Convert emailed work orders into jobs. Work Order PDFs are scanned automatically and populate jobs with information.'
            }
        ],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            eppid: 'd71d8b5a-c112-46b2-9bdd-d6c4e75f939d',
            enabled: true,
            url: 'https://sfbmarketplace.byappdirect.com/api/internal/storefront/v1/cta?productId=293052&type=BUY&editionId=579441&eppid=d71d8b5a-c112-46b2-9bdd-d6c4e75f939d',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [
            {
                content: 'Businesses with up to 5 staff'
            }
        ],
        items: [
            {
                content: 'Unlimited Users'
            },
            {
                content: '150 Credits'
            },
            {
                content: '150 SMS Text Messages'
            }
        ],
        invisible: false,
        leadGen: false,
        revenueType: 'RECURRING'
    },
    {
        name: 'Premium Plus',
        id: 579447,
        primary: false,
        plans: [
            {
                costs: {
                    unitPrice: [
                        {
                            id: 'SMS_TEXT_MESSAGE',
                            usageRange: [
                                {
                                    formattedPrice: 'Free',
                                    formattedFullPrice: 'Free / SMS Text Message / Month',
                                    formattedFullPriceElements: [
                                        'Free',
                                        '/ SMS Text Message',
                                        '/ Month'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        label: 'SMS Text Message',
                                        dependency: '',
                                        id: 'SMS_TEXT_MESSAGE'
                                    },
                                    frequency: {
                                        shortLabel: 'Mo',
                                        label: 'Month',
                                        id: 'MONTHLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 0,
                                        value: 0,
                                        originalValue: 0,
                                        discountValue: 0,
                                        editionStartingPrice: 299,
                                        formatted: {
                                            originalPrice: '$0.00',
                                            originalFullPrice: '$0.00 / SMS Text Message / Month',
                                            originalFullPriceElements: [
                                                '$0.00',
                                                '/ SMS Text Message',
                                                '/ Month'
                                            ]
                                        }
                                    }
                                }
                            ]
                        },
                        {
                            id: 'CREDIT',
                            usageRange: [
                                {
                                    formattedPrice: '$0.20',
                                    formattedFullPrice: '$0.20 / Credit / Month',
                                    formattedFullPriceElements: [
                                        '$0.20',
                                        '/ Credit',
                                        '/ Month'
                                    ],
                                    unit: {
                                        min: 0,
                                        max: null,
                                        label: 'Credit',
                                        dependency: '',
                                        id: 'CREDIT'
                                    },
                                    frequency: {
                                        shortLabel: 'Mo',
                                        label: 'Month',
                                        id: 'MONTHLY'
                                    },
                                    price: {
                                        symbol: '$',
                                        currency: 'USD',
                                        amount: 0.2,
                                        value: 0.2,
                                        originalValue: 0.2,
                                        discountValue: 0,
                                        editionStartingPrice: 299,
                                        formatted: {
                                            originalPrice: '$0.20',
                                            originalFullPrice: '$0.20 / Credit / Month',
                                            originalFullPriceElements: [
                                                '$0.20',
                                                '/ Credit',
                                                '/ Month'
                                            ]
                                        }
                                    }
                                }
                            ]
                        }
                    ],
                    flatPrice: {
                        formattedPrice: '$299.00',
                        formattedFullPrice: '$299.00 / Month',
                        formattedFullPriceElements: [
                            '$299.00',
                            null,
                            '/ Month'
                        ],
                        unit: {
                            min: 0,
                            max: null,
                            dependency: '',
                            id: 'NOT_APPLICABLE'
                        },
                        frequency: {
                            shortLabel: 'Mo',
                            label: 'Month',
                            id: 'MONTHLY'
                        },
                        price: {
                            symbol: '$',
                            currency: 'USD',
                            amount: 299,
                            value: 299,
                            originalValue: 299,
                            discountValue: 0,
                            editionStartingPrice: 299,
                            formatted: {
                                originalPrice: '$299.00',
                                originalFullPrice: '$299.00 / Month',
                                originalFullPriceElements: [
                                    '$299.00',
                                    null,
                                    '/ Month'
                                ]
                            }
                        }
                    }
                },
                setupFees: [],
                contractFees: [],
                contractLength: null,
                formattedContract: false,
                discount: false,
                uuid: '5e2dab7f-ec3d-43d7-a023-618665b16f3f'
            }
        ],
        bullets: [
            {
                content: 'Forms Addon',
                tooltip: 'Create site audits, procedures, and checklists in the field, then immediately turn them into professional PDF reports in the field.'
            },
            {
                content: 'Email Inbox Addon',
                tooltip: 'Convert emailed work orders into jobs. Work Order PDFs are scanned automatically and populate jobs with information.'
            },
            {
                content: 'Job Costing Addon',
                tooltip: 'See how profitable a job will be while you\'re quoting, and how much you\'ve made after the job is completed.'
            },
            {
                content: 'Margin Billing Addon',
                tooltip: 'Quickly set material and labor prices based on margin.'
            }
        ],
        callToAction: {
            label: 'Buy Now',
            tooltip: '',
            isPrimary: false,
            eppid: '5e2dab7f-ec3d-43d7-a023-618665b16f3f',
            enabled: true,
            url: 'https://sfbmarketplace.byappdirect.com/api/internal/storefront/v1/cta?productId=293052&type=BUY&editionId=579447&eppid=5e2dab7f-ec3d-43d7-a023-618665b16f3f',
            actionType: 'BUY',
            popup: {},
            isSecondary: true
        },
        audience: [
            {
                content: 'High Volume Businesses'
            }
        ],
        items: [
            {
                content: 'Unlimited Users'
            },
            {
                content: '1500 Credits'
            },
            {
                content: '1500 SMS Text Messages'
            }
        ],
        invisible: false,
        leadGen: false,
        revenueType: 'RECURRING'
    }
];
