// @appdirect/sfb-theme-components/mocks/application/editions/plans.js
export const monthlyPlanUsd = {
    costs: {
        unitPrice: [],
        flatPrice: {
            formattedPrice: '$20.00',
            formattedFullPrice: '$20.00 / Month',
            formattedFullPriceElements: [
                '$20.00',
                null,
                '/ Month'
            ],
            unit: {
                min: 0,
                max: null,
                dependency: '',
                id: 'NOT_APPLICABLE'
            },
            frequency: {
                shortLabel: 'Mo',
                label: 'Month',
                id: 'MONTHLY'
            },
            price: {
                symbol: '$',
                currency: 'USD',
                amount: 20,
                value: 20,
                originalValue: 20,
                discountValue: 0,
                editionStartingPrice: 20,
                formatted: {
                    originalPrice: '$20.00',
                    originalFullPrice: '$20.00 / Month',
                    originalFullPriceElements: [
                        '$20.00',
                        null,
                        '/ Month'
                    ]
                }
            }
        }
    },
    setupFees: [],
    contractFees: [],
    contractLength: null,
    formattedContract: false,
    discount: false,
    uuid: 'f2dcc6a4-56e9-45da-bc29-da8189ed040a'
};
