// @appdirect/sfb-theme-components/src/constants/prop-types/application/configurator/index.js
import PropTypes from 'prop-types';

export const configuratorPropTypes = PropTypes.shape({
    productId: PropTypes.number.isRequired,
    crossSellManagementEnabled: PropTypes.bool,
    i18n: PropTypes.shape({
        selectedEdition: PropTypes.string,
        selectedAddons: PropTypes.string,
        selectedProduct: PropTypes.string,
        startingAt: PropTypes.string,
        showLess: PropTypes.string,
        more: PropTypes.string,
        cancel: PropTypes.string,
        closeFullscreen: PropTypes.string
    }),
    editionInCartItemId: PropTypes.string
});

export const selectedEditionPropTypes = PropTypes.shape({
    name: PropTypes.string,
    productEditionRefId: PropTypes.string,
    plans: PropTypes.arrayOf(
        PropTypes.shape({
            costs: PropTypes.shape({
                unitPrice: PropTypes.arrayOf(
                    PropTypes.shape({
                        usageRange: PropTypes.arrayOf(
                            PropTypes.shape({
                                formattedFullPriceElements: PropTypes.array
                            })
                        )
                    })
                )
            })
        })
    )
});
