// @appdirect/sfb-theme-components/src/components/featured-product/stories/FeaturedProduct.stories.js
import React from 'react';
import FeaturedProduct from '../FeaturedProduct';
import { featuredProductSettings,
    mockProduct,
    mockProductFree,
    featuredProductSettingsNoImage } from '../../../../mocks/components/featuredProduct';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

export default {
    title: 'Components/Featured Product',
    component: FeaturedProduct,
    argTypes: {
        ...getArgTypes(FeaturedProduct.schema().form)
    },
    parameters: {
        controls: {
            include: Object.keys(FeaturedProduct.schema().form),
            sort: 'alpha'
        }
    }
};

export const Default = {
    render: () => (
        <div>
            <FeaturedProduct settings={featuredProductSettings} />
        </div>
    )
};

export const PriceFree = {
    render: () => (
        <div>
            <FeaturedProduct
                settings={{
                    ...featuredProductSettings,
                    product: mockProductFree
                }}
            />
        </div>
    )
};

export const DefaultImage = {
    render: () => (
        <div>
            <FeaturedProduct settings={featuredProductSettingsNoImage} />
        </div>
    )
};

export const WithSettings = {
    render: args => (
        <div>
            <FeaturedProduct settings={args} />
        </div>
    ),
    args: {
        ...getArgs(FeaturedProduct.schema().form),
        product: mockProduct
    }
};
