// @appdirect/sfb-theme-components/src/components/product-banner/stories/ProductBanner.stories.js
import React from 'react';
import { productBannerMock,
    productBannerMultiButtonMock,
    productBannerCaptionMock,
    productBannerNoImageMock,
    productBannerNoActionMock,
    bundleBannerMock } from '../../../../mocks/product/productBanner';
import ProductBanner, { ProductBannerComponent } from '../ProductBanner';
import BundleBanner from '../BundleBanner';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

const getWrappedArgTypes = () => ({
    ...getArgTypes(ProductBannerComponent.schema().form),
    backgroundImageUrl: {
        control: 'text',
        if: {
            arg: 'backgroundType',
            eq: 'image'
        },
        description: 'The URL of the background image'
    },
    bannerBackgroundColor: {
        control: 'color',
        description: 'The background color of the banner',
        if: {
            arg: 'backgroundType',
            eq: 'color'
        }
    }
});

const getWrappedArgs = () => ({
    backgroundImageUrl: 'https://placehold.co/600x400',
    ...getArgs(ProductBannerComponent.schema().form)
});

export default {
    title: 'components/Product Banner',
    component: ProductBannerComponent,
    argTypes: getWrappedArgTypes(),
    parameters: {
        controls: {
            include: Object.keys(getWrappedArgTypes()),
            exclude: ['backgroundImg']
        }
    }
};

export const Default = {
    render: ({ backgroundImageUrl, ...args }) => (
        <ProductBanner
            data={productBannerMock}
            settings={{
                ...args,
                backgroundImg: { url: backgroundImageUrl }
            }}
        />
    ),
    args: getWrappedArgs()
};

export const WithMissingImage = {
    render: ({ backgroundImageUrl, ...args }) => (
        <ProductBanner
            data={productBannerNoImageMock}
            settings={{
                ...args,
                backgroundImg: { url: backgroundImageUrl }
            }}
        />
    ),
    args: getWrappedArgs()
};

export const WithExtraButtons = {
    render: ({ backgroundImageUrl, ...args }) => (
        <ProductBanner
            data={productBannerMultiButtonMock}
            settings={{
                ...args,
                backgroundImg: { url: backgroundImageUrl }
            }}
        />
    ),
    args: getWrappedArgs()
};

export const WithSubtext = {
    render: ({ backgroundImageUrl, ...args }) => (
        <ProductBanner
            data={productBannerCaptionMock}
            settings={{
                ...args,
                backgroundImg: { url: backgroundImageUrl }
            }}
        />
    ),
    args: getWrappedArgs()
};

export const WithSettings = {
    render: ({ backgroundImageUrl, ...args }) => (
        <ProductBanner
            data={productBannerMock}
            settings={{
                ...args,
                backgroundImg: { url: backgroundImageUrl }
            }}
        />
    ),
    args: getWrappedArgs()
};

export const WithNoActions = {
    render: ({ backgroundImageUrl, ...args }) => (
        <ProductBanner
            data={productBannerNoActionMock}
            settings={{
                ...args,
                backgroundImg: { url: backgroundImageUrl }
            }}
        />
    ),
    args: getWrappedArgs()
};
export const WithBundleBanner = {
    render: ({ backgroundImageUrl, ...args }) => (
        <BundleBanner
            data={bundleBannerMock}
            settings={{
                ...args,
                backgroundImg: { url: backgroundImageUrl }
            }}
        />
    ),
    args: getWrappedArgs()
};
