// @appdirect/sfb-theme-components/src/components/product-additional-info/stories/AdditionalInfo.stories.js
import React from 'react';
import AdditionalInfo from '../AdditionalInfo';
import { defaultMock as additionalInfoMock, noSidebarMock } from '../../../../mocks/components/productAdditionalInfo';
import { getArgTypes, getArgs } from '../../../tools/controlsHelper';

export default {
    title: 'Components/Product Additional Info',
    component: AdditionalInfo,
    argTypes: getArgTypes(AdditionalInfo.schema().form)
};

export const Default = {
    render: args => <AdditionalInfo data={additionalInfoMock} settings={args} />,
    args: getArgs(AdditionalInfo.schema().form)
};

export const NoSidebar = {
    render: args => <AdditionalInfo data={noSidebarMock} settings={args} />,
    args: getArgs(AdditionalInfo.schema().form),
    parameters: {
        controls: {
            exclude: ['sideBar', 'sideBarBorder']
        }
    }
};

export const WithSettings = {
    render: args => <AdditionalInfo data={additionalInfoMock} settings={args} />,
    args: getArgs(AdditionalInfo.schema().form)
};
