// @appdirect/sfb-theme-components/src/composite-demos/list-page/stories/ListPage.stories.jsx
import React from 'react';

import { listingProductsMock } from '../../../../mocks/listingProducts';
import { productFiltersMock } from '../../../../mocks/product/productFilters';

import { ProductListingComponent } from '../../../components/product-listing/ProductListing';
import { CategorySidebarComponent } from '../../../components/category-sidebar/CategorySidebar';
import { ProductListingControlsComponent } from '../../../components/product-listing-controls/ProductListingControls';
import { ListPage } from '../ListPage';

const sidebarMockWithParams = {
    ...productFiltersMock,
    listing: {
        items: listingProductsMock.items,
        pagination: listingProductsMock.pagination,
        i18n: listingProductsMock.i18n
    }
};

export default {
    title: 'Composite Demos/Product Listing Page',
    component: ListPage,
    parameters: {
        layout: 'fullscreen',
        container: 'fluid'
    }
};

export const Default = () => {
    const listing = <ProductListingComponent data={listingProductsMock} />;
    const sidebar = <CategorySidebarComponent data={sidebarMockWithParams} />;
    const controls = <ProductListingControlsComponent data={listingProductsMock} />;

    return <ListPage listing={listing} sidebar={sidebar} controls={controls} />;
};
