// @appdirect/sfb-theme-components/src/atoms/product-edition/list/stories/ProductEditionList.stories.js
import React from 'react';
import { productEditionListMockData as editionsData,
    productEditionListSkipConfiguratorMockData } from '../../../../../mocks/product/productEditions';
import { ProductEditionListComponent as ProductEditionList } from '../ProductEditionList';

export default {
    title: 'Atoms/Product Edition List',
    component: ProductEditionList
};

export const Default = {
    render: () => <ProductEditionList data={editionsData} />,
    name: 'Default'
};

export const Carousel = {
    render: () => (
        <div
            style={{
                maxWidth: 600,
                margin: '0 auto'
            }}
        >
            <ProductEditionList
                data={{
                    ...editionsData,
                    editions: editionsData.editions.slice(0, 4)
                }}
                carouselEnabled
            />
        </div>
    ),

    name: 'Carousel'
};

export const ConfiguratorMode = {
    render: () => <ProductEditionList data={editionsData} configuratorMode />,
    name: 'Configurator mode'
};

export const SkipConfiguratorModeAddToCart = {
    render: () => <ProductEditionList data={productEditionListSkipConfiguratorMockData} />,
    name: 'Skip configurator mode - Add to Cart'
};
