// @appdirect/sfb-theme-components/src/atoms/button/stories/Button.stories.js
import React from 'react';
import Button, { allowedThemes as buttonThemes } from '../Button';
import Icon from '../../icon/Icon';

export default {
    title: 'Atoms/Button',
    component: Button
};

export const WithStyles = {
    render: () => (
        <>
            {buttonThemes.map(type => {
                const states = ['', 'active', 'disabled'];

                return (
                    <div
                        key={type}
                        style={{
                            padding: '5px 10px'
                        }}
                    >
                        {states.map(state => {
                            const props = {
                                type,
                                key: `${type}-${state}`,
                                ...state === 'active' && {
                                    active: true
                                },
                                ...state === 'disabled' && {
                                    disabled: true
                                }
                            };

                            return (
                                <span key={state}>
                                    <Button {...props}>{`${type} ${state}`}</Button>
                                </span>
                            );
                        })}
                    </div>
                );
            })}
        </>
    ),

    name: 'With styles'
};

export const Wide = {
    render: () => (
        <Button type="primary" wide>
            Wide Button
        </Button>
    ),
    name: 'Wide'
};

export const Small = {
    render: () => (
        <Button type="primary" size="small">
            Small button
        </Button>
    ),
    name: 'Small'
};

export const Large = {
    render: () => (
        <Button type="card" size="large">
            Large button
        </Button>
    ),
    name: 'Large'
};

export const Links = {
    render: () => (
        <div
            style={{
                padding: '5px 10px'
            }}
        >
            <Button buttonType="link">Default Link</Button>
        </div>
    ),

    name: 'Links'
};

export const WithIcons = {
    render: () => (
        <>
            <div
                style={{
                    padding: '5px 10px'
                }}
            >
                <Button buttonType="link">
                    View More
                    <Icon
                        type="caretDown"
                        style={{
                            width: '16px',
                            marginLeft: '12px'
                        }}
                    />
                </Button>
            </div>
            <div
                style={{
                    padding: '5px 10px'
                }}
            >
                <Button type="primary" buttonType="link">
                    View More
                    <Icon
                        type="caretDown"
                        style={{
                            width: '16px',
                            marginLeft: '12px'
                        }}
                    />
                </Button>
            </div>
            <div
                style={{
                    padding: '5px 10px'
                }}
            >
                <Button buttonType="icon" aria-label="Close">
                    <Icon type="close" width="24px" />
                </Button>
            </div>
        </>
    ),

    name: 'With icons'
};
