// @appdirect/sfb-theme-components/src/components/domain-app/stories/DomainApp.stories.js
import React from 'react';
import DomainApp, { DomainAppComponent } from '../DomainApp';
import { domainAppMockCheckoutV2,
    domainAppMockCheckoutV2DisabledCta,
    domainAppMockCheckoutV1 } from '../../../../mocks/product/domainApp';
import { fetchSuggestionsMock } from '../../../../mocks/product/domainSuggestionFetch';

export default {
    title: 'Components/Domain App',
    component: DomainApp
};

export const Default = {
    render: () => (
        <div
            style={{
                width: '100%'
            }}
        >
            <DomainAppComponent data={domainAppMockCheckoutV2} fetchSuggestions={fetchSuggestionsMock} />
        </div>
    ),

    name: 'Default'
};

export const WithoutCallToAction = {
    render: () => (
        <div
            style={{
                width: '100%'
            }}
        >
            <DomainAppComponent data={domainAppMockCheckoutV2DisabledCta} fetchSuggestions={fetchSuggestionsMock} />
        </div>
    ),

    name: 'Without Call To Action'
};

export const CheckoutVersion1 = {
    render: () => (
        <div
            style={{
                width: '100%'
            }}
        >
            <DomainAppComponent data={domainAppMockCheckoutV1} fetchSuggestions={fetchSuggestionsMock} />
        </div>
    ),

    name: 'Checkout version 1'
};
