// @appdirect/sfb-theme-components/mocks/comparableProducts/items.js
export const itemsMock = [
    {
        id: 13960,
        name: 'DreamFactory',
        description: 'Open Source REST API Platform',
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/13960/thumbs_64/img2319685171585638435.png',
        iconAlt: 'Logo of DreamFactory',
        myAppLogoIconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/13960/myAppIcon/img2319685171585638435.png',
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/13960/thumbs_112/img5599356460088711979.png',
        features: [
            'Install On Any Server',
            'Connect To Any Database',
            'Develop For Any Device',
            'Enterprise Security',
            'Admin Console',
            'App Hosting',
            'User Management',
            'SQL Services',
            'NoSQL Data',
            'BLOB Storage',
            'External Integration'
        ],
        startingPrice: {
            formattedPrice: '$10.00',
            price: '10.00',
            priceCaption: 'Starting At',
            frequency: {
                label: 'Month',
                shortLabel: 'Mo',
                id: 'MONTHLY'
            },
            symbol: '$',
            currency: 'USD'
        },
        tags: [
            'Mobile Development',
            'Open Source'
        ],
        developer: 'DreamFactory Software, Inc.',
        rating: 5,
        numberOfReview: 9,
        editionNames: [
            'Community Product Support',
            'Individual Product Support',
            'Team Product Support'
        ],
        url: '/en-US/apps/13960/dreamfactory'
    },
    {
        id: 42937,
        name: 'AppWise',
        description: 'Search everything, in one shot',
        iconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_64/img63732497944352803.png',
        iconAlt: 'Logo of AppWise',
        myAppLogoIconUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/myAppIcon/img6640948475444359586.png',
        profileLogoUrl: 'https://d3bql97l1ytoxn.cloudfront.net/app_resources/42937/thumbs_112/img1099429097555068136.png',
        features: [
            'Unified Search'
        ],
        startingPrice: {
            formattedPrice: 'Free',
            price: 'Free',
            priceCaption: '',
            frequency: {
                label: '',
                shortLabel: '',
                id: ''
            }
        },
        tags: [
            'Office Productivity',
            'Private Cloud',
            'Public Cloud',
            'Customer Relationship Management',
            'Help Desk & Ticketing',
            'English (US)',
            'Android',
            'iPhone',
            'iPad',
            'Linux',
            'Mac',
            'Mobile Web App',
            'Web-based',
            'Windows',
            'Windows Phone',
            'Marketing',
            'Single Sign-On',
            'User Provisioning'
        ],
        developer: 'AppWise, Inc.',
        rating: 5,
        numberOfReview: 139,
        editionNames: [
            'Free',
            'Professional',
            'Professional (Annual)',
            'Enterprise'
        ],
        url: '/en-US/apps/42937/appwise'
    },
    {
        id: 255123,
        name: 'New Basic Demo',
        description: null,
        iconUrl: null,
        iconAlt: null,
        myAppLogoIconUrl: null,
        profileLogoUrl: null,
        features: [],
        startingPrice: {
            formattedPrice: '$10.00',
            price: '10.00',
            priceCaption: 'Starting At',
            frequency: {
                label: 'Month',
                shortLabel: 'Mo',
                id: 'MONTHLY'
            },
            symbol: '$',
            currency: 'USD'
        },
        tags: [
            'Configure Price Quote'
        ],
        developer: 'AppDirect',
        rating: null,
        numberOfReview: null,
        editionNames: [
            'Recurring Edition'
        ],
        url: '/en-US/apps/255123/new-basic-demo'
    }
];
