// @appdirect/sfb-theme-components/mocks/product/overview/benefits.js
export const benefitsMock = [
    {
        description: 'Full-text search across 40+ cloud apps. AppWise refines your search as you type, to help you drill into individual apps or look for documents or presentations, wherever they are.',
        descriptionLong: 'Full-text search across 40+ cloud apps. AppWise refines your search as you type, to help you drill into individual apps or look for documents or presentations, wherever they are.',
        title: 'Unified search across multiple accounts',
        type: 'Benefit'
    },
    {
        description: 'Build and save complex filters from service, content type, date range combinations. Get powerful \'proximity searches\' used by legal professionals, or simply refine by author, title or document type.',
        descriptionLong: 'Build and save complex filters from service, content type, date range combinations. Get powerful \'proximity searches\' used by legal professionals, or simply refine by author, title or document type.',
        title: 'Get a personal perspective across all your content',
        type: 'Benefit'
    },
    {
        description: 'Conveniently preview documents, presentations and more without leaving your search.',
        descriptionLong: 'Conveniently preview documents, presentations and more without leaving your search.',
        title: 'Boost productivity by minimizing context switching',
        type: 'Benefit'
    },
    {
        description: 'Don\'t worry about missing critical project updates, or new information about customer prospects. Get pro-actively emailed when your connected accounts are updated about topics you\'re tracking. You can even cc a group email list or an assistant to feed into company workflows.',
        descriptionLong: 'Don\'t worry about missing critical project updates, or new information about customer prospects. Get pro-actively emailed when your connected accounts are updated about topics you\'re tracking. You can even cc a group email list or an assistant to feed into company workflows.',
        title: 'Stay on top of developments across your company',
        type: 'Benefit'
    },
    {
        description: 'AppWise works seamlessly with your existing security and Identity and Access Management infrastructure. AppWise is pre-integrated with AppDirect, Okta, OneLogin and support SAML 2.0.',
        descriptionLong: 'AppWise works seamlessly with your existing security and Identity and Access Management infrastructure. AppWise is pre-integrated with AppDirect, Okta, OneLogin and support SAML 2.0.',
        title: 'Secured and integrated with your company',
        type: 'Benefit'
    },
    {
        description: 'Many companies don\'t work completely in \'the cloud\'. AppWise provides an SDK which can be used to connect on-premises and/or proprietary content sources into AppWise so you get the benefits of cloud-based rapid deployment but with a holistic view over your content.',
        descriptionLong: 'Many companies don\'t work completely in \'the cloud\'. AppWise provides an SDK which can be used to connect on-premises and/or proprietary content sources into AppWise so you get the benefits of cloud-based rapid deployment but with a holistic view over your content.',
        title: 'Bridge On-Prem and Cloud environments',
        type: 'Benefit'
    },
    {
        description: 'AppWise has a comprehensive REST-based API which can be used to integrate with existing user-experiences, enterprise workflows and mobile applications.',
        descriptionLong: 'AppWise has a comprehensive REST-based API which can be used to integrate with existing user-experiences, enterprise workflows and mobile applications.',
        title: 'Enhance existing enterprise workflows',
        type: 'Benefit'
    },
    {
        id: 4132314,
        title: 'Unified Search',
        description: 'Search all your content across all your accounts, with one, powerful search. Save time...',
        descriptionLong: 'Search all your content across all your accounts, with one, powerful search. Save time gathering information and re-using presentations, documents and templates that have already been created inside your company.',
        href: '/en-US/apps/42937/appwise/features',
        linkLabel: 'Learn More',
        slogan: 'Search Everything, in one shot',
        type: 'Feature'
    }
];
