// @appdirect/sfb-theme-components/src/components/header/atoms/common/header-cart-overlay/stories/HeaderCartOverlay.stories.js
import React from 'react';
import HeaderCartOverlay from '../HeaderCartOverlay';
import { cartMock, cartEmptyMock } from '../../../../../../../mocks/cart';

const parentProps = {
    style: {
        maxWidth: 420
    }
};

export default {
    title: 'Global Components/Header/Atoms/Header Cart Overlay',
    component: HeaderCartOverlay
};

export const Default = {
    render: () => (
        <div {...parentProps}>
            <HeaderCartOverlay cart={cartMock} isLoggedIn />
        </div>
    ),

    name: 'Default'
};

export const Empty = {
    render: () => (
        <div {...parentProps}>
            <HeaderCartOverlay cart={cartEmptyMock} isLoggedIn />
        </div>
    ),

    name: 'Empty'
};

export const UserNotConnected = {
    render: () => (
        <div {...parentProps}>
            <HeaderCartOverlay isLoggedIn={false} />
        </div>
    ),

    name: 'User not connected'
};

export const Loading = {
    render: () => (
        <div {...parentProps}>
            <HeaderCartOverlay isLoading />
        </div>
    ),

    name: 'Loading'
};

export const NewProductAdded = {
    render: () => (
        <div {...parentProps}>
            <HeaderCartOverlay cart={cartMock} hasNewProductAdded isLoggedIn />
        </div>
    ),

    name: 'New Product Added'
};

export const AddToCartError = {
    render: () => (
        <div {...parentProps}>
            <HeaderCartOverlay cart={cartMock} addToCartError isLoggedIn />
        </div>
    ),

    name: 'Add to Cart Error'
};

export const AlreadyInCartMessage = {
    render: () => (
        <div {...parentProps}>
            <HeaderCartOverlay cart={cartMock} alreadyInCartNotif isLoggedIn />
        </div>
    ),

    name: 'Already in Cart Message'
};
