// @appdirect/sfb-theme-components/src/atoms/resource-card/stories/ResourceCard.stories.js
import React from 'react';
import ResourceCard from '../ResourceCard';
import { resourceCardMock } from '../../../../mocks/product/resourceCard';
import { i18nMock } from '../../../../mocks/application/resources/i18n';

export default {
    title: 'Atoms/ResourceCard',
    component: ResourceCard,
    render: args => <ResourceCard i18n={i18nMock} {...args} />,
    argTypes: {
        link: {
            control: 'text',
            description: 'The link to the resource'
        },
        cardTitle: {
            control: 'text',
            description: 'The title of the resource'
        },
        info: {
            control: 'text',
            description: 'The info of the resource'
        },
        hidePDFThumbnail: {
            control: 'boolean',
            description: 'Whether to hide the PDF thumbnail'
        }
    },
    parameters: {
        controls: {
            include: ['link', 'cardTitle', 'info', 'hidePDFThumbnail']
        }
    }
};

export const Default = {
    args: {
        link: resourceCardMock.link,
        cardTitle: resourceCardMock.cardTitle,
        info: resourceCardMock.info,
        hidePDFThumbnail: false
    }
};
