// @appdirect/sfb-theme-components/src/atoms/grid/stories/Grid.stories.js
import React from 'react';
import Grid from '../Grid';

const entries = [
    { label: 'one', color: '#EAEAEA' },
    { label: 'two', color: '#CACACA' },
    { label: 'three', color: '#F4F4F4' },
    { label: 'four', color: '#0076ce' },
    { label: 'five', color: '#707070' },
    { label: 'six', color: '#F4F4F4' },
    { label: 'seven', color: '#EAEAEA' }
];

const getEntry = entry => (
    <div key={entry.label} style={{ backgroundColor: entry.color, minHeight: '150px', padding: '7px' }}>
        {entry.label}
    </div>
);

export default {
    title: 'Atoms/Grid',
    component: Grid,
    render: args => (
        <Grid {...args}>
            {entries.map(getEntry)}
        </Grid>
    ),
    argTypes: {
        maxCol: {
            control: 'number',
            description: 'The maximum number of columns to display'
        },
        minWidth: {
            control: 'text',
            description: 'The minimum width of the grid items'
        },
        stretch: {
            control: 'boolean',
            description: 'Whether the grid items should stretch to the full width of the container'
        }
    },
    parameters: {
        controls: {
            include: ['maxCol', 'minWidth', 'stretch']
        }
    }
};

export const GenericColumns = {
    args: {
        maxCol: 3,
        stretch: false
    },
    parameters: {
        controls: {
            exclude: ['minWidth']
        }
    }
};

export const WithMinWidth = {
    args: {
        maxCol: 4,
        minWidth: '302px',
        stretch: false
    },
    parameters: {
        controls: {
            exclude: ['maxWidth']
        }
    }
};
